{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#%load_ext autoreload\n",
    "%reload_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,sys\n",
    "\n",
    "sys.path.insert(0, \"/mnt/f/dev/git/miRExplore/python/\")\n",
    "\n",
    "import time\n",
    "import spacy\n",
    "from spacy import displacy\n",
    "\n",
    "import scispacy\n",
    "\n",
    "%autoreload 2\n",
    "from textdb.MiGenRelDB import MiGenRelDB\n",
    "%autoreload 2\n",
    "from textdb.SentenceDB import SentenceDB\n",
    "%autoreload 2\n",
    "from textmining.MirGeneRelCheck import SentenceRelationClassifier, SentenceRelationChecker\n",
    "\n",
    "from collections import defaultdict\n",
    "from natsort import natsorted\n",
    "from intervaltree import Interval, IntervalTree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp = spacy.load('/mnt/f/spacy/en_core_sci_lg-0.2.4/en_core_sci_lg/en_core_sci_lg-0.2.4/')\n",
    "nlp_ent = spacy.load(\"/mnt/f/spacy/en_ner_bionlp13cg_md-0.2.4/en_ner_bionlp13cg_md/en_ner_bionlp13cg_md-0.2.4\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Building Sentence DB\n",
      "Building Sentence DB\n"
     ]
    }
   ],
   "source": [
    "sentDBTest = SentenceDB.loadFromFile(\"/mnt/f/dev/data/pmid_jun2020/mirtex_benchmark/test/\", \"/mnt/f/dev/data/pmid_jun2020/mirtex_benchmark/development/pmid2sent\", redoPmid2Sent=True)\n",
    "sentDBDev = SentenceDB.loadFromFile(\"/mnt/f/dev/data/pmid_jun2020/mirtex_benchmark/development/\", \"/mnt/f/dev/data/pmid_jun2020/mirtex_benchmark/development/pmid2sent\", redoPmid2Sent=True)\n",
    "#mmuDB = MiGenRelDB.loadFromFile(\"/mnt/f/dev/data/pmid_jun2020/mirtex_benchmark/aggregated_test/mirna_gene.mmu.pmid\", ltype=\"mirna\", rtype=\"gene\")\n",
    "#hsaDB = MiGenRelDB.loadFromFile(\"/mnt/f/dev/data/pmid_jun2020/mirtex_benchmark/aggregated_test/mirna_gene.hsa.pmid\", ltype=\"mirna\", rtype=\"gene\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "relChecker = SentenceRelationChecker(nlp, nlp_ent)\n",
    "relClassifier = SentenceRelationClassifier()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "allSentsTest = sentDBTest.get_all_sentences()\n",
    "allSentsDev = sentDBDev.get_all_sentences()\n",
    "\n",
    "allSents = {}\n",
    "for x in allSentsTest:\n",
    "    allSents[x] = allSentsTest[x]\n",
    "for x in allSentsDev:\n",
    "    allSents[x] = allSentsDev[x]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def analyse(testCase, dist=80):\n",
    "\n",
    "    doc = nlp(testCase[0])\n",
    "    entDoc = nlp_ent(testCase[0])\n",
    "    \n",
    "    enttree = IntervalTree()\n",
    "    for ent in entDoc.ents:\n",
    "        enttree.addi(ent.start_char, ent.end_char, (ent.label_, ent.text))\n",
    "\n",
    "    \n",
    "    lWord = doc[testCase[1]]\n",
    "    rWord = doc[testCase[2]]\n",
    "\n",
    "    for t in doc:\n",
    "        \n",
    "        possTokenEnts = enttree[t.idx: t.idx+len(t.text)]\n",
    "        ents = set([x.data for x in possTokenEnts])\n",
    "        \n",
    "        print(t.idx, t.idx+len(t.text), t.i, t, t.pos_, t.dep_, t.head, t.head.idx, ents)\n",
    "\n",
    "    print(lWord, rWord)\n",
    "\n",
    "    if not \"mir\" in str(lWord).lower() and not \"let\" in str(lWord).lower():\n",
    "        tmp = lWord\n",
    "        lWord = rWord\n",
    "        rWord = tmp\n",
    "        \n",
    "    print(lWord, rWord)\n",
    "    \n",
    "    displacy.render(doc, style=\"dep\", options={\"compact\": True, \"distance\": dist})\n",
    "    \n",
    "    entDoc = nlp_ent(testCase[0])\n",
    "    \n",
    "    \n",
    "    displacy.render(entDoc, style=\"ent\", options={\"compact\": True, \"distance\": dist})\n",
    "    \n",
    "    return enttree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 2 0 In ADP case article 8 set()\n",
      "3 7 1 this DET det article 8 set()\n",
      "8 15 2 article NOUN nmod demonstrate 20 set()\n",
      "15 16 3 , PUNCT punct demonstrate 20 set()\n",
      "17 19 4 we PRON nsubj demonstrate 20 set()\n",
      "20 31 5 demonstrate VERB ROOT demonstrate 20 set()\n",
      "32 36 6 that SCONJ mark regulated 58 set()\n",
      "37 44 7 miR-34a PROPN nsubjpass regulated 58 {('GENE_OR_GENE_PRODUCT', 'miR-34a')}\n",
      "45 48 8 and CCONJ cc miR-34a 37 set()\n",
      "49 53 9 -145 NUM conj miR-34a 37 set()\n",
      "54 57 10 are AUX auxpass regulated 58 set()\n",
      "58 67 11 regulated VERB ccomp demonstrate 20 set()\n",
      "68 70 12 by ADP case p53 71 set()\n",
      "71 74 13 p53 NOUN nmod regulated 58 {('GENE_OR_GENE_PRODUCT', 'p53')}\n",
      "75 78 14 and CCONJ cc regulated 58 set()\n",
      "79 89 15 negatively ADV advmod regulate 90 set()\n",
      "90 98 16 regulate VERB conj regulated 58 set()\n",
      "99 105 17 Twist2 PROPN compound expression 116 {('GENE_OR_GENE_PRODUCT', 'Twist2')}\n",
      "106 109 18 and CCONJ cc Twist2 99 set()\n",
      "110 115 19 c-Maf NOUN conj Twist2 99 {('GENE_OR_GENE_PRODUCT', 'c-Maf')}\n",
      "116 126 20 expression NOUN dobj regulate 90 set()\n",
      "127 129 21 in ADP case microglia 130 set()\n",
      "130 139 22 microglia NOUN nmod expression 116 {('CELL', 'microglia')}\n",
      "140 143 23 and CCONJ cc microglia 130 set()\n",
      "144 147 24 the DET det line 168 set()\n",
      "148 151 25 RAW NOUN compound line 168 {('CELL', 'RAW macrophage cell line')}\n",
      "152 162 26 macrophage NOUN compound line 168 {('CELL', 'RAW macrophage cell line')}\n",
      "163 167 27 cell NOUN compound line 168 {('CELL', 'RAW macrophage cell line')}\n",
      "168 172 28 line NOUN conj microglia 130 {('CELL', 'RAW macrophage cell line')}\n",
      "172 173 29 . PUNCT punct demonstrate 20 set()\n",
      "-145 are\n",
      "are -145\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"6bcebb8ec96c4319b6aaeeb5153586a2-0\" class=\"displacy\" width=\"2290\" height=\"377.0\" direction=\"ltr\" style=\"max-width: none; height: 377.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">In</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"130\">this</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"130\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"210\">article,</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"210\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"290\">we</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"290\">PRON</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"370\">demonstrate</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"370\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"450\">that</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"450\">SCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"530\">miR-34a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"530\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"610\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"610\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"690\">-145</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"690\">NUM</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"770\">are</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"770\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"850\">regulated</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"850\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"930\">by</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"930\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1010\">p53</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1010\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1090\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1090\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1170\">negatively</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1170\">ADV</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1250\">regulate</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1250\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1330\">Twist2</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1330\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1410\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1410\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1490\">c-Maf</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1490\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1570\">expression</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1570\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1650\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1650\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1730\">microglia</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1730\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1810\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1810\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1890\">the</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1890\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1970\">RAW</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1970\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2050\">macrophage</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2050\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2130\">cell</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2130\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2210\">line.</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2210\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-0\" stroke-width=\"2px\" d=\"M62,242.0 62,215.33333333333334 198.0,215.33333333333334 198.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M62,244.0 L58,236.0 66,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-1\" stroke-width=\"2px\" d=\"M142,242.0 142,228.66666666666666 195.0,228.66666666666666 195.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M142,244.0 L138,236.0 146,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-2\" stroke-width=\"2px\" d=\"M222,242.0 222,215.33333333333334 358.0,215.33333333333334 358.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M222,244.0 L218,236.0 226,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-3\" stroke-width=\"2px\" d=\"M302,242.0 302,228.66666666666666 355.0,228.66666666666666 355.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M302,244.0 L298,236.0 306,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-4\" stroke-width=\"2px\" d=\"M462,242.0 462,175.33333333333331 847.0,175.33333333333331 847.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">mark</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M462,244.0 L458,236.0 466,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-5\" stroke-width=\"2px\" d=\"M542,242.0 542,188.66666666666666 844.0,188.66666666666666 844.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubjpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M542,244.0 L538,236.0 546,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-6\" stroke-width=\"2px\" d=\"M542,242.0 542,228.66666666666666 595.0,228.66666666666666 595.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M595.0,244.0 L599.0,236.0 591.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-7\" stroke-width=\"2px\" d=\"M542,242.0 542,215.33333333333334 678.0,215.33333333333334 678.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M678.0,244.0 L682.0,236.0 674.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-8\" stroke-width=\"2px\" d=\"M782,242.0 782,228.66666666666666 835.0,228.66666666666666 835.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">auxpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M782,244.0 L778,236.0 786,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-9\" stroke-width=\"2px\" d=\"M382,242.0 382,162.0 850.0,162.0 850.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">ccomp</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M850.0,244.0 L854.0,236.0 846.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-10\" stroke-width=\"2px\" d=\"M942,242.0 942,228.66666666666666 995.0,228.66666666666666 995.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M942,244.0 L938,236.0 946,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-11\" stroke-width=\"2px\" d=\"M862,242.0 862,215.33333333333334 998.0,215.33333333333334 998.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M998.0,244.0 L1002.0,236.0 994.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-12\" stroke-width=\"2px\" d=\"M862,242.0 862,202.0 1081.0,202.0 1081.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1081.0,244.0 L1085.0,236.0 1077.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-13\" stroke-width=\"2px\" d=\"M1182,242.0 1182,228.66666666666666 1235.0,228.66666666666666 1235.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-13\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">advmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1182,244.0 L1178,236.0 1186,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-14\" stroke-width=\"2px\" d=\"M862,242.0 862,175.33333333333331 1247.0,175.33333333333331 1247.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-14\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1247.0,244.0 L1251.0,236.0 1243.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-15\" stroke-width=\"2px\" d=\"M1342,242.0 1342,202.0 1561.0,202.0 1561.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-15\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1342,244.0 L1338,236.0 1346,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-16\" stroke-width=\"2px\" d=\"M1342,242.0 1342,228.66666666666666 1395.0,228.66666666666666 1395.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-16\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1395.0,244.0 L1399.0,236.0 1391.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-17\" stroke-width=\"2px\" d=\"M1342,242.0 1342,215.33333333333334 1478.0,215.33333333333334 1478.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-17\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1478.0,244.0 L1482.0,236.0 1474.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-18\" stroke-width=\"2px\" d=\"M1262,242.0 1262,188.66666666666666 1564.0,188.66666666666666 1564.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-18\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1564.0,244.0 L1568.0,236.0 1560.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-19\" stroke-width=\"2px\" d=\"M1662,242.0 1662,228.66666666666666 1715.0,228.66666666666666 1715.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-19\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1662,244.0 L1658,236.0 1666,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-20\" stroke-width=\"2px\" d=\"M1582,242.0 1582,215.33333333333334 1718.0,215.33333333333334 1718.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-20\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1718.0,244.0 L1722.0,236.0 1714.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-21\" stroke-width=\"2px\" d=\"M1742,242.0 1742,228.66666666666666 1795.0,228.66666666666666 1795.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-21\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1795.0,244.0 L1799.0,236.0 1791.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-22\" stroke-width=\"2px\" d=\"M1902,242.0 1902,188.66666666666666 2204.0,188.66666666666666 2204.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-22\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1902,244.0 L1898,236.0 1906,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-23\" stroke-width=\"2px\" d=\"M1982,242.0 1982,202.0 2201.0,202.0 2201.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-23\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1982,244.0 L1978,236.0 1986,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-24\" stroke-width=\"2px\" d=\"M2062,242.0 2062,215.33333333333334 2198.0,215.33333333333334 2198.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-24\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2062,244.0 L2058,236.0 2066,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-25\" stroke-width=\"2px\" d=\"M2142,242.0 2142,228.66666666666666 2195.0,228.66666666666666 2195.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-25\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2142,244.0 L2138,236.0 2146,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-26\" stroke-width=\"2px\" d=\"M1742,242.0 1742,162.0 2210.0,162.0 2210.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6bcebb8ec96c4319b6aaeeb5153586a2-0-26\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2210.0,244.0 L2214.0,236.0 2206.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">In this article, we demonstrate that \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em; box-decoration-break: clone; -webkit-box-decoration-break: clone\">\n",
       "    miR-34a\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; text-transform: uppercase; vertical-align: middle; margin-left: 0.5rem\">GENE_OR_GENE_PRODUCT</span>\n",
       "</mark>\n",
       " and -145 are regulated by \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em; box-decoration-break: clone; -webkit-box-decoration-break: clone\">\n",
       "    p53\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; text-transform: uppercase; vertical-align: middle; margin-left: 0.5rem\">GENE_OR_GENE_PRODUCT</span>\n",
       "</mark>\n",
       " and negatively regulate \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em; box-decoration-break: clone; -webkit-box-decoration-break: clone\">\n",
       "    Twist2\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; text-transform: uppercase; vertical-align: middle; margin-left: 0.5rem\">GENE_OR_GENE_PRODUCT</span>\n",
       "</mark>\n",
       " and \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em; box-decoration-break: clone; -webkit-box-decoration-break: clone\">\n",
       "    c-Maf\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; text-transform: uppercase; vertical-align: middle; margin-left: 0.5rem\">GENE_OR_GENE_PRODUCT</span>\n",
       "</mark>\n",
       " expression in \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em; box-decoration-break: clone; -webkit-box-decoration-break: clone\">\n",
       "    microglia\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; text-transform: uppercase; vertical-align: middle; margin-left: 0.5rem\">CELL</span>\n",
       "</mark>\n",
       " and the \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em; box-decoration-break: clone; -webkit-box-decoration-break: clone\">\n",
       "    RAW macrophage cell line\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; text-transform: uppercase; vertical-align: middle; margin-left: 0.5rem\">CELL</span>\n",
       "</mark>\n",
       ".</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "IntervalTree([Interval(37, 44, ('GENE_OR_GENE_PRODUCT', 'miR-34a')), Interval(71, 74, ('GENE_OR_GENE_PRODUCT', 'p53')), Interval(99, 105, ('GENE_OR_GENE_PRODUCT', 'Twist2')), Interval(110, 115, ('GENE_OR_GENE_PRODUCT', 'c-Maf')), Interval(130, 139, ('CELL', 'microglia')), Interval(148, 172, ('CELL', 'RAW macrophage cell line'))])"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "analyse( (allSents[\"24319262.2.12\"], 9, 10) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case let-7\n",
      "add special case Bach1\n",
      "let-7 3 Bach1 6\n",
      "(u\"The effects of let-7 miRNA on Bach1 and HMOX1 gene expression in Huh-7 and HepG2 cells were determined by real-time qRT-PCR, Western blot, and luciferase reporter assays.\", 3, 6)\n",
      "geneword Bach1 mirword let-7\n",
      "Conjuncts [HMOX1, Bach1]\n",
      "neighbour HMOX1 gene\n",
      "neighbour HMOX1 gene\n",
      "Conjunction before fc Bach1\n",
      "fc add children for Bach1\n",
      "fc add children for Bach1\n",
      "Conjuncts [Bach1, HMOX1]\n",
      "neighbour Bach1 and\n",
      "neighbour Bach1 and\n",
      "Conjunction before fc HMOX1\n",
      "Conjuncts [HepG2, Huh-7]\n",
      "neighbour HepG2 cells\n",
      "neighbour HepG2 cells\n",
      "Conjunction before fc Huh-7\n",
      "fc add children for Huh-7\n",
      "fc add children for Huh-7\n",
      "Conjuncts [Huh-7, HepG2]\n",
      "neighbour Huh-7 and\n",
      "neighbour Huh-7 and\n",
      "Conjunction before fc HepG2\n",
      "Conjuncts [blot, assays, qRT-PCR]\n",
      "neighbour blot ,\n",
      "neighbour blot and\n",
      "neighbour assays .\n",
      "neighbour assays None\n",
      "Conjunction before fc qRT-PCR\n",
      "fc add children for qRT-PCR\n",
      "fc add children for qRT-PCR\n",
      "Conjuncts [qRT-PCR, assays, blot]\n",
      "neighbour qRT-PCR ,\n",
      "neighbour qRT-PCR Western\n",
      "neighbour assays .\n",
      "neighbour assays None\n",
      "Conjunction before fc blot\n",
      "fc add children for blot\n",
      "Conjuncts [qRT-PCR, blot, assays]\n",
      "neighbour qRT-PCR ,\n",
      "neighbour qRT-PCR Western\n",
      "neighbour blot ,\n",
      "neighbour blot and\n",
      "Conjunction before fc assays\n",
      "fc add children for assays\n",
      "fc add children for assays\n",
      "Conunct merge check: Bach1 Huh-7\n",
      "expression\n",
      "Conunct merge check: Huh-7 qRT-PCR\n",
      "cells were determined\n",
      "Conjunctions\n",
      "Bach1 [on, Bach1, HMOX1, gene, expression]\n",
      "Huh-7 [in, Huh-7, HepG2, cells]\n",
      "qRT-PCR [by, real-time, qRT-PCR, Western, blot, luciferase, reporter, assays]\n",
      "SDP sent: The effects of let-7 miRNA on Bach1 and HMOX1 gene expression in Huh-7 and HepG2 cells were determined by real-time qRT-PCR, Western blot, and luciferase reporter assays.\n",
      "SDP     : [(let-7, 'PROPN', 'compound', False), (miRNA, 'NOUN', 'nmod', False), (effects, 'NOUN', 'nsubjpass', False), (Bach1, 'PROPN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[HMOX1]\n",
      "DeepCheck Conj Subtree\n",
      "[HepG2]\n",
      "DeepCheck Conj Subtree\n",
      "[Western, blot]\n",
      "DeepCheck Conj Subtree\n",
      "[luciferase, reporter, assays]\n",
      "0 [The, effects, of, let-7, miRNA, on, Bach1, and, HMOX1, gene, expression, in, Huh-7, and, HepG2, cells, were, determined, by, real-time, qRT-PCR, ,, Western, blot, ,, and, luciferase, reporter, assays] 29 0 The effects of let-7 miRNA on Bach1 and HMOX1 gene expression in Huh-7 and HepG2 cells were determined by real-time qRT-PCR, Western blot, and luciferase reporter assays.\n",
      "Recognized overlapping entities: {Interval(30, 35, ('GENE_OR_GENE_PRODUCT', 'Bach1'))}\n",
      "passive? True\n",
      "Counter()\n",
      "Compartment Text The effects of let-7 miRNA on Bach1 and HMOX1 gene expression in Huh-7 and HepG2 cells were determined by real-time qRT-PCR , Western blot , and luciferase reporter assays\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'compartment mir gene by', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence The effects of let-7 miRNA on Bach1 and HMOX1 gene expression in Huh-7 and HepG2 cells were determined by real-time qRT-PCR, Western blot, and luciferase reporter assays.\n",
      "full sentence The effects of <e1>let-7</e1> miRNA on <e2>Bach1</e2> and HMOX1 gene expression in Huh-7 and HepG2 cells were determined by real-time qRT-PCR, Western blot, and luciferase reporter assays.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': True,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': True,\n",
       "   'reg_detect': 'compartment mir gene by',\n",
       "   'reg_detect_major': 'compartment'}},\n",
       " 'tagged_sent': 'The effects of <e1>let-7</e1> miRNA on <e2>Bach1</e2> and HMOX1 gene expression in Huh-7 and HepG2 cells were determined by real-time qRT-PCR, Western blot, and luciferase reporter assays.',\n",
       " 'full_sentence': 'The effects of <e1>let-7</e1> miRNA on <e2>Bach1</e2> and HMOX1 gene expression in Huh-7 and HepG2 cells were determined by real-time qRT-PCR, Western blot, and luciferase reporter assays.',\n",
       " 'entity1': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (15, 20),\n",
       "  'entity_text': 'let-7'},\n",
       " 'entity2': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (30, 35),\n",
       "  'entity_text': 'Bach1'}}"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relChecker = SentenceRelationChecker(nlp, nlp_ent)\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "#  (37, 46)\t(12, 20)\n",
    "\n",
    "\n",
    "relRes = relChecker.check_sentence(allSents[\"22698995.2.4\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (15, 20)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (30, 35)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case tenascin\n",
      "add special case miR-335\n",
      "tenascin 18 miR-335 24\n",
      "(u\"miR-335 suppresses metastasis and migration through targeting of the progenitor cell transcription factor SOX4 and extracellular matrix component tenascin C. Expression of miR-126 and miR-335 is lost in the majority of primary breast tumours from patients who relapse, and the loss of expression of either microRNA is associated with poor distal metastasis-free survival.\", 18, 24)\n",
      "geneword tenascin mirword miR-335\n",
      "Conjuncts [lost, associated, suppresses]\n",
      "neighbour lost in\n",
      "neighbour lost in\n",
      "neighbour associated with\n",
      "neighbour associated with\n",
      "Conjunction before fc suppresses\n",
      "fc add children for suppresses\n",
      "fc add children for suppresses\n",
      "Conjuncts [migration, metastasis]\n",
      "neighbour migration through\n",
      "neighbour migration through\n",
      "Conjunction before fc metastasis\n",
      "Conjuncts [metastasis, migration]\n",
      "neighbour metastasis and\n",
      "neighbour metastasis and\n",
      "Conjunction before fc migration\n",
      "Conjuncts [component, SOX4]\n",
      "neighbour component tenascin\n",
      "neighbour component tenascin\n",
      "Conjunction before fc SOX4\n",
      "fc add children for SOX4\n",
      "fc add children for SOX4\n",
      "fc add children for SOX4\n",
      "fc add children for SOX4\n",
      "fc add children for SOX4\n",
      "Conjuncts [SOX4, component]\n",
      "neighbour SOX4 and\n",
      "neighbour SOX4 and\n",
      "Conjunction before fc component\n",
      "fc add children for component\n",
      "fc add children for component\n",
      "Conjuncts [miR-335, miR-126]\n",
      "neighbour miR-335 is\n",
      "neighbour miR-335 is\n",
      "Conjunction before fc miR-126\n",
      "fc add children for miR-126\n",
      "Conjuncts [miR-126, miR-335]\n",
      "neighbour miR-126 and\n",
      "neighbour miR-126 and\n",
      "Conjunction before fc miR-335\n",
      "Conjuncts [suppresses, associated, lost]\n",
      "neighbour suppresses metastasis\n",
      "neighbour suppresses metastasis\n",
      "neighbour associated with\n",
      "neighbour associated with\n",
      "Conjunction before fc lost\n",
      "fc add children for lost\n",
      "Conjuncts [suppresses, lost, associated]\n",
      "neighbour suppresses metastasis\n",
      "neighbour suppresses metastasis\n",
      "neighbour lost in\n",
      "neighbour lost in\n",
      "Conjunction before fc associated\n",
      "fc add children for associated\n",
      "Conunct merge check: suppresses miR-126\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions\n",
      "suppresses [suppresses, metastasis, migration, through, targeting, of, progenitor, cell, transcription, factor, SOX4, extracellular, matrix, component, lost, in, majority, of, primary, breast, tumours, from, patients, associated, with, poor, distal, metastasis-free, survival]\n",
      "miR-126 [of, miR-126, miR-335]\n",
      "SDP sent: miR-335 suppresses metastasis and migration through targeting of the progenitor cell transcription factor SOX4 and extracellular matrix component tenascin C. Expression of miR-126 and miR-335 is lost in the majority of primary breast tumours from patients who relapse, and the loss of expression of either microRNA is associated with poor distal metastasis-free survival.\n",
      "SDP     : [(tenascin, 'PROPN', 'compound', False), (Expression, 'NOUN', 'nsubjpass', False), (miR-126, 'PROPN', 'nmod', True), (miR-335, 'PROPN', 'conj', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[migration]\n",
      "DeepCheck Conj Subtree\n",
      "[extracellular, matrix, component]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-335]\n",
      "deepcheck connection rule noun found lost tenascin [tenascin, C., Expression, of, miR-126, and, miR-335, is, lost, in, the, majority, of, primary, breast, tumours, from, patients, who, relapse, ,, and, the, loss, of, expression, of, either, microRNA, is, associated, with, poor, distal, metastasis-free, survival]\n",
      "creating subtree at lost conj\n",
      "creating subtree [tenascin, C., Expression, of, miR-126, and, miR-335, is, lost, in, the, majority, of, primary, breast, tumours, from, patients, who, relapse, ,, and, the, loss, of, expression, of, either, microRNA, is, associated, with, poor, distal, metastasis-free, survival]\n",
      "deepcheck connection rule noun found associated loss [the, loss, of, expression, of, either, microRNA, is, associated, with, poor, distal, metastasis-free, survival]\n",
      "creating subtree at associated conj\n",
      "creating subtree [the, loss, of, expression, of, either, microRNA, is, associated, with, poor, distal, metastasis-free, survival]\n",
      "0 [miR-335, suppresses, metastasis, and, migration, through, targeting, of, the, progenitor, cell, transcription, factor, SOX4, and, extracellular, matrix, component] 18 0 miR-335 suppresses metastasis and migration through targeting of the progenitor cell transcription factor SOX4 and extracellular matrix component tenascin C. Expression of miR-126 and miR-335 is lost in the majority of primary breast tumours from patients who relapse, and the loss of expression of either microRNA is associated with poor distal metastasis-free survival.\n",
      "1 [tenascin, C., Expression, of, miR-126, and, miR-335, is, lost, in, the, majority, of, primary, breast, tumours, from, patients, who, relapse] 20 146 tenascin C. Expression of miR-126 and miR-335 is lost in the majority of primary breast tumours from patients who relapse, and the loss of expression of either microRNA is associated with poor distal metastasis-free survival.\n",
      "2 [the, loss, of, expression, of, either, microRNA, is, associated, with, poor, distal, metastasis-free, survival] 14 273 the loss of expression of either microRNA is associated with poor distal metastasis-free survival.\n",
      "Recognized overlapping entities: set()\n",
      "creating subtree at lost conj\n",
      "creating subtree [tenascin, C., Expression, of, miR-126, and, miR-335, is, lost, in, the, majority, of, primary, breast, tumours, from, patients, who, relapse, ,, and, the, loss, of, expression, of, either, microRNA, is, associated, with, poor, distal, metastasis-free, survival]\n",
      "creating subtree at associated conj\n",
      "creating subtree [the, loss, of, expression, of, either, microRNA, is, associated, with, poor, distal, metastasis-free, survival]\n",
      "passive? True\n",
      "Counter()\n",
      "Compartment Text tenascin C. Expression of miR-126 and miR-335 is lost in the majority of primary breast tumours from patients who relapse\n",
      "[tenascin, C., Expression, of, miR-126, and, miR-335, is, lost, in, the, majority, of, primary, breast, tumours, from, patients, who, relapse]\n",
      "TextBetween: tenascin C. Expression of miR-126 and miR-335 is lost in the majority of primary breast tumours from patients who relapse\n",
      "infer direction from token set [tenascin, C., Expression, of, miR-126, and, miR-335, is, lost, in, the, majority, of, primary, breast, tumours, from, patients, who, relapse]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f7ade6a7280>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f7ade6a7ee0>, {'gene': Counter({'NEU': 1}), 'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(Expression, 'NEU'), (lost, 'NEG')]\n",
      "counts between final True\n",
      "['tenascin', 'C.', 'Expression', 'of', 'miR-126', 'and', 'miR-335', 'is', 'lost']\n",
      "Counter({'NEU': 1, 'NEG': 1})\n",
      "[(Expression, 'NEU'), (lost, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-335 suppresses metastasis and migration through targeting of the progenitor cell transcription factor SOX4 and extracellular matrix component tenascin C. Expression of miR-126 and miR-335 is lost in the majority of primary breast tumours from patients who relapse, and the loss of expression of either microRNA is associated with poor distal metastasis-free survival.\n",
      "full sentence miR-335 suppresses metastasis and migration through targeting of the progenitor cell transcription factor SOX4 and extracellular matrix component <e2>tenascin</e2> C. Expression of miR-126 and <e1>miR-335</e1> is lost in the majority of primary breast tumours from patients who relapse, and the loss of expression of either microRNA is associated with poor distal metastasis-free survival.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': True,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'DOWN',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': True,\n",
       "   'reg_detect': 'counts between',\n",
       "   'reg_detect_major': 'counts'}},\n",
       " 'tagged_sent': 'miR-335 suppresses metastasis and migration through targeting of the progenitor cell transcription factor SOX4 and extracellular matrix component <e2>tenascin</e2> C. Expression of miR-126 and <e1>miR-335</e1> is lost in the majority of primary breast tumours from patients who relapse, and the loss of expression of either microRNA is associated with poor distal metastasis-free survival.',\n",
       " 'full_sentence': 'miR-335 suppresses metastasis and migration through targeting of the progenitor cell transcription factor SOX4 and extracellular matrix component <e2>tenascin</e2> C. Expression of miR-126 and <e1>miR-335</e1> is lost in the majority of primary breast tumours from patients who relapse, and the loss of expression of either microRNA is associated with poor distal metastasis-free survival.',\n",
       " 'entity1': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (146, 154),\n",
       "  'entity_text': 'tenascin'},\n",
       " 'entity2': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (184, 191),\n",
       "  'entity_text': 'miR-335'}}"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"18185580.2.5\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (184, 191)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (146, 154)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case mitogen-activated protein kinase kinase 1\n",
      "add special case miR-34a\n",
      "1 8 miR-34a 17\n",
      "(u\"Reporter gene assays identified mitogen-activated protein kinase kinase 1 (MEK1) as a direct target of miR-34a and c-fos as a direct target of miR-221/222.\", 8, 17)\n",
      "geneword 1 mirword miR-34a\n",
      "Conjuncts [c-fos, miR-34a]\n",
      "neighbour c-fos as\n",
      "neighbour c-fos as\n",
      "Conjunction before fc miR-34a\n",
      "fc add children for miR-34a\n",
      "Conjuncts [miR-34a, c-fos]\n",
      "neighbour miR-34a and\n",
      "neighbour miR-34a and\n",
      "Conjunction before fc c-fos\n",
      "MEK1 miR-34a\n",
      ") as a direct target\n",
      "Conjunctions\n",
      "MEK1 [(, )]\n",
      "miR-34a [of, miR-34a, c-fos]\n",
      "SDP sent: Reporter gene assays identified mitogen-activated protein kinase kinase 1 (MEK1) as a direct target of miR-34a and c-fos as a direct target of miR-221/222.\n",
      "SDP     : [(1, 'NUM', 'nummod', False), (kinase, 'NOUN', 'dobj', False), (identified, 'VERB', 'ROOT', False), (target, 'NOUN', 'nmod', False), (miR-34a, 'PROPN', 'nmod', False)]\n",
      "[c-fos]\n",
      "0 [Reporter, gene, assays, identified, mitogen-activated, protein, kinase, kinase, 1, (, MEK1, ), as, a, direct, target, of, miR-34a, and, c-fos, as, a, direct, target, of, miR-221/222] 26 0 Reporter gene assays identified mitogen-activated protein kinase kinase 1 (MEK1) as a direct target of miR-34a and c-fos as a direct target of miR-221/222.\n",
      "Conjuncts [c-fos, miR-34a]\n",
      "neighbour c-fos as\n",
      "neighbour c-fos as\n",
      "Conjunction before fc miR-34a\n",
      "fc add children for miR-34a\n",
      "Conjuncts [miR-34a, c-fos]\n",
      "neighbour miR-34a and\n",
      "neighbour miR-34a and\n",
      "Conjunction before fc c-fos\n",
      "Recognized overlapping entities: {Interval(32, 73, ('GENE_OR_GENE_PRODUCT', 'mitogen-activated protein kinase kinase 1'))}\n",
      "[Interval(72, 73, ('GENE_OR_GENE_PRODUCT', '1', 1))] [Interval(103, 110, ('GENE_OR_GENE_PRODUCT', 'miR-34a', miR-34a))]\n",
      "miR-34a 1\n",
      "[c-fos]\n",
      "[c-fos]\n",
      "passive? False\n",
      "Counter()\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence Reporter gene assays identified mitogen-activated protein kinase kinase 1 (MEK1) as a direct target of miR-34a and c-fos as a direct target of miR-221/222.\n",
      "full sentence Reporter gene assays identified <e2>mitogen-activated protein kinase kinase 1</e2> (MEK1) as a direct target of <e1>miR-34a</e1> and c-fos as a direct target of miR-221/222.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': False,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'GENE_MIR',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'compartment gene mir',\n",
       "   'reg_detect_major': 'compartment'}},\n",
       " 'tagged_sent': 'Reporter gene assays identified <e2>mitogen-activated protein kinase kinase 1</e2> (MEK1) as a direct target of <e1>miR-34a</e1> and c-fos as a direct target of miR-221/222.',\n",
       " 'full_sentence': 'Reporter gene assays identified <e2>mitogen-activated protein kinase kinase 1</e2> (MEK1) as a direct target of <e1>miR-34a</e1> and c-fos as a direct target of miR-221/222.',\n",
       " 'entity1': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (32, 73),\n",
       "  'entity_text': 'mitogen-activated protein kinase kinase 1'},\n",
       " 'entity2': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (103, 110),\n",
       "  'entity_text': 'miR-34a'}}"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"20299489.2.6\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (103,110)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (32,73)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case TGF-betaR2\n",
      "add special case miR-21\n",
      "TGF-betaR2 0 miR-21 5\n",
      "(u\"TGF-betaR2 mRNA, a validated miR-21 target, showed the highest expression in the leukocytes from a subset of the octogenarians.\", 0, 5)\n",
      "geneword TGF-betaR2 mirword miR-21\n",
      "Conjunctions\n",
      "target [a, target]\n",
      "SDP sent: TGF-betaR2 mRNA, a validated miR-21 target, showed the highest expression in the leukocytes from a subset of the octogenarians.\n",
      "SDP     : [(TGF-betaR2, 'PROPN', 'amod', False), (mRNA, 'NOUN', 'nsubj', False), (target, 'NOUN', 'appos', False), (miR-21, 'PROPN', 'compound', False)]\n",
      "0 [TGF-betaR2, mRNA, ,, a, validated, miR-21, target, ,, showed, the, highest, expression, in, the, leukocytes, from, a, subset, of, the, octogenarians] 21 0 TGF-betaR2 mRNA, a validated miR-21 target, showed the highest expression in the leukocytes from a subset of the octogenarians.\n",
      "Recognized overlapping entities: {Interval(0, 10, ('GENE_OR_GENE_PRODUCT', 'TGF-betaR2'))}\n",
      "[Interval(0, 10, ('GENE_OR_GENE_PRODUCT', 'TGF-betaR2', TGF-betaR2))] [Interval(29, 35, ('GENE_OR_GENE_PRODUCT', 'miR-21', miR-21)), Interval(36, 42, ('', 'target', target))]\n",
      "miR-21 TGF-betaR2\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: TGF-betaR2 mRNA , a validated miR-21 target ,\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}})\n",
      " hit sentence TGF-betaR2 mRNA, a validated miR-21 target, showed the highest expression in the leukocytes from a subset of the octogenarians.\n",
      "full sentence <e2>TGF-betaR2</e2> mRNA, a validated <e1>miR-21</e1> target, showed the highest expression in the leukocytes from a subset of the octogenarians.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': False,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'between gene mir',\n",
       "   'reg_detect_major': 'between'}},\n",
       " 'tagged_sent': '<e2>TGF-betaR2</e2> mRNA, a validated <e1>miR-21</e1> target, showed the highest expression in the leukocytes from a subset of the octogenarians.',\n",
       " 'full_sentence': '<e2>TGF-betaR2</e2> mRNA, a validated <e1>miR-21</e1> target, showed the highest expression in the leukocytes from a subset of the octogenarians.',\n",
       " 'entity1': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (0, 10),\n",
       "  'entity_text': 'TGF-betaR2'},\n",
       " 'entity2': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (29, 35),\n",
       "  'entity_text': 'miR-21'}}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"23041385.2.11\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (29,35)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (0,10)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case miRNA-29a\n",
      "add special case E2F1\n",
      "miRNA-29a 9 E2F1 24\n",
      "(u\"In summary, data obtained highlight the role of miRNA-29a in the regulation of osteoblastic cell apoptosis by silencing Bcl-2 and Mcl-1 and inducing E2F1 and E2F3 expression.\", 9, 24)\n",
      "geneword E2F1 mirword miRNA-29a\n",
      "Conjuncts [inducing, silencing]\n",
      "neighbour inducing E2F1\n",
      "neighbour inducing E2F1\n",
      "Conjunction before fc silencing\n",
      "fc add children for silencing\n",
      "Conjuncts [Mcl-1, Bcl-2]\n",
      "neighbour Mcl-1 and\n",
      "neighbour Mcl-1 and\n",
      "Conjunction before fc Bcl-2\n",
      "Conjuncts [Bcl-2, Mcl-1]\n",
      "neighbour Bcl-2 and\n",
      "neighbour Bcl-2 and\n",
      "Conjunction before fc Mcl-1\n",
      "Conjuncts [silencing, inducing]\n",
      "neighbour silencing Bcl-2\n",
      "neighbour silencing Bcl-2\n",
      "Conjunction before fc inducing\n",
      "fc add children for inducing\n",
      "Conjuncts [E2F3, E2F1]\n",
      "neighbour E2F3 expression\n",
      "neighbour E2F3 expression\n",
      "Conjunction before fc E2F1\n",
      "Conjuncts [E2F1, E2F3]\n",
      "neighbour E2F1 and\n",
      "neighbour E2F1 and\n",
      "Conjunction before fc E2F3\n",
      "Conjunctions\n",
      "silencing [silencing, Bcl-2, Mcl-1, inducing, E2F1, E2F3, expression]\n",
      "SDP sent: In summary, data obtained highlight the role of miRNA-29a in the regulation of osteoblastic cell apoptosis by silencing Bcl-2 and Mcl-1 and inducing E2F1 and E2F3 expression.\n",
      "SDP     : [(miRNA-29a, 'PROPN', 'nmod', False), (role, 'NOUN', 'dobj', False), (highlight, 'VERB', 'ROOT', False), (silencing, 'VERB', 'advcl', True), (inducing, 'VERB', 'conj', False), (expression, 'NOUN', 'dobj', False), (E2F1, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule [by, silencing, Bcl-2, and, Mcl-1, and, inducing, E2F1, and, E2F3, expression]\n",
      "DeepCheck advcl\n",
      "[Mcl-1]\n",
      "deepcheck connection rule noun found inducing E2F1 [inducing, E2F1, and, E2F3, expression]\n",
      "deepcheck conj verb\n",
      "[E2F3]\n",
      "0 [In, summary, ,, data, obtained, highlight, the, role, of, miRNA-29a, in, the, regulation, of, osteoblastic, cell, apoptosis, by, silencing, Bcl-2, and, Mcl-1, and, inducing, E2F1, and, E2F3, expression] 28 0 In summary, data obtained highlight the role of miRNA-29a in the regulation of osteoblastic cell apoptosis by silencing Bcl-2 and Mcl-1 and inducing E2F1 and E2F3 expression.\n",
      "Conjuncts [inducing, silencing]\n",
      "neighbour inducing E2F1\n",
      "neighbour inducing E2F1\n",
      "Conjunction before fc silencing\n",
      "fc add children for silencing\n",
      "Conjuncts [Mcl-1, Bcl-2]\n",
      "neighbour Mcl-1 and\n",
      "neighbour Mcl-1 and\n",
      "Conjunction before fc Bcl-2\n",
      "Conjuncts [Bcl-2, Mcl-1]\n",
      "neighbour Bcl-2 and\n",
      "neighbour Bcl-2 and\n",
      "Conjunction before fc Mcl-1\n",
      "Conjuncts [silencing, inducing]\n",
      "neighbour silencing Bcl-2\n",
      "neighbour silencing Bcl-2\n",
      "Conjunction before fc inducing\n",
      "fc add children for inducing\n",
      "Conjuncts [E2F3, E2F1]\n",
      "neighbour E2F3 expression\n",
      "neighbour E2F3 expression\n",
      "Conjunction before fc E2F1\n",
      "Conjuncts [E2F1, E2F3]\n",
      "neighbour E2F1 and\n",
      "neighbour E2F1 and\n",
      "Conjunction before fc E2F3\n",
      "Recognized overlapping entities: {Interval(149, 153, ('GENE_OR_GENE_PRODUCT', 'E2F1'))}\n",
      "[Interval(149, 153, ('GENE_OR_GENE_PRODUCT', 'E2F1', E2F1))] [Interval(48, 57, ('GENE_OR_GENE_PRODUCT', 'miRNA-29a', miRNA-29a))]\n",
      "miRNA-29a E2F1\n",
      "[Mcl-1]\n",
      "deepcheck conj verb\n",
      "[E2F3]\n",
      "[Mcl-1]\n",
      "deepcheck conj verb\n",
      "[E2F3]\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "TextBetween: role of miRNA-29a in the regulation of osteoblastic cell apoptosis by silencing Bcl-2 and Mcl-1 and inducing E2F1 and E2F3\n",
      "counts between 1 Counter({'NEU': 1, 'POS': 1})\n",
      "['role', 'of', 'miRNA-29a', 'in', 'the', 'regulation', 'of', 'osteoblastic', 'cell', 'apoptosis', 'by', 'silencing', 'Bcl-2', 'and', 'Mcl-1', 'and', 'inducing', 'E2F1', 'and', 'E2F3']\n",
      "Counter({'NEU': 1, 'POS': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence In summary, data obtained highlight the role of miRNA-29a in the regulation of osteoblastic cell apoptosis by silencing Bcl-2 and Mcl-1 and inducing E2F1 and E2F3 expression.\n",
      "full sentence In summary, data obtained highlight the role of <e1>miRNA-29a</e1> in the regulation of osteoblastic cell apoptosis by silencing Bcl-2 and Mcl-1 and inducing <e2>E2F1</e2> and E2F3 expression.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': False,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'UP',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'counts between',\n",
       "   'reg_detect_major': 'counts'}},\n",
       " 'tagged_sent': 'In summary, data obtained highlight the role of <e1>miRNA-29a</e1> in the regulation of osteoblastic cell apoptosis by silencing Bcl-2 and Mcl-1 and inducing <e2>E2F1</e2> and E2F3 expression.',\n",
       " 'full_sentence': 'In summary, data obtained highlight the role of <e1>miRNA-29a</e1> in the regulation of osteoblastic cell apoptosis by silencing Bcl-2 and Mcl-1 and inducing <e2>E2F1</e2> and E2F3 expression.',\n",
       " 'entity1': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (48, 57),\n",
       "  'entity_text': 'miRNA-29a'},\n",
       " 'entity2': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (149, 153),\n",
       "  'entity_text': 'E2F1'}}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"23113351.2.10\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (48,57)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (149,153)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case Sp1\n",
      "add special case miR-29b\n",
      "Sp1 2 miR-29b 17\n",
      "(u\"We identified Sp1, a transcription factor endowed with oncogenic activity, as a negative regulator of miR-29b expression in MM cells.\", 2, 17)\n",
      "geneword Sp1 mirword miR-29b\n",
      "Conjunctions\n",
      "factor [a, activity]\n",
      "SDP sent: We identified Sp1, a transcription factor endowed with oncogenic activity, as a negative regulator of miR-29b expression in MM cells.\n",
      "SDP     : [(Sp1, 'PROPN', 'dobj', False), (identified, 'VERB', 'ROOT', False), (regulator, 'NOUN', 'nmod', False), (expression, 'NOUN', 'nmod', False), (miR-29b, 'PROPN', 'compound', False)]\n",
      "0 [We, identified, Sp1, ,, a, transcription, factor, endowed, with, oncogenic, activity, ,, as, a, negative, regulator, of, miR-29b, expression, in, MM, cells] 22 0 We identified Sp1, a transcription factor endowed with oncogenic activity, as a negative regulator of miR-29b expression in MM cells.\n",
      "Recognized overlapping entities: {Interval(14, 17, ('GENE_OR_GENE_PRODUCT', 'Sp1'))}\n",
      "[Interval(14, 17, ('GENE_OR_GENE_PRODUCT', 'Sp1', Sp1))] [Interval(102, 109, ('GENE_OR_GENE_PRODUCT', 'miR-29b', miR-29b))]\n",
      "miR-29b Sp1\n",
      "passive? False\n",
      "Counter()\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence We identified Sp1, a transcription factor endowed with oncogenic activity, as a negative regulator of miR-29b expression in MM cells.\n",
      "full sentence We identified <e2>Sp1</e2>, a transcription factor endowed with oncogenic activity, as a negative regulator of <e1>miR-29b</e1> expression in MM cells.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': False,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'GENE_MIR',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'compartment gene mir',\n",
       "   'reg_detect_major': 'compartment'}},\n",
       " 'tagged_sent': 'We identified <e2>Sp1</e2>, a transcription factor endowed with oncogenic activity, as a negative regulator of <e1>miR-29b</e1> expression in MM cells.',\n",
       " 'full_sentence': 'We identified <e2>Sp1</e2>, a transcription factor endowed with oncogenic activity, as a negative regulator of <e1>miR-29b</e1> expression in MM cells.',\n",
       " 'entity1': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (14, 17),\n",
       "  'entity_text': 'Sp1'},\n",
       " 'entity2': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (102, 109),\n",
       "  'entity_text': 'miR-29b'}}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"23190608.2.4\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (102,109)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (14,17)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case miRNA-424\n",
      "add special case mitogen-activated protein kinase 1\n",
      "miRNA-424 3 1 12\n",
      "(u\"The levels of miRNA-424 and its target gene, mitogen-activated protein kinase 1 (MEK1), and fibroblast growth factor receptor 1 (FGFR1) were examined.\", 3, 12)\n",
      "geneword 1 mirword miRNA-424\n",
      "Conjuncts [receptor, levels]\n",
      "neighbour receptor 1\n",
      "neighbour receptor 1\n",
      "Conjunction before fc levels\n",
      "Conjuncts [gene, miRNA-424]\n",
      "neighbour gene ,\n",
      "neighbour gene mitogen-activated\n",
      "Conjunction before fc miRNA-424\n",
      "fc add children for miRNA-424\n",
      "Conjuncts [miRNA-424, gene]\n",
      "neighbour miRNA-424 and\n",
      "neighbour miRNA-424 and\n",
      "Conjunction before fc gene\n",
      "fc add children for gene\n",
      "fc add children for gene\n",
      "Conjuncts [levels, receptor]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "levels miRNA-424\n",
      "\n",
      "miRNA-424 FGFR1\n",
      "), and fibroblast growth factor receptor 1\n",
      "Conjunctions\n",
      "levels [levels, fibroblast, growth, factor, receptor, 1, FGFR1]\n",
      "miRNA-424 [of, miRNA-424, target, gene, mitogen-activated, protein, kinase, 1, (, MEK1, )]\n",
      "FGFR1 [(, )]\n",
      "['miRNA-424', 'and', 'its', 'target', 'gene', ',', 'mitogen-activated', 'protein', 'kinase']\n",
      "CONJ ITS TARGET RULE ['miRNA-424', 'and', 'its', 'target', 'gene', ',', 'mitogen-activated', 'protein', 'kinase']\n",
      "SDP sent: The levels of miRNA-424 and its target gene, mitogen-activated protein kinase 1 (MEK1), and fibroblast growth factor receptor 1 (FGFR1) were examined.\n",
      "SDP     : [(miRNA-424, 'PROPN', 'nmod', False), (examined, 'VERB', 'ROOT', False), (levels, 'NOUN', 'nsubjpass', False), (miRNA-424, 'PROPN', 'nmod', False)]\n",
      "[its, target, gene, ,, mitogen-activated, protein, kinase, 1, (, MEK1, )]\n",
      "[fibroblast, growth, factor, receptor, 1, (, FGFR1, )]\n",
      "0 [The, levels, of, miRNA-424, and, its, target, gene, ,, mitogen-activated, protein, kinase, 1, (, MEK1, ), ,, and, fibroblast, growth, factor, receptor, 1, (, FGFR1, ), were, examined] 28 0 The levels of miRNA-424 and its target gene, mitogen-activated protein kinase 1 (MEK1), and fibroblast growth factor receptor 1 (FGFR1) were examined.\n",
      "Conjuncts [receptor, levels]\n",
      "neighbour receptor 1\n",
      "neighbour receptor 1\n",
      "Conjunction before fc levels\n",
      "Conjuncts [gene, miRNA-424]\n",
      "neighbour gene ,\n",
      "neighbour gene mitogen-activated\n",
      "Conjunction before fc miRNA-424\n",
      "fc add children for miRNA-424\n",
      "Conjuncts [miRNA-424, gene]\n",
      "neighbour miRNA-424 and\n",
      "neighbour miRNA-424 and\n",
      "Conjunction before fc gene\n",
      "fc add children for gene\n",
      "fc add children for gene\n",
      "Conjuncts [levels, receptor]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "Recognized overlapping entities: {Interval(45, 79, ('GENE_OR_GENE_PRODUCT', 'mitogen-activated protein kinase 1'))}\n",
      "[Interval(78, 79, ('GENE_OR_GENE_PRODUCT', '1', 1))] [Interval(14, 23, ('', 'miRNA-424', miRNA-424))]\n",
      "miRNA-424 1\n",
      "[its, target, gene, ,, mitogen-activated, protein, kinase, 1, (, MEK1, )]\n",
      "[fibroblast, growth, factor, receptor, 1, (, FGFR1, )]\n",
      "[its, target, gene, ,, mitogen-activated, protein, kinase, 1, (, MEK1, )]\n",
      "[fibroblast, growth, factor, receptor, 1, (, FGFR1, )]\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: levels of miRNA-424 and its target gene , mitogen-activated protein kinase 1 ( MEK1\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}})\n",
      " hit sentence The levels of miRNA-424 and its target gene, mitogen-activated protein kinase 1 (MEK1), and fibroblast growth factor receptor 1 (FGFR1) were examined.\n",
      "full sentence The levels of <e1>miRNA-424</e1> and its target gene, <e2>mitogen-activated protein kinase 1</e2> (MEK1), and fibroblast growth factor receptor 1 (FGFR1) were examined.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': True,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'between mir gene',\n",
       "   'reg_detect_major': 'between'}},\n",
       " 'tagged_sent': 'The levels of <e1>miRNA-424</e1> and its target gene, <e2>mitogen-activated protein kinase 1</e2> (MEK1), and fibroblast growth factor receptor 1 (FGFR1) were examined.',\n",
       " 'full_sentence': 'The levels of <e1>miRNA-424</e1> and its target gene, <e2>mitogen-activated protein kinase 1</e2> (MEK1), and fibroblast growth factor receptor 1 (FGFR1) were examined.',\n",
       " 'entity1': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (14, 23),\n",
       "  'entity_text': 'miRNA-424'},\n",
       " 'entity2': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (45, 79),\n",
       "  'entity_text': 'mitogen-activated protein kinase 1'}}"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"23643257.2.2\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (14,23)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (45,79)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case miR-125b\n",
      "add special case SIRT1\n",
      "miR-125b 3 SIRT1 15\n",
      "(u\"Use of antisense miR-125b transcripts enhanced expression of pro-apoptotic p53, repressed expression of anti-apoptotic SIRT1 and, importantly, significantly enhanced dexamethasone-induced cell death responses in MM.\", 3, 15)\n",
      "geneword SIRT1 mirword miR-125b\n",
      "Conjuncts [repressed, enhanced, enhanced]\n",
      "neighbour repressed expression\n",
      "neighbour repressed expression\n",
      "neighbour enhanced dexamethasone-induced\n",
      "neighbour enhanced dexamethasone-induced\n",
      "Conjunction before fc enhanced\n",
      "fc add children for enhanced\n",
      "Conjuncts [enhanced, enhanced, repressed]\n",
      "neighbour enhanced expression\n",
      "neighbour enhanced expression\n",
      "neighbour enhanced dexamethasone-induced\n",
      "neighbour enhanced dexamethasone-induced\n",
      "Conjunction before fc repressed\n",
      "fc add children for repressed\n",
      "Conjuncts [enhanced, repressed, enhanced]\n",
      "neighbour enhanced expression\n",
      "neighbour enhanced expression\n",
      "neighbour repressed expression\n",
      "neighbour repressed expression\n",
      "Conjunction before fc enhanced\n",
      "fc add children for enhanced\n",
      "Conjunctions\n",
      "enhanced [enhanced, expression, repressed, expression, enhanced, dexamethasone-induced, cell, death, responses]\n",
      "SDP sent: Use of antisense miR-125b transcripts enhanced expression of pro-apoptotic p53, repressed expression of anti-apoptotic SIRT1 and, importantly, significantly enhanced dexamethasone-induced cell death responses in MM.\n",
      "SDP     : [(miR-125b, 'PROPN', 'amod', False), (transcripts, 'NOUN', 'nmod', False), (Use, 'NOUN', 'nsubj', False), (enhanced, 'VERB', 'ROOT', True), (repressed, 'VERB', 'conj', False), (expression, 'NOUN', 'dobj', False), (SIRT1, 'PROPN', 'nmod', False)]\n",
      "deepcheck connection rule noun found repressed expression [repressed, expression, of, anti-apoptotic, SIRT1, and]\n",
      "deepcheck conj verb\n",
      "creating subtree at enhanced conj\n",
      "0 [Use, of, antisense, miR-125b, transcripts, enhanced, expression, of, pro-apoptotic, p53, ,, repressed, expression, of, anti-apoptotic, SIRT1] 16 0 Use of antisense miR-125b transcripts enhanced expression of pro-apoptotic p53, repressed expression of anti-apoptotic SIRT1 and, importantly, significantly enhanced dexamethasone-induced cell death responses in MM.\n",
      "1 [importantly, ,, significantly, enhanced, dexamethasone-induced, cell, death, responses, in, MM] 10 130 importantly, significantly enhanced dexamethasone-induced cell death responses in MM.\n",
      "Conjuncts [repressed, enhanced, enhanced]\n",
      "neighbour repressed expression\n",
      "neighbour repressed expression\n",
      "neighbour enhanced dexamethasone-induced\n",
      "neighbour enhanced dexamethasone-induced\n",
      "Conjunction before fc enhanced\n",
      "fc add children for enhanced\n",
      "Conjuncts [enhanced, enhanced, repressed]\n",
      "neighbour enhanced expression\n",
      "neighbour enhanced expression\n",
      "neighbour enhanced dexamethasone-induced\n",
      "neighbour enhanced dexamethasone-induced\n",
      "Conjunction before fc repressed\n",
      "fc add children for repressed\n",
      "Conjuncts [enhanced, repressed, enhanced]\n",
      "neighbour enhanced expression\n",
      "neighbour enhanced expression\n",
      "neighbour repressed expression\n",
      "neighbour repressed expression\n",
      "Conjunction before fc enhanced\n",
      "fc add children for enhanced\n",
      "Recognized overlapping entities: {Interval(119, 124, ('GENE_OR_GENE_PRODUCT', 'SIRT1'))}\n",
      "[Interval(119, 124, ('GENE_OR_GENE_PRODUCT', 'SIRT1', SIRT1))] [Interval(17, 25, ('GENE_OR_GENE_PRODUCT', 'miR-125b', miR-125b))]\n",
      "miR-125b SIRT1\n",
      "deepcheck conj verb\n",
      "creating subtree at enhanced conj\n",
      "deepcheck conj verb\n",
      "creating subtree at enhanced conj\n",
      "passive? False\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "TextBetween: of antisense miR-125b transcripts enhanced expression of pro-apoptotic p53 , repressed expression of anti-apoptotic SIRT1\n",
      "counts between 1 Counter({'NEU': 3, 'POS': 1, 'NEG': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Use of antisense miR-125b transcripts enhanced expression of pro-apoptotic p53, repressed expression of anti-apoptotic SIRT1 and, importantly, significantly enhanced dexamethasone-induced cell death responses in MM.\n",
      "full sentence Use of antisense <e1>miR-125b</e1> transcripts enhanced expression of pro-apoptotic p53, repressed expression of anti-apoptotic <e2>SIRT1</e2> and, importantly, significantly enhanced dexamethasone-induced cell death responses in MM.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': False,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'DOWN',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'counts between equal',\n",
       "   'reg_detect_major': 'counts'}},\n",
       " 'tagged_sent': 'Use of antisense <e1>miR-125b</e1> transcripts enhanced expression of pro-apoptotic p53, repressed expression of anti-apoptotic <e2>SIRT1</e2> and, importantly, significantly enhanced dexamethasone-induced cell death responses in MM.',\n",
       " 'full_sentence': 'Use of antisense <e1>miR-125b</e1> transcripts enhanced expression of pro-apoptotic p53, repressed expression of anti-apoptotic <e2>SIRT1</e2> and, importantly, significantly enhanced dexamethasone-induced cell death responses in MM.',\n",
       " 'entity1': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (17, 25),\n",
       "  'entity_text': 'miR-125b'},\n",
       " 'entity2': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (119, 124),\n",
       "  'entity_text': 'SIRT1'}}"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"23759586.2.8\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (17,25)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (119,124)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case miR-29b\n",
      "add special case leukemia inhibitory factor\n",
      "miR-29b 0 factor 20\n",
      "(u\"miR-29b targeted a cadre of proteins involved in fibrosis, including multiple collagens, matrix metalloproteinases, and leukemia inhibitory factor, insulin-like growth factor 1, and pentraxin 3, 3 predicted targets of miR-29b.\", 0, 20)\n",
      "geneword factor mirword miR-29b\n",
      "Conjuncts [metalloproteinases, factor, factor, pentraxin, targets, collagens]\n",
      "neighbour metalloproteinases ,\n",
      "neighbour metalloproteinases and\n",
      "neighbour factor ,\n",
      "neighbour factor insulin-like\n",
      "neighbour factor 1\n",
      "neighbour factor 1\n",
      "neighbour pentraxin 3\n",
      "neighbour pentraxin 3\n",
      "neighbour targets of\n",
      "neighbour targets of\n",
      "Conjunction before fc collagens\n",
      "fc add children for collagens\n",
      "fc add children for collagens\n",
      "Conjuncts [collagens, factor, factor, pentraxin, targets, metalloproteinases]\n",
      "neighbour collagens ,\n",
      "neighbour collagens matrix\n",
      "neighbour factor ,\n",
      "neighbour factor insulin-like\n",
      "neighbour factor 1\n",
      "neighbour factor 1\n",
      "neighbour pentraxin 3\n",
      "neighbour pentraxin 3\n",
      "neighbour targets of\n",
      "neighbour targets of\n",
      "Conjunction before fc metalloproteinases\n",
      "fc add children for metalloproteinases\n",
      "Conjuncts [collagens, metalloproteinases, factor, pentraxin, targets, factor]\n",
      "neighbour collagens ,\n",
      "neighbour collagens matrix\n",
      "neighbour metalloproteinases ,\n",
      "neighbour metalloproteinases and\n",
      "neighbour factor 1\n",
      "neighbour factor 1\n",
      "neighbour pentraxin 3\n",
      "neighbour pentraxin 3\n",
      "neighbour targets of\n",
      "neighbour targets of\n",
      "Conjunction before fc factor\n",
      "fc add children for factor\n",
      "fc add children for factor\n",
      "Conjuncts [collagens, metalloproteinases, factor, pentraxin, targets, factor]\n",
      "neighbour collagens ,\n",
      "neighbour collagens matrix\n",
      "neighbour metalloproteinases ,\n",
      "neighbour metalloproteinases and\n",
      "neighbour factor ,\n",
      "neighbour factor insulin-like\n",
      "neighbour pentraxin 3\n",
      "neighbour pentraxin 3\n",
      "neighbour targets of\n",
      "neighbour targets of\n",
      "Conjunction before fc factor\n",
      "fc add children for factor\n",
      "fc add children for factor\n",
      "fc add children for factor\n",
      "Conjuncts [collagens, metalloproteinases, factor, factor, targets, pentraxin]\n",
      "neighbour collagens ,\n",
      "neighbour collagens matrix\n",
      "neighbour metalloproteinases ,\n",
      "neighbour metalloproteinases and\n",
      "neighbour factor ,\n",
      "neighbour factor insulin-like\n",
      "neighbour factor 1\n",
      "neighbour factor 1\n",
      "neighbour targets of\n",
      "neighbour targets of\n",
      "Conjunction before fc pentraxin\n",
      "fc add children for pentraxin\n",
      "Conjuncts [collagens, metalloproteinases, factor, factor, pentraxin, targets]\n",
      "neighbour collagens ,\n",
      "neighbour collagens matrix\n",
      "neighbour metalloproteinases ,\n",
      "neighbour metalloproteinases and\n",
      "neighbour factor ,\n",
      "neighbour factor insulin-like\n",
      "neighbour factor 1\n",
      "neighbour factor 1\n",
      "neighbour pentraxin 3\n",
      "neighbour pentraxin 3\n",
      "Conjunction before fc targets\n",
      "fc add children for targets\n",
      "fc add children for targets\n",
      "Conjunctions\n",
      "collagens [including, multiple, collagens, matrix, metalloproteinases, leukemia, inhibitory, factor, insulin-like, growth, factor, 1, pentraxin, 3, 3, predicted, targets]\n",
      "SDP sent: miR-29b targeted a cadre of proteins involved in fibrosis, including multiple collagens, matrix metalloproteinases, and leukemia inhibitory factor, insulin-like growth factor 1, and pentraxin 3, 3 predicted targets of miR-29b.\n",
      "SDP     : [(miR-29b, 'PROPN', 'nsubj', False), (targeted, 'VERB', 'ROOT', False), (cadre, 'NOUN', 'dobj', False), (proteins, 'NOUN', 'nmod', False), (involved, 'VERB', 'acl', False), (fibrosis, 'NOUN', 'nmod', False), (collagens, 'NOUN', 'nmod', True), (factor, 'NOUN', 'conj', False)]\n",
      "[matrix, metalloproteinases]\n",
      "[leukemia, inhibitory, factor]\n",
      "[insulin-like, growth, factor, 1]\n",
      "[pentraxin, 3]\n",
      "[3, predicted, targets, of, miR-29b]\n",
      "0 [miR-29b, targeted, a, cadre, of, proteins, involved, in, fibrosis, ,, including, multiple, collagens, ,, matrix, metalloproteinases, ,, and, leukemia, inhibitory, factor, ,, insulin-like, growth, factor, 1, ,, and, pentraxin, 3, ,, 3, predicted, targets, of, miR-29b] 36 0 miR-29b targeted a cadre of proteins involved in fibrosis, including multiple collagens, matrix metalloproteinases, and leukemia inhibitory factor, insulin-like growth factor 1, and pentraxin 3, 3 predicted targets of miR-29b.\n",
      "Conjuncts [metalloproteinases, factor, factor, pentraxin, targets, collagens]\n",
      "neighbour metalloproteinases ,\n",
      "neighbour metalloproteinases and\n",
      "neighbour factor ,\n",
      "neighbour factor insulin-like\n",
      "neighbour factor 1\n",
      "neighbour factor 1\n",
      "neighbour pentraxin 3\n",
      "neighbour pentraxin 3\n",
      "neighbour targets of\n",
      "neighbour targets of\n",
      "Conjunction before fc collagens\n",
      "fc add children for collagens\n",
      "fc add children for collagens\n",
      "Conjuncts [collagens, factor, factor, pentraxin, targets, metalloproteinases]\n",
      "neighbour collagens ,\n",
      "neighbour collagens matrix\n",
      "neighbour factor ,\n",
      "neighbour factor insulin-like\n",
      "neighbour factor 1\n",
      "neighbour factor 1\n",
      "neighbour pentraxin 3\n",
      "neighbour pentraxin 3\n",
      "neighbour targets of\n",
      "neighbour targets of\n",
      "Conjunction before fc metalloproteinases\n",
      "fc add children for metalloproteinases\n",
      "Conjuncts [collagens, metalloproteinases, factor, pentraxin, targets, factor]\n",
      "neighbour collagens ,\n",
      "neighbour collagens matrix\n",
      "neighbour metalloproteinases ,\n",
      "neighbour metalloproteinases and\n",
      "neighbour factor 1\n",
      "neighbour factor 1\n",
      "neighbour pentraxin 3\n",
      "neighbour pentraxin 3\n",
      "neighbour targets of\n",
      "neighbour targets of\n",
      "Conjunction before fc factor\n",
      "fc add children for factor\n",
      "fc add children for factor\n",
      "Conjuncts [collagens, metalloproteinases, factor, pentraxin, targets, factor]\n",
      "neighbour collagens ,\n",
      "neighbour collagens matrix\n",
      "neighbour metalloproteinases ,\n",
      "neighbour metalloproteinases and\n",
      "neighbour factor ,\n",
      "neighbour factor insulin-like\n",
      "neighbour pentraxin 3\n",
      "neighbour pentraxin 3\n",
      "neighbour targets of\n",
      "neighbour targets of\n",
      "Conjunction before fc factor\n",
      "fc add children for factor\n",
      "fc add children for factor\n",
      "fc add children for factor\n",
      "Conjuncts [collagens, metalloproteinases, factor, factor, targets, pentraxin]\n",
      "neighbour collagens ,\n",
      "neighbour collagens matrix\n",
      "neighbour metalloproteinases ,\n",
      "neighbour metalloproteinases and\n",
      "neighbour factor ,\n",
      "neighbour factor insulin-like\n",
      "neighbour factor 1\n",
      "neighbour factor 1\n",
      "neighbour targets of\n",
      "neighbour targets of\n",
      "Conjunction before fc pentraxin\n",
      "fc add children for pentraxin\n",
      "Conjuncts [collagens, metalloproteinases, factor, factor, pentraxin, targets]\n",
      "neighbour collagens ,\n",
      "neighbour collagens matrix\n",
      "neighbour metalloproteinases ,\n",
      "neighbour metalloproteinases and\n",
      "neighbour factor ,\n",
      "neighbour factor insulin-like\n",
      "neighbour factor 1\n",
      "neighbour factor 1\n",
      "neighbour pentraxin 3\n",
      "neighbour pentraxin 3\n",
      "Conjunction before fc targets\n",
      "fc add children for targets\n",
      "fc add children for targets\n",
      "Recognized overlapping entities: {Interval(120, 176, ('GENE_OR_GENE_PRODUCT', 'leukemia inhibitory factor, insulin-like growth factor 1'))}\n",
      "[Interval(140, 146, ('GENE_OR_GENE_PRODUCT', 'factor', factor))] [Interval(0, 7, ('GENE_OR_GENE_PRODUCT', 'miR-29b', miR-29b))]\n",
      "miR-29b factor\n",
      "[matrix, metalloproteinases]\n",
      "[leukemia, inhibitory, factor]\n",
      "[insulin-like, growth, factor, 1]\n",
      "[pentraxin, 3]\n",
      "[3, predicted, targets, of, miR-29b]\n",
      "[matrix, metalloproteinases]\n",
      "[leukemia, inhibitory, factor]\n",
      "[insulin-like, growth, factor, 1]\n",
      "[pentraxin, 3]\n",
      "[3, predicted, targets, of, miR-29b]\n",
      "passive? False\n",
      "Counter()\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence miR-29b targeted a cadre of proteins involved in fibrosis, including multiple collagens, matrix metalloproteinases, and leukemia inhibitory factor, insulin-like growth factor 1, and pentraxin 3, 3 predicted targets of miR-29b.\n",
      "full sentence <e1>miR-29b</e1> targeted a cadre of proteins involved in fibrosis, including multiple collagens, matrix metalloproteinases, and <e2>leukemia inhibitory factor</e2>, insulin-like growth factor 1, and pentraxin 3, 3 predicted targets of miR-29b.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': False,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'compartment mir gene',\n",
       "   'reg_detect_major': 'compartment'}},\n",
       " 'tagged_sent': '<e1>miR-29b</e1> targeted a cadre of proteins involved in fibrosis, including multiple collagens, matrix metalloproteinases, and <e2>leukemia inhibitory factor</e2>, insulin-like growth factor 1, and pentraxin 3, 3 predicted targets of miR-29b.',\n",
       " 'full_sentence': '<e1>miR-29b</e1> targeted a cadre of proteins involved in fibrosis, including multiple collagens, matrix metalloproteinases, and <e2>leukemia inhibitory factor</e2>, insulin-like growth factor 1, and pentraxin 3, 3 predicted targets of miR-29b.',\n",
       " 'entity1': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (0, 7),\n",
       "  'entity_text': 'miR-29b'},\n",
       " 'entity2': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (120, 146),\n",
       "  'entity_text': 'leukemia inhibitory factor'}}"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"24006456.2.4\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (0,7)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (120,146)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 10 0 Hsa-miR-24 NOUN compound 5p 13 {('GENE_OR_GENE_PRODUCT', 'Hsa-miR-24-2-5p')}\n",
      "10 11 1 - PUNCT punct Hsa-miR-24 0 {('GENE_OR_GENE_PRODUCT', 'Hsa-miR-24-2-5p')}\n",
      "11 12 2 2 NUM nummod 5p 13 {('GENE_OR_GENE_PRODUCT', 'Hsa-miR-24-2-5p')}\n",
      "12 13 3 - PUNCT punct 5p 13 {('GENE_OR_GENE_PRODUCT', 'Hsa-miR-24-2-5p')}\n",
      "13 15 4 5p NOUN nsubj likely 20 {('GENE_OR_GENE_PRODUCT', 'Hsa-miR-24-2-5p')}\n",
      "16 19 5 was AUX cop likely 20 set()\n",
      "20 26 6 likely ADJ ROOT likely 20 set()\n",
      "27 29 7 to PART mark responsible 33 set()\n",
      "30 32 8 be AUX cop responsible 33 set()\n",
      "33 44 9 responsible ADJ xcomp likely 20 set()\n",
      "45 48 10 for ADP case MYC 49 set()\n",
      "49 52 11 MYC PROPN nmod responsible 33 {('GENE_OR_GENE_PRODUCT', 'MYC')}\n",
      "53 56 12 and CCONJ cc MYC 49 set()\n",
      "57 62 13 KCNJ2 NOUN compound genes 63 {('GENE_OR_GENE_PRODUCT', 'KCNJ2')}\n",
      "63 68 14 genes NOUN conj MYC 49 set()\n",
      "69 72 15 and CCONJ cc MYC 49 set()\n",
      "73 87 16 hsa-miR-27a-5p NOUN conj MYC 49 {('GENE_OR_GENE_PRODUCT', 'hsa-miR-27a-5p')}\n",
      "88 91 17 for ADP case ST3GAL6 92 set()\n",
      "92 99 18 ST3GAL6 PROPN nmod MYC 49 {('GENE_OR_GENE_PRODUCT', 'ST3GAL6')}\n",
      "99 100 19 . PUNCT punct likely 20 set()\n",
      "- 2\n",
      "2 -\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"a13c8785a0d04f618c9c2ec047709230-0\" class=\"displacy\" width=\"1410\" height=\"377.0\" direction=\"ltr\" style=\"max-width: none; height: 377.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">Hsa-miR-24-</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"130\">2-</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"130\">NUM</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"210\">5p</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"210\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"290\">was</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"290\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"370\">likely</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"370\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"450\">to</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"450\">PART</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"530\">be</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"530\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"610\">responsible</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"610\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"690\">for</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"690\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"770\">MYC</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"770\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"850\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"850\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"930\">KCNJ2</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"930\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1010\">genes</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1010\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1090\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1090\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1170\">hsa-miR-27a-5p</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1170\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1250\">for</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1250\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"287.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1330\">ST3GAL6.</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1330\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-0\" stroke-width=\"2px\" d=\"M62,242.0 62,215.33333333333334 198.0,215.33333333333334 198.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M62,244.0 L58,236.0 66,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-1\" stroke-width=\"2px\" d=\"M142,242.0 142,228.66666666666666 195.0,228.66666666666666 195.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nummod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M142,244.0 L138,236.0 146,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-2\" stroke-width=\"2px\" d=\"M222,242.0 222,215.33333333333334 358.0,215.33333333333334 358.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M222,244.0 L218,236.0 226,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-3\" stroke-width=\"2px\" d=\"M302,242.0 302,228.66666666666666 355.0,228.66666666666666 355.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cop</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M302,244.0 L298,236.0 306,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-4\" stroke-width=\"2px\" d=\"M462,242.0 462,215.33333333333334 598.0,215.33333333333334 598.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">mark</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M462,244.0 L458,236.0 466,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-5\" stroke-width=\"2px\" d=\"M542,242.0 542,228.66666666666666 595.0,228.66666666666666 595.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cop</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M542,244.0 L538,236.0 546,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-6\" stroke-width=\"2px\" d=\"M382,242.0 382,202.0 601.0,202.0 601.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">xcomp</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M601.0,244.0 L605.0,236.0 597.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-7\" stroke-width=\"2px\" d=\"M702,242.0 702,228.66666666666666 755.0,228.66666666666666 755.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M702,244.0 L698,236.0 706,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-8\" stroke-width=\"2px\" d=\"M622,242.0 622,215.33333333333334 758.0,215.33333333333334 758.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M758.0,244.0 L762.0,236.0 754.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-9\" stroke-width=\"2px\" d=\"M782,242.0 782,228.66666666666666 835.0,228.66666666666666 835.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M835.0,244.0 L839.0,236.0 831.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-10\" stroke-width=\"2px\" d=\"M942,242.0 942,228.66666666666666 995.0,228.66666666666666 995.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M942,244.0 L938,236.0 946,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-11\" stroke-width=\"2px\" d=\"M782,242.0 782,202.0 1001.0,202.0 1001.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1001.0,244.0 L1005.0,236.0 997.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-12\" stroke-width=\"2px\" d=\"M782,242.0 782,188.66666666666666 1084.0,188.66666666666666 1084.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1084.0,244.0 L1088.0,236.0 1080.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-13\" stroke-width=\"2px\" d=\"M782,242.0 782,175.33333333333331 1167.0,175.33333333333331 1167.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-13\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1167.0,244.0 L1171.0,236.0 1163.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-14\" stroke-width=\"2px\" d=\"M1262,242.0 1262,228.66666666666666 1315.0,228.66666666666666 1315.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-14\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1262,244.0 L1258,236.0 1266,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-a13c8785a0d04f618c9c2ec047709230-0-15\" stroke-width=\"2px\" d=\"M782,242.0 782,162.0 1330.0,162.0 1330.0,242.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-a13c8785a0d04f618c9c2ec047709230-0-15\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1330.0,244.0 L1334.0,236.0 1326.0,236.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em; box-decoration-break: clone; -webkit-box-decoration-break: clone\">\n",
       "    Hsa-miR-24-2-5p\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; text-transform: uppercase; vertical-align: middle; margin-left: 0.5rem\">GENE_OR_GENE_PRODUCT</span>\n",
       "</mark>\n",
       " was likely to be responsible for \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em; box-decoration-break: clone; -webkit-box-decoration-break: clone\">\n",
       "    MYC\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; text-transform: uppercase; vertical-align: middle; margin-left: 0.5rem\">GENE_OR_GENE_PRODUCT</span>\n",
       "</mark>\n",
       " and \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em; box-decoration-break: clone; -webkit-box-decoration-break: clone\">\n",
       "    KCNJ2\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; text-transform: uppercase; vertical-align: middle; margin-left: 0.5rem\">GENE_OR_GENE_PRODUCT</span>\n",
       "</mark>\n",
       " genes and \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em; box-decoration-break: clone; -webkit-box-decoration-break: clone\">\n",
       "    hsa-miR-27a-5p\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; text-transform: uppercase; vertical-align: middle; margin-left: 0.5rem\">GENE_OR_GENE_PRODUCT</span>\n",
       "</mark>\n",
       " for \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em; box-decoration-break: clone; -webkit-box-decoration-break: clone\">\n",
       "    ST3GAL6\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; text-transform: uppercase; vertical-align: middle; margin-left: 0.5rem\">GENE_OR_GENE_PRODUCT</span>\n",
       "</mark>\n",
       ".</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "IntervalTree([Interval(0, 15, ('GENE_OR_GENE_PRODUCT', 'Hsa-miR-24-2-5p')), Interval(49, 52, ('GENE_OR_GENE_PRODUCT', 'MYC')), Interval(57, 62, ('GENE_OR_GENE_PRODUCT', 'KCNJ2')), Interval(73, 87, ('GENE_OR_GENE_PRODUCT', 'hsa-miR-27a-5p')), Interval(92, 99, ('GENE_OR_GENE_PRODUCT', 'ST3GAL6'))])"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "analyse( (allSents[\"24219008.2.7\"], 1,2) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case Hsa-miR-24-2\n",
      "add special case ST3GAL6\n",
      "2 2 ST3GAL6 18\n",
      "(u\"Hsa-miR-24-2-5p was likely to be responsible for MYC and KCNJ2 genes and hsa-miR-27a-5p for ST3GAL6.\", 2, 18)\n",
      "geneword ST3GAL6 mirword 2\n",
      "Conjuncts [genes, hsa-miR-27a-5p, MYC]\n",
      "neighbour genes and\n",
      "neighbour genes and\n",
      "neighbour hsa-miR-27a-5p for\n",
      "neighbour hsa-miR-27a-5p for\n",
      "Conjunction before fc MYC\n",
      "fc add children for MYC\n",
      "Conjuncts [MYC, hsa-miR-27a-5p, genes]\n",
      "neighbour MYC and\n",
      "neighbour MYC and\n",
      "neighbour hsa-miR-27a-5p for\n",
      "neighbour hsa-miR-27a-5p for\n",
      "Conjunction before fc genes\n",
      "fc add children for genes\n",
      "Conjuncts [MYC, genes, hsa-miR-27a-5p]\n",
      "neighbour MYC and\n",
      "neighbour MYC and\n",
      "neighbour genes and\n",
      "neighbour genes and\n",
      "Conjunction before fc hsa-miR-27a-5p\n",
      "Conjunctions\n",
      "MYC [for, MYC, KCNJ2, genes, hsa-miR-27a-5p]\n",
      "SDP sent: Hsa-miR-24-2-5p was likely to be responsible for MYC and KCNJ2 genes and hsa-miR-27a-5p for ST3GAL6.\n",
      "SDP     : [(2, 'NUM', 'nummod', False), (5p, 'NOUN', 'nsubj', False), (likely, 'ADJ', 'ROOT', False), (responsible, 'ADJ', 'xcomp', False), (MYC, 'PROPN', 'nmod', False), (ST3GAL6, 'PROPN', 'nmod', False)]\n",
      "[KCNJ2, genes]\n",
      "[hsa-miR-27a-5p]\n",
      "0 [Hsa-miR-24, -, 2, -, 5p, was, likely, to, be, responsible, for, MYC, and, KCNJ2, genes, and, hsa-miR-27a-5p, for, ST3GAL6] 19 0 Hsa-miR-24-2-5p was likely to be responsible for MYC and KCNJ2 genes and hsa-miR-27a-5p for ST3GAL6.\n",
      "Conjuncts [genes, hsa-miR-27a-5p, MYC]\n",
      "neighbour genes and\n",
      "neighbour genes and\n",
      "neighbour hsa-miR-27a-5p for\n",
      "neighbour hsa-miR-27a-5p for\n",
      "Conjunction before fc MYC\n",
      "fc add children for MYC\n",
      "Conjuncts [MYC, hsa-miR-27a-5p, genes]\n",
      "neighbour MYC and\n",
      "neighbour MYC and\n",
      "neighbour hsa-miR-27a-5p for\n",
      "neighbour hsa-miR-27a-5p for\n",
      "Conjunction before fc genes\n",
      "fc add children for genes\n",
      "Conjuncts [MYC, genes, hsa-miR-27a-5p]\n",
      "neighbour MYC and\n",
      "neighbour MYC and\n",
      "neighbour genes and\n",
      "neighbour genes and\n",
      "Conjunction before fc hsa-miR-27a-5p\n",
      "Recognized overlapping entities: {Interval(92, 99, ('GENE_OR_GENE_PRODUCT', 'ST3GAL6'))}\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: Hsa-miR-24 - 2 - 5p was likely to be responsible for MYC and KCNJ2 genes and hsa-miR-27a-5p for ST3GAL6\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence Hsa-miR-24-2-5p was likely to be responsible for MYC and KCNJ2 genes and hsa-miR-27a-5p for ST3GAL6.\n",
      "full sentence <e1>Hsa-miR-24-2</e1>-5p was likely to be responsible for MYC and KCNJ2 genes and hsa-miR-27a-5p for <e2>ST3GAL6</e2>.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': False,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'return',\n",
       "   'reg_detect_major': 'return'}},\n",
       " 'tagged_sent': '<e1>Hsa-miR-24-2</e1>-5p was likely to be responsible for MYC and KCNJ2 genes and hsa-miR-27a-5p for <e2>ST3GAL6</e2>.',\n",
       " 'full_sentence': '<e1>Hsa-miR-24-2</e1>-5p was likely to be responsible for MYC and KCNJ2 genes and hsa-miR-27a-5p for <e2>ST3GAL6</e2>.',\n",
       " 'entity1': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (0, 12),\n",
       "  'entity_text': 'Hsa-miR-24-2'},\n",
       " 'entity2': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (92, 99),\n",
       "  'entity_text': 'ST3GAL6'}}"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"24219008.2.7\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (0,12)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (92,99)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case miR-33\n",
      "add special case SREBP-2\n",
      "miR-33 16 SREBP-2 28\n",
      "(u\"We show that two of these transporters, ABCB11 and ATP8B1, are functional targets of miR-33, a micro-RNA that is expressed from within an intron of SREBP-2.\", 16, 28)\n",
      "geneword SREBP-2 mirword miR-33\n",
      "Conjuncts [ABCB11, ATP8B1, two]\n",
      "neighbour ABCB11 and\n",
      "neighbour ABCB11 and\n",
      "neighbour ATP8B1 ,\n",
      "neighbour ATP8B1 are\n",
      "Conjunction before fc two\n",
      "Conjuncts [two, ATP8B1, ABCB11]\n",
      "neighbour two of\n",
      "neighbour two of\n",
      "neighbour ATP8B1 ,\n",
      "neighbour ATP8B1 are\n",
      "Conjunction before fc ABCB11\n",
      "Conjuncts [two, ABCB11, ATP8B1]\n",
      "neighbour two of\n",
      "neighbour two of\n",
      "neighbour ABCB11 and\n",
      "neighbour ABCB11 and\n",
      "Conjunction before fc ATP8B1\n",
      "two micro-RNA\n",
      "ATP8B1, are functional targets of miR-33,\n",
      "Conjunctions\n",
      "two [two, ABCB11, ATP8B1]\n",
      "micro-RNA [a, SREBP-2]\n",
      "SDP sent: We show that two of these transporters, ABCB11 and ATP8B1, are functional targets of miR-33, a micro-RNA that is expressed from within an intron of SREBP-2.\n",
      "SDP     : [(miR-33, 'PROPN', 'nmod', False), (show, 'VERB', 'ROOT', False), (targets, 'NOUN', 'ccomp', False), (miR-33, 'PROPN', 'nmod', False)]\n",
      "[ABCB11, and, ATP8B1]\n",
      "[ATP8B1]\n",
      "0 [We, show, that, two, of, these, transporters, ,, ABCB11, and, ATP8B1, ,, are, functional, targets, of, miR-33, ,, a, micro-RNA, that, is, expressed, from, within, an, intron, of, SREBP-2] 29 0 We show that two of these transporters, ABCB11 and ATP8B1, are functional targets of miR-33, a micro-RNA that is expressed from within an intron of SREBP-2.\n",
      "Context Fail pre <regex.Match object; span=(5, 15), match=' intron of', fuzzy_counts=(0, 1, 0)> regex.Regex('(intron of){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(148, 155, ('GENE_OR_GENE_PRODUCT', 'SREBP-2'))}\n",
      "[Interval(148, 155, ('GENE_OR_GENE_PRODUCT', 'SREBP-2', SREBP-2))] [Interval(85, 91, ('GENE_OR_GENE_PRODUCT', 'miR-33', miR-33)), Interval(91, 92, ('', ',', ,))]\n",
      "miR-33 SREBP-2\n",
      "[ABCB11, and, ATP8B1]\n",
      "[ATP8B1]\n",
      "[ABCB11, and, ATP8B1]\n",
      "[ATP8B1]\n",
      "passive? False\n",
      "Counter()\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence We show that two of these transporters, ABCB11 and ATP8B1, are functional targets of miR-33, a micro-RNA that is expressed from within an intron of SREBP-2.\n",
      "full sentence We show that two of these transporters, ABCB11 and ATP8B1, are functional targets of <e1>miR-33</e1>, a micro-RNA that is expressed from within an intron of <e2>SREBP-2</e2>.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': False,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': False,\n",
       "  'entity': True,\n",
       "  'passive': False,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'compartment mir gene',\n",
       "   'reg_detect_major': 'compartment'}},\n",
       " 'tagged_sent': 'We show that two of these transporters, ABCB11 and ATP8B1, are functional targets of <e1>miR-33</e1>, a micro-RNA that is expressed from within an intron of <e2>SREBP-2</e2>.',\n",
       " 'full_sentence': 'We show that two of these transporters, ABCB11 and ATP8B1, are functional targets of <e1>miR-33</e1>, a micro-RNA that is expressed from within an intron of <e2>SREBP-2</e2>.',\n",
       " 'entity1': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (85, 91),\n",
       "  'entity_text': 'miR-33'},\n",
       " 'entity2': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (148, 155),\n",
       "  'entity_text': 'SREBP-2'}}"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"22767443.2.3\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (85,91)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (148,155)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case miR-25\n",
      "add special case MYC\n",
      "miR-25 7 MYC 27\n",
      "(u\"In this study, two miRNAs, miR-25 and -32, are identified as p53-repressed miRNAs by p53-dependent negative regulation of their transcriptional regulators, E2F1 and MYC.\", 7, 27)\n",
      "geneword MYC mirword miR-25\n",
      "Conjuncts [-32, miR-25]\n",
      "neighbour -32 ,\n",
      "neighbour -32 are\n",
      "Conjunction before fc miR-25\n",
      "Conjuncts [miR-25, -32]\n",
      "neighbour miR-25 and\n",
      "neighbour miR-25 and\n",
      "Conjunction before fc -32\n",
      "Conjuncts [MYC, E2F1]\n",
      "neighbour MYC .\n",
      "neighbour MYC None\n",
      "Conjunction before fc E2F1\n",
      "Conjuncts [E2F1, MYC]\n",
      "neighbour E2F1 and\n",
      "neighbour E2F1 and\n",
      "Conjunction before fc MYC\n",
      "miR-25 E2F1\n",
      "-32, are identified as p53-repressed miRNAs by p53-dependent negative regulation of their transcriptional regulators,\n",
      "Conjunctions\n",
      "miR-25 [miR-25, -32]\n",
      "E2F1 [E2F1, MYC]\n",
      "SDP sent: In this study, two miRNAs, miR-25 and -32, are identified as p53-repressed miRNAs by p53-dependent negative regulation of their transcriptional regulators, E2F1 and MYC.\n",
      "SDP     : [(miR-25, 'PROPN', 'appos', False), (miRNAs, 'NOUN', 'nsubjpass', False), (identified, 'VERB', 'ROOT', False), (regulation, 'NOUN', 'nmod', False), (regulators, 'NOUN', 'nmod', False), (E2F1, 'PROPN', 'appos', True), (MYC, 'PROPN', 'conj', False)]\n",
      "[-32]\n",
      "[MYC]\n",
      "0 [In, this, study, ,, two, miRNAs, ,, miR-25, and, -32, ,, are, identified, as, p53-repressed, miRNAs, by, p53-dependent, negative, regulation, of, their, transcriptional, regulators, ,, E2F1, and, MYC] 28 0 In this study, two miRNAs, miR-25 and -32, are identified as p53-repressed miRNAs by p53-dependent negative regulation of their transcriptional regulators, E2F1 and MYC.\n",
      "Conjuncts [-32, miR-25]\n",
      "neighbour -32 ,\n",
      "neighbour -32 are\n",
      "Conjunction before fc miR-25\n",
      "Conjuncts [miR-25, -32]\n",
      "neighbour miR-25 and\n",
      "neighbour miR-25 and\n",
      "Conjunction before fc -32\n",
      "Conjuncts [MYC, E2F1]\n",
      "neighbour MYC .\n",
      "neighbour MYC None\n",
      "Conjunction before fc E2F1\n",
      "Conjuncts [E2F1, MYC]\n",
      "neighbour E2F1 and\n",
      "neighbour E2F1 and\n",
      "Conjunction before fc MYC\n",
      "Recognized overlapping entities: {Interval(165, 168, ('GENE_OR_GENE_PRODUCT', 'MYC'))}\n",
      "[Interval(165, 168, ('GENE_OR_GENE_PRODUCT', 'MYC', MYC))] [Interval(27, 33, ('GENE_OR_GENE_PRODUCT', 'miR-25', miR-25))]\n",
      "miR-25 MYC\n",
      "[-32]\n",
      "[MYC]\n",
      "[-32]\n",
      "[MYC]\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: miRNAs , miR-25 and -32 , are identified as p53-repressed miRNAs by p53-dependent negative regulation of their transcriptional regulators , E2F1 and MYC\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene reg corr', 'reg_detect_major': 'between'}})\n",
      " hit sentence In this study, two miRNAs, miR-25 and -32, are identified as p53-repressed miRNAs by p53-dependent negative regulation of their transcriptional regulators, E2F1 and MYC.\n",
      "full sentence In this study, two miRNAs, <e1>miR-25</e1> and -32, are identified as p53-repressed miRNAs by p53-dependent negative regulation of their transcriptional regulators, E2F1 and <e2>MYC</e2>.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': True,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'UP',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'between mir gene reg corr',\n",
       "   'reg_detect_major': 'between'}},\n",
       " 'tagged_sent': 'In this study, two miRNAs, <e1>miR-25</e1> and -32, are identified as p53-repressed miRNAs by p53-dependent negative regulation of their transcriptional regulators, E2F1 and <e2>MYC</e2>.',\n",
       " 'full_sentence': 'In this study, two miRNAs, <e1>miR-25</e1> and -32, are identified as p53-repressed miRNAs by p53-dependent negative regulation of their transcriptional regulators, E2F1 and <e2>MYC</e2>.',\n",
       " 'entity1': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (27, 33),\n",
       "  'entity_text': 'miR-25'},\n",
       " 'entity2': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (165, 168),\n",
       "  'entity_text': 'MYC'}}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"22431589.2.2\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (27,33)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (165,168)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case microRNA-33b\n",
      "add special case SREBF2\n",
      "microRNA-33b 2 SREBF2 19\n",
      "(u\"MicroRNA-33a and microRNA-33b (miR-33a/b) are intronic miRNAs whose encoding regions are embedded in the sterol-response-element-binding protein genes SREBF2 and SREBF1 (refs 3-5), respectively.\", 2, 19)\n",
      "geneword SREBF2 mirword microRNA-33b\n",
      "Conjuncts [microRNA-33b, MicroRNA-33a]\n",
      "neighbour microRNA-33b (\n",
      "neighbour microRNA-33b miR-33a/b\n",
      "Conjunction before fc MicroRNA-33a\n",
      "Conjuncts [MicroRNA-33a, microRNA-33b]\n",
      "neighbour MicroRNA-33a and\n",
      "neighbour MicroRNA-33a and\n",
      "Conjunction before fc microRNA-33b\n",
      "fc add children for microRNA-33b\n",
      "Conjuncts [SREBF1, SREBF2]\n",
      "neighbour SREBF1 (\n",
      "neighbour SREBF1 refs\n",
      "Conjunction before fc SREBF2\n",
      "Conjuncts [SREBF2, SREBF1]\n",
      "neighbour SREBF2 and\n",
      "neighbour SREBF2 and\n",
      "Conjunction before fc SREBF1\n",
      "MicroRNA-33a miR-33a/b\n",
      "\n",
      "miR-33a/b SREBF2\n",
      ") are intronic miRNAs whose encoding regions are embedded in the sterol-response-element-binding protein genes\n",
      "Conjunctions\n",
      "MicroRNA-33a [MicroRNA-33a, microRNA-33b, miR-33a/b]\n",
      "miR-33a/b [(, )]\n",
      "SREBF2 [SREBF2, SREBF1]\n",
      "SDP sent: MicroRNA-33a and microRNA-33b (miR-33a/b) are intronic miRNAs whose encoding regions are embedded in the sterol-response-element-binding protein genes SREBF2 and SREBF1 (refs 3-5), respectively.\n",
      "SDP     : [(microRNA-33b, 'PROPN', 'conj', False), (MicroRNA-33a, 'NOUN', 'nsubj', True), (miRNAs, 'NOUN', 'ROOT', False), (embedded, 'ADJ', 'acl:relcl', False), (genes, 'NOUN', 'nmod', False), (SREBF2, 'PROPN', 'dep', False)]\n",
      "SDP CONTAIN/EMBED/LOCATE RULE\n",
      "[microRNA-33b, (, miR-33a/b, )]\n",
      "[SREBF1]\n",
      "0 [MicroRNA-33a, and, microRNA-33b, (, miR-33a/b, ), are, intronic, miRNAs, whose, encoding, regions, are, embedded, in, the, sterol-response-element-binding, protein, genes, SREBF2, and, SREBF1, (, refs, 3, -, 5, ), ,, respectively] 30 0 MicroRNA-33a and microRNA-33b (miR-33a/b) are intronic miRNAs whose encoding regions are embedded in the sterol-response-element-binding protein genes SREBF2 and SREBF1 (refs 3-5), respectively.\n",
      "Conjuncts [microRNA-33b, MicroRNA-33a]\n",
      "neighbour microRNA-33b (\n",
      "neighbour microRNA-33b miR-33a/b\n",
      "Conjunction before fc MicroRNA-33a\n",
      "Conjuncts [MicroRNA-33a, microRNA-33b]\n",
      "neighbour MicroRNA-33a and\n",
      "neighbour MicroRNA-33a and\n",
      "Conjunction before fc microRNA-33b\n",
      "fc add children for microRNA-33b\n",
      "Conjuncts [SREBF1, SREBF2]\n",
      "neighbour SREBF1 (\n",
      "neighbour SREBF1 refs\n",
      "Conjunction before fc SREBF2\n",
      "Conjuncts [SREBF2, SREBF1]\n",
      "neighbour SREBF2 and\n",
      "neighbour SREBF2 and\n",
      "Conjunction before fc SREBF1\n",
      "Recognized overlapping entities: {Interval(151, 157, ('GENE_OR_GENE_PRODUCT', 'SREBF2'))}\n",
      "[Interval(151, 157, ('GENE_OR_GENE_PRODUCT', 'SREBF2', SREBF2))] [Interval(17, 29, ('GENE_OR_GENE_PRODUCT', 'microRNA-33b', microRNA-33b))]\n",
      "microRNA-33b SREBF2\n",
      "[microRNA-33b, (, miR-33a/b, )]\n",
      "[SREBF1]\n",
      "[microRNA-33b, (, miR-33a/b, )]\n",
      "[SREBF1]\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: MicroRNA-33a and microRNA-33b ( miR-33a/b ) are intronic miRNAs whose encoding regions are embedded in the sterol-response-element-binding protein genes SREBF2 and SREBF1\n",
      "(False, {'conj': True, 'sdp': False, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence MicroRNA-33a and microRNA-33b (miR-33a/b) are intronic miRNAs whose encoding regions are embedded in the sterol-response-element-binding protein genes SREBF2 and SREBF1 (refs 3-5), respectively.\n",
      "full sentence MicroRNA-33a and <e1>microRNA-33b</e1> (miR-33a/b) are intronic miRNAs whose encoding regions are embedded in the sterol-response-element-binding protein genes <e2>SREBF2</e2> and SREBF1 (refs 3-5), respectively.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': False,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': False,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': False,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'return',\n",
       "   'reg_detect_major': 'return'}},\n",
       " 'tagged_sent': 'MicroRNA-33a and <e1>microRNA-33b</e1> (miR-33a/b) are intronic miRNAs whose encoding regions are embedded in the sterol-response-element-binding protein genes <e2>SREBF2</e2> and SREBF1 (refs 3-5), respectively.',\n",
       " 'full_sentence': 'MicroRNA-33a and <e1>microRNA-33b</e1> (miR-33a/b) are intronic miRNAs whose encoding regions are embedded in the sterol-response-element-binding protein genes <e2>SREBF2</e2> and SREBF1 (refs 3-5), respectively.',\n",
       " 'entity1': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (17, 29),\n",
       "  'entity_text': 'microRNA-33b'},\n",
       " 'entity2': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (151, 157),\n",
       "  'entity_text': 'SREBF2'}}"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"22012398.2.4\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (17,29)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (151,157)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case miR-106b\n",
      "add special case CDKN1A/p21\n",
      "miR-106b 0 CDKN1A/p21 16\n",
      "(u\"miR-106b and miR-25 were previously shown to modulate the TGF-beta signaling pathway through their action on CDKN1A/p21 and BCL2L11/Bim.\", 0, 16)\n",
      "geneword CDKN1A/p21 mirword miR-106b\n",
      "Conjuncts [miR-25, miR-106b]\n",
      "neighbour miR-25 were\n",
      "neighbour miR-25 were\n",
      "Conjunction before fc miR-106b\n",
      "Conjuncts [miR-106b, miR-25]\n",
      "neighbour miR-106b and\n",
      "neighbour miR-106b and\n",
      "Conjunction before fc miR-25\n",
      "Conjuncts [BCL2L11/Bim, CDKN1A/p21]\n",
      "neighbour BCL2L11/Bim .\n",
      "neighbour BCL2L11/Bim None\n",
      "Conjunction before fc CDKN1A/p21\n",
      "fc add children for CDKN1A/p21\n",
      "Conjuncts [CDKN1A/p21, BCL2L11/Bim]\n",
      "neighbour CDKN1A/p21 and\n",
      "neighbour CDKN1A/p21 and\n",
      "Conjunction before fc BCL2L11/Bim\n",
      "miR-106b CDKN1A/p21\n",
      "miR-25 were previously shown to modulate the TGF-beta signaling pathway through their action\n",
      "Conjunctions\n",
      "miR-106b [miR-106b, miR-25]\n",
      "CDKN1A/p21 [on, CDKN1A/p21, BCL2L11/Bim]\n",
      "SDP sent: miR-106b and miR-25 were previously shown to modulate the TGF-beta signaling pathway through their action on CDKN1A/p21 and BCL2L11/Bim.\n",
      "SDP     : [(miR-106b, 'PROPN', 'nsubjpass', False), (shown, 'VERB', 'ROOT', False), (modulate, 'VERB', 'xcomp', False), (action, 'NOUN', 'nmod', False), (CDKN1A/p21, 'PROPN', 'nmod', False)]\n",
      "[miR-25]\n",
      "deepcheck connection rule [to, modulate, the, TGF-beta, signaling, pathway, through, their, action, on, CDKN1A/p21, and, BCL2L11/Bim]\n",
      "[BCL2L11/Bim]\n",
      "0 [miR-106b, and, miR-25, were, previously, shown, to, modulate, the, TGF-beta, signaling, pathway, through, their, action, on, CDKN1A/p21, and, BCL2L11/Bim] 19 0 miR-106b and miR-25 were previously shown to modulate the TGF-beta signaling pathway through their action on CDKN1A/p21 and BCL2L11/Bim.\n",
      "Conjuncts [miR-25, miR-106b]\n",
      "neighbour miR-25 were\n",
      "neighbour miR-25 were\n",
      "Conjunction before fc miR-106b\n",
      "Conjuncts [miR-106b, miR-25]\n",
      "neighbour miR-106b and\n",
      "neighbour miR-106b and\n",
      "Conjunction before fc miR-25\n",
      "Conjuncts [BCL2L11/Bim, CDKN1A/p21]\n",
      "neighbour BCL2L11/Bim .\n",
      "neighbour BCL2L11/Bim None\n",
      "Conjunction before fc CDKN1A/p21\n",
      "fc add children for CDKN1A/p21\n",
      "Conjuncts [CDKN1A/p21, BCL2L11/Bim]\n",
      "neighbour CDKN1A/p21 and\n",
      "neighbour CDKN1A/p21 and\n",
      "Conjunction before fc BCL2L11/Bim\n",
      "Recognized overlapping entities: {Interval(109, 119, ('CELL', 'CDKN1A/p21'))}\n",
      "Gene reject: {'CELL'}\n",
      "[miR-25]\n",
      "[BCL2L11/Bim]\n",
      "[miR-25]\n",
      "[BCL2L11/Bim]\n",
      "passive? True\n",
      "Counter()\n",
      "TextBetween: miR-106b and miR-25 were previously shown to modulate the TGF-beta signaling pathway through their action on CDKN1A/p21 and BCL2L11/Bim\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': False, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence miR-106b and miR-25 were previously shown to modulate the TGF-beta signaling pathway through their action on CDKN1A/p21 and BCL2L11/Bim.\n",
      "full sentence <e1>miR-106b</e1> and miR-25 were previously shown to modulate the TGF-beta signaling pathway through their action on <e2>CDKN1A/p21</e2> and BCL2L11/Bim.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': False,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': False,\n",
       "  'passive': True,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': True,\n",
       "   'reg_detect': 'return',\n",
       "   'reg_detect_major': 'return'}},\n",
       " 'tagged_sent': '<e1>miR-106b</e1> and miR-25 were previously shown to modulate the TGF-beta signaling pathway through their action on <e2>CDKN1A/p21</e2> and BCL2L11/Bim.',\n",
       " 'full_sentence': 'CONCLUSION; <e1>miR-106b</e1> and miR-25 were previously shown to modulate the TGF-beta signaling pathway through their action on <e2>CDKN1A/p21</e2> and BCL2L11/Bim.',\n",
       " 'entity1': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (0, 8),\n",
       "  'entity_text': 'miR-106b'},\n",
       " 'entity2': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (109, 119),\n",
       "  'entity_text': 'CDKN1A/p21'}}"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"20637509.2.9\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (12,20)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (121,131)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case miR-1\n",
      "add special case mTOR\n",
      "miR-1 15 mTOR 19\n",
      "(u\"In this study, we report that expression of the muscle-specific micro-RNA (miRNA) miR-1 is regulated by mTOR both in differentiating myoblasts and in mouse regenerating skeletal muscle.\", 15, 19)\n",
      "geneword mTOR mirword miR-1\n",
      "Conjuncts [mouse, myoblasts]\n",
      "neighbour mouse regenerating\n",
      "neighbour mouse regenerating\n",
      "Conjunction before fc myoblasts\n",
      "fc add children for myoblasts\n",
      "fc add children for myoblasts\n",
      "Conjuncts [myoblasts, mouse]\n",
      "neighbour myoblasts and\n",
      "neighbour myoblasts and\n",
      "Conjunction before fc mouse\n",
      "fc add children for mouse\n",
      "fc add children for mouse\n",
      "miRNA myoblasts\n",
      ") miR-1 is regulated by mTOR both\n",
      "Conjunctions\n",
      "miRNA [(, )]\n",
      "myoblasts [in, differentiating, myoblasts, in, mouse, regenerating, skeletal, muscle]\n",
      "SDP sent: In this study, we report that expression of the muscle-specific micro-RNA (miRNA) miR-1 is regulated by mTOR both in differentiating myoblasts and in mouse regenerating skeletal muscle.\n",
      "SDP     : [(miR-1, 'PROPN', 'nmod', False), (expression, 'NOUN', 'nsubjpass', False), (regulated, 'VERB', 'ccomp', False), (mTOR, 'PROPN', 'nmod', False)]\n",
      "deepcheck connection rule noun found regulated expression [that, expression, of, the, muscle-specific, micro-RNA, (, miRNA, ), miR-1, is, regulated, by, mTOR, both, in, differentiating, myoblasts, and, in, mouse, regenerating, skeletal, muscle]\n",
      "creating subtree at regulated ccomp\n",
      "[in, mouse, regenerating, skeletal, muscle]\n",
      "0 [In, this, study, ,, we, report] 6 0 In this study, we report that expression of the muscle-specific micro-RNA (miRNA) miR-1 is regulated by mTOR both in differentiating myoblasts and in mouse regenerating skeletal muscle.\n",
      "1 [expression, of, the, muscle-specific, micro-RNA, (, miRNA, ), miR-1, is, regulated, by, mTOR, both, in, differentiating, myoblasts, and, in, mouse, regenerating, skeletal, muscle] 23 30 expression of the muscle-specific micro-RNA (miRNA) miR-1 is regulated by mTOR both in differentiating myoblasts and in mouse regenerating skeletal muscle.\n",
      "Conjuncts [mouse, myoblasts]\n",
      "neighbour mouse regenerating\n",
      "neighbour mouse regenerating\n",
      "Conjunction before fc myoblasts\n",
      "fc add children for myoblasts\n",
      "fc add children for myoblasts\n",
      "Conjuncts [myoblasts, mouse]\n",
      "neighbour myoblasts and\n",
      "neighbour myoblasts and\n",
      "Conjunction before fc mouse\n",
      "fc add children for mouse\n",
      "fc add children for mouse\n",
      "Recognized overlapping entities: {Interval(104, 108, ('GENE_OR_GENE_PRODUCT', 'mTOR'))}\n",
      "[Interval(104, 108, ('GENE_OR_GENE_PRODUCT', 'mTOR', mTOR))] [Interval(82, 87, ('', 'miR-1', miR-1))]\n",
      "miR-1 mTOR\n",
      "creating subtree at regulated ccomp\n",
      "[in, mouse, regenerating, skeletal, muscle]\n",
      "creating subtree at regulated ccomp\n",
      "[in, mouse, regenerating, skeletal, muscle]\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: miRNA ) miR-1 is regulated by mTOR both in\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'between mir gene rb', 'reg_detect_major': 'between'}})\n",
      " hit sentence In this study, we report that expression of the muscle-specific micro-RNA (miRNA) miR-1 is regulated by mTOR both in differentiating myoblasts and in mouse regenerating skeletal muscle.\n",
      "full sentence In this study, we report that expression of the muscle-specific micro-RNA (miRNA) <e1>miR-1</e1> is regulated by <e2>mTOR</e2> both in differentiating myoblasts and in mouse regenerating skeletal muscle.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': True,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'GENE_MIR',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'between mir gene rb',\n",
       "   'reg_detect_major': 'between'}},\n",
       " 'tagged_sent': 'In this study, we report that expression of the muscle-specific micro-RNA (miRNA) <e1>miR-1</e1> is regulated by <e2>mTOR</e2> both in differentiating myoblasts and in mouse regenerating skeletal muscle.',\n",
       " 'full_sentence': 'In this study, we report that expression of the muscle-specific micro-RNA (miRNA) <e1>miR-1</e1> is regulated by <e2>mTOR</e2> both in differentiating myoblasts and in mouse regenerating skeletal muscle.',\n",
       " 'entity1': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (82, 87),\n",
       "  'entity_text': 'miR-1'},\n",
       " 'entity2': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (104, 108),\n",
       "  'entity_text': 'mTOR'}}"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"20566686.2.2\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (82,87)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (104,108)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case TCF8\n",
      "add special case miR-200c\n",
      "TCF8 27 miR-200c 37\n",
      "(u\"Expression analysis by qRT-PCR of these targets and an additional ten mRNAs (selected using the prediction program ranking alone) revealed four mRNAs, BIN1, TCF8, RND3 and LHFP with an inverse relationship to miR-200c.\", 27, 37)\n",
      "geneword TCF8 mirword miR-200c\n",
      "Conjuncts [mRNAs, qRT-PCR]\n",
      "neighbour mRNAs (\n",
      "neighbour mRNAs selected\n",
      "Conjunction before fc qRT-PCR\n",
      "fc add children for qRT-PCR\n",
      "Conjuncts [qRT-PCR, mRNAs]\n",
      "neighbour qRT-PCR of\n",
      "neighbour qRT-PCR of\n",
      "Conjunction before fc mRNAs\n",
      "fc add children for mRNAs\n",
      "fc add children for mRNAs\n",
      "fc add children for mRNAs\n",
      "Conjuncts [TCF8, mRNAs]\n",
      "neighbour TCF8 ,\n",
      "neighbour TCF8 RND3\n",
      "Conjunction before fc mRNAs\n",
      "fc add children for mRNAs\n",
      "fc add children for mRNAs\n",
      "fc add children for mRNAs\n",
      "Conjuncts [mRNAs, TCF8]\n",
      "neighbour mRNAs ,\n",
      "neighbour mRNAs BIN1\n",
      "Conjunction before fc TCF8\n",
      "Conjuncts [LHFP, RND3]\n",
      "neighbour LHFP with\n",
      "neighbour LHFP with\n",
      "Conjunction before fc RND3\n",
      "Conjuncts [RND3, LHFP]\n",
      "neighbour RND3 and\n",
      "neighbour RND3 and\n",
      "Conjunction before fc LHFP\n",
      "qRT-PCR mRNAs\n",
      "selected using the prediction program ranking alone) revealed\n",
      "Conjunctions\n",
      "qRT-PCR [by, qRT-PCR, additional, ten, mRNAs, selected]\n",
      "mRNAs [four, mRNAs, BIN1, TCF8, RND3, LHFP]\n",
      "SDP sent: Expression analysis by qRT-PCR of these targets and an additional ten mRNAs (selected using the prediction program ranking alone) revealed four mRNAs, BIN1, TCF8, RND3 and LHFP with an inverse relationship to miR-200c.\n",
      "SDP     : [(TCF8, 'PROPN', 'conj', False), (mRNAs, 'NOUN', 'dobj', True), (relationship, 'NOUN', 'nmod', False), (miR-200c, 'PROPN', 'nmod', False)]\n",
      "[an, additional, ten, mRNAs, (, selected, using, the, prediction, program, ranking, alone, )]\n",
      "[TCF8]\n",
      "[LHFP]\n",
      "0 [Expression, analysis, by, qRT-PCR, of, these, targets, and, an, additional, ten, mRNAs, (, selected, ), revealed, four, mRNAs, ,, BIN1, ,, TCF8, ,, RND3, and, LHFP, with, an, inverse, relationship, to, miR-200c] 32 0 Expression analysis by qRT-PCR of these targets and an additional ten mRNAs (selected using the prediction program ranking alone) revealed four mRNAs, BIN1, TCF8, RND3 and LHFP with an inverse relationship to miR-200c.\n",
      "1 [using, the, prediction, program, ranking, alone] 6 86 using the prediction program ranking alone) revealed four mRNAs, BIN1, TCF8, RND3 and LHFP with an inverse relationship to miR-200c.\n",
      "Conjuncts [mRNAs, qRT-PCR]\n",
      "neighbour mRNAs (\n",
      "neighbour mRNAs selected\n",
      "Conjunction before fc qRT-PCR\n",
      "fc add children for qRT-PCR\n",
      "Conjuncts [qRT-PCR, mRNAs]\n",
      "neighbour qRT-PCR of\n",
      "neighbour qRT-PCR of\n",
      "Conjunction before fc mRNAs\n",
      "fc add children for mRNAs\n",
      "fc add children for mRNAs\n",
      "fc add children for mRNAs\n",
      "Conjuncts [TCF8, mRNAs]\n",
      "neighbour TCF8 ,\n",
      "neighbour TCF8 RND3\n",
      "Conjunction before fc mRNAs\n",
      "fc add children for mRNAs\n",
      "fc add children for mRNAs\n",
      "fc add children for mRNAs\n",
      "Conjuncts [mRNAs, TCF8]\n",
      "neighbour mRNAs ,\n",
      "neighbour mRNAs BIN1\n",
      "Conjunction before fc TCF8\n",
      "Conjuncts [LHFP, RND3]\n",
      "neighbour LHFP with\n",
      "neighbour LHFP with\n",
      "Conjunction before fc RND3\n",
      "Conjuncts [RND3, LHFP]\n",
      "neighbour RND3 and\n",
      "neighbour RND3 and\n",
      "Conjunction before fc LHFP\n",
      "Recognized overlapping entities: {Interval(157, 161, ('GENE_OR_GENE_PRODUCT', 'TCF8'))}\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: BIN1 , TCF8 , RND3 and LHFP with an inverse relationship to miR-200c\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir neg corr', 'reg_detect_major': 'between'}})\n",
      " hit sentence Expression analysis by qRT-PCR of these targets and an additional ten mRNAs (selected using the prediction program ranking alone) revealed four mRNAs, BIN1, TCF8, RND3 and LHFP with an inverse relationship to miR-200c.\n",
      "full sentence Expression analysis by qRT-PCR of these targets and an additional ten mRNAs (selected using the prediction program ranking alone) revealed four mRNAs, BIN1, <e2>TCF8</e2>, RND3 and LHFP with an inverse relationship to <e1>miR-200c</e1>.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': True,\n",
       " 'check_results': {'conj': True,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': False,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'DOWN',\n",
       "   'interaction_dir': 'MIR_GENE',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'between gene mir neg corr',\n",
       "   'reg_detect_major': 'between'}},\n",
       " 'tagged_sent': 'Expression analysis by qRT-PCR of these targets and an additional ten mRNAs (selected using the prediction program ranking alone) revealed four mRNAs, BIN1, <e2>TCF8</e2>, RND3 and LHFP with an inverse relationship to <e1>miR-200c</e1>.',\n",
       " 'full_sentence': 'Expression analysis by qRT-PCR of these targets and an additional ten mRNAs (selected using the prediction program ranking alone) revealed four mRNAs, BIN1, <e2>TCF8</e2>, RND3 and LHFP with an inverse relationship to <e1>miR-200c</e1>.',\n",
       " 'entity1': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (157, 161),\n",
       "  'entity_text': 'TCF8'},\n",
       " 'entity2': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (209, 217),\n",
       "  'entity_text': 'miR-200c'}}"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"16929162.2.8\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (209,217)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (157,161)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case MMP-2\n",
      "add special case miR-181b\n",
      "MMP-2 12 miR-181b 21\n",
      "(u\"Furthermore, treatment of SMCCs with SCF+GM-CSF significantly increased matrix metalloproteinases (MMP-2 and MMP-9) messenger RNA as well as miR-181b expression, along with a reduction of metalloproteinase inhibitor 3.\", 12, 21)\n",
      "geneword MMP-2 mirword miR-181b\n",
      "Conjuncts [MMP-9, MMP-2]\n",
      "neighbour MMP-9 )\n",
      "neighbour MMP-9 messenger\n",
      "Conjunction before fc MMP-2\n",
      "Conjuncts [MMP-2, MMP-9]\n",
      "neighbour MMP-2 and\n",
      "neighbour MMP-2 and\n",
      "Conjunction before fc MMP-9\n",
      "Conjuncts [expression, RNA]\n",
      "neighbour expression ,\n",
      "neighbour expression along\n",
      "Conjunction before fc RNA\n",
      "fc add children for RNA\n",
      "fc add children for RNA\n",
      "Conjuncts [RNA, expression]\n",
      "neighbour RNA as\n",
      "neighbour RNA as\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "Conjunctions\n",
      "MMP-2 [matrix, metalloproteinases, MMP-2, MMP-9, messenger, RNA, miR-181b, expression]\n",
      "['MMP-2', 'and', 'MMP-9', ')', 'messenger', 'RNA', 'as', 'well', 'as']\n",
      "achildren {miR-181b}\n",
      "achildren False\n",
      "SDP sent: Furthermore, treatment of SMCCs with SCF+GM-CSF significantly increased matrix metalloproteinases (MMP-2 and MMP-9) messenger RNA as well as miR-181b expression, along with a reduction of metalloproteinase inhibitor 3.\n",
      "SDP     : [(MMP-2, 'PROPN', 'dep', False), (metalloproteinases, 'NOUN', 'compound', False), (RNA, 'NOUN', 'dobj', True), (expression, 'NOUN', 'conj', False), (miR-181b, 'PROPN', 'compound', False)]\n",
      "[MMP-9]\n",
      "[miR-181b, expression]\n",
      "0 [Furthermore, ,, treatment, of, SMCCs, with, SCF+GM-CSF, significantly, increased, matrix, metalloproteinases, (, MMP-2, and, MMP-9, ), messenger, RNA, as, well, as, miR-181b, expression, ,, along, with, a, reduction, of, metalloproteinase, inhibitor, 3] 32 0 Furthermore, treatment of SMCCs with SCF+GM-CSF significantly increased matrix metalloproteinases (MMP-2 and MMP-9) messenger RNA as well as miR-181b expression, along with a reduction of metalloproteinase inhibitor 3.\n",
      "Conjuncts [MMP-9, MMP-2]\n",
      "neighbour MMP-9 )\n",
      "neighbour MMP-9 messenger\n",
      "Conjunction before fc MMP-2\n",
      "Conjuncts [MMP-2, MMP-9]\n",
      "neighbour MMP-2 and\n",
      "neighbour MMP-2 and\n",
      "Conjunction before fc MMP-9\n",
      "Conjuncts [expression, RNA]\n",
      "neighbour expression ,\n",
      "neighbour expression along\n",
      "Conjunction before fc RNA\n",
      "fc add children for RNA\n",
      "fc add children for RNA\n",
      "Conjuncts [RNA, expression]\n",
      "neighbour RNA as\n",
      "neighbour RNA as\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "Recognized overlapping entities: {Interval(99, 104, ('GENE_OR_GENE_PRODUCT', 'MMP-2'))}\n",
      "[Interval(99, 104, ('GENE_OR_GENE_PRODUCT', 'MMP-2', MMP-2))] [Interval(141, 149, ('GENE_OR_GENE_PRODUCT', 'miR-181b', miR-181b))]\n",
      "miR-181b MMP-2\n",
      "[MMP-9]\n",
      "[miR-181b, expression]\n",
      "[MMP-9]\n",
      "[miR-181b, expression]\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: metalloproteinases ( MMP-2 and MMP-9 ) messenger RNA as well as miR-181b expression ,\n",
      "(False, {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence Furthermore, treatment of SMCCs with SCF+GM-CSF significantly increased matrix metalloproteinases (MMP-2 and MMP-9) messenger RNA as well as miR-181b expression, along with a reduction of metalloproteinase inhibitor 3.\n",
      "full sentence Furthermore, treatment of SMCCs with SCF+GM-CSF significantly increased matrix metalloproteinases (<e2>MMP-2</e2> and MMP-9) messenger RNA as well as <e1>miR-181b</e1> expression, along with a reduction of metalloproteinase inhibitor 3.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': False,\n",
       " 'check_results': {'conj': False,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': True,\n",
       "  'passive': False,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'GENE_MIR',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'return',\n",
       "   'reg_detect_major': 'return'}},\n",
       " 'tagged_sent': 'Furthermore, treatment of SMCCs with SCF+GM-CSF significantly increased matrix metalloproteinases (<e2>MMP-2</e2> and MMP-9) messenger RNA as well as <e1>miR-181b</e1> expression, along with a reduction of metalloproteinase inhibitor 3.',\n",
       " 'full_sentence': 'Furthermore, treatment of SMCCs with SCF+GM-CSF significantly increased matrix metalloproteinases (<e2>MMP-2</e2> and MMP-9) messenger RNA as well as <e1>miR-181b</e1> expression, along with a reduction of metalloproteinase inhibitor 3.',\n",
       " 'entity1': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (99, 104),\n",
       "  'entity_text': 'MMP-2'},\n",
       " 'entity2': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (141, 149),\n",
       "  'entity_text': 'miR-181b'}}"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"21932404.2.9\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (141,149)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (99,104)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add special case USP25\n",
      "add special case miR-125b\n",
      "USP25 11 miR-125b-2 26\n",
      "(u\"We found frequent downregulation of two coding genes, SAMSN1 and USP25, as well as of three miRNA genes, miR-99a, let-7c, and miR-125b-2, which reside in the commonly deleted region in human lung cancer.\", 11, 26)\n",
      "geneword USP25 mirword miR-125b-2\n",
      "Conjuncts [USP25, SAMSN1]\n",
      "neighbour USP25 ,\n",
      "neighbour USP25 as\n",
      "Conjunction before fc SAMSN1\n",
      "Conjuncts [SAMSN1, USP25]\n",
      "neighbour SAMSN1 and\n",
      "neighbour SAMSN1 and\n",
      "Conjunction before fc USP25\n",
      "Conjuncts [let-7c, miR-125b-2, genes]\n",
      "neighbour let-7c ,\n",
      "neighbour let-7c and\n",
      "neighbour miR-125b-2 ,\n",
      "neighbour miR-125b-2 which\n",
      "Conjunction before fc genes\n",
      "fc add children for genes\n",
      "fc add children for genes\n",
      "fc add children for genes\n",
      "Conjuncts [genes, miR-125b-2, let-7c]\n",
      "neighbour genes ,\n",
      "neighbour genes miR-99a\n",
      "neighbour miR-125b-2 ,\n",
      "neighbour miR-125b-2 which\n",
      "Conjunction before fc let-7c\n",
      "Conjuncts [genes, let-7c, miR-125b-2]\n",
      "neighbour genes ,\n",
      "neighbour genes miR-99a\n",
      "neighbour let-7c ,\n",
      "neighbour let-7c and\n",
      "Conjunction before fc miR-125b-2\n",
      "SAMSN1 genes\n",
      "USP25, as well as of\n",
      "Double Conjunction Found\n",
      "Conjunctions\n",
      "SAMSN1 [SAMSN1, USP25]\n",
      "genes [three, miRNA, genes, miR-99a, let-7c, miR-125b-2]\n",
      "(genes, 'conj') [SAMSN1, USP25, three, miRNA, genes, miR-99a, let-7c, miR-125b-2]\n",
      "['USP25', ',', 'as', 'well', 'as', 'of', 'three', 'miRNA', 'genes', ',', 'miR-99a', ',', 'let-7c', ',', 'and']\n",
      "achildren {miR-125b-2}\n",
      "achildren False\n",
      "SDP sent: We found frequent downregulation of two coding genes, SAMSN1 and USP25, as well as of three miRNA genes, miR-99a, let-7c, and miR-125b-2, which reside in the commonly deleted region in human lung cancer.\n",
      "SDP     : [(USP25, 'PROPN', 'conj', False), (SAMSN1, 'NOUN', 'appos', True), (genes, 'NOUN', 'nmod', False), (genes, 'NOUN', 'dep', True), (miR-125b-2, 'NOUN', 'conj', False)]\n",
      "[USP25]\n",
      "[let-7c]\n",
      "[miR-125b-2]\n",
      "0 [We, found, frequent, downregulation, of, two, coding, genes, ,, SAMSN1, and, USP25, ,, as, well, as, of, three, miRNA, genes, ,, miR-99a, ,, let-7c, ,, and, miR-125b-2, ,, which, reside, in, the, commonly, deleted, region, in, human, lung, cancer] 39 0 We found frequent downregulation of two coding genes, SAMSN1 and USP25, as well as of three miRNA genes, miR-99a, let-7c, and miR-125b-2, which reside in the commonly deleted region in human lung cancer.\n",
      "Conjuncts [USP25, SAMSN1]\n",
      "neighbour USP25 ,\n",
      "neighbour USP25 as\n",
      "Conjunction before fc SAMSN1\n",
      "Conjuncts [SAMSN1, USP25]\n",
      "neighbour SAMSN1 and\n",
      "neighbour SAMSN1 and\n",
      "Conjunction before fc USP25\n",
      "Conjuncts [let-7c, miR-125b-2, genes]\n",
      "neighbour let-7c ,\n",
      "neighbour let-7c and\n",
      "neighbour miR-125b-2 ,\n",
      "neighbour miR-125b-2 which\n",
      "Conjunction before fc genes\n",
      "fc add children for genes\n",
      "fc add children for genes\n",
      "fc add children for genes\n",
      "Conjuncts [genes, miR-125b-2, let-7c]\n",
      "neighbour genes ,\n",
      "neighbour genes miR-99a\n",
      "neighbour miR-125b-2 ,\n",
      "neighbour miR-125b-2 which\n",
      "Conjunction before fc let-7c\n",
      "Conjuncts [genes, let-7c, miR-125b-2]\n",
      "neighbour genes ,\n",
      "neighbour genes miR-99a\n",
      "neighbour let-7c ,\n",
      "neighbour let-7c and\n",
      "Conjunction before fc miR-125b-2\n",
      "Recognized overlapping entities: {Interval(65, 70, ('GENE_OR_GENE_PRODUCT', 'USP25'))}\n",
      "[Interval(65, 70, ('GENE_OR_GENE_PRODUCT', 'USP25', USP25))] [Interval(126, 136, ('GENE_OR_GENE_PRODUCT', 'miR-125b-2', miR-125b-2))]\n",
      "miR-125b-2 USP25\n",
      "Context SDP Check\n",
      "[(USP25, 'NOUN', 'conj', False), (SAMSN1, 'NOUN', 'appos', True), (genes, 'NOUN', 'nmod', True), (genes, 'NOUN', 'conj', True), (miR-125b-2, 'NOUN', 'conj', False)]\n",
      "(SAMSN1, 'NOUN', 'appos', True)\n",
      "[USP25]\n",
      "[let-7c]\n",
      "[miR-125b-2]\n",
      "[USP25]\n",
      "[let-7c]\n",
      "[miR-125b-2]\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: SAMSN1 and USP25 , as well as of three miRNA genes , miR-99a , let-7c , and miR-125b-2 , which\n",
      "(False, {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence We found frequent downregulation of two coding genes, SAMSN1 and USP25, as well as of three miRNA genes, miR-99a, let-7c, and miR-125b-2, which reside in the commonly deleted region in human lung cancer.\n",
      "full sentence We found frequent downregulation of two coding genes, SAMSN1 and <e2>USP25</e2>, as well as of three miRNA genes, miR-99a, let-7c, and <e1>miR-125b</e1>-2, which reside in the commonly deleted region in human lung cancer.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'accept_relation': False,\n",
       " 'check_results': {'conj': False,\n",
       "  'sdp': True,\n",
       "  'compartment': True,\n",
       "  'context': True,\n",
       "  'entity': False,\n",
       "  'passive': False,\n",
       "  'negation': False,\n",
       "  'classification': {'regulation_dir': 'NEU',\n",
       "   'interaction_dir': 'GENE_MIR',\n",
       "   'passive': False,\n",
       "   'reg_detect': 'return',\n",
       "   'reg_detect_major': 'return'}},\n",
       " 'tagged_sent': 'We found frequent downregulation of two coding genes, SAMSN1 and <e2>USP25</e2>, as well as of three miRNA genes, miR-99a, let-7c, and <e1>miR-125b</e1>-2, which reside in the commonly deleted region in human lung cancer.',\n",
       " 'full_sentence': 'We found frequent downregulation of two coding genes, SAMSN1 and <e2>USP25</e2>, as well as of three miRNA genes, miR-99a, let-7c, and <e1>miR-125b</e1>-2, which reside in the commonly deleted region in human lung cancer.',\n",
       " 'entity1': {'entity_type': 'gene',\n",
       "  'entity_type_token': 'e2',\n",
       "  'entity_location': (65, 70),\n",
       "  'entity_text': 'USP25'},\n",
       " 'entity2': {'entity_type': 'mirna',\n",
       "  'entity_type_token': 'e1',\n",
       "  'entity_location': (126, 134),\n",
       "  'entity_text': 'miR-125b'}}"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relRes = relChecker.check_sentence(allSents[\"18523997.2.6\"]\n",
    "                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": (126,134)} #mirna\n",
    "                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": (65,70)} #gene\n",
    "                                , fix_special_chars=False\n",
    "                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                )\n",
    "relRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
