{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "import os, sys\n",
    "sys.path.insert(0, \"/mnt/d/dev/git/miRExplore/python/\")\n",
    "\n",
    "\n",
    "import scispacy\n",
    "import spacy\n",
    "from spacy import displacy\n",
    "\n",
    "nlp = spacy.load('/mnt/d/spacy/models/en_core_sci_lg-0.2.4/en_core_sci_lg/en_core_sci_lg-0.2.4/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import textmining.MirGeneRelCheck\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def analyse(testCase):\n",
    "    relCheck = textmining.MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "    doc = nlp(testCase[0])\n",
    "    \n",
    "    lWord = doc[testCase[1]]\n",
    "    rWord = doc[testCase[2]]\n",
    "\n",
    "    for t in doc:\n",
    "        print(t.idx, t, t.pos_, t.dep_, t.head, t.head.idx)\n",
    "\n",
    "    print(lWord, rWord)\n",
    "\n",
    "    if not \"mir\" in str(lWord).lower() and not \"let\" in str(lWord).lower():\n",
    "        tmp = lWord\n",
    "        lWord = rWord\n",
    "        rWord = tmp\n",
    "        \n",
    "    print(lWord, rWord)\n",
    "    \n",
    "    checkResults = relCheck.checkRelation(doc, lWord, rWord, verbose=True)\n",
    "    print(checkResults)\n",
    "    \n",
    "    displacy.render(doc, style=\"dep\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Functional ADJ amod association 11\n",
      "11 association NOUN nsubjpass established 72\n",
      "23 between ADP case miR 31\n",
      "31 miR NOUN nmod association 11\n",
      "35 124a NOUN dep miR 31\n",
      "40 and CCONJ cc 124a 35\n",
      "44 CDK6 NOUN conj 124a 35\n",
      "49 in ADP case medulloblastoma 52\n",
      "52 medulloblastoma NOUN nmod association 11\n",
      "68 was AUX auxpass established 72\n",
      "72 established VERB ROOT established 72\n",
      "84 using VERB xcomp established 72\n",
      "90 luciferase NOUN compound assays 101\n",
      "101 assays NOUN dobj using 84\n",
      "107 . PUNCT punct established 72\n",
      "124a CDK6\n",
      "CDK6 124a\n",
      "Conjuncts [CDK6, 124a]\n",
      "neighbour CDK6 in\n",
      "neighbour CDK6 in\n",
      "Conjunction before fc 124a\n",
      "Conjuncts [124a, CDK6]\n",
      "neighbour 124a and\n",
      "neighbour 124a and\n",
      "Conjunction before fc CDK6\n",
      "Conjunctions\n",
      "124a [124a, CDK6]\n",
      "achildren {CDK6}\n",
      "SDP sent: Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\n",
      "SDP     : [(124a, 'NOUN', 'dep', False), (established, 'VERB', 'ROOT', False), (association, 'NOUN', 'nsubjpass', False), (miR, 'NOUN', 'nmod', False), (124a, 'NOUN', 'dep', False)]\n",
      "deepcheck connection rule noun found using luciferase [using, luciferase, assays]\n",
      "0 [Functional, association, between, miR, 124a, and, CDK6, in, medulloblastoma, was, established] 11 0 Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\n",
      "1 [using, luciferase, assays] 3 84 using luciferase assays.\n",
      "Conjuncts [CDK6, 124a]\n",
      "neighbour CDK6 in\n",
      "neighbour CDK6 in\n",
      "Conjunction before fc 124a\n",
      "Conjuncts [124a, CDK6]\n",
      "neighbour 124a and\n",
      "neighbour 124a and\n",
      "Conjunction before fc CDK6\n",
      "(False, {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'passive': True, 'negation': False})\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"4532c7dab8dc4d539d245330bdd9c795-0\" class=\"displacy\" width=\"2500\" height=\"487.0\" direction=\"ltr\" style=\"max-width: none; height: 487.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">Functional</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"225\">association</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"225\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"400\">between</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"400\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"575\">miR</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"575\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"750\">124a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"750\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"925\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"925\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1100\">CDK6</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1100\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1275\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1275\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1450\">medulloblastoma</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1450\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1625\">was</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1625\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1800\">established</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1800\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1975\">using</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1975\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2150\">luciferase</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2150\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2325\">assays.</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2325\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-0\" stroke-width=\"2px\" d=\"M70,352.0 C70,264.5 210.0,264.5 210.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M70,354.0 L62,342.0 78,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-1\" stroke-width=\"2px\" d=\"M245,352.0 C245,2.0 1800.0,2.0 1800.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubjpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M245,354.0 L237,342.0 253,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-2\" stroke-width=\"2px\" d=\"M420,352.0 C420,264.5 560.0,264.5 560.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M420,354.0 L412,342.0 428,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-3\" stroke-width=\"2px\" d=\"M245,352.0 C245,177.0 565.0,177.0 565.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M565.0,354.0 L573.0,342.0 557.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-4\" stroke-width=\"2px\" d=\"M595,352.0 C595,264.5 735.0,264.5 735.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dep</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M735.0,354.0 L743.0,342.0 727.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-5\" stroke-width=\"2px\" d=\"M770,352.0 C770,264.5 910.0,264.5 910.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M910.0,354.0 L918.0,342.0 902.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-6\" stroke-width=\"2px\" d=\"M770,352.0 C770,177.0 1090.0,177.0 1090.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1090.0,354.0 L1098.0,342.0 1082.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-7\" stroke-width=\"2px\" d=\"M1295,352.0 C1295,264.5 1435.0,264.5 1435.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1295,354.0 L1287,342.0 1303,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-8\" stroke-width=\"2px\" d=\"M245,352.0 C245,89.5 1445.0,89.5 1445.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1445.0,354.0 L1453.0,342.0 1437.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-9\" stroke-width=\"2px\" d=\"M1645,352.0 C1645,264.5 1785.0,264.5 1785.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">auxpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1645,354.0 L1637,342.0 1653,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-10\" stroke-width=\"2px\" d=\"M1820,352.0 C1820,264.5 1960.0,264.5 1960.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">xcomp</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1960.0,354.0 L1968.0,342.0 1952.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-11\" stroke-width=\"2px\" d=\"M2170,352.0 C2170,264.5 2310.0,264.5 2310.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2170,354.0 L2162,342.0 2178,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-4532c7dab8dc4d539d245330bdd9c795-0-12\" stroke-width=\"2px\" d=\"M1995,352.0 C1995,177.0 2315.0,177.0 2315.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-4532c7dab8dc4d539d245330bdd9c795-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2315.0,354.0 L2323.0,342.0 2307.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testcase = (u\"Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\", 4, 6)\n",
    "\n",
    "analyse(testcase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Upregulation NOUN nsubj decreased 25\n",
      "13 of ADP case miR-200a 16\n",
      "16 miR-200a NOUN nmod Upregulation 0\n",
      "25 decreased VERB ROOT decreased 25\n",
      "35 the DET det expression 39\n",
      "39 expression NOUN dobj decreased 25\n",
      "50 of ADP case factors 67\n",
      "53 transcription NOUN compound factors 67\n",
      "67 factors NOUN nmod expression 39\n",
      "75 ZEB1 NOUN dep factors 67\n",
      "80 and CCONJ cc ZEB1 75\n",
      "84 SIP1 NOUN conj ZEB1 75\n",
      "88 , PUNCT punct decreased 25\n",
      "90 with ADP case expression 116\n",
      "95 consequent ADJ amod expression 116\n",
      "106 increased VERB amod expression 116\n",
      "116 expression NOUN nmod decreased 25\n",
      "127 of ADP case E-cadherin 130\n",
      "130 E-cadherin NOUN nmod expression 116\n",
      "140 , PUNCT punct E-cadherin 130\n",
      "142 an DET det protein 154\n",
      "145 adhesion NOUN compound protein 154\n",
      "154 protein NOUN appos E-cadherin 130\n",
      "162 associated VERB acl protein 154\n",
      "173 with ADP case differentiation 183\n",
      "178 cell NOUN compound differentiation 183\n",
      "183 differentiation NOUN nmod associated 162\n",
      "198 . PUNCT punct decreased 25\n",
      "miR-200a ZEB1\n",
      "miR-200a ZEB1\n",
      "Conjuncts [SIP1, ZEB1]\n",
      "neighbour SIP1 ,\n",
      "neighbour SIP1 with\n",
      "Conjunction before fc ZEB1\n",
      "Conjuncts [ZEB1, SIP1]\n",
      "neighbour ZEB1 and\n",
      "neighbour ZEB1 and\n",
      "Conjunction before fc SIP1\n",
      "Conjunctions\n",
      "ZEB1 [ZEB1, SIP1]\n",
      "protein [an, differentiation]\n",
      "SDP sent: Upregulation of miR-200a decreased the expression of transcription factors ZEB1 and SIP1, with consequent increased expression of E-cadherin, an adhesion protein associated with cell differentiation.\n",
      "SDP     : [(miR-200a, 'NOUN', 'nmod', False), (Upregulation, 'NOUN', 'nsubj', False), (decreased, 'VERB', 'ROOT', False), (expression, 'NOUN', 'dobj', False), (factors, 'NOUN', 'nmod', False), (ZEB1, 'NOUN', 'dep', False)]\n",
      "0 [Upregulation, of, miR-200a, decreased, the, expression, of, transcription, factors, ZEB1, and, SIP1, ,, with, consequent, increased, expression, of, E-cadherin, ,, an, adhesion, protein, associated, with, cell, differentiation] 27 0 Upregulation of miR-200a decreased the expression of transcription factors ZEB1 and SIP1, with consequent increased expression of E-cadherin, an adhesion protein associated with cell differentiation.\n",
      "Conjuncts [SIP1, ZEB1]\n",
      "neighbour SIP1 ,\n",
      "neighbour SIP1 with\n",
      "Conjunction before fc ZEB1\n",
      "Conjuncts [ZEB1, SIP1]\n",
      "neighbour ZEB1 and\n",
      "neighbour ZEB1 and\n",
      "Conjunction before fc SIP1\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'passive': False, 'negation': False})\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"e48ea98713814cf5bf6aae65d95ab42c-0\" class=\"displacy\" width=\"4425\" height=\"487.0\" direction=\"ltr\" style=\"max-width: none; height: 487.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">Upregulation</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"225\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"225\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"400\">miR-200a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"400\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"575\">decreased</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"575\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"750\">the</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"750\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"925\">expression</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"925\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1100\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1100\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1275\">transcription</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1275\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1450\">factors</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1450\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1625\">ZEB1</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1625\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1800\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1800\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1975\">SIP1,</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1975\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2150\">with</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2150\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2325\">consequent</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2325\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2500\">increased</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2500\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2675\">expression</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2675\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2850\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2850\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3025\">E-cadherin,</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3025\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3200\">an</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3200\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3375\">adhesion</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3375\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3550\">protein</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3550\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3725\">associated</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3725\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3900\">with</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3900\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"4075\">cell</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"4075\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"397.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"4250\">differentiation.</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"4250\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-0\" stroke-width=\"2px\" d=\"M70,352.0 C70,89.5 570.0,89.5 570.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M70,354.0 L62,342.0 78,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-1\" stroke-width=\"2px\" d=\"M245,352.0 C245,264.5 385.0,264.5 385.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M245,354.0 L237,342.0 253,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-2\" stroke-width=\"2px\" d=\"M70,352.0 C70,177.0 390.0,177.0 390.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M390.0,354.0 L398.0,342.0 382.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-3\" stroke-width=\"2px\" d=\"M770,352.0 C770,264.5 910.0,264.5 910.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M770,354.0 L762,342.0 778,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-4\" stroke-width=\"2px\" d=\"M595,352.0 C595,177.0 915.0,177.0 915.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M915.0,354.0 L923.0,342.0 907.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-5\" stroke-width=\"2px\" d=\"M1120,352.0 C1120,177.0 1440.0,177.0 1440.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1120,354.0 L1112,342.0 1128,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-6\" stroke-width=\"2px\" d=\"M1295,352.0 C1295,264.5 1435.0,264.5 1435.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1295,354.0 L1287,342.0 1303,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-7\" stroke-width=\"2px\" d=\"M945,352.0 C945,89.5 1445.0,89.5 1445.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1445.0,354.0 L1453.0,342.0 1437.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-8\" stroke-width=\"2px\" d=\"M1470,352.0 C1470,264.5 1610.0,264.5 1610.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dep</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1610.0,354.0 L1618.0,342.0 1602.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-9\" stroke-width=\"2px\" d=\"M1645,352.0 C1645,264.5 1785.0,264.5 1785.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1785.0,354.0 L1793.0,342.0 1777.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-10\" stroke-width=\"2px\" d=\"M1645,352.0 C1645,177.0 1965.0,177.0 1965.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1965.0,354.0 L1973.0,342.0 1957.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-11\" stroke-width=\"2px\" d=\"M2170,352.0 C2170,89.5 2670.0,89.5 2670.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2170,354.0 L2162,342.0 2178,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-12\" stroke-width=\"2px\" d=\"M2345,352.0 C2345,177.0 2665.0,177.0 2665.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2345,354.0 L2337,342.0 2353,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-13\" stroke-width=\"2px\" d=\"M2520,352.0 C2520,264.5 2660.0,264.5 2660.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-13\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2520,354.0 L2512,342.0 2528,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-14\" stroke-width=\"2px\" d=\"M595,352.0 C595,2.0 2675.0,2.0 2675.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-14\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2675.0,354.0 L2683.0,342.0 2667.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-15\" stroke-width=\"2px\" d=\"M2870,352.0 C2870,264.5 3010.0,264.5 3010.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-15\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2870,354.0 L2862,342.0 2878,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-16\" stroke-width=\"2px\" d=\"M2695,352.0 C2695,177.0 3015.0,177.0 3015.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-16\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3015.0,354.0 L3023.0,342.0 3007.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-17\" stroke-width=\"2px\" d=\"M3220,352.0 C3220,177.0 3540.0,177.0 3540.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-17\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3220,354.0 L3212,342.0 3228,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-18\" stroke-width=\"2px\" d=\"M3395,352.0 C3395,264.5 3535.0,264.5 3535.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-18\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3395,354.0 L3387,342.0 3403,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-19\" stroke-width=\"2px\" d=\"M3045,352.0 C3045,89.5 3545.0,89.5 3545.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-19\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">appos</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3545.0,354.0 L3553.0,342.0 3537.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-20\" stroke-width=\"2px\" d=\"M3570,352.0 C3570,264.5 3710.0,264.5 3710.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-20\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">acl</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3710.0,354.0 L3718.0,342.0 3702.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-21\" stroke-width=\"2px\" d=\"M3920,352.0 C3920,177.0 4240.0,177.0 4240.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-21\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3920,354.0 L3912,342.0 3928,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-22\" stroke-width=\"2px\" d=\"M4095,352.0 C4095,264.5 4235.0,264.5 4235.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-22\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M4095,354.0 L4087,342.0 4103,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-e48ea98713814cf5bf6aae65d95ab42c-0-23\" stroke-width=\"2px\" d=\"M3745,352.0 C3745,89.5 4245.0,89.5 4245.0,352.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-e48ea98713814cf5bf6aae65d95ab42c-0-23\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M4245.0,354.0 L4253.0,342.0 4237.0,342.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testcase = (u\"Upregulation of miR-200a decreased the expression of transcription factors ZEB1 and SIP1, with consequent increased expression of E-cadherin, an adhesion protein associated with cell differentiation.\", 2, 9)\n",
    "\n",
    "analyse(testcase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "doc = nlp(testcase[0])\n",
    "\n",
    "allTokens = {t.i: t for t in doc}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[differentiation]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[x for x in allTokens[23].children]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'allRels' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-9-58076f1c53e7>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mallRels\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'allRels' is not defined"
     ]
    }
   ],
   "source": [
    "allRels[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testCase = (u\"We show here that miR-29a, -29b-1, and -9 can regulate BACE1 expression in vitro.\", 4, 12)\n",
    "testCase = (u\"It has been predicted that miR-146a may target Tata Binding Protein (TBP).\",5, 12)\n",
    "testCase = (u\"Our results showed that the expression of miR-106b was inversely correlated with TβRII protein levels and miR-106b can directly inhibit the TβRII translation in vitro.\", 7, 21)\n",
    "\n",
    "relCheck = textmining.MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "doc = nlp(testCase[0])\n",
    "\n",
    "mirword = doc[testCase[1]]\n",
    "geneword = doc[testCase[2]]\n",
    "\n",
    "verbose=False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "conjResult, conjs = relCheck.checkCommonConj(doc, mirword, geneword, verbose)\n",
    "sdpPass, passive, negated = relCheck.checkSDP(doc, mirword, geneword, verbose)\n",
    "compPass = relCheck.checkCompartments(doc, mirword, geneword, verbose)\n",
    "sigPathway = relCheck.checkSurContext(doc, mirword, geneword, verbose)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(miR-106b, 'NOUN', 'nmod', False),\n",
       " (expression, 'NOUN', 'nsubjpass', False),\n",
       " (correlated, 'VERB', 'ccomp', True),\n",
       " (inhibit, 'VERB', 'conj', False),\n",
       " (translation, 'NOUN', 'dobj', False),\n",
       " (TβRII, 'NOUN', 'compound', False)]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdpRes = relCheck._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "sdpRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testCase = (u\"We show here that miR-29a, -29b-1, and -9 can regulate BACE1 expression in vitro.\", 4, 12)\n",
    "testCase = (u\"It has been predicted that miR-146a may target Tata Binding Protein (TBP).\",5, 12)\n",
    "testCase = (u\"Our results showed that the expression of miR-106b was inversely correlated with TβRII protein levels and miR-106b can directly inhibit the TβRII translation in vitro.\", 7, 21)\n",
    "testCase = (u\"The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 6, 16)\n",
    "testCase = (u\"These findings identify miR-29b as a novel posttranscriptional regulator of PGRN expression, raising the possibility that miR-29b or other miRNAs might be targeted therapeutically to increase hPGRN levels in some FTD patients.\", 3,10)\n",
    "relCheck = textmining.MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "doc = nlp(testCase[0])\n",
    "\n",
    "mirword = doc[testCase[1]]\n",
    "geneword = doc[testCase[2]]\n",
    "\n",
    "verbose=False\n",
    "\n",
    "from collections import Counter\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading stems\n"
     ]
    }
   ],
   "source": [
    "def loadStems():\n",
    "    \n",
    "    print(\"Loading stems\")\n",
    "    allRels = []\n",
    "    stem2class = {}\n",
    "\n",
    "    renameClass = {\n",
    "        'INT': 'NEU',\n",
    "        \"BOOST\": \"POS\",\n",
    "        \"COMBINE\": \"NEU\",\n",
    "        \"CHANGE\": \"NEU\",\n",
    "    }\n",
    "\n",
    "    with open(\"/mnt/d/dev/git/miRExplore/python/allrels.csv\") as fin:\n",
    "\n",
    "        for line in fin:\n",
    "            aline = line.strip().split(\",\", 1)\n",
    "            aregSyns = aline[1].split(\":\")\n",
    "            regSyns = []\n",
    "            if len(aregSyns) > 1:\n",
    "                regSyns = aregSyns[1].split(\"|\")\n",
    "            \n",
    "            aline[1] = aline[1].split(\":\")[0]           \n",
    "\n",
    "            regClass = aline[0]\n",
    "            regStem = aline[1]\n",
    "\n",
    "            if regClass in renameClass:\n",
    "                regClass = renameClass[regClass]\n",
    "                \n",
    "            stem2class[regStem] = regClass\n",
    "                \n",
    "            for regSyn in regSyns:\n",
    "                if not regSyn in stem2class:\n",
    "                    stem2class[regSyn] = regClass\n",
    "\n",
    "\n",
    "            allRels.append((regClass, regStem))\n",
    "\n",
    "    allStems = [x for x in stem2class]\n",
    "    \n",
    "    return allStems, allRels, stem2class\n",
    "\n",
    "allStems, allRels, stem2class = loadStems()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "from textmining.MirGeneRelCheck import SentenceRelationChecker\n",
    "from utils.tmutils import normalize_gene_names\n",
    "from collections import Counter\n",
    "from lxml import etree\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MAP3K7 TAB1\n"
     ]
    }
   ],
   "source": [
    "scaiBase = \"/mnt/d/owncloud/data/miRExplore/scai_corpus/\"\n",
    "\n",
    "base = \"TEST\"\n",
    "\n",
    "if base == \"TRAIN\":\n",
    "    scaiFile = \"miRNA_train_fixed.xml\"\n",
    "    relsFile = \"scai_train_rels.tsv\"\n",
    "elif base == \"TEST\":\n",
    "    scaiFile = \"miRNA_test_fixed.xml\"\n",
    "    relsFile = \"scai_test_rels.tsv\"\n",
    "    \n",
    "normGeneSymbols= normalize_gene_names(path=scaiBase + \"/../obodir/\" + \"/hgnc_no_withdrawn.syn\")\n",
    "\n",
    "relChecker = SentenceRelationChecker(nlp)\n",
    "\n",
    "correctIdentified = 0\n",
    "incorrectIdentified = 0\n",
    "totalChecks = 0\n",
    "incorrectClass = Counter()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {},
   "outputs": [],
   "source": [
    "def loadRelations():\n",
    "    reldir2new = {}\n",
    "    reldir2new[\"TARGET\"] = \"NEU\"\n",
    "    reldir2new[\"INVCORR\"] = \"NEU\"\n",
    "    reldir2new[\"REGULATE\"] = \"NEU\"\n",
    "    relClasses = {}\n",
    "    with open(os.path.join(scaiBase, relsFile)) as fin:\n",
    "        \n",
    "        for lidx, line in enumerate(fin):\n",
    "\n",
    "            if lidx == 0:\n",
    "                continue\n",
    "\n",
    "            line = line.strip().split(\"\\t\")\n",
    "\n",
    "            interactionID = line[0].strip()\n",
    "            relType = line[1].strip()\n",
    "            relRDir = line[2].strip()\n",
    "            relIndirect = line[3].strip()\n",
    "            relPassive = line[4].strip()\n",
    "\n",
    "            if relRDir in reldir2new:\n",
    "                relRDir = reldir2new[relRDir]\n",
    "\n",
    "            relClasses[interactionID] = (relType, relRDir, relIndirect, relPassive)\n",
    "            \n",
    "    return relClasses\n",
    "\n",
    "relClasses = loadRelations()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "def runCheck(numelems):\n",
    "    \n",
    "    allStems, allRels, stem2class = loadStems()\n",
    "    \n",
    "    with open(os.path.join(scaiBase, scaiFile), 'r') as fin:\n",
    "        tree = etree.parse(fin)\n",
    "        root = tree.getroot()\n",
    "        scaiPairs = []\n",
    "        totalChecks = 0\n",
    "        correctIdentified = 0\n",
    "        incorrectIdentified = 0\n",
    "        \n",
    "        incorrectClassified = 0\n",
    "        correctClassified = 0\n",
    "        \n",
    "        errorByDetect = Counter()\n",
    "\n",
    "        for elem in root.findall(\".//document\"):\n",
    "\n",
    "            pmid = elem.attrib['origId']\n",
    "\n",
    "            for sentElem in elem:\n",
    "\n",
    "                allEntities = sentElem.findall(\".//entity\")\n",
    "                allPairs = sentElem.findall(\".//pair\")\n",
    "\n",
    "                sentText = sentElem.attrib[\"text\"]\n",
    "\n",
    "                entId2elem = {}\n",
    "\n",
    "                for entity in allEntities:\n",
    "                    entId = entity.attrib['id']\n",
    "                    entText = entity.attrib['text']\n",
    "                    entType = entity.attrib['type']\n",
    "                    entOffset = tuple([int(x) for x in entity.attrib['charOffset'].split(\"-\")])\n",
    "\n",
    "                    if entType in [\"Specific_miRNAs\", \"Genes/Proteins\"]:\n",
    "\n",
    "                        if \"Genes\" in entType:\n",
    "                            if entText in normGeneSymbols:\n",
    "                                entText = normGeneSymbols[entText]\n",
    "                            elif entText.upper() in normGeneSymbols:\n",
    "                                gene = normGeneSymbols[entText.upper()]\n",
    "                        else:\n",
    "                            try:\n",
    "                                entText = miRNA(entText).getStringFromParts([miRNAPART.MATURE, miRNAPART.ID, miRNAPART.PRECURSOR])\n",
    "                            except:\n",
    "                                pass\n",
    "\n",
    "                        entTuple = (entText, entType, (entOffset[0], entOffset[1]+1))\n",
    "                        entId2elem[entId] = entTuple\n",
    "\n",
    "\n",
    "                        sentEntText = sentText[entTuple[2][0]:entTuple[2][1]]\n",
    "\n",
    "                for pair in allPairs:\n",
    "\n",
    "                    validInteraction = pair.attrib['interaction'].lower() == \"true\"\n",
    "                    pairE1 = pair.attrib['e1']\n",
    "                    pairE2 = pair.attrib['e2']\n",
    "\n",
    "                    #if pairInt == 'true':\n",
    "                    if pairE1 in entId2elem and pairE2 in entId2elem:\n",
    "\n",
    "                        totalChecks += 1\n",
    "\n",
    "                        e1 = entId2elem[pairE1]\n",
    "                        e2 = entId2elem[pairE2]\n",
    "\n",
    "                        if not e1[1] in [\"Specific_miRNAs\"]:\n",
    "\n",
    "                            tmp=e1\n",
    "                            e1=e2\n",
    "                            e2=tmp\n",
    "\n",
    "\n",
    "                        relRes = relChecker.check_sentence(sentText\n",
    "                                                            , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": e1[2]}\n",
    "                                                            , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": e2[2]}\n",
    "                                                            , fix_special_chars=False\n",
    "                                                            , relClassifier=testRelClassifier2, verbose=False\n",
    "                                                            )\n",
    "\n",
    "\n",
    "                        fullsentence = relRes['full_sentence']\n",
    "                        acceptInteraction = relRes['accept_relation']\n",
    "\n",
    "                        if not acceptInteraction == validInteraction:\n",
    "\n",
    "                            incorrectClass[(validInteraction, acceptInteraction)]+=1\n",
    "                            incorrectIdentified += 1\n",
    "                        else:\n",
    "                            correctIdentified += 1\n",
    "\n",
    "\n",
    "                            if acceptInteraction:\n",
    "                                #print(relRes[\"check_results\"])                                \n",
    "                                \n",
    "                                foundClasses = relRes[\"check_results\"][\"classification\"] #{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'regulation_keys': set()}\n",
    "                                foundTuple = (foundClasses[\"interaction_dir\"], foundClasses[\"regulation_dir\"])                                \n",
    "                                relationID = pair.attrib[\"id\"]\n",
    "                                testTuple = relClasses[relationID]\n",
    "                                \n",
    "                                if testTuple[0] != foundTuple[0] or testTuple[1] != foundTuple[1]:\n",
    "                                    print(sentText)\n",
    "                                    print(e1, e2)\n",
    "                                    print(foundTuple, foundClasses[\"reg_detect\"], \"passive?\", foundClasses[\"passive\"])\n",
    "                                    print(testTuple)\n",
    "                                    \n",
    "                                    print()\n",
    "                                    print()\n",
    "                                    print()\n",
    "                                    \n",
    "                                    errorByDetect[foundClasses[\"reg_detect\"]] += 1\n",
    "                                    \n",
    "                                    incorrectClassified += 1\n",
    "\n",
    "\n",
    "\n",
    "                                    numelems -= 1\n",
    "                                    if numelems == 0:\n",
    "                                        return\n",
    "                                else:\n",
    "                                    \n",
    "                                    correctClassified += 1\n",
    "\n",
    "\n",
    "\n",
    "        def printStats(outfile):\n",
    "            print(\"Total:     \", totalChecks, file=outfile)\n",
    "            print(\"Correct:   \", correctIdentified, correctIdentified/totalChecks, file=outfile)\n",
    "            print(\"Incorrect: \", incorrectIdentified, incorrectIdentified/totalChecks, file=outfile)\n",
    "            print(\"classes\", incorrectClass, file=outfile)\n",
    "            \n",
    "            print(\"Correct classified:   \", correctClassified, correctClassified/totalChecks, file=outfile)\n",
    "            print(\"Incorrect classified:   \", incorrectClassified, incorrectClassified/totalChecks, file=outfile)\n",
    "            print(errorByDetect)\n",
    "\n",
    "        printStats(sys.stdout)\n",
    "        #printStats(sys.stderr)\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def getTokenDist(tk1, tk2):\n",
    "    return abs(tk1.i-tk2.i)\n",
    "\n",
    "def countStems(tokens, geneword, mirword, keepNEU=False):\n",
    "    stemTypes2Count = Counter()\n",
    "    mirgene2stem = defaultdict(lambda: Counter())\n",
    "\n",
    "    for tidx,x in enumerate(tokens):\n",
    "        tknStem, matchStem, wasFound = getstem(x)\n",
    "\n",
    "        if wasFound and (tknStem != \"NEU\" or keepNEU):\n",
    "            \n",
    "            if tidx >= len(tokens)-2 and str(tokens[tidx-1]) == \"to\":\n",
    "                continue\n",
    "            \n",
    "            geneDist = getTokenDist(x, geneword)\n",
    "            mirDist = getTokenDist(x, mirword)\n",
    "                        \n",
    "            if geneDist < mirDist and geneDist < 3:\n",
    "                mirgene2stem[\"gene\"][tknStem] += 1\n",
    "            elif mirDist < geneDist and mirDist < 3:\n",
    "                mirgene2stem[\"mir\"][tknStem] += 1\n",
    "\n",
    "        if wasFound and (tknStem != \"NEU\" or keepNEU):\n",
    "            stemTypes2Count[tknStem] += 1\n",
    "\n",
    "    if len(stemTypes2Count) == 0:\n",
    "        if \"regulated\" in str(geneword):\n",
    "            stemTypes2Count[\"NEU\"] += 1\n",
    "\n",
    "    return stemTypes2Count,mirgene2stem\n",
    "\n",
    "def getRelsForTokens(tokens, desc,verbose):\n",
    "    rels = []\n",
    "    verbSet = set([\"VERB\"])\n",
    "    for x in tokens:\n",
    "        \n",
    "        tokenStemClass, s, _ = getstem(x)\n",
    "        rels.append((s, None, tokenStemClass, x.pos_, x))\n",
    "        if verbose:\n",
    "            print(desc, x, s, tokenStemClass, x.pos_, x)\n",
    "                    \n",
    "    hasVerb = len(list(set([x[3] for x in rels]) & verbSet)) > 0\n",
    "    \n",
    "    if hasVerb:\n",
    "        rels = [x for x in rels if x[3] in verbSet]\n",
    "    \n",
    "    return rels\n",
    "\n",
    "def evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg):\n",
    "\n",
    "    invcor = False\n",
    "    regdir = None\n",
    "\n",
    "    #if dirReg == 0:\n",
    "    if (mirPos > 0 and geneNeg > 0) or (genePos > 0 and mirNeg > 0):\n",
    "        return \"DOWN\", True\n",
    "\n",
    "    #else:\n",
    "    if ((mirPos > 0 and genePos > 0) and(mirNeg == 0 and geneNeg == 0)) or ((mirNeg > 0 and geneNeg >0) and(mirPos == 0 and genePos == 0)):\n",
    "        return \"UP\", False\n",
    "\n",
    "    return None, None\n",
    "\n",
    "def scoreWord(tkn):\n",
    "    \n",
    "    jtkstem, s, _ = getstem(tkn)\n",
    "                        \n",
    "    if jtkstem == None:\n",
    "\n",
    "        if jtk.startswith((\"loss\", \"decreas\", \"low\")):\n",
    "            return 0,1\n",
    "\n",
    "        elif jtk.startswith((\"increas\", \"high\", \"posit\")):\n",
    "            return 1,0\n",
    "\n",
    "    else:\n",
    "        if jtkstem == \"POS\":\n",
    "            return 1,0\n",
    "        elif jtkstem == \"NEG\":\n",
    "            return 0,1\n",
    "        \n",
    "    return 0,0\n",
    "\n",
    "dir2opp = {\"MIR_GENE\": \"GENE_MIR\", \"GENE_MIR\": \"MIR_GENE\"}\n",
    "idir2opp = {\"POS\": \"NEG\", \"NEG\": \"POS\"}\n",
    "rdir2opp = {\"DOWN\": \"UP\", \"UP\": \"DOWN\"}\n",
    "\n",
    "def testRelClassifier2(doc, mirword, geneword, relchecker, verbose):\n",
    "    isPassive = relchecker.checkPassive(doc, mirword, geneword, verbose=False)\n",
    "    compartment = relchecker.getCompartment(doc, mirword, geneword, verbose=False)\n",
    "    \n",
    "    if verbose:\n",
    "        print(\"passive?\", isPassive)\n",
    "        \n",
    "        \n",
    "    sdpRes = relchecker._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "        \n",
    "    stemTypes2Count = Counter()\n",
    "    \n",
    "    for x in sdpRes:\n",
    "        tknStem, matchStem, _ = getstem(x[0])\n",
    "        \n",
    "        if tknStem != \"NEU\":\n",
    "            stemTypes2Count[tknStem] += 1\n",
    "            \n",
    "    print(stemTypes2Count)\n",
    "        \n",
    "    if len(stemTypes2Count) > 0:\n",
    "        regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "        \n",
    "        if regStem == \"NEG\":\n",
    "            regStem = \"DOWN\"\n",
    "        elif regStem == \"POS\":\n",
    "            regStem = \"UP\"\n",
    "            \n",
    "        if stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"]:\n",
    "            regStem = \"DOWN\"\n",
    "            \n",
    "        idir = \"MIR_GENE\"\n",
    "        if isPassive and mirword.i < geneword.i:\n",
    "            idir = \"GENE_MIR\"\n",
    "            \n",
    "        #return {\"regulation_dir\": regStem, \"interaction_dir\": idir, \"passive\": isPassive, \"reg_detect\": \"sdp counts\"}\n",
    "        \n",
    "    targetWords = [\"targets of\", \"targeting\", \"direct regulator\"]\n",
    "        \n",
    "    if compartment != None:\n",
    "        tokenBetween = compartment\n",
    "        textBetween = ' '.join(\" \".join([str(x) for x in tokenBetween]).split())\n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"compartment gene mir pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"compartment gene mir neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment gene mir\"}\n",
    "            \n",
    "        elif mirword.i < geneword.i:\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment mir gene pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment mir gene neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment mir gene\"}\n",
    "        \n",
    "\n",
    "\n",
    "    if True:\n",
    "        \n",
    "        tokSrc = compartment if compartment != None else doc\n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "            tokenBetween = [x for x in tokSrc if geneword.i-3 < x.i < mirword.i+3]\n",
    "            tokenBetweenStrict = [x for x in tokSrc if geneword.i < x.i < mirword.i]\n",
    "            textBetween = ' '.join(\" \".join([str(x) for x in tokenBetween]).split())\n",
    "            textAfter = ' '.join(\" \".join([str(x) for x in [y for y in doc if mirword.i <= y.i < mirword.i+5]]).split())\n",
    "            \n",
    "            print(textBetween)\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between gene mir pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between gene mir neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"negative regulation of\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between gene mir reg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords]) or any([x in textBetween for x in [\"target\"]]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between gene mir\"}\n",
    "            \n",
    "            #stemTypes2Count = Counter()\n",
    "            #for x in tokenBetween:\n",
    "            #    tknStem, matchStem = getstem(x)\n",
    "            #    if tknStem != \"NEU\":\n",
    "            #        stemTypes2Count[tknStem] += 1\n",
    "                    \n",
    "            if False and len(stemTypes2Count) > 0:\n",
    "                regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                if regStem == \"NEG\":\n",
    "                    regStem = \"DOWN\"\n",
    "                elif regStem == \"POS\":\n",
    "                    regStem = \"UP\"\n",
    "                    \n",
    "                if stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"]:\n",
    "                    regStem = \"DOWN\"\n",
    "\n",
    "                idir = \"MIR_GENE\"\n",
    "                if isPassive and geneword.i < mirword.i:\n",
    "                    idir = \"GENE_MIR\"\n",
    "\n",
    "                return {\"regulation_dir\": regStem, \"interaction_dir\": idir, \"passive\": isPassive, \"reg_detect\": \"counts between gene mir\"}\n",
    "            \n",
    "        else:\n",
    "            #mirword.i < geneword.i\n",
    "            tokenBetween = [x for x in tokSrc if mirword.i-3 < x.i < geneword.i+3]\n",
    "            tokenBetweenStrict = [x for x in tokSrc if mirword.i < x.i < geneword.i]\n",
    "            textBetween = ' '.join(\" \".join([str(x) for x in tokenBetween]).split())\n",
    "            \n",
    "            print(textBetween)\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]) and stemTypes2Count.get(\"NEG\", 0) == 0:\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"negative regulation of\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene reg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"directly inhibits\", \"directly represses\", \"translational inhibition\", ]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords + [\"directly regulated\"]]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"negatively regulated by\", \"negative regulation of\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between mir gene nrb\"}\n",
    "            \n",
    "            \n",
    "        if len(tokenBetween) > 0:\n",
    "                       \n",
    "            stemTypes2Count,mirgene2stem = countStems(tokenBetween, geneword, mirword)\n",
    "                    \n",
    "            if len(stemTypes2Count) > 0:\n",
    "                regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                \n",
    "                if len(stemTypes2Count) >= 2 and len(mirgene2stem[\"gene\"]) >0 and len(mirgene2stem[\"mir\"])>0 and mirgene2stem[\"gene\"].most_common(1)[0][0] == idir2opp[mirgene2stem[\"mir\"].most_common(1)[0][0]]:\n",
    "                    print(mirgene2stem)\n",
    "                    if geneword.i < mirword.i:\n",
    "                        if mirgene2stem[\"gene\"].most_common(1)[0][0] == \"POS\":\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                        else:\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                    else:\n",
    "                                                \n",
    "                        if mirgene2stem[\"mir\"].most_common(1)[0][0] == \"NEG\":\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                                                        \n",
    "                        else:\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                            \n",
    "                        if \"after\" in textBetween:\n",
    "                            regDir = dir2opp[regDir]\n",
    "                            \n",
    "                    if isPassive:\n",
    "                        regDir = dir2opp[regDir]\n",
    "                        regStem = rdir2opp[regStem]\n",
    "                        \n",
    "                    return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts opp\"}\n",
    "                    \n",
    "                    \n",
    "                else:\n",
    "                    stemTypes2Count,mirgene2stem = countStems(tokenBetween, geneword, mirword, keepNEU=True)\n",
    "                    print(\"counts between 1\", stemTypes2Count)\n",
    "                    regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                    \n",
    "                    if stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"] == 1 and stemTypes2Count[\"NEU\"] <= 2:\n",
    "                        regDir = \"MIR_GENE\"\n",
    "                        regStem = \"DOWN\"\n",
    "                        \n",
    "                        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts between equal\"}\n",
    "                    \n",
    "                    if stemTypes2Count[\"POS\"] == 2 and stemTypes2Count[\"NEG\"] == 1 and stemTypes2Count[\"NEU\"] <= 2:\n",
    "                        regDir = \"MIR_GENE\"\n",
    "                        regStem = \"DOWN\"\n",
    "                        \n",
    "                        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts between alternating\"}\n",
    "                    \n",
    "                    if regStem == \"NEU\":\n",
    "                        if stemTypes2Count[\"POS\"] > 0 and stemTypes2Count[\"POS\"] > stemTypes2Count[\"NEG\"]:\n",
    "                            regStem = \"POS\"\n",
    "                            \n",
    "                        elif stemTypes2Count[\"NEG\"] > 0 and stemTypes2Count[\"NEG\"] > stemTypes2Count[\"POS\"]:\n",
    "                            regStem = \"NEG\"\n",
    "                        \n",
    "                    \n",
    "                    #regStem = \"NEU\"\n",
    "\n",
    "                    if mirword.i < geneword.i:\n",
    "                        regDir = \"MIR_GENE\"\n",
    "                    else:\n",
    "                        regDir = \"GENE_MIR\"\n",
    "                        \n",
    "                    if mirword.i < geneword.i and \"-stressed\" in textBetween:\n",
    "                        regDir = \"GENE_MIR\"\n",
    "                        \n",
    "                    if mirword.i < geneword.i and str(doc[geneword.i-1]) == \"by\":\n",
    "                        regDir = \"GENE_MIR\"\n",
    "\n",
    "                        stemTypes2Count,mirgene2stem = countStems(tokenBetweenStrict, geneword, mirword)\n",
    "\n",
    "                        if len(stemTypes2Count) > 0:\n",
    "                            regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                        else:\n",
    "                            regStem = \"NEU\"\n",
    "                        print(\"counts between 2\", stemTypes2Count, regStem)\n",
    "                            \n",
    "                    elif geneword.i < mirword.i and str(doc[mirword.i-1]) == \"by\":\n",
    "                        regDir = \"MIR_GENE\"\n",
    "\n",
    "                        stemTypes2Count,mirgene2stem = countStems(tokenBetweenStrict, geneword, mirword)\n",
    "                        if len(stemTypes2Count) > 0:\n",
    "                            regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                        else:\n",
    "                            regStem = \"NEU\"\n",
    "                        print(\"counts between 2\", stemTypes2Count, regStem)\n",
    "                            \n",
    "                    elif isPassive and geneword.i < mirword.i:\n",
    "                        regDir = dir2opp[regDir]\n",
    "\n",
    "                    if stemTypes2Count[\"POS\"] > 0 and stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"]:\n",
    "                        regStem = \"NEU\"                       \n",
    "                        \n",
    "                    \n",
    "                        #regStem = idir2opp[regStem]\n",
    "                    \n",
    "\n",
    "                        \n",
    "                    if geneword.i < mirword.i:\n",
    "                        if str(geneword).endswith(\"-regulating\"):\n",
    "                            regStem = \"NEU\"\n",
    "                            \n",
    "                    if mirword.i < geneword.i and \"after the addition\" in textBetween:\n",
    "                        regStem = \"DOWN\"\n",
    "                        regDir = \"GENE_MIR\"\n",
    "                            \n",
    "                        \n",
    "                    if regStem == \"NEG\":\n",
    "                        regStem = \"DOWN\"\n",
    "                    elif regStem == \"POS\":\n",
    "                        regStem = \"UP\"\n",
    "\n",
    "                        \n",
    "                    \n",
    "                    return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts between \"}\n",
    "    \n",
    "        \n",
    "    regDir = \"MIR_GENE\"\n",
    "    regStem = \"NEU\"\n",
    "    \n",
    "    regeluateWords = [\"mediated\",]\n",
    "    \n",
    "    if mirword.i < geneword.i and \"mediated\" in str(mirword):\n",
    "        regStem = \"DOWN\"\n",
    "        \n",
    "    if mirword.i<geneword.i and str(doc[geneword.i-1]) == \"by\":\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"GENE_MIR\"\n",
    "        \n",
    "    if geneword.i < mirword.i:\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"GENE_MIR\"\n",
    "    \n",
    "    if geneword.i < mirword.i and str(doc[mirword.i-1]) == \"by\":\n",
    "        regStem = \"DOWN\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "        \n",
    "    if mirword.i < geneword.i and str(doc[mirword.i+1]) == \"target\":\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "    \n",
    "    #if mirword.i < geneword.i:\n",
    "    #    regDir = \"MIR_GENE\"\n",
    "    #else:\n",
    "    #    regDir = \"GENE_MIR\"\n",
    "    \n",
    "    #if isPassive:\n",
    "    #    if mirword.i < geneword.i:\n",
    "    #        regDir = \"GENE_MIR\"\n",
    "    #    else:\n",
    "    #        regDir = \"MIR_GENE\"\n",
    "        \n",
    "    return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"return\"}\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getstem(tkn):\n",
    "    tknstr = str(tkn).lower()\n",
    "    for sidx, s in enumerate(stem2class):\n",
    "        if tknstr.startswith(s):\n",
    "            \n",
    "            if tknstr in [\"enhancer\",\"inhibitor\"]:\n",
    "                continue\n",
    "\n",
    "            tokenStemClass = stem2class[s]\n",
    "            \n",
    "            if tokenStemClass == \"NEU\":\n",
    "                for i in range(1,2):\n",
    "\n",
    "                    if tkn.i-i<0:\n",
    "                        continue\n",
    "\n",
    "                    #tknPre = str(tkn.doc[tkn.i-i])\n",
    "                    #if tknPre.startswith((\"high\", \"elevat\", \"increas\")):\n",
    "                    #    tokenStemClass = \"POS\"\n",
    "\n",
    "                    #if tknPre.startswith((\"low\", \"decreas\")):\n",
    "                    #    tokenStemClass = \"NEG\"\n",
    "                                                \n",
    "            return tokenStemClass, s, True\n",
    "        \n",
    "        elif tknstr.endswith((\"-stressed\")):\n",
    "            return \"POS\", s, True\n",
    "        \n",
    "    return \"NEU\", str(tkn), False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading stems\n",
      "NEG\n",
      "True\n",
      "expressed ('NEU', 'express', True)\n",
      ", ('NEU', ',', False)\n",
      "miR-34a ('NEU', 'miR-34a', False)\n",
      ", ('NEU', ',', False)\n",
      "was ('NEU', 'was', False)\n",
      "studied ('NEU', 'studied', False)\n",
      "further ('NEU', 'further', False)\n",
      "because ('NEU', 'because', False)\n",
      "sequence ('NEU', 'sequence', False)\n",
      "analysis ('NEU', 'analysis', False)\n",
      "suggested ('NEU', 'suggested', False)\n",
      "a ('NEU', 'a', False)\n",
      "likely ('NEU', 'likely', False)\n",
      "interaction ('NEU', 'interact', True)\n",
      "with ('NEU', 'with', False)\n",
      "the ('NEU', 'the', False)\n",
      "3'-untranslated ('NEU', \"3'-untranslated\", False)\n",
      "region ('NEU', 'region', False)\n",
      "of ('NEU', 'of', False)\n",
      "bcl2 ('NEU', 'bcl2', False)\n",
      "mRNA ('NEU', 'mRNA', False)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(Counter(),\n",
       " defaultdict(<function __main__.countStems.<locals>.<lambda>()>, {}))"
      ]
     },
     "execution_count": 218,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allStems, allRels, stem2class = loadStems()\n",
    "\n",
    "print(stem2class[\"underexpress\"])\n",
    "print(\"underexpression\".startswith(\"underexpress\"))\n",
    "\n",
    "doct = nlp(\"expressed , miR-34a , was studied further because sequence analysis suggested a likely interaction with the 3'-untranslated region of bcl2 mRNA\")\n",
    "\n",
    "toks = [x for x in doct]\n",
    "\n",
    "for tok in toks:\n",
    "    print(tok, getstem(tok))\n",
    "\n",
    "countStems(toks, toks[0], toks[-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading stems\n",
      "['miRNA-corp.d1.s0.p2', 'miRNA-corp.d1.s0.p3', 'miRNA-corp.d1.s0.p4', 'miRNA-corp.d1.s1.p0', 'miRNA-corp.d1.s1.p1']\n",
      "Loading stems\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "down-regulation of miRNA-128 may contribute to glioma and GBM , in part , by coordinately up-regulating ARP5 ( ANGPTL6\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f6533622268>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "down-regulation of miRNA-128 may contribute to glioma and GBM , in part , by coordinately up-regulating ARP5 ( ANGPTL6 ) ,\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1, 'POS': 1})\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "down-regulation of miRNA-128 may contribute to glioma and GBM , in part , by coordinately up-regulating ARP5 ( ANGPTL6 ) , Bmi-1 and E2F-3a\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1, 'POS': 1})\n",
      "Counter({'POS': 1})\n",
      "of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines ( a ) restored ' homeostatic ' ARP5 ( ANGPTL6\n",
      "Counter({'POS': 1})\n",
      "of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines ( a ) restored ' homeostatic ' ARP5 ( ANGPTL6 ) ,\n",
      "Counter({'POS': 1})\n",
      "of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines ( a ) restored ' homeostatic ' ARP5 ( ANGPTL6 ) , Bmi-1 and E2F-3a\n",
      "Counter()\n",
      "Three bioinformatics-verified miRNA-128 targets , angiopoietin-related growth factor protein 5 ( ARP5\n",
      "Counter()\n",
      "Three bioinformatics-verified miRNA-128 targets , angiopoietin-related growth factor protein 5 ( ARP5 ; ANGPTL6\n",
      "Counter()\n",
      "Three bioinformatics-verified miRNA-128 targets , angiopoietin-related growth factor protein 5 ( ARP5 ; ANGPTL6 ) ,\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "Three bioinformatics-verified miRNA-128 targets , angiopoietin-related growth factor protein 5 ( ARP5 ; ANGPTL6 ) , a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation , Bmi-1 , and\n",
      "counts between 1 Counter({'NEU': 3, 'NEG': 3, 'POS': 1})\n",
      "Counter({'NEG': 1})\n",
      "Micro-RNA-128 ( miRNA-128 ) down-regulation in glioblastoma targets ARP5 ( ANGPTL6\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1})\n",
      "Micro-RNA-128 ( miRNA-128 ) down-regulation in glioblastoma targets ARP5 ( ANGPTL6 ) ,\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1})\n",
      "Micro-RNA-128 ( miRNA-128 ) down-regulation in glioblastoma targets ARP5 ( ANGPTL6 ) , Bmi-1 and E2F-3a\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1})\n",
      "Micro-RNA-128 ( miRNA-128 ) down-regulation in glioblastoma targets ARP5 ( ANGPTL6\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1})\n",
      "Micro-RNA-128 ( miRNA-128 ) down-regulation in glioblastoma targets ARP5 ( ANGPTL6 ) ,\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1})\n",
      "Micro-RNA-128 ( miRNA-128 ) down-regulation in glioblastoma targets ARP5 ( ANGPTL6 ) , Bmi-1 and E2F-3a\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "Reduction of miR-21 induces glioma cell apoptosis via activating caspase 9 and 3\n",
      "counts between 1 Counter({'NEG': 1, 'POS': 1, 'NEU': 1})\n",
      "Counter()\n",
      "activating caspase 9 and 3 . Extensive data indicate that miR-21 plays a\n",
      "Counter()\n",
      "activating caspase 9 and 3 . Extensive data indicate that miR-21 plays a critical role in gliomagenesis , however , knowledge is limited on the mechanism of action of miR-21 , including\n",
      "Counter()\n",
      "reduction of miR-21 activated caspase 9 and 3\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter()\n",
      "reduction of miR-21 activated caspase 9 and 3 , which may be mediated by modulating multiple potential target genes , such as TIMP3\n",
      "counts between 1 Counter({'NEU': 4, 'NEG': 1})\n",
      "Counter({'NEG': 1})\n",
      "affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and\n",
      "counts between 1 Counter({'NEU': 2, 'POS': 1})\n",
      "The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration. \n",
      "('miR-433', 'Specific_miRNAs', (33, 40)) ('FGF20', 'Genes/Proteins', (81, 86))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which\n",
      "counts between 1 Counter({'NEU': 2, 'POS': 2})\n",
      "The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration. \n",
      "('miR-433', 'Specific_miRNAs', (33, 40)) ('alpha-synuclein', 'Genes/Proteins', (118, 133))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "FGF20 rs12720208 SNP and microRNA-433 variation\n",
      "Counter()\n",
      "of the apl-1 transcription are modulated by the activity of let-7 family microRNAs\n",
      "Counter()\n",
      "C. elegans apl-1 shows significant genetic interactions with let-7 family microRNAs\n",
      "Counter()\n",
      "interactions with let-7 family microRNAs and let-7-targeted heterochronic genes , hbl-1 , lin-41\n",
      "Counter()\n",
      "interactions with let-7 family microRNAs and let-7-targeted heterochronic genes , hbl-1 , lin-41 and lin-42\n",
      "Counter()\n",
      "C. elegans apl-1 shows significant genetic interactions with let-7 family microRNAs and let-7-targeted heterochronic genes\n",
      "Counter()\n",
      "microRNAs and let-7-targeted heterochronic genes , hbl-1 , lin-41\n",
      "Counter()\n",
      "microRNAs and let-7-targeted heterochronic genes , hbl-1 , lin-41 and lin-42\n",
      "Counter()\n",
      "SRF and myocardin regulate a cardiovascular-specific microRNA ( miRNA ) cluster encoding miR-143 and miR-145\n",
      "Counter()\n",
      "SRF and myocardin regulate a cardiovascular-specific microRNA ( miRNA ) cluster encoding miR-143 and miR-145\n",
      "Counter()\n",
      "SRF and myocardin regulate a cardiovascular-specific microRNA ( miRNA ) cluster encoding miR-143 and miR-145\n",
      "Counter()\n",
      "SRF and myocardin regulate a cardiovascular-specific microRNA ( miRNA ) cluster encoding miR-143 and miR-145\n",
      "Counter()\n",
      "modulators of SRF activity and actin dynamics by miR-143 and miR-145\n",
      "These abnormalities reflect the regulation of a cadre of modulators of SRF activity and actin dynamics by miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (106, 113)) ('SRF', 'Genes/Proteins', (71, 74))\n",
      "('MIR_GENE', 'DOWN') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "modulators of SRF activity and actin dynamics by miR-143 and miR-145\n",
      "These abnormalities reflect the regulation of a cadre of modulators of SRF activity and actin dynamics by miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (118, 125)) ('SRF', 'Genes/Proteins', (71, 74))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "Thus , miR-143 and miR-145 act as integral components of the regulatory network whereby SRF controls cytoskeletal\n",
      "Counter()\n",
      "miR-143 and miR-145 act as integral components of the regulatory network whereby SRF controls cytoskeletal\n",
      "Counter({'NEG': 1})\n",
      "an miR-30-based shRNA construct prevents Oas1 pathway associated\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'POS': 1})\n",
      "naturally occurring microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in\n",
      "counts between 1 Counter({'POS': 1})\n",
      "Counter({'POS': 1})\n",
      "microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in\n",
      "counts between 1 Counter({'POS': 1})\n",
      "Counter({'NEG': 1})\n",
      "Downregulation of Pdcd4 by mir-21 facilitates glioblastoma\n",
      "counts between 1 Counter({'NEG': 1})\n",
      "counts between 2 Counter() NEU\n",
      "Downregulation of Pdcd4 by mir-21 facilitates glioblastoma proliferation in vivo.  \n",
      "('mir-21', 'Specific_miRNAs', (27, 33)) ('Pdcd4', 'Genes/Proteins', (18, 23))\n",
      "('MIR_GENE', 'NEU') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1})\n",
      "downregulation of mir-21 in glioblastoma-derived cell lines results in increased expression of its target , programmed cell death 4 ( Pdcd4\n",
      "counts between 1 Counter({'NEU': 3, 'NEG': 1, 'POS': 1})\n",
      "We demonstrate that downregulation of mir-21 in glioblastoma-derived cell lines results in increased expression of its target, programmed cell death 4 (Pdcd4), a known tumor-suppressor gene. \n",
      "('mir-21', 'Specific_miRNAs', (38, 44)) ('programmed cell death 4', 'Genes/Proteins', (127, 150))\n",
      "('MIR_GENE', 'NEU') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1})\n",
      "downregulation of mir-21 in glioblastoma-derived cell lines results in increased expression of its target , programmed cell death 4 ( Pdcd4 ) ,\n",
      "counts between 1 Counter({'NEU': 3, 'NEG': 1, 'POS': 1})\n",
      "We demonstrate that downregulation of mir-21 in glioblastoma-derived cell lines results in increased expression of its target, programmed cell death 4 (Pdcd4), a known tumor-suppressor gene. \n",
      "('mir-21', 'Specific_miRNAs', (38, 44)) ('Pdcd4', 'Genes/Proteins', (152, 157))\n",
      "('MIR_GENE', 'NEU') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1})\n",
      "in which mir-21 levels are downregulated or Pdcd4 is over-expressed\n",
      "counts between 1 Counter({'NEU': 1, 'NEG': 1, 'POS': 1})\n",
      "Counter({'NEG': 1})\n",
      "downregulation of mir-21 or overexpression of its target , Pdcd4 , in\n",
      "counts between 1 Counter({'NEG': 1, 'POS': 1, 'NEU': 1})\n",
      "Counter()\n",
      "linkage between mir-21 and Pdcd4\n",
      "Counter({'NEG': 1})\n",
      "linkage between mir-21 and Pdcd4 and further elucidate the molecular mechanisms by which the known high level of mir-21 expression in glioblastoma can attribute to tumorigenesis -- namely , inhibition of Pdcd4 and its\n",
      "counts between 1 Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter()\n",
      "mir-21 and Pdcd4 and further elucidate the molecular mechanisms by which the known high level of mir-21 expression in\n",
      "Counter({'NEG': 1})\n",
      "level of mir-21 expression in glioblastoma can attribute to tumorigenesis -- namely , inhibition of Pdcd4 and its\n",
      "counts between 1 Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter()\n",
      "identified that miR-145-mediated phenotypic modulation of VSMCs is through its target gene KLF5 and its\n",
      "We have further identified that miR-145-mediated phenotypic modulation of VSMCs is through its target gene KLF5 and its downstream signaling molecule, myocardin. \n",
      "('miR-145', 'Specific_miRNAs', (32, 39)) ('KLF5', 'Genes/Proteins', (107, 111))\n",
      "('MIR_GENE', 'DOWN') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "identified that miR-145-mediated phenotypic modulation of VSMCs is through its target gene KLF5 and its downstream signaling molecule , myocardin\n",
      "Counter()\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets ; IQ motif containing GTPase activating protein 1 ( IQGAP1 ) ,\n",
      "Counter()\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets ; IQ motif containing GTPase activating protein 1 ( IQGAP1\n",
      "Counter()\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets ; IQ motif containing GTPase activating protein 1 ( IQGAP1 ) , laminin γ1 ( LAMC1\n",
      "Counter()\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets ; IQ motif containing GTPase activating protein 1 ( IQGAP1 ) , laminin γ1 ( LAMC1 ) and\n",
      "Counter()\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets ; IQ motif containing GTPase activating protein 1 ( IQGAP1 ) , laminin γ1 ( LAMC1 ) and integrin β1 ( ITGB1\n",
      "Counter()\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets ; IQ motif containing GTPase activating protein 1 ( IQGAP1 ) , laminin γ1 ( LAMC1 ) and integrin β1 ( ITGB1 ) were\n",
      "Counter()\n",
      "levels of MCP-2 and increased levels of mir-146a were observed\n",
      "counts between 1 Counter({'NEU': 2, 'POS': 1})\n",
      "Counter()\n",
      "levels of MCP-2 and increased levels of mir-146a were observed , suggesting a role for mir-146a in the\n",
      "counts between 1 Counter({'NEU': 2, 'POS': 1})\n",
      "Counter()\n",
      "show that CCL8/MCP-2 is a target for mir-146a in HIV-1\n",
      "Counter()\n",
      "show that CCL8/MCP-2 is a target for mir-146a in HIV-1\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "target for mir-146a in HIV-1 infected microglia , as overexpression of mir-146a prevented HIV-induced secretion of MCP-2 chemokine .\n",
      "counts between 1 Counter({'POS': 2, 'NEU': 1, 'NEG': 1})\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "show that CCL8/MCP-2 is a target for mir-146a in HIV-1 infected microglia , as overexpression of mir-146a prevented HIV-induced\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "show that CCL8/MCP-2 is a target for mir-146a in HIV-1 infected microglia , as overexpression of mir-146a prevented HIV-induced\n",
      "Counter({'POS': 2, 'NEG': 1})\n",
      "overexpression of mir-146a prevented HIV-induced secretion of MCP-2 chemokine\n",
      "counts between 1 Counter({'POS': 2, 'NEG': 1})\n",
      "Counter()\n",
      "CCL8/MCP-2 is a target for mir-146a in HIV-1-infected\n",
      "CCL8/MCP-2 is a target for mir-146a in HIV-1-infected human microglial cells.  \n",
      "('mir-146a', 'Specific_miRNAs', (27, 35)) ('CCL8', 'Genes/Proteins', (0, 4))\n",
      "('MIR_GENE', 'NEU') between gene mir passive? False\n",
      "('GENE_MIR', 'UP', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "CCL8/MCP-2 is a target for mir-146a in HIV-1-infected\n",
      "CCL8/MCP-2 is a target for mir-146a in HIV-1-infected human microglial cells.  \n",
      "('mir-146a', 'Specific_miRNAs', (27, 35)) ('CCL8', 'Genes/Proteins', (5, 10))\n",
      "('MIR_GENE', 'NEU') between gene mir passive? False\n",
      "('GENE_MIR', 'UP', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "targets for mir-146a , the chemokine CCL8/MCP-2 is a\n",
      "Counter()\n",
      "targets for mir-146a , the chemokine CCL8/MCP-2 is a\n",
      "Counter()\n",
      "targets for mir-146a , the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a\n",
      "Counter()\n",
      "targets for mir-146a , the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry\n",
      "Counter()\n",
      "targets for mir-146a , the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry\n",
      "Counter({'NEG': 1})\n",
      "Downregulation of ABCG2 expression in glioblastoma cancer stem cells with miRNA-328 may decrease\n",
      "counts between 1 Counter({'NEG': 2, 'NEU': 1})\n",
      "Downregulation of ABCG2 expression in glioblastoma cancer stem cells with miRNA-328 may decrease their chemoresistance.  \n",
      "('miRNA-328', 'Specific_miRNAs', (74, 83)) ('ABCG2', 'Genes/Proteins', (18, 23))\n",
      "('GENE_MIR', 'DOWN') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "Thus, we hypothesize that modulating ABCG2 expression by targeting miRNA-328 in glioblastoma cancer stem cells could represent a promising strategy for therapeutic manipulation to increase the efficacy of chemotherapeutic agents for glioblastoma, a highly lethal type of cancer.   \n",
      "('miRNA-328', 'Specific_miRNAs', (67, 76)) ('ABCG2', 'Genes/Proteins', (37, 42))\n",
      "('MIR_GENE', 'NEU') compartment gene mir passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "shown that ABCG2 is the target gene for miRNA-328\n",
      "Counter()\n",
      "these is miR-9 , which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of STAT3 .\n",
      "counts between 1 Counter({'NEG': 3, 'NEU': 2})\n",
      "Counter({'POS': 1})\n",
      "Co-suppression of miR-221/222 directly resulted in the up-regulation of p27Kip1 in the\n",
      "counts between 1 Counter({'NEU': 1, 'POS': 1})\n",
      "Co-suppression of miR-221/222 directly resulted in the up-regulation of p27Kip1 in the tested cells, consequently, affects their growth potential by reducing a G1 to S shift in the cell cycle. \n",
      "('miR-221/222', 'Specific_miRNAs', (18, 29)) ('p27Kip1', 'Genes/Proteins', (72, 79))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'POS': 1})\n",
      "Consistently , miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251\n",
      "counts between 1 Counter({'POS': 1})\n",
      "Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1. \n",
      "('miR-221/222', 'Specific_miRNAs', (14, 25)) ('p27Kip1', 'Genes/Proteins', (91, 98))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "Consistently , miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1 .\n",
      "counts between 1 Counter({'POS': 1, 'NEG': 1, 'NEU': 1})\n",
      "Counter()\n",
      "suggest that miR-221/222 is a regulator of the tumor suppressor gene p27Kip1\n",
      "counts between 1 Counter({'NEU': 1, 'NEG': 1})\n",
      "Our results suggest that miR-221/222 is a regulator of the tumor suppressor gene p27Kip1, and co-suppression of miR-221/222 expression in advanced gliomas may inhibit glioma cell proliferation by a mechanism involving the up-regulation of p27Kip1 in vitro and in vivo.   \n",
      "('miR-221/222', 'Specific_miRNAs', (25, 36)) ('p27Kip1', 'Genes/Proteins', (81, 88))\n",
      "('MIR_GENE', 'DOWN') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "suggest that miR-221/222 is a regulator of the tumor suppressor gene p27Kip1 , and co-suppression of miR-221/222 expression in advanced gliomas may inhibit glioma cell proliferation by a mechanism involving the up-regulation of p27Kip1 in vitro\n",
      "counts between 1 Counter({'NEU': 2, 'NEG': 2, 'POS': 1})\n",
      "Counter({'NEG': 1})\n",
      "suppressor gene p27Kip1 , and co-suppression of miR-221/222 expression in\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "co-suppression of miR-221/222 expression in advanced gliomas may inhibit glioma cell proliferation by a mechanism involving the up-regulation of p27Kip1 in vitro\n",
      "counts between 1 Counter({'NEU': 1, 'NEG': 1, 'POS': 1})\n",
      "Counter({'NEG': 1})\n",
      "Co-suppression of miR-221/222 cluster suppresses human glioma cell growth by targeting p27kip1 in vitro and in vivo.  \n",
      "('miR-221/222', 'Specific_miRNAs', (18, 29)) ('p27kip1', 'Genes/Proteins', (87, 94))\n",
      "('MIR_GENE', 'NEU') compartment mir gene passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "expression of miR-221/miR-222 and the cell cycle inhibitor p27Kip1 was identified\n",
      "Counter()\n",
      "expression of miR-221/miR-222 and the cell cycle inhibitor p27Kip1 was identified\n",
      "Counter()\n",
      "loop between miR-21 and STAT3 might provide\n",
      "Counter()\n",
      "loop between miR-21 and STAT3 might provide an insight into the mechanism of modulating EGFR/STAT3 signaling\n",
      "Counter()\n",
      "loop between miR-21 and STAT3 might provide an insight into the mechanism of modulating EGFR/STAT3 signaling\n",
      "Counter({'NEG': 1})\n",
      "miR-29b downregulates the expression of luciferase through hPGRN or mouse PGRN ( mPGRN\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 2})\n",
      "or mouse PGRN ( mPGRN ) 3'UTRs , and the regulation was abolished by mutations in the miR-29b binding site\n",
      "counts between 1 Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter()\n",
      "site for microRNA-29b ( miR-29b ) in the 3 ' untranslated region ( 3'UTR ) of the human PGRN ( hPGRN\n",
      "Counter()\n",
      "microRNA-29b ( miR-29b ) in the 3 ' untranslated region ( 3'UTR ) of the human PGRN ( hPGRN\n",
      "Counter()\n",
      "MicroRNA-29b regulates the expression level of human progranulin , a\n",
      "Counter()\n",
      "findings identify miR-29b as a novel posttranscriptional regulator of PGRN expression\n",
      "Counter()\n",
      "regulator of PGRN expression , raising the possibility that miR-29b or other\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "overexpression of miR-21 protects against ischemic neuronal death , and that downregulation of FASLG , a\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f6535a23b70>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "overexpression of miR-21 protects against ischemic neuronal death , and that downregulation of FASLG , a tumor necrosis factor-α family member\n",
      "counts between 1 Counter({'POS': 1, 'NEG': 1})\n",
      "Counter()\n",
      "downregulation of FASLG , a tumor necrosis factor-α family member and an important cell death-inducing ligand whose gene is targeted by miR-21 , probably\n",
      "Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor-α family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect. \n",
      "('miR-21', 'Specific_miRNAs', (234, 240)) ('FASLG', 'Genes/Proteins', (117, 122))\n",
      "('MIR_GENE', 'NEU') between gene mir passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "tumor necrosis factor-α family member and an important cell death-inducing ligand whose gene is targeted by miR-21 , probably\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "overexpression of miR-21 in neurons significantly reduced FASLG levels , and introduction of an miR-21 mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of Faslg .\n",
      "counts between 1 Counter({'NEU': 3, 'NEG': 2, 'POS': 1})\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "overexpression of miR-21 in neurons significantly reduced FASLG levels\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f6533622268>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Counter({'NEG': 1})\n",
      "of an miR-21 mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of Faslg\n",
      "counts between 1 Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter({'NEG': 2})\n",
      "significantly reduced FASLG levels , and introduction of an miR-21 mimic into\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'POS': 1})\n",
      "lesions ( microRNA-34a , microRNA-155 and microRNA-326 ) targeted the 3'-untranslated region of CD47 in reporter\n",
      "Counter({'POS': 1})\n",
      "microRNA-34a , microRNA-155 and microRNA-326 ) targeted the 3'-untranslated region of CD47 in reporter\n",
      "Counter({'POS': 1})\n",
      "microRNA-155 and microRNA-326 ) targeted the 3'-untranslated region of CD47 in reporter\n",
      "Counter()\n",
      "region of CD47 in reporter assays , with microRNA-155 even at\n",
      "Counter()\n",
      "The miRNA miR-23a was predicted to bind XIAP mRNA\n",
      "Counter()\n",
      "miR-23a directly bound the 3 ' UTR of XIAP , and\n",
      "Counter({'POS': 1})\n",
      "miR-23a directly bound the 3 ' UTR of XIAP , and miR-23a inhibition led to an increase in XIAP mRNA in\n",
      "counts between 1 Counter({'NEU': 1, 'NEG': 1, 'POS': 1})\n",
      "Counter()\n",
      "miR-23a directly bound the 3 ' UTR of XIAP , and miR-23a inhibition led to an increase in XIAP mRNA in vitro , demonstrating that XIAP is a\n",
      "counts between 1 Counter({'NEU': 1, 'NEG': 1, 'POS': 1})\n",
      "Counter()\n",
      "UTR of XIAP , and miR-23a\n",
      "Counter({'POS': 1})\n",
      ", and miR-23a inhibition led to an increase in XIAP mRNA in\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f6533622268>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "Counter()\n",
      ", and miR-23a inhibition led to an increase in XIAP mRNA in vitro , demonstrating that XIAP is a\n",
      "counts between 1 Counter({'NEG': 1, 'POS': 1})\n",
      "Counter()\n",
      "UTR of XIAP , and miR-23a inhibition led to an increase in XIAP mRNA in vitro , demonstrating that XIAP is a previously uncharacterized target for miR-23a .\n",
      "Counter({'POS': 1})\n",
      "increase in XIAP mRNA in vitro , demonstrating that XIAP is a previously uncharacterized target for miR-23a .\n",
      "Counter()\n",
      "demonstrating that XIAP is a previously uncharacterized target for miR-23a\n",
      "Counter()\n",
      "miR-23a regulation of X-linked inhibitor of apoptosis ( XIAP\n",
      "Counter()\n",
      "miR-23a regulation of X-linked inhibitor of apoptosis ( XIAP ) contributes\n",
      "Counter({'NEG': 1})\n",
      "Transfection of let-7 miRNA reduced expression of pan-RAS , N-RAS\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1})\n",
      "Transfection of let-7 miRNA reduced expression of pan-RAS , N-RAS , and\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1})\n",
      "Transfection of let-7 miRNA reduced expression of pan-RAS , N-RAS , and K-RAS in the\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter()\n",
      "α-actin , SM22 , and calponin ) and muscle-enriched microRNAs ( miR-143/145 and miR-1/133a\n",
      "Counter()\n",
      ", and calponin ) and muscle-enriched microRNAs ( miR-143/145 and miR-1/133a\n",
      "Counter({'NEG': 1})\n",
      "activity of myocardin as indicated by reduced expression of SMC contractile marker genes ( SM α-actin , SM22 , and calponin ) and muscle-enriched microRNAs ( miR-143/145 and miR-1/133a\n",
      "counts between 1 Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter({'NEG': 1})\n",
      "microRNAs ( miR-143/145 and miR-1/133a ) , and reduced contractility of human vascular smooth muscle cells ( SMCs ) embedded in collagen gel lattices , suggesting that myocardin degradation is\n",
      "counts between 1 Counter({'NEG': 2})\n",
      "Counter({'POS': 1})\n",
      "accumulation of myocardin in response to proteasome inhibition by MG132 or lactacystin resulted in decrease of transcriptional activity of myocardin as indicated by reduced expression of SMC contractile marker genes ( SM α-actin , SM22 , and calponin ) and muscle-enriched microRNAs ( miR-143/145 and miR-1/133a\n",
      "counts between 1 Counter({'NEU': 5, 'NEG': 3, 'POS': 1})\n",
      "Counter()\n",
      "α-actin , SM22 , and calponin ) and muscle-enriched microRNAs ( miR-143/145 and miR-1/133a\n",
      "Counter()\n",
      ", and calponin ) and muscle-enriched microRNAs ( miR-143/145 and miR-1/133a\n",
      "Counter({'NEG': 1})\n",
      "activity of myocardin as indicated by reduced expression of SMC contractile marker genes ( SM α-actin , SM22 , and calponin ) and muscle-enriched microRNAs ( miR-143/145 and miR-1/133a ) ,\n",
      "counts between 1 Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter({'NEG': 1})\n",
      "miR-143/145 and miR-1/133a ) , and reduced contractility of human vascular smooth muscle cells ( SMCs ) embedded in collagen gel lattices , suggesting that myocardin degradation is\n",
      "counts between 1 Counter({'NEG': 2})\n",
      "Counter({'POS': 1})\n",
      "accumulation of myocardin in response to proteasome inhibition by MG132 or lactacystin resulted in decrease of transcriptional activity of myocardin as indicated by reduced expression of SMC contractile marker genes ( SM α-actin , SM22 , and calponin ) and muscle-enriched microRNAs ( miR-143/145 and miR-1/133a ) ,\n",
      "counts between 1 Counter({'NEU': 5, 'NEG': 3, 'POS': 1})\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f6539d93378>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Counter({'POS': 2, 'NEG': 1})\n",
      "Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels , whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-synuclein 3'-untranslated region\n",
      "counts between 1 Counter({'POS': 2, 'NEG': 2, 'NEU': 2})\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'POS': 2, 'NEG': 1})\n",
      "mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels , whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-synuclein 3'-untranslated region\n",
      "counts between 1 Counter({'NEG': 2, 'NEU': 2, 'POS': 1})\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "reduces endogenous alpha-synuclein levels , whereas inhibition of mir-7 and mir-153\n",
      "counts between 1 Counter({'NEG': 2, 'NEU': 1})\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-synuclein 3'-untranslated region\n",
      "counts between 1 Counter({'NEG': 1, 'POS': 1, 'NEU': 1})\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "reduces endogenous alpha-synuclein levels , whereas inhibition of mir-7 and mir-153 enhances translation\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f653a321598>, {'gene': Counter({'NEG': 1}), 'mir': Counter({'POS': 1})})\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-synuclein 3'-untranslated region\n",
      "counts between 1 Counter({'POS': 1, 'NEU': 1})\n",
      "Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-synuclein 3'-untranslated region in primary neurons. \n",
      "('mir-153', 'Specific_miRNAs', (125, 132)) ('alpha-synuclein', 'Genes/Proteins', (192, 207))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      ", namely mir-7 and mir-153 , have been identified to regulate alpha-synuclein levels post-transcriptionally\n",
      "Counter()\n",
      "mir-7 and mir-153 , have been identified to regulate alpha-synuclein levels post-transcriptionally\n",
      "Counter()\n",
      "regulation of alpha-synuclein expression by mir-7 and mir-153\n",
      "Post-transcriptional regulation of alpha-synuclein expression by mir-7 and mir-153.  \n",
      "('mir-7', 'Specific_miRNAs', (65, 70)) ('alpha-synuclein', 'Genes/Proteins', (35, 50))\n",
      "('MIR_GENE', 'DOWN') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "regulation of alpha-synuclein expression by mir-7 and mir-153\n",
      "Post-transcriptional regulation of alpha-synuclein expression by mir-7 and mir-153.  \n",
      "('mir-153', 'Specific_miRNAs', (75, 82)) ('alpha-synuclein', 'Genes/Proteins', (35, 50))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1})\n",
      "rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding ,\n",
      "counts between 1 Counter({'NEU': 2, 'NEG': 1})\n",
      "We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele. \n",
      "('miR-211', 'Specific_miRNAs', (94, 101)) ('angiopoietin-1', 'Genes/Proteins', (51, 65))\n",
      "('GENE_MIR', 'DOWN') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "in the miR-211-binding site of angiopoietin-1\n",
      "Counter()\n",
      "3'-UTR of angiopoietin-1 using bioinformatics ; variant rs2507800 was identified to be located in the miR-211-binding site of\n",
      "Counter({'NEG': 2})\n",
      "3'-UTR of angiopoietin-1 might reduce stroke risk by interfering with the binding efficiency of microRNA 211\n",
      "counts between 1 Counter({'NEG': 2, 'NEU': 1})\n",
      "A functional variant in the 3'-UTR of angiopoietin-1 might reduce stroke risk by interfering with the binding efficiency of microRNA 211.  \n",
      "('microRNA 211', 'Specific_miRNAs', (124, 136)) ('angiopoietin-1', 'Genes/Proteins', (38, 52))\n",
      "('GENE_MIR', 'DOWN') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 2})\n",
      "3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding\n",
      "counts between 1 Counter({'NEG': 2, 'NEU': 1})\n",
      "Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.   \n",
      "('miR-211', 'Specific_miRNAs', (146, 153)) ('angiopoietin-1', 'Genes/Proteins', (79, 93))\n",
      "('GENE_MIR', 'DOWN') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "miR-107 contributes to GRN expression regulation\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "decreased miR-107 with augmentation of neuronal GRN expression\n",
      "counts between 1 Counter({'NEG': 1, 'POS': 1, 'NEU': 1})\n",
      "Counter()\n",
      "showed that GRN is the strongest target for miR-107 in human\n",
      "Counter()\n",
      "miR-107 has been implicated in Alzheimer 's disease pathogenesis , and sequence elements in the open reading frame-rather than the 3 ' untranslated region-of GRN mRNA are\n",
      "Counter()\n",
      "untranslated region-of GRN mRNA are recognized by miR-107\n",
      "miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of GRN mRNA are recognized by miR-107 and are highly conserved among vertebrate species. \n",
      "('miR-107', 'Specific_miRNAs', (182, 189)) ('GRN', 'Genes/Proteins', (155, 158))\n",
      "('MIR_GENE', 'DOWN') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "miR-107 regulates granulin/progranulin with implications\n",
      "Counter()\n",
      "miR-107 regulates granulin/progranulin with implications\n",
      "Counter()\n",
      "with Argonaute 2 . In vitro and in vivo studies indicate that regulation of GRN by miR-107 may be\n",
      "Counter()\n",
      "GRN mRNA interacts preferentially with Argonaute 2 . In vitro and in vivo studies indicate that regulation of GRN by miR-107 may be\n",
      "Counter()\n",
      "regulation of GRN by miR-107 may be\n",
      "GRN mRNA interacts preferentially with Argonaute 2. In vitro and in vivo studies indicate that regulation of GRN by miR-107 may be functionally important. \n",
      "('miR-107', 'Specific_miRNAs', (116, 123)) ('GRN', 'Genes/Proteins', (109, 112))\n",
      "('MIR_GENE', 'DOWN') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "to increased miR-107 levels also results in decreased GRN expression ,\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f653a321598>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Counter({'POS': 1})\n",
      "to increased miR-107 levels also results in decreased GRN expression , including changes in cell compartmentation and decreased secretion of GRN protein\n",
      "counts between 1 Counter({'NEU': 4, 'POS': 2, 'NEG': 2})\n",
      "Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein. \n",
      "('miR-107', 'Specific_miRNAs', (66, 73)) ('GRN', 'Genes/Proteins', (193, 196))\n",
      "('MIR_GENE', 'NEU') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1})\n",
      "MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1})\n",
      "MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide by decreasing Bax/Bcl-2 ratio and\n",
      "counts between 1 Counter({'NEG': 1})\n",
      "Counter({'NEG': 1})\n",
      "MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide by decreasing Bax/Bcl-2 ratio and\n",
      "counts between 1 Counter({'NEG': 1})\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "cells with miR-21 overexpression before , which appeared to downregulate the Bax expression\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f653a321730>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Counter({'POS': 2, 'NEG': 1})\n",
      "cells with miR-21 overexpression before , which appeared to downregulate the Bax expression , upregulate the Bcl-2 expression and\n",
      "counts between 1 Counter({'POS': 2, 'NEU': 2, 'NEG': 1})\n",
      "Counter({'POS': 2, 'NEG': 2})\n",
      "cells with miR-21 overexpression before , which appeared to downregulate the Bax expression , upregulate the Bcl-2 expression and decrease caspase-3 activity .\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f6539d931e0>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Counter({'NEG': 2})\n",
      "that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells , at least in part , by decreasing Bax/Bcl-2 ratio and\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f65336221e0>, {'mir': Counter({'POS': 1, 'NEG': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Counter({'NEG': 2})\n",
      "that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells , at least in part , by decreasing Bax/Bcl-2 ratio and\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f653a321a60>, {'mir': Counter({'POS': 1, 'NEG': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Counter({'NEG': 2})\n",
      "that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells , at least in part , by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "counts between 1 Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "by decreasing Bax/Bcl-2 ratio and caspase-3 activity , which highlighted the possibility of miR-21 overexpression in\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f653a3af950>, {'gene': Counter({'NEG': 1}), 'mir': Counter({'POS': 1})})\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "by decreasing Bax/Bcl-2 ratio and caspase-3 activity , which highlighted the possibility of miR-21 overexpression in\n",
      "defaultdict(<function countStems.<locals>.<lambda> at 0x7f6533622268>, {'gene': Counter({'NEG': 1}), 'mir': Counter({'POS': 1})})\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "ratio and caspase-3 activity , which highlighted the possibility of miR-21 overexpression in\n",
      "counts between 1 Counter({'NEU': 1, 'POS': 1})\n",
      "Counter()\n",
      "in vivo ephrin-A3 and neuronal pentraxin 1 are potentially regulated by miR-210\n",
      "Hypoxia induces microRNA miR-210 in vitro and in vivo ephrin-A3 and neuronal pentraxin 1 are potentially regulated by miR-210.  \n",
      "('miR-210', 'Specific_miRNAs', (118, 125)) ('ephrin-A3', 'Genes/Proteins', (54, 63))\n",
      "('MIR_GENE', 'DOWN') return passive? True\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "neuronal pentraxin 1 are potentially regulated by miR-210\n",
      "Hypoxia induces microRNA miR-210 in vitro and in vivo ephrin-A3 and neuronal pentraxin 1 are potentially regulated by miR-210.  \n",
      "('miR-210', 'Specific_miRNAs', (118, 125)) ('neuronal pentraxin 1', 'Genes/Proteins', (68, 88))\n",
      "('MIR_GENE', 'DOWN') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'POS': 1})\n",
      "induces microRNA miR-210 in vitro and in vivo ephrin-A3 and neuronal\n",
      "counts between 1 Counter({'POS': 1})\n",
      "Counter({'POS': 1})\n",
      "induces microRNA miR-210 in vitro and in vivo ephrin-A3 and neuronal pentraxin 1 are potentially\n",
      "counts between 1 Counter({'POS': 1})\n",
      "Counter({'POS': 1})\n",
      "of the HIF-1alpha induced the expression of miR-210\n",
      "counts between 1 Counter({'POS': 1, 'NEU': 1})\n",
      "Counter({'POS': 1})\n",
      "expression of miR-210 , showing the involvement of the HIF-1 signaling pathway\n",
      "Counter({'POS': 1})\n",
      "of the HIF-1alpha induced the expression of miR-210 , showing the involvement of the HIF-1 signaling pathway in miR-210 gene transcription\n",
      "counts between 1 Counter({'NEU': 3, 'POS': 1})\n",
      "Counter()\n",
      "of the HIF-1 signaling pathway in miR-210 gene transcription\n",
      "Counter()\n",
      "Furthermore , miR-210 was shown to bind to the predicted target sites of ephrin-A3 or neuronal\n",
      "Counter()\n",
      "Furthermore , miR-210 was shown to bind to the predicted target sites of ephrin-A3 or neuronal pentraxin 1\n",
      "Counter({'NEG': 1})\n",
      "miR-124 directly inhibited the transcription factor CCAAT/enhancer-binding protein-α ( C/EBP-α\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1})\n",
      "miR-124 directly inhibited the transcription factor CCAAT/enhancer-binding protein-α ( C/EBP-α ) and\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1})\n",
      "miR-124 directly inhibited the transcription factor CCAAT/enhancer-binding protein-α ( C/EBP-α ) and its downstream target PU.1\n",
      "counts between 1 Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter()\n",
      "MiR-451 levels are regulated by glucose ; under conditions of abundant energy miR-451 expression is high , and the suppression of AMPK signaling allows cells to maintain elevated proliferation rates via unrestrained mTOR activation .\n",
      "counts between 1 Counter({'NEU': 5, 'NEG': 1, 'POS': 1})\n",
      "Counter()\n",
      "abundant energy miR-451 expression is high , and the suppression of AMPK signaling allows cells to maintain elevated proliferation rates via unrestrained mTOR activation .\n",
      "counts between 1 Counter({'NEU': 3, 'NEG': 1, 'POS': 1})\n",
      "Counter()\n",
      "miR-206 mediates these effects at least in part through histone deacetylase 4 and fibroblast\n",
      "Counter()\n",
      "CDK6 is regulated by microRNA 124 in medulloblastoma\n",
      "Our data strongly indicate that CDK6 is regulated by microRNA 124 in medulloblastoma and that miR 124 modulates medulloblastoma cell growth.   \n",
      "('microRNA 124', 'Specific_miRNAs', (53, 65)) ('CDK6', 'Genes/Proteins', (32, 36))\n",
      "('GENE_MIR', 'NEU') return passive? True\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "indicate that CDK6 is regulated by microRNA 124 in medulloblastoma and that miR-124 modulates medulloblastoma\n",
      "Counter()\n",
      "in the CDK6 including microRNA 124a , a\n",
      "We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA. \n",
      "('microRNA 124a', 'Specific_miRNAs', (55, 68)) ('CDK6', 'Genes/Proteins', (40, 44))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1})\n",
      "re-expression of miR-124a in medulloblastoma cells decreased expression of CDK6 protein\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter()\n",
      "association between miR-124a and CDK6 in medulloblastoma\n",
      "Counter()\n",
      "dependent kinase 6 by microRNA 124 in medulloblastoma\n",
      "Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.  \n",
      "('microRNA 124', 'Specific_miRNAs', (43, 55)) ('cyclin dependent kinase 6', 'Genes/Proteins', (14, 39))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "Counter()\n",
      "Counter()\n",
      "Counter()\n",
      "Counter()\n",
      "Counter()\n",
      "Counter()\n",
      "Counter()\n",
      "Counter()\n",
      "the microRNA miR-128a , which can tune paired helical filament Tau levels in\n",
      "Counter()\n",
      "BAG2 levels in cells are under the physiological control of the microRNA miR-128a , which\n",
      "BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons. \n",
      "('miR-128a', 'Specific_miRNAs', (73, 81)) ('BAG2', 'Genes/Proteins', (0, 4))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "Counter()\n",
      "Counter()\n",
      "Counter()\n",
      "Counter({'POS': 2})\n",
      "overexpression of Hsa-miR-30b , Hsa-miR-30d and KHDRBS3 at 8q24.22-q24.23\n",
      "counts between 1 Counter({'POS': 1})\n",
      "Counter()\n",
      "Hsa-miR-30b , Hsa-miR-30d and KHDRBS3 at 8q24.22-q24.23\n",
      "Counter()\n",
      "expression of hsa-miR-30b , hsa-miR-30d and KHDRBS3 correlated with\n",
      "Counter()\n",
      "hsa-miR-30b , hsa-miR-30d and KHDRBS3 correlated with\n",
      "Counter()\n",
      "genes ( ZFAT1 , LOC286094 , KHDRBS3 ) and two genes encoding micro-RNAs ( hsa-miR-30b , hsa-miR-30d\n",
      "Counter()\n",
      "LOC286094 , KHDRBS3 ) and two genes encoding micro-RNAs ( hsa-miR-30b , hsa-miR-30d\n",
      "Counter()\n",
      "genes ( ZFAT1 , LOC286094 , KHDRBS3 ) and two genes encoding micro-RNAs ( hsa-miR-30b , hsa-miR-30d\n",
      "Counter()\n",
      "LOC286094 , KHDRBS3 ) and two genes encoding micro-RNAs ( hsa-miR-30b , hsa-miR-30d\n",
      "Counter()\n",
      "data implicate hsa-miR-30b , hsa-miR-30d and KHDRBS3 as putative\n",
      "Counter()\n",
      "hsa-miR-30b , hsa-miR-30d and KHDRBS3 as putative\n",
      "Counter()\n",
      "miR-29b and miR-125a regulate podoplanin\n",
      "Counter()\n",
      "miR-29b and miR-125a regulate podoplanin\n",
      "Counter()\n",
      "focused on miR-29b and miR-125a , which were predicted to regulate PDPN\n",
      "Counter()\n",
      "focused on miR-29b and miR-125a , which were predicted to regulate PDPN , and demonstrated that these microRNAs directly target the 3 ' untranslated region of PDPN and inhibit\n",
      "counts between 1 Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter()\n",
      "miR-29b and miR-125a , which were predicted to regulate PDPN\n",
      "Counter()\n",
      "miR-29b and miR-125a , which were predicted to regulate PDPN , and demonstrated that these microRNAs directly target the 3 ' untranslated region of PDPN and inhibit\n",
      "counts between 1 Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter({'POS': 1})\n",
      "Counter()\n",
      "in the miRNA-433 binding site of FGF20 confers risk\n",
      "Counter({'POS': 1})\n",
      "in the miRNA-433 binding site of FGF20 confers risk for Parkinson disease by overexpression of alpha-synuclein\n",
      "counts between 1 Counter({'NEU': 1, 'POS': 1})\n",
      "Variation in the miRNA-433 binding site of FGF20 confers risk for Parkinson disease by overexpression of alpha-synuclein.  \n",
      "('miRNA-433', 'Specific_miRNAs', (17, 26)) ('alpha-synuclein', 'Genes/Proteins', (105, 120))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter()\n",
      "site for microRNA-433 , increasing translation of FGF20 in vitro\n",
      "counts between 1 Counter({'POS': 1, 'NEU': 1})\n",
      "Total:      232\n",
      "Correct:    218 0.9396551724137931\n",
      "Incorrect:  14 0.0603448275862069\n",
      "classes Counter({(True, False): 32, (False, True): 24})\n",
      "Correct classified:    97 0.41810344827586204\n",
      "Incorrect classified:    33 0.14224137931034483\n",
      "Counter({'counts between ': 15, 'return': 13, 'between gene mir': 3, 'compartment gene mir': 1, 'compartment mir gene': 1})\n"
     ]
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClasses = loadRelations()\n",
    "allStems, allRels, stem2class = loadStems()\n",
    "\n",
    "print([x for x in relClasses][:5])\n",
    "\n",
    "runCheck(-1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "def testRelClassifier(doc, mirword, geneword, relchecker, verbose):\n",
    "    \n",
    "    if verbose:\n",
    "        print()\n",
    "        print()\n",
    "        print()\n",
    "        print(\"Rel Classifier\")\n",
    "    \n",
    "        print(doc.text_with_ws)\n",
    "    \n",
    "    foundStems = set()\n",
    "    sdpRes = relchecker._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "    if verbose:\n",
    "        for x in sdpRes:\n",
    "            print(x)\n",
    "        \n",
    "        \n",
    "    sdpTokens = [x[0] for x in sdpRes]\n",
    "    #print(sdpRes)\n",
    "    \n",
    "    sdpVerbRels = set()\n",
    "\n",
    "    for sidx, x in enumerate(sdpRes):\n",
    "        if x[1] == \"VERB\":\n",
    "            for s in allStems:\n",
    "                if s in str(x[0]):\n",
    "                    sdpVerbRels.add((s, sidx, stem2class[s]))\n",
    "                    if verbose:\n",
    "                        print(\"sdpVerbRels\", x[0], s, sidx, stem2class[s])\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "            \n",
    "    docRels = set()\n",
    "    for tkn in doc:\n",
    "\n",
    "        if tkn in sdpTokens:\n",
    "            continue\n",
    "\n",
    "        for s in allStems:\n",
    "            if str(tkn).lower().startswith(s):\n",
    "                docRels.add((s, sidx, stem2class[s]))\n",
    "                if verbose:    \n",
    "                    print(\"docRels\", tkn, s, sidx, stem2class[s])\n",
    "            \n",
    "            \n",
    "    mirAncests = [x for x in mirword.ancestors]\n",
    "    geneAncests = [x for x in geneword.ancestors if not x in mirAncests]\n",
    "    \n",
    "    if len(mirAncests) > 1:\n",
    "        useMirAncests = mirAncests[0:1]\n",
    "    else:\n",
    "        useMirAncests = mirAncests\n",
    "        \n",
    "    if len(geneAncests) > 1:\n",
    "        useGeneAncests = geneAncests[0:1]\n",
    "    else:\n",
    "        useGeneAncests = geneAncests\n",
    "        \n",
    "    useGeneAncests = [x for x in useGeneAncests if not x in useMirAncests]\n",
    "    \n",
    "    if verbose:\n",
    "        print(\"mirc\", mirAncests)\n",
    "        print(\"genec\", geneAncests)\n",
    "    \n",
    "    invCor = False\n",
    "    regdir = \"NEU\"\n",
    "    intdir = \"MIR_GENE\"\n",
    "    isPassive = relchecker.checkPassive(doc, mirword, geneword, verbose=False)\n",
    "    if verbose:\n",
    "        print(\"passive?\", isPassive)\n",
    "    \n",
    "    ancestHandled = False\n",
    "    #if len(mirAncests) > 0 and len(geneAncests) > 0:\n",
    "    if True:\n",
    "        mirRels = []\n",
    "        geneRels = []\n",
    "        \n",
    "        mirRels = getRelsForTokens(useMirAncests, \"ancestRels mir\", verbose)     \n",
    "        neuMirs = [x for x in mirRels if x[2] in [\"NEU\"]]\n",
    "        #if len(neuMirs) > 0 or len(mirRels) == 0:\n",
    "        #    mirRels = getRelsForTokens(mirAncests, \"ancestRels all mir\", verbose)\n",
    "            \n",
    "        geneRels = getRelsForTokens(useGeneAncests, \"ancestRels gene\", verbose)       \n",
    "        neuGenes = [x for x in geneRels if x[2] in [\"NEU\"]]\n",
    "                \n",
    "        if False and ((len(neuGenes) == len(geneRels)) or len(geneRels) == 0):\n",
    "            geneRels = getRelsForTokens([x for x in geneAncests if not x in useMirAncests], \"ancestRels all gene\", verbose)\n",
    "            neuGenes = [x for x in geneRels if x[2] in [\"NEU\"]]\n",
    "            \n",
    "            if (len(neuGenes) == len(geneRels)) or len(geneRels) == 0:\n",
    "                \n",
    "                if mirword.i < geneword.i:\n",
    "                    tks = [x for x in doc if geneword.i -8 < x.i < geneword.i if not x in useMirAncests]\n",
    "                else:\n",
    "                    tks = [x for x in doc if geneword.i < x.i < geneword.i+5 if not x in useMirAncests]\n",
    "                \n",
    "                geneRels = getRelsForTokens(tks, \"ancestRels tks gene\", verbose)\n",
    "                \n",
    "                                        \n",
    "        posRegs = [x for x in mirRels+geneRels if x[2] in [\"POS\"]]\n",
    "        negRegs = [x for x in mirRels+geneRels if x[2] in [\"NEG\"]]\n",
    "        \n",
    "        mirPos = len([x for x in mirRels if x[2] in [\"POS\"]])\n",
    "        mirNeu = [x for x in mirRels if x[2] in [\"NEU\"]]\n",
    "        mirNeg = len([x for x in mirRels if x[2] in [\"NEG\"]])\n",
    "        \n",
    "        genePos = len([x for x in geneRels if x[2] in [\"POS\"]])\n",
    "        geneNeu = [x for x in geneRels if x[2] in [\"NEU\"]]\n",
    "        geneNeg = len([x for x in geneRels if x[2] in [\"NEG\"]])\n",
    "    \n",
    "        dirReg = (mirPos - geneNeg) + (genePos-mirNeg)\n",
    "        \n",
    "        invCor = False\n",
    "        \n",
    "        if verbose:\n",
    "            print(mirword, geneword)\n",
    "            print(\"ancestRel counts\", mirPos, mirNeg, genePos, geneNeg)\n",
    "                          \n",
    "            \n",
    "        nregdir, ninvcor = evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg)\n",
    "        if verbose:\n",
    "            print(\"evalcount\")\n",
    "            print(nregdir, ninvcor)\n",
    "        \n",
    "        if nregdir != None:\n",
    "            regdir = nregdir\n",
    "            invcor = ninvcor\n",
    "            \n",
    "            ancestHandled = True\n",
    "            \n",
    "            if verbose:\n",
    "                print(\"accepted by evalcount\")\n",
    "            \n",
    "        else:\n",
    "            lookBehind = 8\n",
    "\n",
    "                                \n",
    "            if len(mirNeu) > 0 or len(geneNeu) > 0:\n",
    "                if verbose:\n",
    "                    print(\"ancestRel neu part\")\n",
    "\n",
    "                if mirword.i < geneword.i:\n",
    "                    for x in mirNeu:\n",
    "                        if x[0].startswith((\"correlat\", \"affect\")):\n",
    "\n",
    "                            for j in range(1,lookBehind):\n",
    "                                if str(doc[x[4].i-j]).startswith((\"posit\", \"high\")):\n",
    "                                    mirPos += 1\n",
    "                                    \n",
    "                                elif str(doc[x[4].i-j]).startswith((\"negat\", \"low\")):\n",
    "                                    mirNeg += 1\n",
    "                else:\n",
    "                    for x in geneNeu:\n",
    "                        if x[0].startswith((\"correlat\")):\n",
    "                            \n",
    "                            for j in range(1,lookBehind):\n",
    "                                if str(doc[x[4].i-j]).startswith((\"posit\", \"high\")):\n",
    "                                    genePos += 1\n",
    "                                elif str(doc[x[4].i-j]).startswith((\"negat\", \"low\")):\n",
    "                                    geneNeg += 1    \n",
    "            \n",
    "                                \n",
    "            if not ancestHandled:    \n",
    "                \n",
    "                                \n",
    "                    \n",
    "                if verbose:\n",
    "                    print(\"ancestRel counts stem check\", mirPos, mirNeg, genePos, geneNeg)\n",
    "                    \n",
    "                nregdir, ninvcor = evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg)\n",
    "                if nregdir != None:\n",
    "                    regdir = nregdir\n",
    "                    invcor = ninvcor\n",
    "                    ancestHandled = True\n",
    "                    \n",
    "                    if verbose:\n",
    "                        print(\"accept after before stem check\")\n",
    "                    \n",
    "                if not ancestHandled:\n",
    "                    for j in range(1,lookBehind):\n",
    "                        \n",
    "                        if mirword.i-j < 0:\n",
    "                            continue\n",
    "                        \n",
    "                        jtk = doc[mirword.i-j]\n",
    "                        if verbose:\n",
    "                            print(\"word check mir\", j, jtk)\n",
    "                        pos, neg = scoreWord(jtk)\n",
    "                        \n",
    "                        mirPos += pos\n",
    "                        mirNeg += neg\n",
    "                            \n",
    "                    for j in range(1,lookBehind):\n",
    "                        \n",
    "                        if geneword.i-j < 0:\n",
    "                            continue\n",
    "                        \n",
    "                        jtk = doc[geneword.i-j]\n",
    "                        if verbose:\n",
    "                            print(\"word check gene\", j, jtk)\n",
    "                            \n",
    "                        pos, neg = scoreWord(jtk)\n",
    "                        \n",
    "                        genePos += pos\n",
    "                        geneNeg += neg\n",
    "                            \n",
    "                            \n",
    "                if verbose:\n",
    "                    print(\"ancestRel counts word check\", mirPos, mirNeg, genePos, geneNeg)\n",
    "                \n",
    "                nregdir, ninvcor = evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg)\n",
    "                if nregdir != None:\n",
    "                    regdir = nregdir\n",
    "                    invcor = ninvcor\n",
    "                    ancestHandled = True\n",
    "                    \n",
    "                    if verbose:\n",
    "                        print(\"accept after before word check\")\n",
    "                    \n",
    "           \n",
    "\n",
    "                \n",
    "                \n",
    "        if verbose:\n",
    "            print(\"ancestRels\", dirReg, invCor)\n",
    "            \n",
    "    if ancestHandled:\n",
    "        \n",
    "        if isPassive:\n",
    "            intdir = \"GENE_MIR\"\n",
    "            \n",
    "    \n",
    "    if not ancestHandled:\n",
    "        \n",
    "        if verbose:\n",
    "            print(\"ancestHandled part\")\n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "            lword = geneword\n",
    "            rword = mirword\n",
    "            useTks = [x for x in doc if geneword.i < x.i < mirword.i]\n",
    "        else:\n",
    "            lword = mirword\n",
    "            rword = geneword\n",
    "            useTks = [x for x in doc if mirword.i < x.i < geneword.i]\n",
    "            \n",
    "        if verbose:\n",
    "            print(useTks)\n",
    "        \n",
    "        betRels = getRelsForTokens(useTks, \"ancestRels useTks\", verbose)\n",
    "        \n",
    "        if verbose:\n",
    "            print(betRels)\n",
    "        \n",
    "        betPos = len([x for x in betRels if x[2] in [\"POS\"]])\n",
    "        betNeg = len([x for x in betRels if x[2] in [\"NEG\"]])\n",
    "        \n",
    "        if betPos + betNeg == 1:\n",
    "            ancestHandled = True\n",
    "            \n",
    "            if verbose:\n",
    "                print(\"ancest bet handled\", mirword.i, geneword.i, isPassive)\n",
    "            \n",
    "            if geneword.i < mirword.i:\n",
    "                \n",
    "                if isPassive:\n",
    "                    intdir = \"MIR_GENE\"\n",
    "                    regdir = \"DOWN\" if betNeg > 0 else \"UP\"\n",
    "                    \n",
    "                else:\n",
    "                    intdir = \"GENE_MIR\"\n",
    "                    regdir = \"UP\" if betPos > 0 else \"DOWN\"\n",
    "                    \n",
    "            else:\n",
    "                \n",
    "                if isPassive:\n",
    "                    intdir = \"GENE_MIR\"\n",
    "                    regdir = \"UP\" if betPos > 0 else \"DOWN\"\n",
    "                    \n",
    "                else:\n",
    "                    intdir = \"MIR_GENE\"\n",
    "                    regdir = \"UP\" if betPos > 0 else \"DOWN\"\n",
    "                    \n",
    "        else:\n",
    "            \n",
    "            isMediated = False\n",
    "            \n",
    "            for x in [lword] + useTks[0:2]:\n",
    "                for y in [\"mediated\", \"-mediated\"]:\n",
    "                    \n",
    "                    if y in str(x):\n",
    "                        isMediated = True\n",
    "                        break\n",
    "                if isMediated:\n",
    "                    break\n",
    "                                    \n",
    "            if isMediated:\n",
    "                \n",
    "                ancestHandled = True\n",
    "                \n",
    "                if verbose:\n",
    "                    print(\"ancest mediated handled\")\n",
    "                \n",
    "                if geneword.i < mirword.i:\n",
    "                    intdir = \"GENE_MIR\"\n",
    "                else:\n",
    "                    intdir = \"MIR_GENE\"\n",
    "                    \n",
    "                if betPos > betNeg:\n",
    "                    regdir = \"UP\"\n",
    "                elif betNeg > betPos:\n",
    "                    regdir = \"DOWN\"\n",
    "                        \n",
    "            \n",
    "            \n",
    "            \n",
    "\n",
    "    #classCounter = Counter()\n",
    "    #if len(classCounter) > 0:\n",
    "    #    regdir = classCounter.most_common(1)[0][0]\n",
    "    #else:\n",
    "    #    regdir = \"NEU\"\n",
    "        \n",
    "    #if invCor:\n",
    "    #    regdir = \"DOWN\"       \n",
    "              \n",
    "    \n",
    "    if not ancestHandled:\n",
    "        \n",
    "        if verbose:\n",
    "            print(\"in final resolution\")\n",
    "        \n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "        \n",
    "            textBetween = \" \".join([str(x) for x in doc if geneword.i < x.i < mirword.i])\n",
    "            textBetween = ' '.join(textBetween.split())\n",
    "            \n",
    "            if verbose:\n",
    "                print(textBetween)\n",
    "            \n",
    "\n",
    "            if len(textBetween) < 10 and any([x in textBetween for x in [\"by\"]]):\n",
    "                sdpRels = set()\n",
    "                for sidx,x  in enumerate(sdpRes):\n",
    "                    for s in allStems:\n",
    "                        if str(x[0]).lower().startswith(s):\n",
    "                            sdpRels.add((s, sidx, stem2class[s]))\n",
    "                            if verbose:\n",
    "                                print(\"sdpRels\", x[0], s, sidx, stem2class[s])\n",
    "\n",
    "                posRegs = [x for x in sdpRels if x[2] in [\"POS\"]]\n",
    "                negRegs = [x for x in sdpRels if x[2] in [\"NEG\"]]\n",
    "\n",
    "                dirReg = len(posRegs)-len(negRegs)\n",
    "\n",
    "                if dirReg > 0:\n",
    "                    regdir  = \"UP\"\n",
    "                elif dirReg < 0:\n",
    "                    regdir = \"DOWN\"\n",
    "                else:\n",
    "                    regdir = \"NEU\"\n",
    "                \n",
    "                intdir = \"MIR_GENE\"\n",
    "\n",
    "            elif any([x in textBetween for x in [\"target of\", \"target for\", \"targeting\"]]):\n",
    "                intdir = \"MIR_GENE\"\n",
    "                regdir = \"NEU\"\n",
    "\n",
    "            else:                \n",
    "                intdir = \"GENE_MIR\"\n",
    "    \n",
    "        elif mirword.i < geneword.i:\n",
    "            textBetween = \" \".join([str(x) for x in doc if geneword.i-5 < x.i < geneword.i])\n",
    "\n",
    "            if any([x in textBetween for x in [\"by\"]]):\n",
    "                intdir = \"GENE_MIR\"\n",
    "                regdir = \"UP\"\n",
    "        \n",
    "    \n",
    "    regstems = foundStems\n",
    "    \n",
    "    #displacy.render(doc, style=\"dep\")\n",
    "    \n",
    "    if verbose:\n",
    "        print((intdir, regdir, \"NA\", isPassive))\n",
    "        \n",
    "    return {\"regulation_dir\": regdir, \"interaction_dir\": intdir, \"regulation_keys\": regstems}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
