{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "import os, sys\n",
    "sys.path.insert(0, \"/mnt/f/dev/git/miRExplore/python/\")\n",
    "\n",
    "\n",
    "import scispacy\n",
    "import spacy\n",
    "from spacy import displacy\n",
    "\n",
    "#nlp = spacy.load('/mnt/d/spacy/models/en_core_sci_lg-0.2.4/en_core_sci_lg/en_core_sci_lg-0.2.4/')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp_weblg = spacy.load('/mnt/d/spacy/models/en_core_web_lg-2.2.0/en_core_web_lg/en_core_web_lg-2.2.0/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp = spacy.load('/mnt/f/spacy/en_core_sci_lg-0.2.4/en_core_sci_lg/en_core_sci_lg-0.2.4/')\n",
    "nlp_ent = spacy.load(\"/mnt/f/spacy/en_ner_bionlp13cg_md-0.2.4/en_ner_bionlp13cg_md/en_ner_bionlp13cg_md-0.2.4\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp = spacy.load('/mnt/f/spacy/en_core_sci_lg-0.2.5/en_core_sci_lg/en_core_sci_lg-0.2.5/')\n",
    "nlp_ent = spacy.load(\"/mnt/f/spacy/en_ner_bionlp13cg_md-0.2.5/en_ner_bionlp13cg_md/en_ner_bionlp13cg_md-0.2.5\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!ls /mnt/d/spacy/models/en_core_web_lg-2.2.0/en_core_web_lg/en_core_web_lg-2.2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import textmining.MirGeneRelCheck\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def analyse(testCase, dist=80):\n",
    "    relCheck = textmining.MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "    doc = nlp(testCase[0])\n",
    "    \n",
    "    #print(\"conjuncts\", doc.conjuncts)\n",
    "    \n",
    "    lWord = doc[testCase[1]]\n",
    "    rWord = doc[testCase[2]]\n",
    "\n",
    "    for t in doc:\n",
    "        print(t.idx, t.i, t, t.pos_, t.dep_, t.head, t.head.idx, t.conjuncts)\n",
    "\n",
    "    print(lWord, rWord)\n",
    "\n",
    "    if not \"mir\" in str(lWord).lower() and not \"let\" in str(lWord).lower():\n",
    "        tmp = lWord\n",
    "        lWord = rWord\n",
    "        rWord = tmp\n",
    "        \n",
    "    print(lWord, rWord)\n",
    "    \n",
    "    checkResults = relCheck.checkRelation(doc, lWord, rWord, verbose=True)\n",
    "    print(checkResults)\n",
    "    \n",
    "    displacy.render(doc, style=\"dep\", options={\"compact\": True, \"distance\": dist})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 Disfunction NOUN nsubjpass described 46 ()\n",
      "12 1 of ADP case neutrophils 15 ()\n",
      "15 2 neutrophils NOUN nmod Disfunction 0 ()\n",
      "27 3 in ADP case sepsis 30 ()\n",
      "30 4 sepsis NOUN nmod Disfunction 0 ()\n",
      "37 5 has AUX aux described 46 ()\n",
      "41 6 been AUX auxpass described 46 ()\n",
      "46 7 described VERB ROOT described 46 ()\n",
      "56 8 in ADP case studies 68 ()\n",
      "59 9 previous ADJ amod studies 68 ()\n",
      "68 10 studies NOUN nmod described 46 ()\n",
      "75 11 , PUNCT punct known 96 ()\n",
      "77 12 however ADV advmod known 96 ()\n",
      "84 13 , PUNCT punct known 96 ()\n",
      "86 14 little ADJ nsubjpass known 96 ()\n",
      "93 15 is AUX auxpass known 96 ()\n",
      "96 16 known VERB parataxis described 46 (participate,)\n",
      "102 17 about ADP case role 112 ()\n",
      "108 18 the DET det role 112 ()\n",
      "112 19 role NOUN nmod known 96 ()\n",
      "117 20 of ADP case microRNA-let-7b 120 ()\n",
      "120 21 microRNA-let-7b NOUN nmod role 112 (receptor, activity)\n",
      "136 22 ( PUNCT punct miR-let-7b 137 ()\n",
      "137 23 miR-let-7b NOUN appos microRNA-let-7b 120 ()\n",
      "147 24 ) PUNCT punct miR-let-7b 137 ()\n",
      "148 25 , PUNCT punct microRNA-let-7b 120 ()\n",
      "150 26 toll-like ADJ amod receptor 160 ()\n",
      "160 27 receptor NOUN conj microRNA-let-7b 120 (microRNA-let-7b, activity)\n",
      "169 28 4 NUM nummod receptor 160 ()\n",
      "171 29 ( PUNCT punct TLR4 172 ()\n",
      "172 30 TLR4 NOUN appos receptor 160 ()\n",
      "176 31 ) PUNCT punct TLR4 172 ()\n",
      "177 32 , PUNCT punct microRNA-let-7b 120 ()\n",
      "179 33 and CCONJ cc microRNA-let-7b 120 ()\n",
      "183 34 nuclear ADJ amod B 204 ()\n",
      "191 35 factor NOUN compound B 204 ()\n",
      "198 36 kappa NOUN compound B 204 ()\n",
      "204 37 B NOUN compound activity 218 ()\n",
      "206 38 ( PUNCT punct NF-kappaB 207 ()\n",
      "207 39 NF-kappaB NOUN appos B 204 ()\n",
      "216 40 ) PUNCT punct NF-kappaB 207 ()\n",
      "218 41 activity NOUN conj microRNA-let-7b 120 (microRNA-let-7b, receptor)\n",
      "227 42 in ADP case neutrophils 230 ()\n",
      "230 43 neutrophils NOUN nmod role 112 ()\n",
      "242 44 and CCONJ cc known 96 ()\n",
      "246 45 how ADV advmod participate 255 ()\n",
      "250 46 they PRON nsubj participate 255 ()\n",
      "255 47 participate VERB conj known 96 (known,)\n",
      "267 48 in ADP case development 274 ()\n",
      "270 49 the DET det development 274 ()\n",
      "274 50 development NOUN nmod participate 255 ()\n",
      "286 51 of ADP case sepsis 289 ()\n",
      "289 52 sepsis NOUN nmod development 274 ()\n",
      "295 53 . PUNCT punct described 46 ()\n",
      "microRNA-let-7b NF-kappaB\n",
      "microRNA-let-7b NF-kappaB\n",
      "Conjuncts [participate, known]\n",
      "neighbour participate in\n",
      "neighbour participate in\n",
      "Conjunction before fc known\n",
      "fc add children for known\n",
      "Conjuncts [receptor, activity, microRNA-let-7b]\n",
      "neighbour receptor 4\n",
      "neighbour receptor 4\n",
      "neighbour activity in\n",
      "neighbour activity in\n",
      "Conjunction before fc microRNA-let-7b\n",
      "fc add children for microRNA-let-7b\n",
      "fc add children for microRNA-let-7b\n",
      "Conjuncts [microRNA-let-7b, activity, receptor]\n",
      "neighbour microRNA-let-7b (\n",
      "neighbour microRNA-let-7b miR-let-7b\n",
      "neighbour activity in\n",
      "neighbour activity in\n",
      "Conjunction before fc receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "Conjuncts [microRNA-let-7b, receptor, activity]\n",
      "neighbour microRNA-let-7b (\n",
      "neighbour microRNA-let-7b miR-let-7b\n",
      "neighbour receptor 4\n",
      "neighbour receptor 4\n",
      "Conjunction before fc activity\n",
      "fc add children for activity\n",
      "Conjuncts [known, participate]\n",
      "neighbour known about\n",
      "neighbour known about\n",
      "Conjunction before fc participate\n",
      "fc add children for participate\n",
      "Conunct merge check: known miR-let-7b\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: miR-let-7b TLR4\n",
      "), toll-like receptor 4\n",
      "Conunct merge check: TLR4 NF-kappaB\n",
      "), and nuclear factor kappa B\n",
      "Double Conjunction Found\n",
      "Conjunctions\n",
      "known [known, about, role, of, microRNA-let-7b, miR-let-7b, toll-like, receptor, 4, TLR4, nuclear, factor, kappa, B, NF-kappaB, activity, in, neutrophils, participate, in, development, of, sepsis]\n",
      "miR-let-7b [(, )]\n",
      "TLR4 [(, )]\n",
      "NF-kappaB [(, )]\n",
      "(NF-kappaB, 'conj') [(, ), (, )]\n",
      "achildren {microRNA-let-7b}\n",
      "achildren False\n",
      "SDP sent: Disfunction of neutrophils in sepsis has been described in previous studies, however, little is known about the role of microRNA-let-7b (miR-let-7b), toll-like receptor 4 (TLR4), and nuclear factor kappa B (NF-kappaB) activity in neutrophils and how they participate in the development of sepsis.\n",
      "SDP     : [(microRNA-let-7b, 'NOUN', 'nmod', False), (described, 'VERB', 'ROOT', False), (known, 'VERB', 'parataxis', False), (role, 'NOUN', 'nmod', False), (microRNA-let-7b, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[toll-like, receptor, 4, (, TLR4, )]\n",
      "DeepCheck Conj Subtree\n",
      "[nuclear, factor, kappa, B, (, NF-kappaB, ), activity]\n",
      "0 [Disfunction, of, neutrophils, in, sepsis, has, been, described, in, previous, studies] 11 0 Disfunction of neutrophils in sepsis has been described in previous studies, however, little is known about the role of microRNA-let-7b (miR-let-7b), toll-like receptor 4 (TLR4), and nuclear factor kappa B (NF-kappaB) activity in neutrophils and how they participate in the development of sepsis.\n",
      "1 [however, ,, little, is, known, about, the, role, of, microRNA-let-7b, (, miR-let-7b, ), ,, toll-like, receptor, 4, (, TLR4, ), ,, and, nuclear, factor, kappa, B, (, NF-kappaB, ), activity, in, neutrophils] 32 77 however, little is known about the role of microRNA-let-7b (miR-let-7b), toll-like receptor 4 (TLR4), and nuclear factor kappa B (NF-kappaB) activity in neutrophils and how they participate in the development of sepsis.\n",
      "2 [how, they, participate, in, the, development, of, sepsis] 8 246 how they participate in the development of sepsis.\n",
      "(False, {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False})\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"457d5ebe068f4cd89e252a6bf897a93b-0\" class=\"displacy\" width=\"3490\" height=\"537.0\" direction=\"ltr\" style=\"max-width: none; height: 537.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">Disfunction</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"130\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"130\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"210\">neutrophils</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"210\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"290\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"290\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"370\">sepsis</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"370\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"450\">has</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"450\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"530\">been</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"530\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"610\">described</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"610\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"690\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"690\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"770\">previous</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"770\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"850\">studies,</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"850\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"930\">however,</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"930\">ADV</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1010\">little</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1010\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1090\">is</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1090\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1170\">known</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1170\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1250\">about</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1250\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1330\">the</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1330\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1410\">role</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1410\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1490\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1490\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1570\">microRNA-let-7b (</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1570\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1650\">miR-let-7b),</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1650\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1730\">toll-like</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1730\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1810\">receptor</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1810\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1890\">4 (</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1890\">NUM</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1970\">TLR4),</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1970\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2050\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2050\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2130\">nuclear</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2130\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2210\">factor</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2210\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2290\">kappa</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2290\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2370\">B (</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2370\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2450\">NF-kappaB)</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2450\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2530\">activity</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2530\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2610\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2610\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2690\">neutrophils</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2690\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2770\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2770\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2850\">how</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2850\">ADV</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2930\">they</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2930\">PRON</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3010\">participate</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3010\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3090\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3090\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3170\">the</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3170\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3250\">development</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3250\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3330\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3330\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"447.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3410\">sepsis.</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3410\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-0\" stroke-width=\"2px\" d=\"M62,402.0 62,322.0 598.0,322.0 598.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubjpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M62,404.0 L58,396.0 66,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-1\" stroke-width=\"2px\" d=\"M142,402.0 142,388.6666666666667 183.0,388.6666666666667 183.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M142,404.0 L138,396.0 146,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-2\" stroke-width=\"2px\" d=\"M62,402.0 62,375.3333333333333 186.0,375.3333333333333 186.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M186.0,404.0 L190.0,396.0 182.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-3\" stroke-width=\"2px\" d=\"M302,402.0 302,388.6666666666667 343.0,388.6666666666667 343.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M302,404.0 L298,396.0 306,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-4\" stroke-width=\"2px\" d=\"M62,402.0 62,348.6666666666667 352.0,348.6666666666667 352.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M352.0,404.0 L356.0,396.0 348.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-5\" stroke-width=\"2px\" d=\"M462,402.0 462,375.3333333333333 586.0,375.3333333333333 586.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">aux</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M462,404.0 L458,396.0 466,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-6\" stroke-width=\"2px\" d=\"M542,402.0 542,388.6666666666667 583.0,388.6666666666667 583.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">auxpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M542,404.0 L538,396.0 546,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-7\" stroke-width=\"2px\" d=\"M702,402.0 702,375.3333333333333 826.0,375.3333333333333 826.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M702,404.0 L698,396.0 706,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-8\" stroke-width=\"2px\" d=\"M782,402.0 782,388.6666666666667 823.0,388.6666666666667 823.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M782,404.0 L778,396.0 786,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-9\" stroke-width=\"2px\" d=\"M622,402.0 622,362.0 829.0,362.0 829.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M829.0,404.0 L833.0,396.0 825.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-10\" stroke-width=\"2px\" d=\"M942,402.0 942,362.0 1149.0,362.0 1149.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">advmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M942,404.0 L938,396.0 946,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-11\" stroke-width=\"2px\" d=\"M1022,402.0 1022,375.3333333333333 1146.0,375.3333333333333 1146.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubjpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1022,404.0 L1018,396.0 1026,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-12\" stroke-width=\"2px\" d=\"M1102,402.0 1102,388.6666666666667 1143.0,388.6666666666667 1143.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">auxpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1102,404.0 L1098,396.0 1106,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-13\" stroke-width=\"2px\" d=\"M622,402.0 622,322.0 1158.0,322.0 1158.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-13\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">parataxis</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1158.0,404.0 L1162.0,396.0 1154.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-14\" stroke-width=\"2px\" d=\"M1262,402.0 1262,375.3333333333333 1386.0,375.3333333333333 1386.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-14\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1262,404.0 L1258,396.0 1266,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-15\" stroke-width=\"2px\" d=\"M1342,402.0 1342,388.6666666666667 1383.0,388.6666666666667 1383.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-15\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1342,404.0 L1338,396.0 1346,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-16\" stroke-width=\"2px\" d=\"M1182,402.0 1182,362.0 1389.0,362.0 1389.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-16\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1389.0,404.0 L1393.0,396.0 1385.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-17\" stroke-width=\"2px\" d=\"M1502,402.0 1502,388.6666666666667 1543.0,388.6666666666667 1543.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-17\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1502,404.0 L1498,396.0 1506,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-18\" stroke-width=\"2px\" d=\"M1422,402.0 1422,375.3333333333333 1546.0,375.3333333333333 1546.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-18\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1546.0,404.0 L1550.0,396.0 1542.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-19\" stroke-width=\"2px\" d=\"M1582,402.0 1582,388.6666666666667 1623.0,388.6666666666667 1623.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-19\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">appos</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1623.0,404.0 L1627.0,396.0 1619.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-20\" stroke-width=\"2px\" d=\"M1742,402.0 1742,388.6666666666667 1783.0,388.6666666666667 1783.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-20\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1742,404.0 L1738,396.0 1746,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-21\" stroke-width=\"2px\" d=\"M1582,402.0 1582,362.0 1789.0,362.0 1789.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-21\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1789.0,404.0 L1793.0,396.0 1785.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-22\" stroke-width=\"2px\" d=\"M1822,402.0 1822,388.6666666666667 1863.0,388.6666666666667 1863.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-22\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nummod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1863.0,404.0 L1867.0,396.0 1859.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-23\" stroke-width=\"2px\" d=\"M1822,402.0 1822,375.3333333333333 1946.0,375.3333333333333 1946.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-23\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">appos</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1946.0,404.0 L1950.0,396.0 1942.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-24\" stroke-width=\"2px\" d=\"M1582,402.0 1582,335.3333333333333 2035.0,335.3333333333333 2035.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-24\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2035.0,404.0 L2039.0,396.0 2031.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-25\" stroke-width=\"2px\" d=\"M2142,402.0 2142,362.0 2349.0,362.0 2349.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-25\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2142,404.0 L2138,396.0 2146,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-26\" stroke-width=\"2px\" d=\"M2222,402.0 2222,375.3333333333333 2346.0,375.3333333333333 2346.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-26\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2222,404.0 L2218,396.0 2226,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-27\" stroke-width=\"2px\" d=\"M2302,402.0 2302,388.6666666666667 2343.0,388.6666666666667 2343.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-27\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2302,404.0 L2298,396.0 2306,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-28\" stroke-width=\"2px\" d=\"M2382,402.0 2382,375.3333333333333 2506.0,375.3333333333333 2506.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-28\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2382,404.0 L2378,396.0 2386,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-29\" stroke-width=\"2px\" d=\"M2382,402.0 2382,388.6666666666667 2423.0,388.6666666666667 2423.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-29\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">appos</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2423.0,404.0 L2427.0,396.0 2419.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-30\" stroke-width=\"2px\" d=\"M1582,402.0 1582,308.6666666666667 2521.0,308.6666666666667 2521.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-30\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2521.0,404.0 L2525.0,396.0 2517.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-31\" stroke-width=\"2px\" d=\"M2622,402.0 2622,388.6666666666667 2663.0,388.6666666666667 2663.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-31\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2622,404.0 L2618,396.0 2626,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-32\" stroke-width=\"2px\" d=\"M1422,402.0 1422,295.3333333333333 2684.0,295.3333333333333 2684.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-32\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2684.0,404.0 L2688.0,396.0 2680.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-33\" stroke-width=\"2px\" d=\"M1182,402.0 1182,282.0 2767.0,282.0 2767.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-33\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2767.0,404.0 L2771.0,396.0 2763.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-34\" stroke-width=\"2px\" d=\"M2862,402.0 2862,375.3333333333333 2986.0,375.3333333333333 2986.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-34\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">advmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2862,404.0 L2858,396.0 2866,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-35\" stroke-width=\"2px\" d=\"M2942,402.0 2942,388.6666666666667 2983.0,388.6666666666667 2983.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-35\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2942,404.0 L2938,396.0 2946,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-36\" stroke-width=\"2px\" d=\"M1182,402.0 1182,268.66666666666663 3010.0,268.66666666666663 3010.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-36\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3010.0,404.0 L3014.0,396.0 3006.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-37\" stroke-width=\"2px\" d=\"M3102,402.0 3102,375.3333333333333 3226.0,375.3333333333333 3226.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-37\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3102,404.0 L3098,396.0 3106,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-38\" stroke-width=\"2px\" d=\"M3182,402.0 3182,388.6666666666667 3223.0,388.6666666666667 3223.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-38\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3182,404.0 L3178,396.0 3186,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-39\" stroke-width=\"2px\" d=\"M3022,402.0 3022,362.0 3229.0,362.0 3229.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-39\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3229.0,404.0 L3233.0,396.0 3225.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-40\" stroke-width=\"2px\" d=\"M3342,402.0 3342,388.6666666666667 3383.0,388.6666666666667 3383.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-40\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3342,404.0 L3338,396.0 3346,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-457d5ebe068f4cd89e252a6bf897a93b-0-41\" stroke-width=\"2px\" d=\"M3262,402.0 3262,375.3333333333333 3386.0,375.3333333333333 3386.0,402.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-457d5ebe068f4cd89e252a6bf897a93b-0-41\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3386.0,404.0 L3390.0,396.0 3382.0,396.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testcase = (u\"Disfunction of neutrophils in sepsis has been described in previous studies, however, little is known about the role of microRNA-let-7b (miR-let-7b), toll-like receptor 4 (TLR4), and nuclear factor kappa B (NF-kappaB) activity in neutrophils and how they participate in the development of sepsis.\", 21, 39)\n",
    "\n",
    "analyse(testcase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 Functional ADJ amod association 11\n",
      "11 1 association NOUN nsubjpass established 72\n",
      "23 2 between ADP case miR 31\n",
      "31 3 miR NOUN nmod association 11\n",
      "35 4 124a NOUN dep miR 31\n",
      "40 5 and CCONJ cc 124a 35\n",
      "44 6 CDK6 NOUN conj 124a 35\n",
      "49 7 in ADP case medulloblastoma 52\n",
      "52 8 medulloblastoma NOUN nmod association 11\n",
      "68 9 was AUX auxpass established 72\n",
      "72 10 established VERB ROOT established 72\n",
      "84 11 using VERB xcomp established 72\n",
      "90 12 luciferase NOUN compound assays 101\n",
      "101 13 assays NOUN dobj using 84\n",
      "107 14 . PUNCT punct established 72\n",
      "124a CDK6\n",
      "CDK6 124a\n",
      "Conjuncts [CDK6, 124a]\n",
      "neighbour CDK6 in\n",
      "neighbour CDK6 in\n",
      "Conjunction before fc 124a\n",
      "Conjuncts [124a, CDK6]\n",
      "neighbour 124a and\n",
      "neighbour 124a and\n",
      "Conjunction before fc CDK6\n",
      "Conjunctions\n",
      "124a [124a, CDK6]\n",
      "['124a', 'and']\n",
      "achildren {CDK6}\n",
      "achildren False\n",
      "SDP sent: Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\n",
      "SDP     : [(124a, 'NOUN', 'dep', False), (established, 'VERB', 'ROOT', False), (association, 'NOUN', 'nsubjpass', False), (miR, 'NOUN', 'nmod', False), (124a, 'NOUN', 'dep', False)]\n",
      "[CDK6]\n",
      "deepcheck connection rule noun found using luciferase [using, luciferase, assays]\n",
      "creating subtree at using xcomp\n",
      "0 [Functional, association, between, miR, 124a, and, CDK6, in, medulloblastoma, was, established] 11 0 Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\n",
      "1 [using, luciferase, assays] 3 84 using luciferase assays.\n",
      "Conjuncts [CDK6, 124a]\n",
      "neighbour CDK6 in\n",
      "neighbour CDK6 in\n",
      "Conjunction before fc 124a\n",
      "Conjuncts [124a, CDK6]\n",
      "neighbour 124a and\n",
      "neighbour 124a and\n",
      "Conjunction before fc CDK6\n",
      "(False, {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': True, 'negation': False})\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"d33adb73d4294392814217648fceafe6-0\" class=\"displacy\" width=\"1170\" height=\"297.0\" direction=\"ltr\" style=\"max-width: none; height: 297.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">Functional</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"130\">association</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"130\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"210\">between</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"210\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"290\">miR</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"290\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"370\">124a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"370\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"450\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"450\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"530\">CDK6</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"530\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"610\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"610\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"690\">medulloblastoma</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"690\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"770\">was</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"770\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"850\">established</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"850\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"930\">using</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"930\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1010\">luciferase</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1010\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1090\">assays.</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1090\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-0\" stroke-width=\"2px\" d=\"M62,162.0 62,148.66666666666666 121.0,148.66666666666666 121.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M62,164.0 L58,156.0 66,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-1\" stroke-width=\"2px\" d=\"M142,162.0 142,108.66666666666666 850.0,108.66666666666666 850.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubjpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M142,164.0 L138,156.0 146,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-2\" stroke-width=\"2px\" d=\"M222,162.0 222,148.66666666666666 281.0,148.66666666666666 281.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M222,164.0 L218,156.0 226,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-3\" stroke-width=\"2px\" d=\"M142,162.0 142,135.33333333333334 284.0,135.33333333333334 284.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M284.0,164.0 L288.0,156.0 280.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-4\" stroke-width=\"2px\" d=\"M302,162.0 302,148.66666666666666 361.0,148.66666666666666 361.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dep</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M361.0,164.0 L365.0,156.0 357.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-5\" stroke-width=\"2px\" d=\"M382,162.0 382,148.66666666666666 441.0,148.66666666666666 441.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M441.0,164.0 L445.0,156.0 437.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-6\" stroke-width=\"2px\" d=\"M382,162.0 382,135.33333333333334 524.0,135.33333333333334 524.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M524.0,164.0 L528.0,156.0 520.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-7\" stroke-width=\"2px\" d=\"M622,162.0 622,148.66666666666666 681.0,148.66666666666666 681.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M622,164.0 L618,156.0 626,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-8\" stroke-width=\"2px\" d=\"M142,162.0 142,122.0 687.0,122.0 687.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M687.0,164.0 L691.0,156.0 683.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-9\" stroke-width=\"2px\" d=\"M782,162.0 782,148.66666666666666 841.0,148.66666666666666 841.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">auxpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M782,164.0 L778,156.0 786,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-10\" stroke-width=\"2px\" d=\"M862,162.0 862,148.66666666666666 921.0,148.66666666666666 921.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">xcomp</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M921.0,164.0 L925.0,156.0 917.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-11\" stroke-width=\"2px\" d=\"M1022,162.0 1022,148.66666666666666 1081.0,148.66666666666666 1081.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1022,164.0 L1018,156.0 1026,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-d33adb73d4294392814217648fceafe6-0-12\" stroke-width=\"2px\" d=\"M942,162.0 942,135.33333333333334 1084.0,135.33333333333334 1084.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-d33adb73d4294392814217648fceafe6-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1084.0,164.0 L1088.0,156.0 1080.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testcase = (u\"Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\", 4, 6)\n",
    "\n",
    "analyse(testcase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 Overexpression NOUN nsubj reduces 50\n",
      "15 1 of ADP case mir-7 18\n",
      "18 2 mir-7 PROPN nmod Overexpression 0\n",
      "24 3 and CCONJ cc mir-7 18\n",
      "28 4 mir-153 PROPN conj mir-7 18\n",
      "36 5 significantly ADV advmod reduces 50\n",
      "50 6 reduces VERB ROOT reduces 50\n",
      "58 7 endogenous ADJ amod levels 85\n",
      "69 8 alpha-synuclein PROPN amod levels 85\n",
      "85 9 levels NOUN dobj reduces 50\n",
      "91 10 , PUNCT punct reduces 50\n",
      "93 11 whereas SCONJ dep reduces 50\n",
      "101 12 inhibition NOUN nsubj enhances 133\n",
      "112 13 of ADP case mir-7 115\n",
      "115 14 mir-7 PROPN nmod inhibition 101\n",
      "121 15 and CCONJ cc mir-7 115\n",
      "125 16 mir-153 PROPN conj mir-7 115\n",
      "133 17 enhances VERB parataxis reduces 50\n",
      "142 18 translation NOUN dobj enhances 133\n",
      "154 19 of ADP case construct 170\n",
      "157 20 a DET det construct 170\n",
      "159 21 luciferase NOUN compound construct 170\n",
      "170 22 construct NOUN nmod translation 142\n",
      "180 23 bearing VERB acl construct 170\n",
      "188 24 the DET det alpha-s 192\n",
      "192 25 alpha-s NOUN dobj bearing 180\n",
      "reduces ,\n",
      ", reduces\n",
      "Conjuncts [mir-153, mir-7]\n",
      "neighbour mir-153 significantly\n",
      "neighbour mir-153 significantly\n",
      "Conjunction before fc mir-7\n",
      "fc add children for mir-7\n",
      "Conjuncts [mir-7, mir-153]\n",
      "neighbour mir-7 and\n",
      "neighbour mir-7 and\n",
      "Conjunction before fc mir-153\n",
      "Conjuncts [mir-153, mir-7]\n",
      "neighbour mir-153 enhances\n",
      "neighbour mir-153 enhances\n",
      "Conjunction before fc mir-7\n",
      "fc add children for mir-7\n",
      "Conjuncts [mir-7, mir-153]\n",
      "neighbour mir-7 and\n",
      "neighbour mir-7 and\n",
      "Conjunction before fc mir-153\n",
      "mir-7 mir-7\n",
      "mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition\n",
      "Conjunctions\n",
      "mir-7 [of, mir-7, mir-153]\n",
      "mir-7 [of, mir-7, mir-153]\n",
      "SDP sent: Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\n",
      "SDP     : [(reduces, 'VERB', 'ROOT', False), (reduces, 'VERB', 'ROOT', False)]\n",
      "[mir-153]\n",
      "[mir-153]\n",
      "deepcheck connection rule noun found enhances inhibition [inhibition, of, mir-7, and, mir-153, enhances, translation, of, a, luciferase, construct, bearing, the, alpha-s]\n",
      "0 [Overexpression, of, mir-7, and, mir-153, significantly, reduces, endogenous, alpha-synuclein, levels] 10 0 Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\n",
      "1 [inhibition, of, mir-7, and, mir-153, enhances, translation, of, a, luciferase, construct, bearing, the, alpha-s] 14 101 inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\n",
      "Conjuncts [mir-153, mir-7]\n",
      "neighbour mir-153 significantly\n",
      "neighbour mir-153 significantly\n",
      "Conjunction before fc mir-7\n",
      "fc add children for mir-7\n",
      "Conjuncts [mir-7, mir-153]\n",
      "neighbour mir-7 and\n",
      "neighbour mir-7 and\n",
      "Conjunction before fc mir-153\n",
      "Conjuncts [mir-153, mir-7]\n",
      "neighbour mir-153 enhances\n",
      "neighbour mir-153 enhances\n",
      "Conjunction before fc mir-7\n",
      "fc add children for mir-7\n",
      "Conjuncts [mir-7, mir-153]\n",
      "neighbour mir-7 and\n",
      "neighbour mir-7 and\n",
      "Conjunction before fc mir-153\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False})\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"fd36a89503b646fda43baaf9da2eb5da-0\" class=\"displacy\" width=\"2050\" height=\"417.0\" direction=\"ltr\" style=\"max-width: none; height: 417.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">Overexpression</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"130\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"130\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"210\">mir-7</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"210\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"290\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"290\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"370\">mir-153</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"370\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"450\">significantly</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"450\">ADV</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"530\">reduces</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"530\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"610\">endogenous</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"610\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"690\">alpha-synuclein</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"690\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"770\">levels,</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"770\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"850\">whereas</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"850\">SCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"930\">inhibition</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"930\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1010\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1010\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1090\">mir-7</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1090\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1170\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1170\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1250\">mir-153</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1250\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1330\">enhances</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1330\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1410\">translation</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1410\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1490\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1490\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1570\">a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1570\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1650\">luciferase</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1650\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1730\">construct</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1730\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1810\">bearing</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1810\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1890\">the</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1890\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1970\">alpha-s</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1970\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-0\" stroke-width=\"2px\" d=\"M62,282.0 62,202.0 527.0,202.0 527.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M62,284.0 L58,276.0 66,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-1\" stroke-width=\"2px\" d=\"M142,282.0 142,268.6666666666667 192.0,268.6666666666667 192.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M142,284.0 L138,276.0 146,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-2\" stroke-width=\"2px\" d=\"M62,282.0 62,255.33333333333334 195.0,255.33333333333334 195.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M195.0,284.0 L199.0,276.0 191.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-3\" stroke-width=\"2px\" d=\"M222,282.0 222,268.6666666666667 272.0,268.6666666666667 272.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M272.0,284.0 L276.0,276.0 268.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-4\" stroke-width=\"2px\" d=\"M222,282.0 222,255.33333333333334 355.0,255.33333333333334 355.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M355.0,284.0 L359.0,276.0 351.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-5\" stroke-width=\"2px\" d=\"M462,282.0 462,268.6666666666667 512.0,268.6666666666667 512.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">advmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M462,284.0 L458,276.0 466,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-6\" stroke-width=\"2px\" d=\"M622,282.0 622,255.33333333333334 755.0,255.33333333333334 755.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M622,284.0 L618,276.0 626,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-7\" stroke-width=\"2px\" d=\"M702,282.0 702,268.6666666666667 752.0,268.6666666666667 752.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M702,284.0 L698,276.0 706,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-8\" stroke-width=\"2px\" d=\"M542,282.0 542,242.0 758.0,242.0 758.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M758.0,284.0 L762.0,276.0 754.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-9\" stroke-width=\"2px\" d=\"M542,282.0 542,228.66666666666666 841.0,228.66666666666666 841.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dep</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M841.0,284.0 L845.0,276.0 837.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-10\" stroke-width=\"2px\" d=\"M942,282.0 942,215.33333333333331 1324.0,215.33333333333331 1324.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M942,284.0 L938,276.0 946,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-11\" stroke-width=\"2px\" d=\"M1022,282.0 1022,268.6666666666667 1072.0,268.6666666666667 1072.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1022,284.0 L1018,276.0 1026,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-12\" stroke-width=\"2px\" d=\"M942,282.0 942,255.33333333333334 1075.0,255.33333333333334 1075.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1075.0,284.0 L1079.0,276.0 1071.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-13\" stroke-width=\"2px\" d=\"M1102,282.0 1102,268.6666666666667 1152.0,268.6666666666667 1152.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-13\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1152.0,284.0 L1156.0,276.0 1148.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-14\" stroke-width=\"2px\" d=\"M1102,282.0 1102,255.33333333333334 1235.0,255.33333333333334 1235.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-14\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1235.0,284.0 L1239.0,276.0 1231.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-15\" stroke-width=\"2px\" d=\"M542,282.0 542,188.66666666666669 1330.0,188.66666666666669 1330.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-15\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">parataxis</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1330.0,284.0 L1334.0,276.0 1326.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-16\" stroke-width=\"2px\" d=\"M1342,282.0 1342,268.6666666666667 1392.0,268.6666666666667 1392.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-16\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1392.0,284.0 L1396.0,276.0 1388.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-17\" stroke-width=\"2px\" d=\"M1502,282.0 1502,242.0 1718.0,242.0 1718.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-17\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1502,284.0 L1498,276.0 1506,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-18\" stroke-width=\"2px\" d=\"M1582,282.0 1582,255.33333333333334 1715.0,255.33333333333334 1715.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-18\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1582,284.0 L1578,276.0 1586,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-19\" stroke-width=\"2px\" d=\"M1662,282.0 1662,268.6666666666667 1712.0,268.6666666666667 1712.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-19\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1662,284.0 L1658,276.0 1666,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-20\" stroke-width=\"2px\" d=\"M1422,282.0 1422,228.66666666666666 1721.0,228.66666666666666 1721.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-20\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1721.0,284.0 L1725.0,276.0 1717.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-21\" stroke-width=\"2px\" d=\"M1742,282.0 1742,268.6666666666667 1792.0,268.6666666666667 1792.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-21\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">acl</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1792.0,284.0 L1796.0,276.0 1788.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-22\" stroke-width=\"2px\" d=\"M1902,282.0 1902,268.6666666666667 1952.0,268.6666666666667 1952.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-22\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1902,284.0 L1898,276.0 1906,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-fd36a89503b646fda43baaf9da2eb5da-0-23\" stroke-width=\"2px\" d=\"M1822,282.0 1822,255.33333333333334 1955.0,255.33333333333334 1955.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-fd36a89503b646fda43baaf9da2eb5da-0-23\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1955.0,284.0 L1959.0,276.0 1951.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testcase = (u\"Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\", 6, 10)\n",
    "\n",
    "analyse(testcase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 The DET det relevance 13\n",
      "4 1 clinical ADJ amod relevance 13\n",
      "13 2 relevance NOUN nsubjpass evaluated 43\n",
      "23 3 of ADP case findings 30\n",
      "26 4 our PRON nmod:poss findings 30\n",
      "30 5 findings NOUN nmod relevance 13\n",
      "39 6 was AUX auxpass evaluated 43\n",
      "43 7 evaluated VERB ROOT evaluated 43\n",
      "53 8 in ADP case samples 86\n",
      "56 9 HIV-encephalitis NOUN compound samples 86\n",
      "73 10 ( PUNCT punct HIVE 74\n",
      "74 11 HIVE NOUN appos HIV-encephalitis 56\n",
      "78 12 ) PUNCT punct HIVE 74\n",
      "80 13 brain NOUN compound samples 86\n",
      "86 14 samples NOUN nmod evaluated 43\n",
      "94 15 in ADP case which 97\n",
      "97 16 which PRON nmod observed 167\n",
      "103 17 decreased VERB amod levels 113\n",
      "113 18 levels NOUN nsubjpass observed 167\n",
      "120 19 of ADP case MCP-2 123\n",
      "123 20 MCP-2 PROPN nmod levels 113\n",
      "129 21 and CCONJ cc levels 113\n",
      "133 22 increased VERB amod levels 143\n",
      "143 23 levels NOUN conj levels 113\n",
      "150 24 of ADP case mir-146a 153\n",
      "153 25 mir-146a PROPN nmod levels 143\n",
      "162 26 were AUX auxpass observed 167\n",
      "167 27 observed VERB acl:relcl samples 86\n",
      "175 28 , PUNCT punct evaluated 43\n",
      "177 29 suggesting VERB xcomp evaluated 43\n",
      "188 30 a DET det role 190\n",
      "190 31 role NOUN dobj suggesting 177\n",
      "195 32 for ADP case mir-146a 199\n",
      "199 33 mir-146a PROPN nmod role 190\n",
      "208 34 in ADP case maintenance 215\n",
      "211 35 the DET det maintenance 215\n",
      "215 36 maintenance NOUN nmod role 190\n",
      "227 37 of ADP case inflammation 251\n",
      "230 38 HIV-mediated ADJ amod inflammation 251\n",
      "243 39 chronic ADJ amod inflammation 251\n",
      "251 40 inflammation NOUN nmod maintenance 215\n",
      "264 41 of ADP case brain 271\n",
      "267 42 the DET det brain 271\n",
      "271 43 brain NOUN nmod inflammation 251\n",
      "276 44 . PUNCT punct evaluated 43\n",
      "MCP-2 mir-146a\n",
      "mir-146a MCP-2\n",
      "Conjuncts [levels, levels]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, levels]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "HIVE levels\n",
      ") brain samples in which\n",
      "Conjunctions\n",
      "HIVE [(, )]\n",
      "levels [decreased, levels, of, MCP-2, increased, levels, of, mir-146a]\n",
      "achildren {mir-146a}\n",
      "achildren False\n",
      "SDP sent: The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "SDP     : [(MCP-2, 'PROPN', 'nmod', False), (levels, 'NOUN', 'nsubjpass', True), (levels, 'NOUN', 'conj', False), (mir-146a, 'PROPN', 'nmod', False)]\n",
      "[increased, levels, of, mir-146a]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "0 [The, clinical, relevance, of, our, findings, was, evaluated, in, HIV-encephalitis, (, HIVE, ), brain, samples, in, which, decreased, levels, of, MCP-2, and, increased, levels, of, mir-146a, were, observed] 28 0 The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "1 [suggesting, a, role, for, mir-146a, in, the, maintenance, of, HIV-mediated, chronic, inflammation, of, the, brain] 15 177 suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "Conjuncts [levels, levels]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, levels]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "(False, {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': True, 'negation': False})\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"340528f3349948fc85a69897dc8555fb-0\" class=\"displacy\" width=\"3330\" height=\"497.0\" direction=\"ltr\" style=\"max-width: none; height: 497.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">The</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"130\">clinical</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"130\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"210\">relevance</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"210\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"290\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"290\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"370\">our</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"370\">PRON</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"450\">findings</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"450\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"530\">was</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"530\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"610\">evaluated</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"610\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"690\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"690\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"770\">HIV-encephalitis (</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"770\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"850\">HIVE)</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"850\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"930\">brain</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"930\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1010\">samples</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1010\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1090\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1090\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1170\">which</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1170\">PRON</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1250\">decreased</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1250\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1330\">levels</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1330\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1410\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1410\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1490\">MCP-2</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1490\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1570\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1570\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1650\">increased</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1650\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1730\">levels</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1730\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1810\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1810\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1890\">mir-146a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1890\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1970\">were</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1970\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2050\">observed,</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2050\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2130\">suggesting</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2130\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2210\">a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2210\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2290\">role</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2290\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2370\">for</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2370\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2450\">mir-146a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2450\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2530\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2530\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2610\">the</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2610\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2690\">maintenance</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2690\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2770\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2770\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2850\">HIV-mediated</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2850\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2930\">chronic</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2930\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3010\">inflammation</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3010\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3090\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3090\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3170\">the</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3170\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3250\">brain.</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3250\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-0\" stroke-width=\"2px\" d=\"M62,362.0 62,335.3333333333333 189.0,335.3333333333333 189.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M62,364.0 L58,356.0 66,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-1\" stroke-width=\"2px\" d=\"M142,362.0 142,348.6666666666667 186.0,348.6666666666667 186.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M142,364.0 L138,356.0 146,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-2\" stroke-width=\"2px\" d=\"M222,362.0 222,295.3333333333333 598.0,295.3333333333333 598.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubjpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M222,364.0 L218,356.0 226,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-3\" stroke-width=\"2px\" d=\"M302,362.0 302,335.3333333333333 429.0,335.3333333333333 429.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M302,364.0 L298,356.0 306,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-4\" stroke-width=\"2px\" d=\"M382,362.0 382,348.6666666666667 426.0,348.6666666666667 426.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod:poss</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M382,364.0 L378,356.0 386,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-5\" stroke-width=\"2px\" d=\"M222,362.0 222,322.0 432.0,322.0 432.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M432.0,364.0 L436.0,356.0 428.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-6\" stroke-width=\"2px\" d=\"M542,362.0 542,348.6666666666667 586.0,348.6666666666667 586.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">auxpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M542,364.0 L538,356.0 546,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-7\" stroke-width=\"2px\" d=\"M702,362.0 702,308.6666666666667 995.0,308.6666666666667 995.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M702,364.0 L698,356.0 706,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-8\" stroke-width=\"2px\" d=\"M782,362.0 782,322.0 992.0,322.0 992.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M782,364.0 L778,356.0 786,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-9\" stroke-width=\"2px\" d=\"M782,362.0 782,348.6666666666667 826.0,348.6666666666667 826.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">appos</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M826.0,364.0 L830.0,356.0 822.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-10\" stroke-width=\"2px\" d=\"M942,362.0 942,348.6666666666667 986.0,348.6666666666667 986.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M942,364.0 L938,356.0 946,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-11\" stroke-width=\"2px\" d=\"M622,362.0 622,295.3333333333333 998.0,295.3333333333333 998.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M998.0,364.0 L1002.0,356.0 994.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-12\" stroke-width=\"2px\" d=\"M1102,362.0 1102,348.6666666666667 1146.0,348.6666666666667 1146.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1102,364.0 L1098,356.0 1106,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-13\" stroke-width=\"2px\" d=\"M1182,362.0 1182,268.6666666666667 2044.0,268.6666666666667 2044.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-13\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1182,364.0 L1178,356.0 1186,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-14\" stroke-width=\"2px\" d=\"M1262,362.0 1262,348.6666666666667 1306.0,348.6666666666667 1306.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-14\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1262,364.0 L1258,356.0 1266,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-15\" stroke-width=\"2px\" d=\"M1342,362.0 1342,282.0 2041.0,282.0 2041.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-15\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubjpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1342,364.0 L1338,356.0 1346,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-16\" stroke-width=\"2px\" d=\"M1422,362.0 1422,348.6666666666667 1466.0,348.6666666666667 1466.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-16\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1422,364.0 L1418,356.0 1426,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-17\" stroke-width=\"2px\" d=\"M1342,362.0 1342,335.3333333333333 1469.0,335.3333333333333 1469.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-17\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1469.0,364.0 L1473.0,356.0 1465.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-18\" stroke-width=\"2px\" d=\"M1342,362.0 1342,322.0 1552.0,322.0 1552.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-18\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1552.0,364.0 L1556.0,356.0 1548.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-19\" stroke-width=\"2px\" d=\"M1662,362.0 1662,348.6666666666667 1706.0,348.6666666666667 1706.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-19\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1662,364.0 L1658,356.0 1666,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-20\" stroke-width=\"2px\" d=\"M1342,362.0 1342,295.3333333333333 1718.0,295.3333333333333 1718.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-20\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1718.0,364.0 L1722.0,356.0 1714.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-21\" stroke-width=\"2px\" d=\"M1822,362.0 1822,348.6666666666667 1866.0,348.6666666666667 1866.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-21\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1822,364.0 L1818,356.0 1826,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-22\" stroke-width=\"2px\" d=\"M1742,362.0 1742,335.3333333333333 1869.0,335.3333333333333 1869.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-22\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1869.0,364.0 L1873.0,356.0 1865.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-23\" stroke-width=\"2px\" d=\"M1982,362.0 1982,348.6666666666667 2026.0,348.6666666666667 2026.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-23\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">auxpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1982,364.0 L1978,356.0 1986,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-24\" stroke-width=\"2px\" d=\"M1022,362.0 1022,255.33333333333331 2047.0,255.33333333333331 2047.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-24\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">acl:relcl</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2047.0,364.0 L2051.0,356.0 2043.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-25\" stroke-width=\"2px\" d=\"M622,362.0 622,242.0 2130.0,242.0 2130.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-25\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">xcomp</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2130.0,364.0 L2134.0,356.0 2126.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-26\" stroke-width=\"2px\" d=\"M2222,362.0 2222,348.6666666666667 2266.0,348.6666666666667 2266.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-26\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2222,364.0 L2218,356.0 2226,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-27\" stroke-width=\"2px\" d=\"M2142,362.0 2142,335.3333333333333 2269.0,335.3333333333333 2269.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-27\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2269.0,364.0 L2273.0,356.0 2265.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-28\" stroke-width=\"2px\" d=\"M2382,362.0 2382,348.6666666666667 2426.0,348.6666666666667 2426.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-28\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2382,364.0 L2378,356.0 2386,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-29\" stroke-width=\"2px\" d=\"M2302,362.0 2302,335.3333333333333 2429.0,335.3333333333333 2429.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-29\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2429.0,364.0 L2433.0,356.0 2425.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-30\" stroke-width=\"2px\" d=\"M2542,362.0 2542,335.3333333333333 2669.0,335.3333333333333 2669.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-30\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2542,364.0 L2538,356.0 2546,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-31\" stroke-width=\"2px\" d=\"M2622,362.0 2622,348.6666666666667 2666.0,348.6666666666667 2666.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-31\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2622,364.0 L2618,356.0 2626,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-32\" stroke-width=\"2px\" d=\"M2302,362.0 2302,295.3333333333333 2678.0,295.3333333333333 2678.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-32\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2678.0,364.0 L2682.0,356.0 2674.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-33\" stroke-width=\"2px\" d=\"M2782,362.0 2782,322.0 2992.0,322.0 2992.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-33\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2782,364.0 L2778,356.0 2786,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-34\" stroke-width=\"2px\" d=\"M2862,362.0 2862,335.3333333333333 2989.0,335.3333333333333 2989.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-34\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2862,364.0 L2858,356.0 2866,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-35\" stroke-width=\"2px\" d=\"M2942,362.0 2942,348.6666666666667 2986.0,348.6666666666667 2986.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-35\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2942,364.0 L2938,356.0 2946,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-36\" stroke-width=\"2px\" d=\"M2702,362.0 2702,308.6666666666667 2995.0,308.6666666666667 2995.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-36\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2995.0,364.0 L2999.0,356.0 2991.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-37\" stroke-width=\"2px\" d=\"M3102,362.0 3102,335.3333333333333 3229.0,335.3333333333333 3229.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-37\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3102,364.0 L3098,356.0 3106,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-38\" stroke-width=\"2px\" d=\"M3182,362.0 3182,348.6666666666667 3226.0,348.6666666666667 3226.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-38\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3182,364.0 L3178,356.0 3186,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-340528f3349948fc85a69897dc8555fb-0-39\" stroke-width=\"2px\" d=\"M3022,362.0 3022,322.0 3232.0,322.0 3232.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-340528f3349948fc85a69897dc8555fb-0-39\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3232.0,364.0 L3236.0,356.0 3228.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "analyse((u\"The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\", 20, 25)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "doc = nlp(testcase[0])\n",
    "\n",
    "allTokens = {t.i: t for t in doc}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "[x for x in allTokens[23].children]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "allRels[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testCase = (u\"We show here that miR-29a, -29b-1, and -9 can regulate BACE1 expression in vitro.\", 4, 12)\n",
    "testCase = (u\"It has been predicted that miR-146a may target Tata Binding Protein (TBP).\",5, 12)\n",
    "testCase = (u\"Our results showed that the expression of miR-106b was inversely correlated with TβRII protein levels and miR-106b can directly inhibit the TβRII translation in vitro.\", 7, 21)\n",
    "\n",
    "relCheck = textmining.MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "doc = nlp(testCase[0])\n",
    "\n",
    "mirword = doc[testCase[1]]\n",
    "geneword = doc[testCase[2]]\n",
    "\n",
    "verbose=False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conjResult, conjs = relCheck.checkCommonConj(doc, mirword, geneword, verbose)\n",
    "sdpPass, passive, negated = relCheck.checkSDP(doc, mirword, geneword, verbose)\n",
    "compPass = relCheck.checkCompartments(doc, mirword, geneword, verbose)\n",
    "sigPathway = relCheck.checkSurContext(doc, mirword, geneword, verbose)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sdpRes = relCheck._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "sdpRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testCase = (u\"We show here that miR-29a, -29b-1, and -9 can regulate BACE1 expression in vitro.\", 4, 12)\n",
    "testCase = (u\"It has been predicted that miR-146a may target Tata Binding Protein (TBP).\",5, 12)\n",
    "testCase = (u\"Our results showed that the expression of miR-106b was inversely correlated with TβRII protein levels and miR-106b can directly inhibit the TβRII translation in vitro.\", 7, 21)\n",
    "testCase = (u\"The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 6, 16)\n",
    "testCase = (u\"These findings identify miR-29b as a novel posttranscriptional regulator of PGRN expression, raising the possibility that miR-29b or other miRNAs might be targeted therapeutically to increase hPGRN levels in some FTD patients.\", 3,10)\n",
    "relCheck = textmining.MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "doc = nlp(testCase[0])\n",
    "\n",
    "mirword = doc[testCase[1]]\n",
    "geneword = doc[testCase[2]]\n",
    "\n",
    "verbose=False\n",
    "\n",
    "from collections import Counter\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading stems\n"
     ]
    }
   ],
   "source": [
    "def loadStems():\n",
    "    \n",
    "    print(\"Loading stems\")\n",
    "    allRels = []\n",
    "    stem2class = {}\n",
    "\n",
    "    renameClass = {\n",
    "        'INT': 'NEU',\n",
    "        \"BOOST\": \"POS\",\n",
    "        \"COMBINE\": \"NEU\",\n",
    "        \"CHANGE\": \"NEU\",\n",
    "    }\n",
    "\n",
    "    with open(\"/mnt/d/dev/git/miRExplore/python/allrels.csv\") as fin:\n",
    "\n",
    "        for line in fin:\n",
    "            aline = line.strip().split(\",\", 1)\n",
    "            aregSyns = aline[1].split(\":\")\n",
    "            regSyns = []\n",
    "            if len(aregSyns) > 1:\n",
    "                regSyns = aregSyns[1].split(\"|\")\n",
    "            \n",
    "            aline[1] = aline[1].split(\":\")[0]           \n",
    "\n",
    "            regClass = aline[0]\n",
    "            regStem = aline[1]\n",
    "\n",
    "            if regClass in renameClass:\n",
    "                regClass = renameClass[regClass]\n",
    "                \n",
    "            stem2class[regStem] = regClass\n",
    "                \n",
    "            for regSyn in regSyns:\n",
    "                if not regSyn in stem2class:\n",
    "                    stem2class[regSyn] = regClass\n",
    "\n",
    "\n",
    "            allRels.append((regClass, regStem))\n",
    "\n",
    "    allStems = [x for x in stem2class]\n",
    "    \n",
    "    return allStems, allRels, stem2class\n",
    "\n",
    "allStems, allRels, stem2class = loadStems()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "from textmining.MirGeneRelCheck import SentenceRelationChecker, SentenceRelationClassifier\n",
    "from utils.tmutils import normalize_gene_names\n",
    "from collections import Counter\n",
    "from lxml import etree\n",
    "from sklearn.metrics import multilabel_confusion_matrix, classification_report\n",
    "from itertools import chain, combinations\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "scaiBase = \"/mnt/d/owncloud/data/miRExplore/scai_corpus/\"\n",
    "scaiBase = \"/mnt/f/dev/git/miRExplore/python/scai_corpus/\"\n",
    "\n",
    "base = \"TRAIN\"\n",
    "\n",
    "if base == \"TRAIN\":\n",
    "    scaiFile = \"miRNA_train_fixed.xml\"\n",
    "    relsFile = \"scai_train_rels.tsv\"\n",
    "elif base == \"TEST\":\n",
    "    scaiFile = \"miRNA_test_fixed.xml\"\n",
    "    relsFile = \"scai_test_rels.tsv\"\n",
    "    \n",
    "normGeneSymbols= normalize_gene_names(path=\"/mnt/d/owncloud/data/miRExplore/obodir/\" + \"/hgnc_no_withdrawn.syn\")\n",
    "\n",
    "relChecker = SentenceRelationChecker(nlp)\n",
    "\n",
    "correctIdentified = 0\n",
    "incorrectIdentified = 0\n",
    "totalChecks = 0\n",
    "incorrectClass = Counter()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "loadRelations() missing 1 required positional argument: 'relsFile'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-28-5d6986e8447d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mrelClasses\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mloadRelations\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m: loadRelations() missing 1 required positional argument: 'relsFile'"
     ]
    }
   ],
   "source": [
    "relClasses = loadRelations()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp = spacy.load('/mnt/f/spacy/en_core_sci_lg-0.2.4/en_core_sci_lg/en_core_sci_lg-0.2.4/')\n",
    "nlp_ent = spacy.load(\"/mnt/f/spacy/en_ner_bionlp13cg_md-0.2.4/en_ner_bionlp13cg_md/en_ner_bionlp13cg_md-0.2.4\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "scaiBase = \"/mnt/d/owncloud/data/miRExplore/scai_corpus/\"\n",
    "\n",
    "normGeneSymbols= normalize_gene_names(path=scaiBase + \"/../obodir/\" + \"/hgnc_no_withdrawn.syn\")\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "def loadRelations(relsFile):\n",
    "    reldir2new = {}\n",
    "    reldir2new[\"TARGET\"] = \"NEU\"\n",
    "    reldir2new[\"INVCORR\"] = \"NEU\"\n",
    "    reldir2new[\"REGULATE\"] = \"NEU\"\n",
    "    relClasses = {}\n",
    "    with open(os.path.join(scaiBase, relsFile)) as fin:\n",
    "        \n",
    "        print(\"Loading relations\", relsFile)\n",
    "        \n",
    "        for lidx, line in enumerate(fin):\n",
    "\n",
    "            if lidx == 0:\n",
    "                continue\n",
    "\n",
    "            line = line.strip().split(\"\\t\")\n",
    "\n",
    "            interactionID = line[0].strip()\n",
    "            relType = line[1].strip()\n",
    "            relRDir = line[2].strip()\n",
    "            relIndirect = line[3].strip()\n",
    "            relPassive = line[4].strip()\n",
    "\n",
    "            if relRDir in reldir2new:\n",
    "                relRDir = reldir2new[relRDir]\n",
    "\n",
    "            relClasses[interactionID] = (relType, relRDir, relIndirect, relPassive)\n",
    "            \n",
    "        print(\"Interactions\", len(relClasses))\n",
    "        print(\"Non NA Interactions\", len([x for x in relClasses if not relClasses[x][0] == \"NA\"]))\n",
    "            \n",
    "    return relClasses\n",
    "\n",
    "def all_subsets(ss):\n",
    "    \n",
    "    return chain(*map(lambda x: combinations(ss, x), range(0, len(ss)+1)))\n",
    "\n",
    "def runCheck(numelems, base):\n",
    "    relChecker = SentenceRelationChecker(nlp, nlp_ent=nlp_ent)\n",
    "    relClassifier = SentenceRelationClassifier()\n",
    "    \n",
    "    if base == \"TRAIN\":\n",
    "        scaiFile = \"miRNA_train_fixed.xml\"\n",
    "        relsFile = \"scai_train_rels.tsv\"\n",
    "    elif base == \"TEST\":\n",
    "        scaiFile = \"miRNA_test_fixed.xml\"\n",
    "        relsFile = \"scai_test_rels.tsv\"\n",
    "    \n",
    "    relClasses = loadRelations(relsFile)\n",
    "    \n",
    "    check2results = {}\n",
    "    checkSubsets = [x for x in all_subsets(relClassifier.major_checks)]\n",
    "    checkSubsets = [checkSubsets[-1]]\n",
    "    \n",
    "    print(\"Will test {} subsets.\".format(len(checkSubsets)))\n",
    "    \n",
    "    for subset_checks in checkSubsets:\n",
    "    \n",
    "        print(\"Testing checks\", subset_checks)\n",
    "        \n",
    "        #relClassifier.active_checks = subset_checks\n",
    "\n",
    "        #allStems, allRels, stem2class = loadStems()\n",
    "\n",
    "        with open(os.path.join(scaiBase, scaiFile), 'r') as fin:\n",
    "            tree = etree.parse(fin)\n",
    "            root = tree.getroot()\n",
    "            scaiPairs = []\n",
    "            totalChecks = 0\n",
    "            correctIdentified = 0\n",
    "            incorrectIdentified = 0\n",
    "\n",
    "            incorrectClassified = 0\n",
    "            correctClassified = 0\n",
    "            totalClassified = 0\n",
    "            correctInteractClassified = 0\n",
    "            totalValidClassified = 0\n",
    "            totalClassifiable = 0\n",
    "            totalMirGeneDownClassified = 0\n",
    "\n",
    "            errorByDetect = Counter()\n",
    "            classifiedByDetect = Counter()\n",
    "            elemCaseCounter = Counter()\n",
    "            incorrectClass = Counter()\n",
    "\n",
    "            classifyTrue = []\n",
    "            classifyPred = []\n",
    "\n",
    "            for elem in root.findall(\".//document\"):\n",
    "\n",
    "                pmid = elem.attrib['origId']\n",
    "\n",
    "                for sentElem in elem:\n",
    "\n",
    "                    allEntities = sentElem.findall(\".//entity\")\n",
    "                    allPairs = sentElem.findall(\".//pair\")\n",
    "\n",
    "                    sentText = sentElem.attrib[\"text\"]\n",
    "\n",
    "                    entId2elem = {}\n",
    "\n",
    "                    for entity in allEntities:\n",
    "                        entId = entity.attrib['id']\n",
    "                        entText = entity.attrib['text']\n",
    "                        entType = entity.attrib['type']\n",
    "                        entOffset = tuple([int(x) for x in entity.attrib['charOffset'].split(\"-\")])\n",
    "\n",
    "                        if entType in [\"Specific_miRNAs\", \"Genes/Proteins\"]:\n",
    "\n",
    "                            if \"Genes\" in entType:\n",
    "                                if entText in normGeneSymbols:\n",
    "                                    entText = normGeneSymbols[entText]\n",
    "                                elif entText.upper() in normGeneSymbols:\n",
    "                                    gene = normGeneSymbols[entText.upper()]\n",
    "                            else:\n",
    "                                try:\n",
    "                                    entText = miRNA(entText).getStringFromParts([miRNAPART.MATURE, miRNAPART.ID, miRNAPART.PRECURSOR])\n",
    "                                except:\n",
    "                                    pass\n",
    "\n",
    "                            entTuple = (entText, entType, (entOffset[0], entOffset[1]+1))\n",
    "                            entId2elem[entId] = entTuple\n",
    "\n",
    "\n",
    "                            sentEntText = sentText[entTuple[2][0]:entTuple[2][1]]\n",
    "\n",
    "                    for pair in allPairs:\n",
    "\n",
    "                        validInteraction = pair.attrib['interaction'].lower() == \"true\"\n",
    "                        pairE1 = pair.attrib['e1']\n",
    "                        pairE2 = pair.attrib['e2']\n",
    "\n",
    "                        #if pairInt == 'true':\n",
    "                        if pairE1 in entId2elem and pairE2 in entId2elem:\n",
    "\n",
    "                            totalChecks += 1\n",
    "\n",
    "                            e1 = entId2elem[pairE1]\n",
    "                            e2 = entId2elem[pairE2]\n",
    "\n",
    "                            if not e1[1] in [\"Specific_miRNAs\"]:\n",
    "\n",
    "                                tmp=e1\n",
    "                                e1=e2\n",
    "                                e2=tmp\n",
    "\n",
    "\n",
    "                            relRes = relChecker.check_sentence(sentText\n",
    "                                                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": e1[2]}\n",
    "                                                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": e2[2]}\n",
    "                                                                , fix_special_chars=False\n",
    "                                                                , relClassifier=relClassifier.classify, verbose=False\n",
    "                                                                )\n",
    "\n",
    "\n",
    "                            fullsentence = relRes['full_sentence']\n",
    "                            acceptInteraction = relRes['accept_relation']\n",
    "\n",
    "                            if not acceptInteraction == validInteraction:\n",
    "\n",
    "                                incorrectIdentified += 1\n",
    "                                \n",
    "                                \"\"\"\n",
    "                                relResV=relChecker.check_sentence(sentText\n",
    "                                                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": e1[2]}\n",
    "                                                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": e2[2]}\n",
    "                                                                , fix_special_chars=False\n",
    "                                                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                                                )\n",
    "                                \n",
    "                                print(relResV)\n",
    "                                \"\"\"\n",
    "                            else:\n",
    "                                correctIdentified += 1\n",
    "\n",
    "                            elemCaseCounter[(validInteraction, acceptInteraction)]+=1\n",
    "\n",
    "                            totalClassified += 1\n",
    "\n",
    "                            if validInteraction:# and acceptInteraction:\n",
    "                                #print(relRes[\"check_results\"])   \n",
    "                                totalValidClassified += 1\n",
    "\n",
    "                                foundClasses = relRes[\"check_results\"][\"classification\"] #{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'regulation_keys': set()}\n",
    "                                foundTuple = (foundClasses[\"interaction_dir\"], foundClasses[\"regulation_dir\"])                                \n",
    "                                relationID = pair.attrib[\"id\"]\n",
    "                                testTuple = relClasses[relationID]\n",
    "\n",
    "                                classifyTrue.append((testTuple[0], testTuple[1]))\n",
    "                                classifyPred.append((foundTuple[0], foundTuple[1]))\n",
    "\n",
    "                                if testTuple[0] == \"MIR_GENE\" and testTuple[1] == \"DOWN\":\n",
    "                                    totalMirGeneDownClassified += 1\n",
    "\n",
    "                                classifiedByDetect[foundClasses[\"reg_detect\"]] += 1\n",
    "\n",
    "                                if testTuple[0] != foundTuple[0] or testTuple[1] != foundTuple[1]:\n",
    "\n",
    "\n",
    "                                    #print(sentText)\n",
    "                                    #print(e1, e2)\n",
    "                                    #print(foundTuple, foundClasses[\"reg_detect\"], \"passive?\", foundClasses[\"passive\"])\n",
    "                                    #print(testTuple)\n",
    "\n",
    "                                    print()\n",
    "                                    print()\n",
    "                                    print()\n",
    "                                    \n",
    "                                    print(relationID)\n",
    "                                    print(\"IS: \", foundTuple)\n",
    "                                    print(\"SB: \", testTuple)\n",
    "                                    print(relRes)\n",
    "                                    \n",
    "                                    relResV=relChecker.check_sentence(sentText\n",
    "                                                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": e1[2]}\n",
    "                                                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": e2[2]}\n",
    "                                                                , fix_special_chars=False\n",
    "                                                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                                                )\n",
    "                                \n",
    "                                    print(relResV)\n",
    "                                    \n",
    "                                    print()\n",
    "                                    print()\n",
    "                                    print()\n",
    "\n",
    "                                    errorByDetect[foundClasses[\"reg_detect\"]] += 1\n",
    "                                    \n",
    "                                    \n",
    "\n",
    "                                    incorrectClassified += 1\n",
    "\n",
    "\n",
    "\n",
    "                                    #numelems -= 1\n",
    "                                    #if numelems == 0:\n",
    "                                    #    return\n",
    "                                else:\n",
    "                                    \n",
    "                                    if foundClasses[\"reg_detect_major\"] == \"return\":\n",
    "                                        print(foundClasses)\n",
    "                                        print(sentText)\n",
    "                                        print(e1, e2)\n",
    "                                        print(foundTuple, foundClasses[\"reg_detect\"], \"passive?\", foundClasses[\"passive\"])\n",
    "                                        print(testTuple)\n",
    "\n",
    "                                        print()\n",
    "                                        print()\n",
    "                                        print()\n",
    "                                    correctClassified += 1\n",
    "\n",
    "                                    if acceptInteraction:\n",
    "                                        correctInteractClassified += 1\n",
    "\n",
    "\n",
    "\n",
    "            def printStats(outfile):\n",
    "                print(\"Total:     \", totalChecks, file=outfile)\n",
    "                print(\"Correct:   \", correctIdentified, correctIdentified/totalChecks, file=outfile)\n",
    "                print(\"Incorrect: \", incorrectIdentified, incorrectIdentified/totalChecks, file=outfile)\n",
    "                print(\"classes\", incorrectClass, file=outfile)\n",
    "                \n",
    "                precision = elemCaseCounter[(True, True)] / (elemCaseCounter[(True, True)]+elemCaseCounter[(True, False)])\n",
    "                recall = elemCaseCounter[(True, True)] / (elemCaseCounter[(True, True)]+elemCaseCounter[(False, True)])\n",
    "\n",
    "                f1 = 2* precision * recall / (precision+recall)\n",
    "\n",
    "                specificity = elemCaseCounter[(False, False)] / (elemCaseCounter[(True, False)] + elemCaseCounter[(False, False)])\n",
    "\n",
    "                print(\"precision\", precision, file=outfile)\n",
    "                print(\"recall\", recall, file=outfile)\n",
    "                print(\"specificity\", specificity, file=outfile)\n",
    "                print(\"f1\", f1, file=outfile)\n",
    "\n",
    "                print(\"Correct classified:     \", correctClassified, correctClassified/totalClassified, correctClassified/totalValidClassified,file=outfile)\n",
    "                print(\"Incorrect classified:   \", incorrectClassified, incorrectClassified/totalClassified, incorrectClassified/totalValidClassified,file=outfile)\n",
    "                print(\"Random classified: \", totalMirGeneDownClassified, totalMirGeneDownClassified / totalClassified, totalMirGeneDownClassified/totalValidClassified, file=outfile)\n",
    "                print(errorByDetect)\n",
    "\n",
    "                print(\"Correct interaction&classified:     \", correctInteractClassified, correctInteractClassified/totalClassified, correctInteractClassified/totalValidClassified,file=outfile)\n",
    "\n",
    "                print(\"Classified by: \", classifiedByDetect)\n",
    "\n",
    "                lClassifyTrue = [\"_\".join(x) for x in classifyTrue]\n",
    "                lClassifyPred = [\"_\".join(x) for x in classifyPred]\n",
    "\n",
    "                allLabels = sorted(set(lClassifyTrue+lClassifyPred))\n",
    "                matrix = multilabel_confusion_matrix(lClassifyTrue, lClassifyPred, labels=allLabels)\n",
    "                print(matrix)\n",
    "                print(classification_report(lClassifyTrue,lClassifyPred))\n",
    "                \n",
    "                return classification_report(lClassifyTrue, lClassifyPred, output_dict=True)\n",
    "\n",
    "            rep = printStats(sys.stdout)\n",
    "            check2results[subset_checks] = rep\n",
    "        #printStats(sys.stderr)\n",
    "    return check2results\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading relations scai_test_rels.tsv\n",
      "Interactions 232\n",
      "Non NA Interactions 138\n",
      "Will test 1 subsets.\n",
      "Testing checks ('compartment', 'between', 'counts', 'return')\n",
      "down-regulation that of\n",
      "contribute may to\n",
      "up-regulating coordinately ARP5\n",
      "down-regulation that of\n",
      "contribute may to\n",
      "up-regulating coordinately ARP5\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('APOBEC3C', 'Genes/Proteins', (99, 103))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('ANGPTL6', 'Genes/Proteins', (105, 112))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('Bmi-1', 'Genes/Proteins', (115, 120))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Reduction . of\n",
      "induces miR-21 glioma\n",
      "activating via caspase\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d9.s1.p0\n",
      "IS:  ('MIR_GENE', 'UP')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 'full_sentence': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'miR-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (81, 86), 'entity_text': 'FGF20'}}\n",
      "add special case miR-433\n",
      "add special case FGF20\n",
      "miR-433 6 FGF20 16\n",
      "(u\"The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 6, 16)\n",
      "geneword FGF20 mirword miR-433\n",
      "Conjuncts [upregulation, expression]\n",
      "neighbour upregulation of\n",
      "neighbour upregulation of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, upregulation]\n",
      "neighbour expression and\n",
      "neighbour expression and\n",
      "Conjunction before fc upregulation\n",
      "fc add children for upregulation\n",
      "Conjunctions\n",
      "expression [in, increased, FGF20, expression, upregulation, of, alpha-synuclein]\n",
      "SDP sent: The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "SDP     : [(miR-433, 'PROPN', 'nmod', False), (affinity, 'NOUN', 'nmod', False), (reduction, 'NOUN', 'nsubj', False), (result, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', False), (FGF20, 'PROPN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "0 [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration] 31 0 The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "Recognized overlapping entities: {Interval(81, 86, ('GENE_OR_GENE_PRODUCT', 'FGF20'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text The reduction of the affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which could in turn promote dopaminergic neurons degeneration\n",
      "[The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "TextBetween: The reduction of the affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which could in turn promote dopaminergic neurons degeneration\n",
      "infer direction from token set [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "stemTypes2Count: Counter({'POS': 3, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8125346670>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8125346ee0>, {'gene': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(result, 'NEU'), (increased, 'POS'), (expression, 'NEU')]\n",
      "Between Tokens [(result, 'NEU'), (increased, 'POS')]\n",
      "counts between final False\n",
      "['affinity', 'of', 'miR-433', 'to', 'the', '3', \"'\", 'UTR', 'would', 'result', 'in', 'increased', 'FGF20', 'expression', 'and']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "[(result, 'NEU'), (increased, 'POS'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "full sentence The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 'full_sentence': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'miR-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (81, 86), 'entity_text': 'FGF20'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d9.s1.p1\n",
      "IS:  ('MIR_GENE', 'UP')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\", 'full_sentence': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'miR-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (118, 133), 'entity_text': 'alpha-synuclein'}}\n",
      "add special case miR-433\n",
      "add special case alpha-synuclein\n",
      "miR-433 6 alpha-synuclein 21\n",
      "(u\"The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 6, 21)\n",
      "geneword alpha-synuclein mirword miR-433\n",
      "Conjuncts [upregulation, expression]\n",
      "neighbour upregulation of\n",
      "neighbour upregulation of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, upregulation]\n",
      "neighbour expression and\n",
      "neighbour expression and\n",
      "Conjunction before fc upregulation\n",
      "fc add children for upregulation\n",
      "Conjunctions\n",
      "expression [in, increased, FGF20, expression, upregulation, of, alpha-synuclein]\n",
      "SDP sent: The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "SDP     : [(miR-433, 'PROPN', 'nmod', False), (affinity, 'NOUN', 'nmod', False), (reduction, 'NOUN', 'nsubj', False), (result, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', True), (upregulation, 'NOUN', 'conj', False), (alpha-synuclein, 'PROPN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "0 [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration] 31 0 The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "Recognized overlapping entities: {Interval(118, 133, ('GENE_OR_GENE_PRODUCT', 'alpha-synuclein'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "Compartment Text The reduction of the affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which could in turn promote dopaminergic neurons degeneration\n",
      "[The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "TextBetween: The reduction of the affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which could in turn promote dopaminergic neurons degeneration\n",
      "infer direction from token set [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "stemTypes2Count: Counter({'POS': 3, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8125356550>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f81253564c0>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(result, 'NEU'), (increased, 'POS'), (expression, 'NEU'), (upregulation, 'POS')]\n",
      "Between Tokens [(result, 'NEU'), (increased, 'POS'), (expression, 'NEU'), (upregulation, 'POS')]\n",
      "counts between final False\n",
      "['affinity', 'of', 'miR-433', 'to', 'the', '3', \"'\", 'UTR', 'would', 'result', 'in', 'increased', 'FGF20', 'expression', 'and', 'upregulation', 'of', 'alpha-synuclein', ',', 'which']\n",
      "Counter({'NEU': 2, 'POS': 2})\n",
      "[(result, 'NEU'), (increased, 'POS'), (expression, 'NEU'), (upregulation, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "full sentence The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\", 'full_sentence': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'miR-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (118, 133), 'entity_text': 'alpha-synuclein'}}\n",
      "\n",
      "\n",
      "\n",
      "downregulation that of\n",
      "results lines in\n",
      "increased in expression\n",
      "expression increased of\n",
      "target its ,\n",
      "downregulation that of\n",
      "results lines in\n",
      "increased in expression\n",
      "expression increased of\n",
      "target its ,\n",
      "levels mir-21 are\n",
      "downregulated are or\n",
      "over-expressed is exhibit\n",
      "downregulation either of\n",
      "overexpression or of\n",
      "target its ,\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d48.s1.p3\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}}, 'tagged_sent': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'full_sentence': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 25), 'entity_text': 'miR-221/222'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (199, 206), 'entity_text': 'p27Kip1'}}\n",
      "add special case miR-221/222\n",
      "add special case p27Kip1\n",
      "miR-221/222 2 p27Kip1 25\n",
      "(u\"Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\", 2, 25)\n",
      "geneword p27Kip1 mirword miR-221/222\n",
      "Conjuncts [reduced, p27Kip1]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conjuncts [p27Kip1, reduced]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "fc add children for reduced\n",
      "Conjunctions\n",
      "p27Kip1 [increased, p27Kip1, in, U251, glioma, subcutaneous, mice, reduced, tumor, growth, through, up, regulation, of, p27Kip1]\n",
      "SDP sent: Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "SDP     : [(miR-221/222, 'PROPN', 'amod', False), (knocked-down, 'VERB', 'nsubj', False), (increased, 'VERB', 'amod', False), (p27Kip1, 'PROPN', 'ROOT', True), (reduced, 'VERB', 'conj', False), (regulation, 'NOUN', 'nmod', False), (p27Kip1, 'PROPN', 'nmod', False)]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "Recognized overlapping entities: {Interval(199, 206, ('GENE_OR_GENE_PRODUCT', 'p27Kip1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "TextBetween: Consistently , miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1 .\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}})\n",
      " hit sentence Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "full sentence Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}}, 'tagged_sent': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'full_sentence': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 25), 'entity_text': 'miR-221/222'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (199, 206), 'entity_text': 'p27Kip1'}}\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "The data strongly suggested that a regulatory loop between miR-21 and STAT3 might provide an insight into the mechanism of modulating EGFR/STAT3 signaling. \n",
      "('miR-21', 'Specific_miRNAs', (59, 65)) ('STAT3', 'Genes/Proteins', (70, 75))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('microRNA-29b', 'Specific_miRNAs', (59, 71)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('miR-29b', 'Specific_miRNAs', (73, 80)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "overexpression that of\n",
      "downregulation that of\n",
      "bound directly the\n",
      "inhibition miR-23a led\n",
      "increase an in\n",
      "bound directly the\n",
      "inhibition miR-23a led\n",
      "increase an in\n",
      "inhibition miR-23a led\n",
      "increase an in\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d66.s2.p0\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.', 'full_sentence': 'We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (51, 65), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (94, 101), 'entity_text': 'miR-211'}}\n",
      "add special case angiopoietin-1\n",
      "add special case miR-211\n",
      "angiopoietin-1 9 miR-211 13\n",
      "(u\"We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\", 9, 13)\n",
      "geneword angiopoietin-1 mirword miR-211\n",
      "Conjuncts [allele, binding]\n",
      "neighbour allele .\n",
      "neighbour allele None\n",
      "Conjunction before fc binding\n",
      "fc add children for binding\n",
      "Conjuncts [binding, allele]\n",
      "neighbour binding ,\n",
      "neighbour binding but\n",
      "Conjunction before fc allele\n",
      "fc add children for allele\n",
      "Conjunctions\n",
      "binding [miR-211, binding, T, allele]\n",
      "SDP sent: We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "SDP     : [(angiopoietin-1, 'PROPN', 'compound', False), (translation, 'NOUN', 'dobj', False), (suppressed, 'VERB', 'ccomp', False), (facilitating, 'VERB', 'advcl', False), (binding, 'NOUN', 'dobj', False), (miR-211, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule [by, facilitating, miR-211, binding, ,, but, not, the, T, allele]\n",
      "DeepCheck Conj Subtree\n",
      "[the, T, allele]\n",
      "0 [We, found] 2 0 We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "1 [the, A, allele, of, rs2507800, suppressed, angiopoietin-1, translation, by, facilitating, miR-211, binding, ,, but, not, the, T, allele] 18 14 the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "Recognized overlapping entities: {Interval(51, 65, ('GENE_OR_GENE_PRODUCT', 'angiopoietin-1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding , but not the T allele\n",
      "[the, A, allele, of, rs2507800, suppressed, angiopoietin-1, translation, by, facilitating, miR-211, binding, ,, but, not, the, T, allele]\n",
      "TextBetween: the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding , but not the T allele\n",
      "infer direction from token set [the, A, allele, of, rs2507800, suppressed, angiopoietin-1, translation, by, facilitating, miR-211, binding, ,, but, not, the, T, allele]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8126030c10>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8126030af0>, {'gene': Counter({'NEG': 1, 'NEU': 1}), 'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(suppressed, 'NEG'), (translation, 'NEU'), (binding, 'NEU')]\n",
      "counts between 3 Counter() NEG\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['rs2507800', 'suppressed', 'angiopoietin-1', 'translation', 'by', 'facilitating', 'miR-211', 'binding', ',']\n",
      "Counter()\n",
      "[]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "full sentence We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.', 'full_sentence': 'We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (51, 65), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (94, 101), 'entity_text': 'miR-211'}}\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "To test our hypothesis, a predicted microRNA-binding site was found in the 3'-UTR of angiopoietin-1 using bioinformatics; variant rs2507800 was identified to be located in the miR-211-binding site of angiopoietin-1. \n",
      "('miR-211', 'Specific_miRNAs', (176, 183)) ('angiopoietin-1', 'Genes/Proteins', (200, 214))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d66.s9.p1\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\", 'full_sentence': \"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (79, 93), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (146, 153), 'entity_text': 'miR-211'}}\n",
      "add special case angiopoietin-1\n",
      "add special case miR-211\n",
      "angiopoietin-1 15 miR-211 25\n",
      "(u\"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\", 15, 25)\n",
      "geneword angiopoietin-1 mirword miR-211\n",
      "Conjunctions\n",
      "rs2507800 [(, )]\n",
      "SDP sent: Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "SDP     : [(angiopoietin-1, 'PROPN', 'nmod', False), (3'-UTR, 'NOUN', 'nmod', False), (genotype, 'NOUN', 'nsubj', False), (reduce, 'VERB', 'ccomp', False), (interfering, 'VERB', 'advcl', False), (binding, 'NOUN', 'nmod', False), (miR-211, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule [by, interfering, with, miR-211, binding]\n",
      "0 [Our, results, provide, evidence] 4 0 Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "1 [the, TT, genotype, (, rs2507800, ), in, the, 3'-UTR, of, angiopoietin-1, might, reduce, the, risk, of, stroke, by, interfering, with, miR-211, binding] 22 34 the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "Recognized overlapping entities: {Interval(79, 93, ('GENE_OR_GENE_PRODUCT', 'angiopoietin-1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text the TT genotype ( rs2507800 ) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding\n",
      "[the, TT, genotype, (, rs2507800, ), in, the, 3'-UTR, of, angiopoietin-1, might, reduce, the, risk, of, stroke, by, interfering, with, miR-211, binding]\n",
      "TextBetween: the TT genotype ( rs2507800 ) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding\n",
      "infer direction from token set [the, TT, genotype, (, rs2507800, ), in, the, 3'-UTR, of, angiopoietin-1, might, reduce, the, risk, of, stroke, by, interfering, with, miR-211, binding]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8126030550>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8126030af0>, {'gene': Counter({'NEG': 1}), 'mir': Counter({'NEU': 2})})\n",
      "counts between 1 stemTypes2Count [(reduce, 'NEG'), (interfering, 'NEU'), (binding, 'NEU')]\n",
      "counts between 3 Counter({'NEG': 1}) NEG\n",
      "Between Tokens [(reduce, 'NEG')]\n",
      "counts between final False\n",
      "[\"3'-UTR\", 'of', 'angiopoietin-1', 'might', 'reduce', 'the', 'risk', 'of', 'stroke', 'by', 'interfering', 'with', 'miR-211', 'binding']\n",
      "Counter({'NEG': 1})\n",
      "[(reduce, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "full sentence Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\", 'full_sentence': \"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (79, 93), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (146, 153), 'entity_text': 'miR-211'}}\n",
      "\n",
      "\n",
      "\n",
      "decreased show miR-107\n",
      "augmentation with of\n",
      "expression GRN .\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d69.s11.p1\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.', 'full_sentence': 'Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (66, 73), 'entity_text': 'miR-107'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (193, 196), 'entity_text': 'GRN'}}\n",
      "add special case miR-107\n",
      "add special case GRN\n",
      "miR-107 9 GRN 27\n",
      "(u\"Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\", 9, 27)\n",
      "geneword GRN mirword miR-107\n",
      "Conjuncts [secretion, expression]\n",
      "neighbour secretion of\n",
      "neighbour secretion of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, secretion]\n",
      "neighbour expression ,\n",
      "neighbour expression including\n",
      "Conjunction before fc secretion\n",
      "fc add children for secretion\n",
      "fc add children for secretion\n",
      "Conjunctions\n",
      "expression [in, decreased, GRN, expression, including, changes, in, cell, compartmentation, decreased, secretion, of, GRN, protein]\n",
      "SDP sent: Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\n",
      "SDP     : [(miR-107, 'PROPN', 'compound', False), (levels, 'NOUN', 'nmod', False), (leads, 'VERB', 'acl:relcl', False), (cells, 'NOUN', 'nmod', False), (supplementation, 'NOUN', 'nsubj', False), (results, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', True), (secretion, 'NOUN', 'conj', False), (protein, 'NOUN', 'nmod', False), (GRN, 'PROPN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[decreased, secretion, of, GRN, protein]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "0 [Glucose, supplementation, in, cultured, cells, that, leads, to, increased, miR-107, levels, also, results, in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein] 29 0 Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression , including changes in cell compartmentation and decreased secretion of GRN protein\n",
      "[Glucose, supplementation, in, cultured, cells, that, leads, to, increased, miR-107, levels, also, results, in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein]\n",
      "TextBetween: Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression , including changes in cell compartmentation and decreased secretion of GRN protein\n",
      "infer direction from token set [Glucose, supplementation, in, cultured, cells, that, leads, to, increased, miR-107, levels, also, results, in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein]\n",
      "stemTypes2Count: Counter({'POS': 2, 'NEG': 2})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f815b4fa040>, {'mir': Counter({'POS': 1}), 'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 4, 'POS': 2, 'NEG': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8126030c10>, {'mir': Counter({'POS': 1, 'NEU': 1}), 'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(increased, 'POS'), (levels, 'NEU'), (results, 'NEU'), (decreased, 'NEG'), (expression, 'NEU'), (including, 'NEU'), (decreased, 'NEG'), (secretion, 'POS')]\n",
      "Between Tokens [(levels, 'NEU'), (results, 'NEU'), (decreased, 'NEG'), (expression, 'NEU'), (including, 'NEU'), (decreased, 'NEG'), (secretion, 'POS')]\n",
      "counts between final False\n",
      "['to', 'increased', 'miR-107', 'levels', 'also', 'results', 'in', 'decreased', 'GRN', 'expression', ',', 'including', 'changes', 'in', 'cell', 'compartmentation', 'and', 'decreased', 'secretion', 'of', 'GRN', 'protein']\n",
      "Counter({'NEU': 4, 'POS': 2, 'NEG': 2})\n",
      "[(increased, 'POS'), (levels, 'NEU'), (results, 'NEU'), (decreased, 'NEG'), (expression, 'NEU'), (including, 'NEU'), (decreased, 'NEG'), (secretion, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\n",
      "full sentence Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.', 'full_sentence': 'Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (66, 73), 'entity_text': 'miR-107'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (193, 196), 'entity_text': 'GRN'}}\n",
      "\n",
      "\n",
      "\n",
      "induced temozolomide apoptosis\n",
      "decreasing by Bax/Bcl-2\n",
      "activity caspase-3 .\n",
      "induced temozolomide apoptosis\n",
      "decreasing by Bax/Bcl-2\n",
      "induced temozolomide apoptosis\n",
      "decreasing by Bax/Bcl-2\n",
      "expression miR-451 is\n",
      "suppression the of\n",
      "signaling AMPK allows\n",
      "elevated maintain proliferation\n",
      "activation mTOR .\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d80.s5.p1\n",
      "IS:  ('GENE_MIR', 'NEU')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.', 'full_sentence': 'We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (40, 44), 'entity_text': 'CDK6'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (55, 68), 'entity_text': 'microRNA 124a'}}\n",
      "add special case CDK6\n",
      "add special case microRNA 124a\n",
      "CDK6 7 124a 10\n",
      "(u\"We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\", 7, 10)\n",
      "geneword CDK6 mirword 124a\n",
      "Conjunctions\n",
      "microRNA [a, microRNA]\n",
      "SDP sent: We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\n",
      "SDP     : [(CDK6, 'PROPN', 'nmod', False), (identified, 'VERB', 'ROOT', False), (sites, 'NOUN', 'dobj', False), (CDK6, 'PROPN', 'nmod', False)]\n",
      "0 [We, identified, putative, miR, sites, in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA] 16 0 We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\n",
      "Recognized overlapping entities: {Interval(40, 44, ('GENE_OR_GENE_PRODUCT', 'CDK6'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text We identified putative miR sites in the CDK6 including microRNA 124a , a brain enriched microRNA\n",
      "[We, identified, putative, miR, sites, in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA]\n",
      "TextBetween: We identified putative miR sites in the CDK6 including microRNA 124a , a brain enriched microRNA\n",
      "infer direction from token set [We, identified, putative, miR, sites, in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8126030700>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8126030dc0>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(including, 'NEU')]\n",
      "Between Tokens [(including, 'NEU')]\n",
      "counts between final False\n",
      "['in', 'the', 'CDK6', 'including', 'microRNA', '124a', ',', 'a']\n",
      "Counter({'NEU': 1})\n",
      "[(including, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\n",
      "full sentence We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.', 'full_sentence': 'We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (40, 44), 'entity_text': 'CDK6'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (55, 68), 'entity_text': 'microRNA 124a'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d80.s9.p1\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.', 'full_sentence': 'Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (14, 39), 'entity_text': 'cyclin dependent kinase 6'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (43, 55), 'entity_text': 'microRNA 124'}}\n",
      "add special case cyclin dependent kinase 6\n",
      "add special case microRNA 124\n",
      "6 5 124 8\n",
      "(u\"Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\", 5, 8)\n",
      "geneword 6 mirword 124\n",
      "Conjunctions\n",
      "SDP sent: Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\n",
      "SDP     : [(6, 'NUM', 'nummod', False), (kinase, 'NOUN', 'nmod', False), (Regulation, 'NOUN', 'ROOT', False), (microRNA, 'NOUN', 'nmod', False), (124, 'NUM', 'nummod', False)]\n",
      "0 [Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma] 11 0 Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\n",
      "Recognized overlapping entities: {Interval(14, 39, ('GENE_OR_GENE_PRODUCT', 'cyclin dependent kinase 6'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma\n",
      "[Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma]\n",
      "TextBetween: Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma\n",
      "infer direction from token set [Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f81242af940>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f815b4f9ca0>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(dependent, 'NEU')]\n",
      "counts between 3 Counter() NEU\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['dependent', 'kinase', '6', 'by', 'microRNA', '124', 'in', 'medulloblastoma']\n",
      "Counter()\n",
      "[]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\n",
      "full sentence Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.', 'full_sentence': 'Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (14, 39), 'entity_text': 'cyclin dependent kinase 6'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (43, 55), 'entity_text': 'microRNA 124'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d87.s9.p3\n",
      "IS:  ('GENE_MIR', 'NEU')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.', 'full_sentence': '<e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (0, 4), 'entity_text': 'BAG2'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (73, 81), 'entity_text': 'miR-128a'}}\n",
      "add special case BAG2\n",
      "add special case miR-128a\n",
      "BAG2 0 miR-128a 12\n",
      "(u\"BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\", 0, 12)\n",
      "geneword BAG2 mirword miR-128a\n",
      "Conjunctions\n",
      "SDP sent: BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "SDP     : [(BAG2, 'PROPN', 'compound', False), (levels, 'NOUN', 'nsubj', False), (control, 'NOUN', 'ROOT', False), (miR-128a, 'PROPN', 'nmod', False)]\n",
      "0 [BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons] 24 0 BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "Recognized overlapping entities: {Interval(0, 4, ('GENE_OR_GENE_PRODUCT', 'BAG2'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text BAG2 levels in cells are under the physiological control of the microRNA miR-128a , which can tune paired helical filament Tau levels in neurons\n",
      "[BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons]\n",
      "TextBetween: BAG2 levels in cells are under the physiological control of the microRNA miR-128a , which can tune paired helical filament Tau levels in neurons\n",
      "infer direction from token set [BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f815b4f9af0>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8126030670>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(levels, 'NEU'), (control, 'NEU')]\n",
      "Between Tokens [(levels, 'NEU'), (control, 'NEU')]\n",
      "counts between final False\n",
      "['BAG2', 'levels', 'in', 'cells', 'are', 'under', 'the', 'physiological', 'control', 'of', 'the', 'microRNA', 'miR-128a', ',', 'which']\n",
      "Counter({'NEU': 2})\n",
      "[(levels, 'NEU'), (control, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "full sentence <e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.', 'full_sentence': '<e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (0, 4), 'entity_text': 'BAG2'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (73, 81), 'entity_text': 'miR-128a'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d96.s2.p1\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'full_sentence': '<e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 7), 'entity_text': 'miR-29b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (30, 40), 'entity_text': 'podoplanin'}}\n",
      "add special case miR-29b\n",
      "add special case podoplanin\n",
      "miR-29b 0 podoplanin 4\n",
      "(u\"miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\", 0, 4)\n",
      "geneword podoplanin mirword miR-29b\n",
      "Conjuncts [miR-125a, miR-29b]\n",
      "neighbour miR-125a regulate\n",
      "neighbour miR-125a regulate\n",
      "Conjunction before fc miR-29b\n",
      "Conjuncts [miR-29b, miR-125a]\n",
      "neighbour miR-29b and\n",
      "neighbour miR-29b and\n",
      "Conjunction before fc miR-125a\n",
      "Conjuncts [suppress, regulate]\n",
      "neighbour suppress invasion\n",
      "neighbour suppress invasion\n",
      "Conjunction before fc regulate\n",
      "fc add children for regulate\n",
      "Conjuncts [regulate, suppress]\n",
      "neighbour regulate podoplanin\n",
      "neighbour regulate podoplanin\n",
      "Conjunction before fc suppress\n",
      "fc add children for suppress\n",
      "Conunct merge check: miR-29b regulate\n",
      "miR-125a\n",
      "Conjunctions\n",
      "miR-29b [miR-29b, miR-125a]\n",
      "regulate [regulate, podoplanin, suppress, invasion, in, glioblastoma]\n",
      "SDP sent: miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "SDP     : [(miR-29b, 'PROPN', 'nsubj', False), (regulate, 'VERB', 'ROOT', False), (podoplanin, 'PROPN', 'dobj', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-125a]\n",
      "deepcheck connection rule noun found suppress invasion [suppress, invasion, in, glioblastoma]\n",
      "deepcheck conj verb\n",
      "0 [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma] 10 0 miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "Recognized overlapping entities: {Interval(30, 40, ('GENE_OR_GENE_PRODUCT', 'podoplanin'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma\n",
      "[miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma]\n",
      "TextBetween: miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma\n",
      "infer direction from token set [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f815eb8a160>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f815b4f9e50>, {'gene': Counter({'NEU': 1, 'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(regulate, 'NEU'), (suppress, 'NEG')]\n",
      "Between Tokens [(regulate, 'NEU')]\n",
      "counts between final False\n",
      "['miR-29b', 'and', 'miR-125a', 'regulate', 'podoplanin', 'and', 'suppress']\n",
      "Counter({'NEU': 1, 'NEG': 1})\n",
      "[(regulate, 'NEU'), (suppress, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "full sentence <e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'full_sentence': '<e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 7), 'entity_text': 'miR-29b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (30, 40), 'entity_text': 'podoplanin'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d96.s2.p3\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'full_sentence': 'miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (12, 20), 'entity_text': 'miR-125a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (30, 40), 'entity_text': 'podoplanin'}}\n",
      "add special case miR-125a\n",
      "add special case podoplanin\n",
      "miR-125a 2 podoplanin 4\n",
      "(u\"miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\", 2, 4)\n",
      "geneword podoplanin mirword miR-125a\n",
      "Conjuncts [miR-125a, miR-29b]\n",
      "neighbour miR-125a regulate\n",
      "neighbour miR-125a regulate\n",
      "Conjunction before fc miR-29b\n",
      "Conjuncts [miR-29b, miR-125a]\n",
      "neighbour miR-29b and\n",
      "neighbour miR-29b and\n",
      "Conjunction before fc miR-125a\n",
      "Conjuncts [suppress, regulate]\n",
      "neighbour suppress invasion\n",
      "neighbour suppress invasion\n",
      "Conjunction before fc regulate\n",
      "fc add children for regulate\n",
      "Conjuncts [regulate, suppress]\n",
      "neighbour regulate podoplanin\n",
      "neighbour regulate podoplanin\n",
      "Conjunction before fc suppress\n",
      "fc add children for suppress\n",
      "Conunct merge check: miR-29b regulate\n",
      "miR-125a\n",
      "Conjunctions\n",
      "miR-29b [miR-29b, miR-125a]\n",
      "regulate [regulate, podoplanin, suppress, invasion, in, glioblastoma]\n",
      "SDP sent: miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "SDP     : [(miR-125a, 'PROPN', 'conj', False), (miR-29b, 'PROPN', 'nsubj', True), (regulate, 'VERB', 'ROOT', False), (podoplanin, 'PROPN', 'dobj', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-125a]\n",
      "deepcheck connection rule noun found suppress invasion [suppress, invasion, in, glioblastoma]\n",
      "deepcheck conj verb\n",
      "0 [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma] 10 0 miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "Recognized overlapping entities: {Interval(30, 40, ('GENE_OR_GENE_PRODUCT', 'podoplanin'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma\n",
      "[miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma]\n",
      "TextBetween: miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma\n",
      "infer direction from token set [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f815b4fa160>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8125feea60>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(regulate, 'NEU'), (suppress, 'NEG')]\n",
      "Between Tokens [(regulate, 'NEU')]\n",
      "counts between final False\n",
      "['miR-29b', 'and', 'miR-125a', 'regulate', 'podoplanin', 'and', 'suppress']\n",
      "Counter({'NEU': 1, 'NEG': 1})\n",
      "[(regulate, 'NEU'), (suppress, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "full sentence miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'full_sentence': 'miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (12, 20), 'entity_text': 'miR-125a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (30, 40), 'entity_text': 'podoplanin'}}\n",
      "\n",
      "\n",
      "\n",
      "Total:      232\n",
      "Correct:    217 0.9353448275862069\n",
      "Incorrect:  15 0.06465517241379311\n",
      "classes Counter()\n",
      "precision 0.927536231884058\n",
      "recall 0.9624060150375939\n",
      "specificity 0.898989898989899\n",
      "f1 0.9446494464944649\n",
      "Correct classified:      127 0.5474137931034483 0.9202898550724637\n",
      "Incorrect classified:    11 0.04741379310344827 0.07971014492753623\n",
      "Random classified:  52 0.22413793103448276 0.37681159420289856\n",
      "Counter({'counts between': 10, 'between mir gene': 1})\n",
      "Correct interaction&classified:      118 0.5086206896551724 0.855072463768116\n",
      "Classified by:  Counter({'counts between': 65, 'counts between equal': 10, 'between gene mir': 10, 'counts opp': 9, 'compartment mir gene by': 8, 'return': 7, 'between mir gene': 6, 'compartment mir gene': 6, 'compartment gene mir': 5, 'compartment gene mir by': 4, 'compartment mir gene neg corr': 4, 'between mir gene reg corr': 3, 'counts between alternating': 1})\n",
      "[[[132   2]\n",
      "  [  0   4]]\n",
      "\n",
      " [[137   0]\n",
      "  [  0   1]]\n",
      "\n",
      " [[ 82   4]\n",
      "  [  5  47]]\n",
      "\n",
      " [[ 54   3]\n",
      "  [  6  75]]\n",
      "\n",
      " [[136   2]\n",
      "  [  0   0]]]\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      " GENE_MIR_NEU       0.67      1.00      0.80         4\n",
      "  GENE_MIR_UP       1.00      1.00      1.00         1\n",
      "MIR_GENE_DOWN       0.92      0.90      0.91        52\n",
      " MIR_GENE_NEU       0.96      0.93      0.94        81\n",
      "  MIR_GENE_UP       0.00      0.00      0.00         0\n",
      "\n",
      "     accuracy                           0.92       138\n",
      "    macro avg       0.71      0.77      0.73       138\n",
      " weighted avg       0.94      0.92      0.93       138\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "test_results = runCheck(-1, \"TEST\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading relations scai_test_rels.tsv\n",
      "Interactions 232\n",
      "Non NA Interactions 140\n",
      "Will test 1 subsets.\n",
      "Testing checks ('compartment', 'between', 'counts', 'return')\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('APOBEC3C', 'Genes/Proteins', (99, 103))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('ANGPTL6', 'Genes/Proteins', (105, 112))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('Bmi-1', 'Genes/Proteins', (115, 120))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('angiopoietin-related growth factor protein 5', 'Genes/Proteins', (49, 93))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('APOBEC3C', 'Genes/Proteins', (95, 99))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('ANGPTL6', 'Genes/Proteins', (101, 108))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "['miR-21', 'activated', 'caspase']\n",
      "add special case miR-21\n",
      "add special case caspase 9\n",
      "miR-21 4 9 7\n",
      "(u\"Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\", 4, 7)\n",
      "geneword 9 mirword miR-21\n",
      "Conjuncts [3, caspase]\n",
      "neighbour 3 ,\n",
      "neighbour 3 which\n",
      "Conjunction before fc caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "Conjuncts [caspase, 3]\n",
      "neighbour caspase 9\n",
      "neighbour caspase 9\n",
      "Conjunction before fc 3\n",
      "reduction caspase\n",
      "\n",
      "Conjunctions\n",
      "reduction [reduction, .]\n",
      "caspase [of, miR-21, activated, caspase, 9, 3]\n",
      "['miR-21', 'activated', 'caspase']\n",
      "achildren {miR-21}\n",
      "achildren False\n",
      "SDP sent: Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "SDP     : [(miR-21, 'PROPN', 'compound', False), (caspase, 'NOUN', 'nmod', False), (9, 'NUM', 'nummod', False)]\n",
      "[3]\n",
      "deepcheck connection rule [by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3]\n",
      "0 [Moreover, ,, reduction, of, miR-21, activated, caspase, 9, and, 3, ,, which, may, be, mediated, by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3] 25 0 Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "Conjuncts [3, caspase]\n",
      "neighbour 3 ,\n",
      "neighbour 3 which\n",
      "Conjunction before fc caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "Conjuncts [caspase, 3]\n",
      "neighbour caspase 9\n",
      "neighbour caspase 9\n",
      "Conjunction before fc 3\n",
      "Context Fail for Noun Rule\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: reduction of miR-21 activated caspase 9 and 3\n",
      "counts between 1 Counter({'NEG': 1, 'NEU': 1})\n",
      "['reduction', 'of', 'miR-21', 'activated', 'caspase', '9', 'and', '3']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "(False, {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "full sentence Moreover, reduction of <e1>miR-21</e1> activated <e2>caspase 9</e2> and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Moreover, reduction of <e1>miR-21</e1> activated <e2>caspase 9</e2> and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.', 'full_sentence': 'Moreover, reduction of <e1>miR-21</e1> activated <e2>caspase 9</e2> and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (23, 29), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (40, 49), 'entity_text': 'caspase 9'}}\n",
      "['FGF20', 'rs12720208', 'SNP', 'and']\n",
      "['let-7', 'family', 'microRNAs', 'and', 'let-7-targeted', 'heterochronic', 'genes', ',']\n",
      "['let-7', 'family', 'microRNAs', 'and', 'let-7-targeted', 'heterochronic', 'genes', ',', 'hbl-1', ',']\n",
      "['let-7-targeted', 'heterochronic', 'genes', ',']\n",
      "['let-7-targeted', 'heterochronic', 'genes', ',', 'hbl-1', ',']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (100, 107)) ('SRF', 'Genes/Proteins', (13, 16))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (100, 107)) ('myocardin', 'Genes/Proteins', (21, 30))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (112, 119)) ('SRF', 'Genes/Proteins', (13, 16))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (112, 119)) ('myocardin', 'Genes/Proteins', (21, 30))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These abnormalities reflect the regulation of a cadre of modulators of SRF activity and actin dynamics by miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (106, 113)) ('SRF', 'Genes/Proteins', (71, 74))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These abnormalities reflect the regulation of a cadre of modulators of SRF activity and actin dynamics by miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (118, 125)) ('SRF', 'Genes/Proteins', (71, 74))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case microRNA-30\n",
      "add special case Oas1\n",
      "microRNA-30 12 Oas1 18\n",
      "(u\"Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\", 12, 18)\n",
      "geneword Oas1 mirword microRNA-30\n",
      "Conjunctions\n",
      "SDP sent: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "SDP     : [(microRNA-30, 'PROPN', 'compound', False), (precursor, 'NOUN', 'nmod', False), (features, 'NOUN', 'dobj', False), (implementing, 'VERB', 'advcl', False), (Modification, 'NOUN', 'nsubj', False), (avoided, 'VERB', 'ROOT', False), (induction, 'NOUN', 'dobj', False), (Oas1, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "TextBetween: naturally occurring microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in\n",
      "counts between 1 Counter({'POS': 1})\n",
      "['naturally', 'occurring', 'microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "Counter({'POS': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "full sentence Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'full_sentence': 'Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (88, 99), 'entity_text': 'microRNA-30'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (127, 131), 'entity_text': 'Oas1'}}\n",
      "add special case miR-30\n",
      "add special case Oas1\n",
      "miR-30 14 Oas1 18\n",
      "(u\"Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\", 14, 18)\n",
      "geneword Oas1 mirword miR-30\n",
      "Conjunctions\n",
      "SDP sent: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "SDP     : [(miR-30, 'PROPN', 'compound', False), (precursor, 'NOUN', 'nmod', False), (features, 'NOUN', 'dobj', False), (implementing, 'VERB', 'advcl', False), (Modification, 'NOUN', 'nsubj', False), (avoided, 'VERB', 'ROOT', False), (induction, 'NOUN', 'dobj', False), (Oas1, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "TextBetween: microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in\n",
      "counts between 1 Counter({'POS': 1})\n",
      "['microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "Counter({'POS': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "full sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'full_sentence': 'Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (101, 107), 'entity_text': 'miR-30'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (127, 131), 'entity_text': 'Oas1'}}\n",
      "['mir-21', 'and']\n",
      "add special case mir-21\n",
      "add special case Pdcd4\n",
      "mir-21 11 Pdcd4 13\n",
      "(u\"These critical in vivo findings demonstrate an important functional linkage between mir-21 and Pdcd4 and further elucidate the molecular mechanisms by which the known high level of mir-21 expression in glioblastoma can attribute to tumorigenesis--namely, inhibition of Pdcd4 and its tumor-suppressive functions.\", 11, 13)\n",
      "geneword Pdcd4 mirword mir-21\n",
      "Conjuncts [elucidate, demonstrate]\n",
      "neighbour elucidate the\n",
      "neighbour elucidate the\n",
      "Conjunction before fc demonstrate\n",
      "fc add children for demonstrate\n",
      "Conjuncts [Pdcd4, mir-21]\n",
      "neighbour Pdcd4 and\n",
      "neighbour Pdcd4 and\n",
      "Conjunction before fc mir-21\n",
      "fc add children for mir-21\n",
      "Conjuncts [mir-21, Pdcd4]\n",
      "neighbour mir-21 and\n",
      "neighbour mir-21 and\n",
      "Conjunction before fc Pdcd4\n",
      "Conjuncts [demonstrate, elucidate]\n",
      "neighbour demonstrate an\n",
      "neighbour demonstrate an\n",
      "Conjunction before fc elucidate\n",
      "fc add children for elucidate\n",
      "fc add children for elucidate\n",
      "Conjuncts [functions, inhibition]\n",
      "neighbour functions .\n",
      "neighbour functions None\n",
      "Conjunction before fc inhibition\n",
      "Conjuncts [inhibition, functions]\n",
      "neighbour inhibition of\n",
      "neighbour inhibition of\n",
      "Conjunction before fc functions\n",
      "fc add children for functions\n",
      "demonstrate mir-21\n",
      "\n",
      "Conjunctions\n",
      "demonstrate [demonstrate, important, functional, linkage, elucidate, molecular, mechanisms, inhibition, tumor-suppressive, functions]\n",
      "mir-21 [between, mir-21, Pdcd4]\n",
      "['mir-21', 'and']\n",
      "Conjunction case rule [between]\n",
      "SDP sent: These critical in vivo findings demonstrate an important functional linkage between mir-21 and Pdcd4 and further elucidate the molecular mechanisms by which the known high level of mir-21 expression in glioblastoma can attribute to tumorigenesis--namely, inhibition of Pdcd4 and its tumor-suppressive functions.\n",
      "SDP     : [(mir-21, 'PROPN', 'nmod', False), (demonstrate, 'VERB', 'ROOT', False), (linkage, 'NOUN', 'dobj', False), (mir-21, 'PROPN', 'nmod', False)]\n",
      "[Pdcd4]\n",
      "[its, tumor-suppressive, functions]\n",
      "0 [These, critical, in, vivo, findings, demonstrate, an, important, functional, linkage, between, mir-21, and, Pdcd4] 14 0 These critical in vivo findings demonstrate an important functional linkage between mir-21 and Pdcd4 and further elucidate the molecular mechanisms by which the known high level of mir-21 expression in glioblastoma can attribute to tumorigenesis--namely, inhibition of Pdcd4 and its tumor-suppressive functions.\n",
      "1 [further, elucidate, the, molecular, mechanisms, by, which, the, known, high, level, of, mir-21, expression, in, glioblastoma, can, attribute, to, tumorigenesis, --, namely, ,, inhibition, of, Pdcd4, and, its, tumor-suppressive, functions] 30 105 further elucidate the molecular mechanisms by which the known high level of mir-21 expression in glioblastoma can attribute to tumorigenesis--namely, inhibition of Pdcd4 and its tumor-suppressive functions.\n",
      "Conjuncts [elucidate, demonstrate]\n",
      "neighbour elucidate the\n",
      "neighbour elucidate the\n",
      "Conjunction before fc demonstrate\n",
      "fc add children for demonstrate\n",
      "Conjuncts [Pdcd4, mir-21]\n",
      "neighbour Pdcd4 and\n",
      "neighbour Pdcd4 and\n",
      "Conjunction before fc mir-21\n",
      "fc add children for mir-21\n",
      "Conjuncts [mir-21, Pdcd4]\n",
      "neighbour mir-21 and\n",
      "neighbour mir-21 and\n",
      "Conjunction before fc Pdcd4\n",
      "Conjuncts [demonstrate, elucidate]\n",
      "neighbour demonstrate an\n",
      "neighbour demonstrate an\n",
      "Conjunction before fc elucidate\n",
      "fc add children for elucidate\n",
      "fc add children for elucidate\n",
      "Conjuncts [functions, inhibition]\n",
      "neighbour functions .\n",
      "neighbour functions None\n",
      "Conjunction before fc inhibition\n",
      "Conjuncts [inhibition, functions]\n",
      "neighbour inhibition of\n",
      "neighbour inhibition of\n",
      "Conjunction before fc functions\n",
      "fc add children for functions\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: linkage between mir-21 and Pdcd4\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence These critical in vivo findings demonstrate an important functional linkage between mir-21 and Pdcd4 and further elucidate the molecular mechanisms by which the known high level of mir-21 expression in glioblastoma can attribute to tumorigenesis--namely, inhibition of Pdcd4 and its tumor-suppressive functions.\n",
      "full sentence These critical in vivo findings demonstrate an important functional linkage between <e1>mir-21</e1> and <e2>Pdcd4</e2> and further elucidate the molecular mechanisms by which the known high level of mir-21 expression in glioblastoma can attribute to tumorigenesis--namely, inhibition of Pdcd4 and its tumor-suppressive functions.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}}, 'tagged_sent': 'These critical in vivo findings demonstrate an important functional linkage between <e1>mir-21</e1> and <e2>Pdcd4</e2> and further elucidate the molecular mechanisms by which the known high level of mir-21 expression in glioblastoma can attribute to tumorigenesis--namely, inhibition of Pdcd4 and its tumor-suppressive functions.', 'full_sentence': 'These critical in vivo findings demonstrate an important functional linkage between <e1>mir-21</e1> and <e2>Pdcd4</e2> and further elucidate the molecular mechanisms by which the known high level of mir-21 expression in glioblastoma can attribute to tumorigenesis--namely, inhibition of Pdcd4 and its tumor-suppressive functions.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (84, 90), 'entity_text': 'mir-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (95, 100), 'entity_text': 'Pdcd4'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('IQGAP1', 'Genes/Proteins', (132, 138))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('IQ motif containing GTPase activating protein 1', 'Genes/Proteins', (83, 130))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('laminin γ1', 'Genes/Proteins', (141, 151))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('LAMC1', 'Genes/Proteins', (153, 158))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('integrin β1', 'Genes/Proteins', (164, 175))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('ITGB1', 'Genes/Proteins', (177, 182))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case MCP-2\n",
      "add special case mir-146a\n",
      "MCP-2 20 mir-146a 25\n",
      "(u\"The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\", 20, 25)\n",
      "geneword MCP-2 mirword mir-146a\n",
      "Conjuncts [levels, levels]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, levels]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "HIVE levels\n",
      ") brain samples in which\n",
      "Conjunctions\n",
      "HIVE [(, )]\n",
      "levels [decreased, levels, increased, levels]\n",
      "SDP sent: The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "SDP     : [(MCP-2, 'PROPN', 'nmod', False), (levels, 'NOUN', 'nsubjpass', True), (levels, 'NOUN', 'conj', False), (mir-146a, 'PROPN', 'nmod', False)]\n",
      "[increased, levels, of, mir-146a]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "0 [The, clinical, relevance, of, our, findings, was, evaluated, in, HIV-encephalitis, (, HIVE, ), brain, samples, in, which, decreased, levels, of, MCP-2, and, increased, levels, of, mir-146a, were, observed] 28 0 The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "1 [suggesting, a, role, for, mir-146a, in, the, maintenance, of, HIV-mediated, chronic, inflammation, of, the, brain] 15 177 suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "Conjuncts [levels, levels]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, levels]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: levels of MCP-2 and increased levels of mir-146a were observed\n",
      "counts between 1 Counter({'NEU': 2, 'POS': 1})\n",
      "['levels', 'of', 'MCP-2', 'and', 'increased', 'levels', 'of', 'mir-146a', 'were', 'observed']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "full sentence The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of <e2>MCP-2</e2> and increased levels of <e1>mir-146a</e1> were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of <e2>MCP-2</e2> and increased levels of <e1>mir-146a</e1> were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.', 'full_sentence': 'The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of <e2>MCP-2</e2> and increased levels of <e1>mir-146a</e1> were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (123, 128), 'entity_text': 'MCP-2'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (153, 161), 'entity_text': 'mir-146a'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('chemokine CCL8', 'Genes/Proteins', (51, 65))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CCL8', 'Genes/Proteins', (66, 71))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CCR5 chemokine receptor', 'Genes/Proteins', (92, 115))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CD4', 'Genes/Proteins', (142, 145))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CCR5', 'Genes/Proteins', (146, 150))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case miR-221/222\n",
      "add special case p27Kip1\n",
      "miR-221/222 2 p27Kip1 25\n",
      "(u\"Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\", 2, 25)\n",
      "geneword p27Kip1 mirword miR-221/222\n",
      "Conjuncts [reduced, p27Kip1]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conjuncts [p27Kip1, reduced]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "Conjunctions\n",
      "p27Kip1 [increased, p27Kip1, reduced, tumor, growth]\n",
      "SDP sent: Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "SDP     : [(miR-221/222, 'PROPN', 'amod', False), (knocked-down, 'VERB', 'nsubj', False), (increased, 'VERB', 'amod', False), (p27Kip1, 'PROPN', 'ROOT', True), (reduced, 'VERB', 'conj', False), (regulation, 'NOUN', 'nmod', False), (p27Kip1, 'PROPN', 'nmod', False)]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "Conjuncts [reduced, p27Kip1]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conjuncts [p27Kip1, reduced]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "passive? False\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "TextBetween: Consistently , miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1 .\n",
      "counts between 1 Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "['Consistently', ',', 'miR-221/222', 'knocked-down', 'through', 'antisense', \"2'-OME-oligonucleotides\", 'increased', 'p27Kip1', 'in', 'U251', 'glioma', 'subcutaneous', 'mice', 'and', 'strongly', 'reduced', 'tumor', 'growth', 'in', 'vivo', 'through', 'up', 'regulation', 'of', 'p27Kip1', '.']\n",
      "Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "full sentence Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'full_sentence': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 25), 'entity_text': 'miR-221/222'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (199, 206), 'entity_text': 'p27Kip1'}}\n",
      "['miR-221/miR-222', 'and', 'the', 'cell', 'cycle', 'inhibitor']\n",
      "['miR-221/miR-222', 'and', 'the', 'cell', 'cycle', 'inhibitor']\n",
      "['miR-21', 'and']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "The data strongly suggested that a regulatory loop between miR-21 and STAT3 might provide an insight into the mechanism of modulating EGFR/STAT3 signaling. \n",
      "('miR-21', 'Specific_miRNAs', (59, 65)) ('STAT3', 'Genes/Proteins', (70, 75))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('microRNA-29b', 'Specific_miRNAs', (59, 71)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('miR-29b', 'Specific_miRNAs', (73, 80)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "MicroRNA-29b regulates the expression level of human progranulin, a secreted glycoprotein implicated in frontotemporal dementia.  \n",
      "('MicroRNA-29b', 'Specific_miRNAs', (0, 12)) ('progranulin', 'Genes/Proteins', (53, 64))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites. \n",
      "('microRNA-34a', 'Specific_miRNAs', (65, 77)) ('CD47', 'Genes/Proteins', (149, 153))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites. \n",
      "('microRNA-155', 'Specific_miRNAs', (79, 91)) ('CD47', 'Genes/Proteins', (149, 153))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites. \n",
      "('microRNA-326', 'Specific_miRNAs', (96, 108)) ('CD47', 'Genes/Proteins', (149, 153))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case CD47\n",
      "add special case microRNA-155\n",
      "CD47 20 microRNA-155 26\n",
      "(u\"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\", 20, 26)\n",
      "geneword CD47 mirword microRNA-155\n",
      "Conjuncts [microRNA-155, microRNA-326, microRNA-34a]\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-34a\n",
      "Conjuncts [microRNA-34a, microRNA-326, microRNA-155]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-155\n",
      "Conjuncts [microRNA-34a, microRNA-155, microRNA-326]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "Conjunction before fc microRNA-326\n",
      "Conjunctions\n",
      "microRNA-34a [microRNA-34a, microRNA-155, microRNA-326]\n",
      "SDP sent: Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "SDP     : [(CD47, 'PROPN', 'nmod', False), (region, 'NOUN', 'dobj', False), (targeted, 'VERB', 'ROOT', False), (microRNA-155, 'PROPN', 'nmod', False)]\n",
      "[microRNA-155]\n",
      "[microRNA-326]\n",
      "0 [Three, microRNA, upregulated, in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, ), targeted, the, 3'-untranslated, region, of, CD47, in, reporter, assays, ,, with, microRNA-155, even, at, two, distinct, sites] 32 0 Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "Conjuncts [microRNA-155, microRNA-326, microRNA-34a]\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-34a\n",
      "Conjuncts [microRNA-34a, microRNA-326, microRNA-155]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-155\n",
      "Conjuncts [microRNA-34a, microRNA-155, microRNA-326]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "Conjunction before fc microRNA-326\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: region of CD47 in reporter assays , with microRNA-155 even at\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "full sentence Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}}, 'tagged_sent': \"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\", 'full_sentence': \"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (149, 153), 'entity_text': 'CD47'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (179, 191), 'entity_text': 'microRNA-155'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "The miRNA miR-23a was predicted to bind XIAP mRNA. \n",
      "('miR-23a', 'Specific_miRNAs', (10, 17)) ('XIAP', 'Genes/Proteins', (40, 44))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a. \n",
      "('miR-23a', 'Specific_miRNAs', (0, 7)) ('XIAP', 'Genes/Proteins', (37, 41))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case XIAP\n",
      "add special case miR-23a\n",
      "XIAP 8 miR-23a 11\n",
      "(u\"miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 8, 11)\n",
      "geneword XIAP mirword miR-23a\n",
      "Conjuncts [led, bound]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc bound\n",
      "fc add children for bound\n",
      "Conjuncts [miR-23a, UTR]\n",
      "neighbour miR-23a inhibition\n",
      "neighbour miR-23a inhibition\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [UTR, miR-23a]\n",
      "neighbour UTR of\n",
      "neighbour UTR of\n",
      "Conjunction before fc miR-23a\n",
      "Conjuncts [bound, led]\n",
      "neighbour bound the\n",
      "neighbour bound the\n",
      "Conjunction before fc led\n",
      "Conjunctions\n",
      "bound [bound, 3, UTR, miR-23a, led]\n",
      "SDP sent: miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "SDP     : [(XIAP, 'PROPN', 'nmod', False), (UTR, 'NOUN', 'dobj', True), (miR-23a, 'PROPN', 'conj', False)]\n",
      "[miR-23a]\n",
      "deepcheck connection rule noun found led inhibition [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro]\n",
      "0 [miR-23a, directly, bound, the, 3, ', UTR, of, XIAP, ,, and, miR-23a] 12 0 miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "1 [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro] 10 55 inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "2 [demonstrating, that, XIAP, is, a, previously, uncharacterized, target, for, miR-23a] 10 108 demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "Conjuncts [led, bound]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc bound\n",
      "fc add children for bound\n",
      "Conjuncts [miR-23a, UTR]\n",
      "neighbour miR-23a inhibition\n",
      "neighbour miR-23a inhibition\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [UTR, miR-23a]\n",
      "neighbour UTR of\n",
      "neighbour UTR of\n",
      "Conjunction before fc miR-23a\n",
      "Conjuncts [bound, led]\n",
      "neighbour bound the\n",
      "neighbour bound the\n",
      "Conjunction before fc led\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: UTR of XIAP , and miR-23a\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "full sentence miR-23a directly bound the 3' UTR of <e2>XIAP</e2>, and <e1>miR-23a</e1> inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}}, 'tagged_sent': \"miR-23a directly bound the 3' UTR of <e2>XIAP</e2>, and <e1>miR-23a</e1> inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 'full_sentence': \"miR-23a directly bound the 3' UTR of <e2>XIAP</e2>, and <e1>miR-23a</e1> inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (37, 41), 'entity_text': 'XIAP'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (47, 54), 'entity_text': 'miR-23a'}}\n",
      "add special case miR-23a\n",
      "add special case XIAP\n",
      "miR-23a 11 XIAP 18\n",
      "(u\"miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 11, 18)\n",
      "geneword XIAP mirword miR-23a\n",
      "Conjuncts [led, bound]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc bound\n",
      "fc add children for bound\n",
      "Conjuncts [miR-23a, UTR]\n",
      "neighbour miR-23a inhibition\n",
      "neighbour miR-23a inhibition\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [UTR, miR-23a]\n",
      "neighbour UTR of\n",
      "neighbour UTR of\n",
      "Conjunction before fc miR-23a\n",
      "Conjuncts [bound, led]\n",
      "neighbour bound the\n",
      "neighbour bound the\n",
      "Conjunction before fc led\n",
      "Conjunctions\n",
      "bound [bound, 3, UTR, miR-23a, led]\n",
      "SDP sent: miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "SDP     : [(miR-23a, 'PROPN', 'conj', False), (UTR, 'NOUN', 'dobj', True), (bound, 'VERB', 'ROOT', True), (led, 'VERB', 'conj', False), (increase, 'NOUN', 'nmod', False), (mRNA, 'NOUN', 'nmod', False), (XIAP, 'PROPN', 'compound', False)]\n",
      "[miR-23a]\n",
      "deepcheck connection rule noun found led inhibition [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro]\n",
      "0 [miR-23a, directly, bound, the, 3, ', UTR, of, XIAP, ,, and, miR-23a] 12 0 miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "1 [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro] 10 55 inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "2 [demonstrating, that, XIAP, is, a, previously, uncharacterized, target, for, miR-23a] 10 108 demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "Conjuncts [led, bound]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc bound\n",
      "fc add children for bound\n",
      "Conjuncts [miR-23a, UTR]\n",
      "neighbour miR-23a inhibition\n",
      "neighbour miR-23a inhibition\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [UTR, miR-23a]\n",
      "neighbour UTR of\n",
      "neighbour UTR of\n",
      "Conjunction before fc miR-23a\n",
      "Conjuncts [bound, led]\n",
      "neighbour bound the\n",
      "neighbour bound the\n",
      "Conjunction before fc led\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "TextBetween: , and miR-23a inhibition led to an increase in XIAP mRNA in\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7f8247389790>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "full sentence miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 'full_sentence': \"miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (47, 54), 'entity_text': 'miR-23a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (88, 92), 'entity_text': 'XIAP'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-23a regulation of X-linked inhibitor of apoptosis (XIAP) contributes to sex differences in the response to cerebral ischemia.  \n",
      "('miR-23a', 'Specific_miRNAs', (0, 7)) ('X-linked inhibitor of apoptosis', 'Genes/Proteins', (22, 53))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-23a regulation of X-linked inhibitor of apoptosis (XIAP) contributes to sex differences in the response to cerebral ischemia.  \n",
      "('miR-23a', 'Specific_miRNAs', (0, 7)) ('XIAP', 'Genes/Proteins', (55, 59))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "['SM22', ',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(']\n",
      "['calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(']\n",
      "['SM22', ',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and']\n",
      "['calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Here, two microRNAs, namely mir-7 and mir-153, have been identified to regulate alpha-synuclein levels post-transcriptionally. \n",
      "('mir-7', 'Specific_miRNAs', (28, 33)) ('alpha-synuclein', 'Genes/Proteins', (80, 95))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Here, two microRNAs, namely mir-7 and mir-153, have been identified to regulate alpha-synuclein levels post-transcriptionally. \n",
      "('mir-153', 'Specific_miRNAs', (38, 45)) ('alpha-synuclein', 'Genes/Proteins', (80, 95))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Post-transcriptional regulation of alpha-synuclein expression by mir-7 and mir-153.  \n",
      "('mir-7', 'Specific_miRNAs', (65, 70)) ('alpha-synuclein', 'Genes/Proteins', (35, 50))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Post-transcriptional regulation of alpha-synuclein expression by mir-7 and mir-153.  \n",
      "('mir-153', 'Specific_miRNAs', (75, 82)) ('alpha-synuclein', 'Genes/Proteins', (35, 50))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These findings indicate that miR-107 contributes to GRN expression regulation with implications for brain disorders.   \n",
      "('miR-107', 'Specific_miRNAs', (29, 36)) ('GRN', 'Genes/Proteins', (52, 55))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case miR-107\n",
      "add special case GRN\n",
      "miR-107 17 GRN 22\n",
      "(u\"In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\", 17, 22)\n",
      "geneword GRN mirword miR-107\n",
      "Conjuncts [decreased, subjacent]\n",
      "neighbour decreased miR-107\n",
      "neighbour decreased miR-107\n",
      "Conjunction before fc subjacent\n",
      "Conjuncts [subjacent, decreased]\n",
      "neighbour subjacent to\n",
      "neighbour subjacent to\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "Conjunctions\n",
      "subjacent [subjacent, decreased, miR-107]\n",
      "SDP sent: In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "SDP     : [(miR-107, 'PROPN', 'dobj', False), (decreased, 'VERB', 'conj', False), (augmentation, 'NOUN', 'nmod', False), (expression, 'NOUN', 'nmod', False), (GRN, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule noun found decreased miR-107 [decreased, miR-107, with, augmentation, of, neuronal, GRN, expression]\n",
      "deepcheck conj verb\n",
      "0 [In, brain, tissue, subjacent, to, 1.0, mm, depth, controlled, cortical, impact, ,, decreased, miR-107, with, augmentation, of, neuronal, GRN, expression] 20 0 In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "1 [surviving, hippocampal, neurons] 3 70 surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "2 [show] 1 100 show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "Conjuncts [decreased, subjacent]\n",
      "neighbour decreased miR-107\n",
      "neighbour decreased miR-107\n",
      "Conjunction before fc subjacent\n",
      "Conjuncts [subjacent, decreased]\n",
      "neighbour subjacent to\n",
      "neighbour subjacent to\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "passive? False\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "TextBetween: decreased miR-107 with augmentation of neuronal GRN expression\n",
      "counts between 1 Counter({'NEG': 1, 'POS': 1, 'NEU': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "full sentence In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased <e1>miR-107</e1> with augmentation of neuronal <e2>GRN</e2> expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased <e1>miR-107</e1> with augmentation of neuronal <e2>GRN</e2> expression.', 'full_sentence': 'In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased <e1>miR-107</e1> with augmentation of neuronal <e2>GRN</e2> expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (115, 122), 'entity_text': 'miR-107'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (153, 156), 'entity_text': 'GRN'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of GRN mRNA are recognized by miR-107 and are highly conserved among vertebrate species. \n",
      "('miR-107', 'Specific_miRNAs', (182, 189)) ('GRN', 'Genes/Proteins', (155, 158))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-107 regulates granulin/progranulin with implications for traumatic brain injury and neurodegenerative disease.  \n",
      "('miR-107', 'Specific_miRNAs', (0, 7)) ('granulin', 'Genes/Proteins', (18, 26))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-107 regulates granulin/progranulin with implications for traumatic brain injury and neurodegenerative disease.  \n",
      "('miR-107', 'Specific_miRNAs', (0, 7)) ('progranulin', 'Genes/Proteins', (27, 38))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "GRN mRNA interacts preferentially with Argonaute 2. In vitro and in vivo studies indicate that regulation of GRN by miR-107 may be functionally important. \n",
      "('miR-107', 'Specific_miRNAs', (116, 123)) ('GRN', 'Genes/Proteins', (109, 112))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case MiR-21\n",
      "add special case Bax\n",
      "MiR-21 0 Bax/Bcl-2 14\n",
      "(u\"MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\", 0, 14)\n",
      "geneword Bax/Bcl-2 mirword MiR-21\n",
      "Conjuncts [induced, protected]\n",
      "neighbour induced apoptosis\n",
      "neighbour induced apoptosis\n",
      "Conjunction before fc protected\n",
      "fc add children for protected\n",
      "Conjuncts [protected, induced]\n",
      "neighbour protected human\n",
      "neighbour protected human\n",
      "Conjunction before fc induced\n",
      "fc add children for induced\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity .\n",
      "neighbour activity None\n",
      "Conjunction before fc ratio\n",
      "fc add children for ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "fc add children for activity\n",
      "protected ratio\n",
      "apoptosis by decreasing\n",
      "Conjunctions\n",
      "protected [protected, human, glioblastoma, U87MG, cells, induced, apoptosis]\n",
      "ratio [Bax/Bcl-2, ratio, caspase-3, activity]\n",
      "SDP sent: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "SDP     : [(MiR-21, 'PROPN', 'nsubj', False), (protected, 'VERB', 'ROOT', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', False), (Bax/Bcl-2, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found induced apoptosis [induced, apoptosis]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "[caspase-3, activity]\n",
      "0 [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 19 0 MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "Context Fail <regex.Match object; span=(8, 14), match=' ratio'> regex.Regex('( ratio){e<=0}', flags=regex.V0)\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "TextBetween: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and\n",
      "counts between 1 Counter({'POS': 1, 'NEG': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "full sentence <e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing <e2>Bax</e2>/Bcl-2 ratio and caspase-3 activity.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing <e2>Bax</e2>/Bcl-2 ratio and caspase-3 activity.', 'full_sentence': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing <e2>Bax</e2>/Bcl-2 ratio and caspase-3 activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 6), 'entity_text': 'MiR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (120, 123), 'entity_text': 'Bax'}}\n",
      "add special case MiR-21\n",
      "add special case Bcl-2\n",
      "MiR-21 0 Bax/Bcl-2 14\n",
      "(u\"MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\", 0, 14)\n",
      "geneword Bax/Bcl-2 mirword MiR-21\n",
      "Conjuncts [induced, protected]\n",
      "neighbour induced apoptosis\n",
      "neighbour induced apoptosis\n",
      "Conjunction before fc protected\n",
      "fc add children for protected\n",
      "Conjuncts [protected, induced]\n",
      "neighbour protected human\n",
      "neighbour protected human\n",
      "Conjunction before fc induced\n",
      "fc add children for induced\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity .\n",
      "neighbour activity None\n",
      "Conjunction before fc ratio\n",
      "fc add children for ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "fc add children for activity\n",
      "protected ratio\n",
      "apoptosis by decreasing\n",
      "Conjunctions\n",
      "protected [protected, human, glioblastoma, U87MG, cells, induced, apoptosis]\n",
      "ratio [Bax/Bcl-2, ratio, caspase-3, activity]\n",
      "SDP sent: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "SDP     : [(MiR-21, 'PROPN', 'nsubj', False), (protected, 'VERB', 'ROOT', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', False), (Bax/Bcl-2, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found induced apoptosis [induced, apoptosis]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "[caspase-3, activity]\n",
      "0 [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 19 0 MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "Context Fail <regex.Match object; span=(8, 14), match=' ratio'> regex.Regex('( ratio){e<=0}', flags=regex.V0)\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "TextBetween: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and\n",
      "counts between 1 Counter({'POS': 1, 'NEG': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "full sentence <e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/<e2>Bcl-2</e2> ratio and caspase-3 activity.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/<e2>Bcl-2</e2> ratio and caspase-3 activity.', 'full_sentence': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/<e2>Bcl-2</e2> ratio and caspase-3 activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 6), 'entity_text': 'MiR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (124, 129), 'entity_text': 'Bcl-2'}}\n",
      "add special case miR-21\n",
      "add special case caspase-3\n",
      "miR-21 8 caspase-3 27\n",
      "(u\"Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\", 8, 27)\n",
      "geneword caspase-3 mirword miR-21\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity ,\n",
      "neighbour activity which\n",
      "Conjunction before fc ratio\n",
      "fc add children for ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "fc add children for activity\n",
      "Conjunctions\n",
      "ratio [Bax/Bcl-2, ratio, caspase-3, activity]\n",
      "SDP sent: Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "SDP     : [(miR-21, 'PROPN', 'nsubj', False), (inhibit, 'VERB', 'ccomp', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', True), (activity, 'NOUN', 'conj', False), (caspase-3, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity, ,, which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "[caspase-3, activity]\n",
      "0 [these, results, suggested, that, over-express, miR-21, could, inhibit, TMZ-induced, apoptosis, in, U87MG, cells, ,, at, least, in, part, ,, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 26 16 these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "1 [Taken, together] 2 0 Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "2 [] 0\n",
      "empty compartment []\n",
      "3 [which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ] 16 189 which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity ,\n",
      "neighbour activity which\n",
      "Conjunction before fc ratio\n",
      "fc add children for ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "fc add children for activity\n",
      "passive? False\n",
      "Counter({'NEG': 2})\n",
      "TextBetween: that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells , at least in part , by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "counts between 1 Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "['that', 'over-express', 'miR-21', 'could', 'inhibit', 'TMZ-induced', 'apoptosis', 'in', 'U87MG', 'cells', ',', 'at', 'least', 'in', 'part', ',', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and', 'caspase-3', 'activity']\n",
      "Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "full sentence Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.', 'full_sentence': 'Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (58, 64), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (169, 178), 'entity_text': 'caspase-3'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Hypoxia induces microRNA miR-210 in vitro and in vivo ephrin-A3 and neuronal pentraxin 1 are potentially regulated by miR-210.  \n",
      "('miR-210', 'Specific_miRNAs', (118, 125)) ('ephrin-A3', 'Genes/Proteins', (54, 63))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Hypoxia induces microRNA miR-210 in vitro and in vivo ephrin-A3 and neuronal pentraxin 1 are potentially regulated by miR-210.  \n",
      "('miR-210', 'Specific_miRNAs', (118, 125)) ('neuronal pentraxin 1', 'Genes/Proteins', (68, 88))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Furthermore, miR-210 was shown to bind to the predicted target sites of ephrin-A3 or neuronal pentraxin 1, causing repression in luciferase reporter activity. \n",
      "('miR-210', 'Specific_miRNAs', (13, 20)) ('ephrin-A3', 'Genes/Proteins', (72, 81))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Furthermore, miR-210 was shown to bind to the predicted target sites of ephrin-A3 or neuronal pentraxin 1, causing repression in luciferase reporter activity. \n",
      "('miR-210', 'Specific_miRNAs', (13, 20)) ('neuronal pentraxin 1', 'Genes/Proteins', (85, 105))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Our data strongly indicate that CDK6 is regulated by microRNA 124 in medulloblastoma and that miR 124 modulates medulloblastoma cell growth.   \n",
      "('microRNA 124', 'Specific_miRNAs', (53, 65)) ('CDK6', 'Genes/Proteins', (32, 36))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "['miR-124a', 'and']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays. \n",
      "('miR 124a', 'Specific_miRNAs', (31, 39)) ('CDK6', 'Genes/Proteins', (44, 48))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons. \n",
      "('miR-128a', 'Specific_miRNAs', (73, 81)) ('Tau', 'Genes/Proteins', (122, 125))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "['Hsa-miR-30b', ',', 'Hsa-miR-30d', 'and']\n",
      "['Hsa-miR-30d', 'and']\n",
      "['hsa-miR-30b', ',', 'hsa-miR-30d', 'and']\n",
      "['hsa-miR-30d', 'and']\n",
      "['ZFAT1', ',', 'LOC286094', ',', 'KHDRBS3', ')', 'and', 'two', 'genes', 'encoding', 'micro-RNAs', '(']\n",
      "['KHDRBS3', ')', 'and', 'two', 'genes', 'encoding', 'micro-RNAs', '(']\n",
      "['ZFAT1', ',', 'LOC286094', ',', 'KHDRBS3', ')', 'and', 'two', 'genes', 'encoding', 'micro-RNAs', '(', 'hsa-miR-30b', ',']\n",
      "['KHDRBS3', ')', 'and', 'two', 'genes', 'encoding', 'micro-RNAs', '(', 'hsa-miR-30b', ',']\n",
      "['hsa-miR-30b', ',', 'hsa-miR-30d', 'and']\n",
      "['hsa-miR-30d', 'and']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "In this study, we focused on miR-29b and miR-125a, which were predicted to regulate PDPN, and demonstrated that these microRNAs directly target the 3' untranslated region of PDPN and inhibit invasion, apoptosis, and proliferation of glioblastomas. \n",
      "('miR-29b', 'Specific_miRNAs', (29, 36)) ('PDPN', 'Genes/Proteins', (84, 88))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "In this study, we focused on miR-29b and miR-125a, which were predicted to regulate PDPN, and demonstrated that these microRNAs directly target the 3' untranslated region of PDPN and inhibit invasion, apoptosis, and proliferation of glioblastomas. \n",
      "('miR-125a', 'Specific_miRNAs', (41, 49)) ('PDPN', 'Genes/Proteins', (84, 88))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case microRNA-433\n",
      "add special case FGF20\n",
      "microRNA-433 17 FGF20 22\n",
      "(u\"We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\", 17, 22)\n",
      "geneword FGF20 mirword microRNA-433\n",
      "Conjunctions\n",
      "SDP sent: We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "SDP     : [(microRNA-433, 'PROPN', 'nmod', False), (site, 'NOUN', 'dobj', False), (disrupts, 'VERB', 'ccomp', False), (translation, 'NOUN', 'nmod', False), (FGF20, 'PROPN', 'nmod', False)]\n",
      "0 [We, show, in, several, functional, assays] 6 0 We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "1 [the, risk, allele, for, rs12720208, disrupts, a, binding, site, for, microRNA-433, ,, increasing, translation, of, FGF20, in, vitro, and, in, vivo] 21 42 the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "passive? False\n",
      "Counter()\n",
      "TextBetween: site for microRNA-433 , increasing translation of FGF20 in vitro\n",
      "counts between 1 Counter({'POS': 1, 'NEU': 1})\n",
      "['site', 'for', 'microRNA-433', ',', 'increasing', 'translation', 'of', 'FGF20', 'in', 'vitro']\n",
      "Counter({'POS': 1, 'NEU': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "full sentence We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.', 'full_sentence': 'We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (101, 113), 'entity_text': 'microRNA-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (141, 146), 'entity_text': 'FGF20'}}\n",
      "Total:      232\n",
      "Correct:    218 0.9396551724137931\n",
      "Incorrect:  14 0.0603448275862069\n",
      "classes Counter()\n",
      "precision 0.9642857142857143\n",
      "recall 0.9375\n",
      "specificity 0.9431818181818182\n",
      "f1 0.9507042253521126\n",
      "Correct classified:      126 0.5431034482758621 0.9\n",
      "Incorrect classified:    14 0.0603448275862069 0.1\n",
      "Random classified:  53 0.22844827586206898 0.37857142857142856\n",
      "Counter({'counts between': 9, 'return': 3, 'compartment gene mir': 1, 'counts between alternating': 1})\n",
      "Correct interaction&classified:      121 0.521551724137931 0.8642857142857143\n",
      "Classified by:  Counter({'return': 55, 'counts between': 32, 'compartment mir gene': 12, 'counts between equal': 11, 'counts opp': 10, 'between gene mir': 9, 'between mir gene': 4, 'compartment mir gene neg corr': 4, 'counts between alternating': 2, 'compartment gene mir': 1})\n",
      "[[[133   3]\n",
      "  [  0   4]]\n",
      "\n",
      " [[139   0]\n",
      "  [  0   1]]\n",
      "\n",
      " [[ 81   6]\n",
      "  [  5  48]]\n",
      "\n",
      " [[ 57   2]\n",
      "  [  8  73]]\n",
      "\n",
      " [[136   3]\n",
      "  [  1   0]]]\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      " GENE_MIR_NEU       0.57      1.00      0.73         4\n",
      "  GENE_MIR_UP       1.00      1.00      1.00         1\n",
      "MIR_GENE_DOWN       0.89      0.91      0.90        53\n",
      " MIR_GENE_NEU       0.97      0.90      0.94        81\n",
      "  MIR_GENE_UP       0.00      0.00      0.00         1\n",
      "\n",
      "     accuracy                           0.90       140\n",
      "    macro avg       0.69      0.76      0.71       140\n",
      " weighted avg       0.92      0.90      0.91       140\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "test_results = runCheck(-1, \"TEST\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading relations scai_test_rels.tsv\n",
      "Interactions 232\n",
      "Non NA Interactions 138\n",
      "Will test 1 subsets.\n",
      "Testing checks ('compartment', 'between', 'counts', 'return')\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('APOBEC3C', 'Genes/Proteins', (99, 103))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('ANGPTL6', 'Genes/Proteins', (105, 112))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('Bmi-1', 'Genes/Proteins', (115, 120))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('angiopoietin-related growth factor protein 5', 'Genes/Proteins', (49, 93))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('APOBEC3C', 'Genes/Proteins', (95, 99))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('ANGPTL6', 'Genes/Proteins', (101, 108))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case miRNA-128\n",
      "add special case Bmi-1\n",
      "miRNA-128 2 Bmi-1 39\n",
      "(u\"Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\", 2, 39)\n",
      "geneword Bmi-1 mirword miRNA-128\n",
      "Conjuncts [inhibits, Bmi-1, found, promotes]\n",
      "neighbour inhibits the\n",
      "neighbour inhibits the\n",
      "neighbour Bmi-1 ,\n",
      "neighbour Bmi-1 and\n",
      "neighbour found to\n",
      "neighbour found to\n",
      "Conjunction before fc promotes\n",
      "fc add children for promotes\n",
      "Conjuncts [promotes, Bmi-1, found, inhibits]\n",
      "neighbour promotes stem\n",
      "neighbour promotes stem\n",
      "neighbour Bmi-1 ,\n",
      "neighbour Bmi-1 and\n",
      "neighbour found to\n",
      "neighbour found to\n",
      "Conjunction before fc inhibits\n",
      "fc add children for inhibits\n",
      "Conjuncts [differentiation, senescence]\n",
      "neighbour differentiation ,\n",
      "neighbour differentiation Bmi-1\n",
      "Conjunction before fc senescence\n",
      "fc add children for senescence\n",
      "Conjuncts [senescence, differentiation]\n",
      "neighbour senescence and\n",
      "neighbour senescence and\n",
      "Conjunction before fc differentiation\n",
      "Conjuncts [promotes, inhibits, found, Bmi-1]\n",
      "neighbour promotes stem\n",
      "neighbour promotes stem\n",
      "neighbour inhibits the\n",
      "neighbour inhibits the\n",
      "neighbour found to\n",
      "neighbour found to\n",
      "Conjunction before fc Bmi-1\n",
      "Conjuncts [promotes, inhibits, Bmi-1, found]\n",
      "neighbour promotes stem\n",
      "neighbour promotes stem\n",
      "neighbour inhibits the\n",
      "neighbour inhibits the\n",
      "neighbour Bmi-1 ,\n",
      "neighbour Bmi-1 and\n",
      "Conjunction before fc found\n",
      "Conjunctions\n",
      "protein [angiopoietin-related, a, up-regulated]\n",
      "ARP5 [(, )]\n",
      "ANGPTL6 [ANGPTL6]\n",
      "promotes [promotes, renewal, inhibits, expression, of, known, genes, involved, in, senescence, differentiation, Bmi-1, found]\n",
      "E2F-3a [E2F-3a]\n",
      "SDP sent: Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\n",
      "SDP     : [(miRNA-128, 'PROPN', 'compound', False), (targets, 'NOUN', 'ROOT', False), (protein, 'NOUN', 'appos', False), (suppressor, 'NOUN', 'appos', False), (promotes, 'VERB', 'acl:relcl', True), (inhibits, 'VERB', 'conj', True), (Bmi-1, 'PROPN', 'conj', False)]\n",
      "deepcheck connection rule noun found found transcription [a, transcription, factor, critical, for, the, control, of, cell-cycle, progression, ,, E2F-3a, ,, were, found, to, be, up-regulated]\n",
      "deepcheck connection rule [to, be, up-regulated]\n",
      "0 [Three, bioinformatics-verified, miRNA-128, targets, ,, angiopoietin-related, growth, factor, protein, 5, (, ARP5, ;, ANGPTL6, ), ,, a, transcription, suppressor, that, promotes, stem, cell, renewal] 24 0 Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\n",
      "1 [inhibits, the, expression, of, known, tumor, suppressor, genes, involved, in, senescence, and, differentiation, ,, Bmi-1] 15 174 inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\n",
      "2 [a, transcription, factor, critical, for, the, control, of, cell-cycle, progression, ,, E2F-3a, ,, were, found, to, be, up-regulated] 18 285 a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\n",
      "Conjuncts [inhibits, Bmi-1, found, promotes]\n",
      "neighbour inhibits the\n",
      "neighbour inhibits the\n",
      "neighbour Bmi-1 ,\n",
      "neighbour Bmi-1 and\n",
      "neighbour found to\n",
      "neighbour found to\n",
      "Conjunction before fc promotes\n",
      "fc add children for promotes\n",
      "Conjuncts [promotes, Bmi-1, found, inhibits]\n",
      "neighbour promotes stem\n",
      "neighbour promotes stem\n",
      "neighbour Bmi-1 ,\n",
      "neighbour Bmi-1 and\n",
      "neighbour found to\n",
      "neighbour found to\n",
      "Conjunction before fc inhibits\n",
      "fc add children for inhibits\n",
      "Conjuncts [differentiation, senescence]\n",
      "neighbour differentiation ,\n",
      "neighbour differentiation Bmi-1\n",
      "Conjunction before fc senescence\n",
      "fc add children for senescence\n",
      "Conjuncts [senescence, differentiation]\n",
      "neighbour senescence and\n",
      "neighbour senescence and\n",
      "Conjunction before fc differentiation\n",
      "Conjuncts [promotes, inhibits, found, Bmi-1]\n",
      "neighbour promotes stem\n",
      "neighbour promotes stem\n",
      "neighbour inhibits the\n",
      "neighbour inhibits the\n",
      "neighbour found to\n",
      "neighbour found to\n",
      "Conjunction before fc Bmi-1\n",
      "Conjuncts [promotes, inhibits, Bmi-1, found]\n",
      "neighbour promotes stem\n",
      "neighbour promotes stem\n",
      "neighbour inhibits the\n",
      "neighbour inhibits the\n",
      "neighbour Bmi-1 ,\n",
      "neighbour Bmi-1 and\n",
      "Conjunction before fc found\n",
      "Recognized overlapping entities: {Interval(274, 279, ('GENE_OR_GENE_PRODUCT', 'Bmi-1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "Three bioinformatics-verified miRNA-128 targets , angiopoietin-related growth factor protein 5 ( ARP5 ; ANGPTL6 ) , a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation , Bmi-1 , and\n",
      "counts between 1 Counter({'NEU': 3, 'NEG': 3, 'POS': 1})\n",
      "['Three', 'bioinformatics-verified', 'miRNA-128', 'targets', ',', 'angiopoietin-related', 'growth', 'factor', 'protein', '5', '(', 'ARP5', ';', 'ANGPTL6', ')', ',', 'a', 'transcription', 'suppressor', 'that', 'promotes', 'stem', 'cell', 'renewal', 'and', 'inhibits', 'the', 'expression', 'of', 'known', 'tumor', 'suppressor', 'genes', 'involved', 'in', 'senescence', 'and', 'differentiation', ',', 'Bmi-1', ',', 'and']\n",
      "Counter({'NEU': 3, 'NEG': 3, 'POS': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\n",
      "full sentence Three bioinformatics-verified <e1>miRNA-128</e1> targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, <e2>Bmi-1</e2>, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Three bioinformatics-verified <e1>miRNA-128</e1> targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, <e2>Bmi-1</e2>, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.', 'full_sentence': 'Three bioinformatics-verified <e1>miRNA-128</e1> targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, <e2>Bmi-1</e2>, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (30, 39), 'entity_text': 'miRNA-128'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (274, 279), 'entity_text': 'Bmi-1'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (100, 107)) ('SRF', 'Genes/Proteins', (13, 16))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (100, 107)) ('myocardin', 'Genes/Proteins', (21, 30))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (112, 119)) ('SRF', 'Genes/Proteins', (13, 16))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (112, 119)) ('myocardin', 'Genes/Proteins', (21, 30))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These abnormalities reflect the regulation of a cadre of modulators of SRF activity and actin dynamics by miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (106, 113)) ('SRF', 'Genes/Proteins', (71, 74))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These abnormalities reflect the regulation of a cadre of modulators of SRF activity and actin dynamics by miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (118, 125)) ('SRF', 'Genes/Proteins', (71, 74))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case microRNA-30\n",
      "add special case Oas1\n",
      "microRNA-30 12 Oas1 18\n",
      "(u\"Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\", 12, 18)\n",
      "geneword Oas1 mirword microRNA-30\n",
      "Conjunctions\n",
      "SDP sent: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "SDP     : [(microRNA-30, 'PROPN', 'compound', False), (precursor, 'NOUN', 'nmod', False), (features, 'NOUN', 'dobj', False), (implementing, 'VERB', 'advcl', False), (Modification, 'NOUN', 'nsubj', False), (avoided, 'VERB', 'ROOT', False), (induction, 'NOUN', 'dobj', False), (Oas1, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "Recognized overlapping entities: {Interval(127, 131, ('GENE_OR_GENE_PRODUCT', 'Oas1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "naturally occurring microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in\n",
      "counts between 1 Counter({'POS': 1})\n",
      "['naturally', 'occurring', 'microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "Counter({'POS': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "full sentence Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'full_sentence': 'Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (88, 99), 'entity_text': 'microRNA-30'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (127, 131), 'entity_text': 'Oas1'}}\n",
      "add special case miR-30\n",
      "add special case Oas1\n",
      "miR-30 14 Oas1 18\n",
      "(u\"Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\", 14, 18)\n",
      "geneword Oas1 mirword miR-30\n",
      "Conjunctions\n",
      "SDP sent: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "SDP     : [(miR-30, 'PROPN', 'compound', False), (precursor, 'NOUN', 'nmod', False), (features, 'NOUN', 'dobj', False), (implementing, 'VERB', 'advcl', False), (Modification, 'NOUN', 'nsubj', False), (avoided, 'VERB', 'ROOT', False), (induction, 'NOUN', 'dobj', False), (Oas1, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "Recognized overlapping entities: {Interval(127, 131, ('GENE_OR_GENE_PRODUCT', 'Oas1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in\n",
      "counts between 1 Counter({'POS': 1})\n",
      "['microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "Counter({'POS': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "full sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'full_sentence': 'Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (101, 107), 'entity_text': 'miR-30'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (127, 131), 'entity_text': 'Oas1'}}\n",
      "add special case mir-21\n",
      "add special case Pdcd4\n",
      "mir-21 10 Pdcd4 17\n",
      "(u\"In addition, our data indicate that either downregulation of mir-21 or overexpression of its target, Pdcd4, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.\", 10, 17)\n",
      "geneword Pdcd4 mirword mir-21\n",
      "Conjuncts [overexpression, mir-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc mir-21\n",
      "fc add children for mir-21\n",
      "Conjuncts [mir-21, overexpression]\n",
      "neighbour mir-21 or\n",
      "neighbour mir-21 or\n",
      "Conjunction before fc overexpression\n",
      "fc add children for overexpression\n",
      "Conjuncts [decreased, leads]\n",
      "neighbour decreased colony\n",
      "neighbour decreased colony\n",
      "Conjunction before fc leads\n",
      "fc add children for leads\n",
      "Conjuncts [apoptosis, proliferation]\n",
      "neighbour apoptosis ,\n",
      "neighbour apoptosis and\n",
      "Conjunction before fc proliferation\n",
      "fc add children for proliferation\n",
      "fc add children for proliferation\n",
      "Conjuncts [proliferation, apoptosis]\n",
      "neighbour proliferation ,\n",
      "neighbour proliferation increased\n",
      "Conjunction before fc apoptosis\n",
      "fc add children for apoptosis\n",
      "Conjuncts [leads, decreased]\n",
      "neighbour leads to\n",
      "neighbour leads to\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "fc add children for decreased\n",
      "Conjunctions\n",
      "mir-21 [of, mir-21, overexpression, of, target, Pdcd4]\n",
      "leads [leads, to, decreased, proliferation, increased, apoptosis, decreased, formation, in, soft, agar]\n",
      "achildren {mir-21}\n",
      "achildren False\n",
      "SDP sent: In addition, our data indicate that either downregulation of mir-21 or overexpression of its target, Pdcd4, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.\n",
      "SDP     : [(mir-21, 'PROPN', 'nmod', False), (indicate, 'VERB', 'ROOT', False), (leads, 'VERB', 'ccomp', False), (downregulation, 'NOUN', 'nsubj', False), (mir-21, 'PROPN', 'nmod', False)]\n",
      "deepcheck connection rule noun found decreased colony [decreased, colony, formation, in, soft, agar]\n",
      "0 [In, addition, ,, our, data, indicate] 6 0 In addition, our data indicate that either downregulation of mir-21 or overexpression of its target, Pdcd4, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.\n",
      "1 [downregulation, of, mir-21, or, overexpression, of, its, target, ,, Pdcd4, ,, in, glioblastoma-derived, cell, lines, leads, to, decreased, proliferation, ,, increased, apoptosis] 22 43 downregulation of mir-21 or overexpression of its target, Pdcd4, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.\n",
      "2 [decreased, colony, formation, in, soft, agar] 6 202 decreased colony formation in soft agar.\n",
      "Conjuncts [overexpression, mir-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc mir-21\n",
      "fc add children for mir-21\n",
      "Conjuncts [mir-21, overexpression]\n",
      "neighbour mir-21 or\n",
      "neighbour mir-21 or\n",
      "Conjunction before fc overexpression\n",
      "fc add children for overexpression\n",
      "Conjuncts [decreased, leads]\n",
      "neighbour decreased colony\n",
      "neighbour decreased colony\n",
      "Conjunction before fc leads\n",
      "fc add children for leads\n",
      "Conjuncts [apoptosis, proliferation]\n",
      "neighbour apoptosis ,\n",
      "neighbour apoptosis and\n",
      "Conjunction before fc proliferation\n",
      "fc add children for proliferation\n",
      "fc add children for proliferation\n",
      "Conjuncts [proliferation, apoptosis]\n",
      "neighbour proliferation ,\n",
      "neighbour proliferation increased\n",
      "Conjunction before fc apoptosis\n",
      "fc add children for apoptosis\n",
      "Conjuncts [leads, decreased]\n",
      "neighbour leads to\n",
      "neighbour leads to\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "fc add children for decreased\n",
      "Recognized overlapping entities: {Interval(101, 106, ('GENE_OR_GENE_PRODUCT', 'Pdcd4'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "downregulation of mir-21 or overexpression of its target , Pdcd4 , in\n",
      "counts between 1 Counter({'NEG': 1, 'POS': 1, 'NEU': 1})\n",
      "(False, {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence In addition, our data indicate that either downregulation of mir-21 or overexpression of its target, Pdcd4, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.\n",
      "full sentence In addition, our data indicate that either downregulation of <e1>mir-21</e1> or overexpression of its target, <e2>Pdcd4</e2>, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'In addition, our data indicate that either downregulation of <e1>mir-21</e1> or overexpression of its target, <e2>Pdcd4</e2>, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.', 'full_sentence': 'In addition, our data indicate that either downregulation of <e1>mir-21</e1> or overexpression of its target, <e2>Pdcd4</e2>, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (61, 67), 'entity_text': 'mir-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (101, 106), 'entity_text': 'Pdcd4'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('IQGAP1', 'Genes/Proteins', (132, 138))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('IQ motif containing GTPase activating protein 1', 'Genes/Proteins', (83, 130))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('laminin γ1', 'Genes/Proteins', (141, 151))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('LAMC1', 'Genes/Proteins', (153, 158))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('integrin β1', 'Genes/Proteins', (164, 175))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('ITGB1', 'Genes/Proteins', (177, 182))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('chemokine CCL8', 'Genes/Proteins', (51, 65))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CCL8', 'Genes/Proteins', (66, 71))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CCR5 chemokine receptor', 'Genes/Proteins', (92, 115))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CD4', 'Genes/Proteins', (142, 145))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CCR5', 'Genes/Proteins', (146, 150))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case miR-9\n",
      "add special case STAT3\n",
      "miR-9 4 STAT3 22\n",
      "(u\"Foremost among these is miR-9, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of STAT3.\", 4, 22)\n",
      "geneword STAT3 mirword miR-9\n",
      "Conjuncts [inhibiting, downregulating]\n",
      "neighbour inhibiting activation\n",
      "neighbour inhibiting activation\n",
      "Conjunction before fc downregulating\n",
      "fc add children for downregulating\n",
      "Conjuncts [downregulating, inhibiting]\n",
      "neighbour downregulating expression\n",
      "neighbour downregulating expression\n",
      "Conjunction before fc inhibiting\n",
      "fc add children for inhibiting\n",
      "Conjunctions\n",
      "downregulating [downregulating, expression, of, kinases, inhibiting, activation, of, STAT3]\n",
      "SDP sent: Foremost among these is miR-9, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of STAT3.\n",
      "SDP     : [(miR-9, 'PROPN', 'ROOT', False), (miR-9, 'PROPN', 'ROOT', False)]\n",
      "deepcheck connection rule [by, downregulating, expression, of, JAK, kinases, and, inhibiting, activation, of, STAT3]\n",
      "deepcheck connection rule noun found inhibiting activation [inhibiting, activation, of, STAT3]\n",
      "0 [Foremost, among, these, is, miR-9, ,, which, suppresses, mesenchymal, differentiation, in, glioblastoma, by, downregulating, expression, of, JAK, kinases] 18 0 Foremost among these is miR-9, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of STAT3.\n",
      "1 [inhibiting, activation, of, STAT3] 4 140 inhibiting activation of STAT3.\n",
      "Conjuncts [inhibiting, downregulating]\n",
      "neighbour inhibiting activation\n",
      "neighbour inhibiting activation\n",
      "Conjunction before fc downregulating\n",
      "fc add children for downregulating\n",
      "Conjuncts [downregulating, inhibiting]\n",
      "neighbour downregulating expression\n",
      "neighbour downregulating expression\n",
      "Conjunction before fc inhibiting\n",
      "fc add children for inhibiting\n",
      "Recognized overlapping entities: {Interval(165, 170, ('GENE_OR_GENE_PRODUCT', 'STAT3'))}\n",
      "passive? False\n",
      "Counter()\n",
      "these is miR-9 , which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of STAT3 .\n",
      "counts between 1 Counter({'NEG': 3, 'NEU': 2})\n",
      "['these', 'is', 'miR-9', ',', 'which', 'suppresses', 'mesenchymal', 'differentiation', 'in', 'glioblastoma', 'by', 'downregulating', 'expression', 'of', 'JAK', 'kinases', 'and', 'inhibiting', 'activation', 'of', 'STAT3', '.']\n",
      "Counter({'NEG': 3, 'NEU': 2})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Foremost among these is miR-9, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of STAT3.\n",
      "full sentence Foremost among these is <e1>miR-9</e1>, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of <e2>STAT3</e2>.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Foremost among these is <e1>miR-9</e1>, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of <e2>STAT3</e2>.', 'full_sentence': 'Foremost among these is <e1>miR-9</e1>, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of <e2>STAT3</e2>.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (24, 29), 'entity_text': 'miR-9'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (165, 170), 'entity_text': 'STAT3'}}\n",
      "add special case miR-221/222\n",
      "add special case p27Kip1\n",
      "miR-221/222 2 p27Kip1 25\n",
      "(u\"Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\", 2, 25)\n",
      "geneword p27Kip1 mirword miR-221/222\n",
      "Conjuncts [reduced, p27Kip1]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conjuncts [p27Kip1, reduced]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "fc add children for reduced\n",
      "Conjunctions\n",
      "p27Kip1 [increased, p27Kip1, in, subcutaneous, mice, reduced, growth, through, up, regulation, of, p27Kip1]\n",
      "SDP sent: Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "SDP     : [(miR-221/222, 'PROPN', 'amod', False), (knocked-down, 'VERB', 'nsubj', False), (increased, 'VERB', 'amod', False), (p27Kip1, 'PROPN', 'ROOT', True), (reduced, 'VERB', 'conj', False), (regulation, 'NOUN', 'nmod', False), (p27Kip1, 'PROPN', 'nmod', False)]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "Conjuncts [reduced, p27Kip1]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conjuncts [p27Kip1, reduced]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "fc add children for reduced\n",
      "Recognized overlapping entities: {Interval(199, 206, ('GENE_OR_GENE_PRODUCT', 'p27Kip1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "Consistently , miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1 .\n",
      "counts between 1 Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "['Consistently', ',', 'miR-221/222', 'knocked-down', 'through', 'antisense', \"2'-OME-oligonucleotides\", 'increased', 'p27Kip1', 'in', 'U251', 'glioma', 'subcutaneous', 'mice', 'and', 'strongly', 'reduced', 'tumor', 'growth', 'in', 'vivo', 'through', 'up', 'regulation', 'of', 'p27Kip1', '.']\n",
      "Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "full sentence Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'full_sentence': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 25), 'entity_text': 'miR-221/222'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (199, 206), 'entity_text': 'p27Kip1'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "The data strongly suggested that a regulatory loop between miR-21 and STAT3 might provide an insight into the mechanism of modulating EGFR/STAT3 signaling. \n",
      "('miR-21', 'Specific_miRNAs', (59, 65)) ('STAT3', 'Genes/Proteins', (70, 75))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('microRNA-29b', 'Specific_miRNAs', (59, 71)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('miR-29b', 'Specific_miRNAs', (73, 80)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "MicroRNA-29b regulates the expression level of human progranulin, a secreted glycoprotein implicated in frontotemporal dementia.  \n",
      "('MicroRNA-29b', 'Specific_miRNAs', (0, 12)) ('progranulin', 'Genes/Proteins', (53, 64))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites. \n",
      "('microRNA-34a', 'Specific_miRNAs', (65, 77)) ('CD47', 'Genes/Proteins', (149, 153))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites. \n",
      "('microRNA-155', 'Specific_miRNAs', (79, 91)) ('CD47', 'Genes/Proteins', (149, 153))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites. \n",
      "('microRNA-326', 'Specific_miRNAs', (96, 108)) ('CD47', 'Genes/Proteins', (149, 153))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case CD47\n",
      "add special case microRNA-155\n",
      "CD47 20 microRNA-155 26\n",
      "(u\"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\", 20, 26)\n",
      "geneword CD47 mirword microRNA-155\n",
      "Conjuncts [microRNA-155, microRNA-326, microRNA-34a]\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-34a\n",
      "Conjuncts [microRNA-34a, microRNA-326, microRNA-155]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-155\n",
      "Conjuncts [microRNA-34a, microRNA-155, microRNA-326]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "Conjunction before fc microRNA-326\n",
      "Conjunctions\n",
      "microRNA-34a [microRNA-34a, microRNA-155, microRNA-326]\n",
      "SDP sent: Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "SDP     : [(CD47, 'PROPN', 'nmod', False), (region, 'NOUN', 'dobj', False), (targeted, 'VERB', 'ROOT', False), (microRNA-155, 'PROPN', 'nmod', False)]\n",
      "0 [Three, microRNA, upregulated, in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, ), targeted, the, 3'-untranslated, region, of, CD47, in, reporter, assays, ,, with, microRNA-155, even, at, two, distinct, sites] 32 0 Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "Conjuncts [microRNA-155, microRNA-326, microRNA-34a]\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-34a\n",
      "Conjuncts [microRNA-34a, microRNA-326, microRNA-155]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-155\n",
      "Conjuncts [microRNA-34a, microRNA-155, microRNA-326]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "Conjunction before fc microRNA-326\n",
      "Recognized overlapping entities: {Interval(149, 153, ('GENE_OR_GENE_PRODUCT', 'CD47'))}\n",
      "passive? False\n",
      "Counter()\n",
      "region of CD47 in reporter assays , with microRNA-155 even at\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "full sentence Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}}, 'tagged_sent': \"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\", 'full_sentence': \"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (149, 153), 'entity_text': 'CD47'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (179, 191), 'entity_text': 'microRNA-155'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "The miRNA miR-23a was predicted to bind XIAP mRNA. \n",
      "('miR-23a', 'Specific_miRNAs', (10, 17)) ('XIAP', 'Genes/Proteins', (40, 44))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a. \n",
      "('miR-23a', 'Specific_miRNAs', (0, 7)) ('XIAP', 'Genes/Proteins', (37, 41))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case miR-23a\n",
      "add special case XIAP\n",
      "miR-23a 11 XIAP 18\n",
      "(u\"miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 11, 18)\n",
      "geneword XIAP mirword miR-23a\n",
      "Conjuncts [led, bound]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc bound\n",
      "fc add children for bound\n",
      "Conjuncts [miR-23a, UTR]\n",
      "neighbour miR-23a inhibition\n",
      "neighbour miR-23a inhibition\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [UTR, miR-23a]\n",
      "neighbour UTR of\n",
      "neighbour UTR of\n",
      "Conjunction before fc miR-23a\n",
      "Conjuncts [bound, led]\n",
      "neighbour bound the\n",
      "neighbour bound the\n",
      "Conjunction before fc led\n",
      "fc add children for led\n",
      "Conjunctions\n",
      "bound [bound, 3, UTR, of, XIAP, miR-23a, led, to, increase, in, mRNA]\n",
      "SDP sent: miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "SDP     : [(miR-23a, 'PROPN', 'conj', False), (UTR, 'NOUN', 'dobj', True), (bound, 'VERB', 'ROOT', True), (led, 'VERB', 'conj', False), (increase, 'NOUN', 'nmod', False), (mRNA, 'NOUN', 'nmod', False), (XIAP, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule noun found led inhibition [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro]\n",
      "0 [miR-23a, directly, bound, the, 3, ', UTR, of, XIAP, ,, and, miR-23a] 12 0 miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "1 [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro] 10 55 inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "2 [demonstrating, that, XIAP, is, a, previously, uncharacterized, target, for, miR-23a] 10 108 demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "Conjuncts [led, bound]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc bound\n",
      "fc add children for bound\n",
      "Conjuncts [miR-23a, UTR]\n",
      "neighbour miR-23a inhibition\n",
      "neighbour miR-23a inhibition\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [UTR, miR-23a]\n",
      "neighbour UTR of\n",
      "neighbour UTR of\n",
      "Conjunction before fc miR-23a\n",
      "Conjuncts [bound, led]\n",
      "neighbour bound the\n",
      "neighbour bound the\n",
      "Conjunction before fc led\n",
      "fc add children for led\n",
      "Recognized overlapping entities: {Interval(88, 92, ('GENE_OR_GENE_PRODUCT', 'XIAP'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      ", and miR-23a inhibition led to an increase in XIAP mRNA in\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7ff758df31f0>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "full sentence miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 'full_sentence': \"miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (47, 54), 'entity_text': 'miR-23a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (88, 92), 'entity_text': 'XIAP'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-23a regulation of X-linked inhibitor of apoptosis (XIAP) contributes to sex differences in the response to cerebral ischemia.  \n",
      "('miR-23a', 'Specific_miRNAs', (0, 7)) ('X-linked inhibitor of apoptosis', 'Genes/Proteins', (22, 53))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-23a regulation of X-linked inhibitor of apoptosis (XIAP) contributes to sex differences in the response to cerebral ischemia.  \n",
      "('miR-23a', 'Specific_miRNAs', (0, 7)) ('XIAP', 'Genes/Proteins', (55, 59))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Here, two microRNAs, namely mir-7 and mir-153, have been identified to regulate alpha-synuclein levels post-transcriptionally. \n",
      "('mir-7', 'Specific_miRNAs', (28, 33)) ('alpha-synuclein', 'Genes/Proteins', (80, 95))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Here, two microRNAs, namely mir-7 and mir-153, have been identified to regulate alpha-synuclein levels post-transcriptionally. \n",
      "('mir-153', 'Specific_miRNAs', (38, 45)) ('alpha-synuclein', 'Genes/Proteins', (80, 95))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Post-transcriptional regulation of alpha-synuclein expression by mir-7 and mir-153.  \n",
      "('mir-7', 'Specific_miRNAs', (65, 70)) ('alpha-synuclein', 'Genes/Proteins', (35, 50))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Post-transcriptional regulation of alpha-synuclein expression by mir-7 and mir-153.  \n",
      "('mir-153', 'Specific_miRNAs', (75, 82)) ('alpha-synuclein', 'Genes/Proteins', (35, 50))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These findings indicate that miR-107 contributes to GRN expression regulation with implications for brain disorders.   \n",
      "('miR-107', 'Specific_miRNAs', (29, 36)) ('GRN', 'Genes/Proteins', (52, 55))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case miR-107\n",
      "add special case GRN\n",
      "miR-107 17 GRN 22\n",
      "(u\"In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\", 17, 22)\n",
      "geneword GRN mirword miR-107\n",
      "Conjuncts [decreased, subjacent]\n",
      "neighbour decreased miR-107\n",
      "neighbour decreased miR-107\n",
      "Conjunction before fc subjacent\n",
      "fc add children for subjacent\n",
      "Conjuncts [subjacent, decreased]\n",
      "neighbour subjacent to\n",
      "neighbour subjacent to\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "fc add children for decreased\n",
      "Conjunctions\n",
      "subjacent [subjacent, to, depth, controlled, cortical, impact, decreased, miR-107, with, augmentation, of, neuronal, expression]\n",
      "SDP sent: In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "SDP     : [(miR-107, 'PROPN', 'dobj', False), (decreased, 'VERB', 'conj', False), (augmentation, 'NOUN', 'nmod', False), (expression, 'NOUN', 'nmod', False), (GRN, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule noun found decreased miR-107 [decreased, miR-107, with, augmentation, of, neuronal, GRN, expression]\n",
      "0 [In, brain, tissue, subjacent, to, 1.0, mm, depth, controlled, cortical, impact, ,, show] 13 0 In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "1 [surviving, hippocampal, neurons] 3 70 surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "2 [] 0\n",
      "empty compartment []\n",
      "3 [decreased, miR-107, with, augmentation, of, neuronal, GRN, expression] 8 105 decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "Conjuncts [decreased, subjacent]\n",
      "neighbour decreased miR-107\n",
      "neighbour decreased miR-107\n",
      "Conjunction before fc subjacent\n",
      "fc add children for subjacent\n",
      "Conjuncts [subjacent, decreased]\n",
      "neighbour subjacent to\n",
      "neighbour subjacent to\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "fc add children for decreased\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "decreased miR-107 with augmentation of neuronal GRN expression\n",
      "counts between 1 Counter({'NEG': 1, 'POS': 1, 'NEU': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "full sentence In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased <e1>miR-107</e1> with augmentation of neuronal <e2>GRN</e2> expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased <e1>miR-107</e1> with augmentation of neuronal <e2>GRN</e2> expression.', 'full_sentence': 'In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased <e1>miR-107</e1> with augmentation of neuronal <e2>GRN</e2> expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (115, 122), 'entity_text': 'miR-107'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (153, 156), 'entity_text': 'GRN'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of GRN mRNA are recognized by miR-107 and are highly conserved among vertebrate species. \n",
      "('miR-107', 'Specific_miRNAs', (182, 189)) ('GRN', 'Genes/Proteins', (155, 158))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-107 regulates granulin/progranulin with implications for traumatic brain injury and neurodegenerative disease.  \n",
      "('miR-107', 'Specific_miRNAs', (0, 7)) ('granulin', 'Genes/Proteins', (18, 26))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-107 regulates granulin/progranulin with implications for traumatic brain injury and neurodegenerative disease.  \n",
      "('miR-107', 'Specific_miRNAs', (0, 7)) ('progranulin', 'Genes/Proteins', (27, 38))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "GRN mRNA interacts preferentially with Argonaute 2. In vitro and in vivo studies indicate that regulation of GRN by miR-107 may be functionally important. \n",
      "('miR-107', 'Specific_miRNAs', (116, 123)) ('GRN', 'Genes/Proteins', (109, 112))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case MiR-21\n",
      "add special case Bax\n",
      "MiR-21 0 Bax/Bcl-2 14\n",
      "(u\"MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\", 0, 14)\n",
      "geneword Bax/Bcl-2 mirword MiR-21\n",
      "Conjuncts [induced, protected]\n",
      "neighbour induced apoptosis\n",
      "neighbour induced apoptosis\n",
      "Conjunction before fc protected\n",
      "fc add children for protected\n",
      "fc add children for protected\n",
      "Conjuncts [protected, induced]\n",
      "neighbour protected human\n",
      "neighbour protected human\n",
      "Conjunction before fc induced\n",
      "fc add children for induced\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity .\n",
      "neighbour activity None\n",
      "Conjunction before fc ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "Conjunctions\n",
      "protected [protected, human, cells, from, chemotherapeutic, temozolomide, induced, apoptosis]\n",
      "ratio [ratio, activity]\n",
      "SDP sent: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "SDP     : [(MiR-21, 'PROPN', 'nsubj', False), (protected, 'VERB', 'ROOT', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', False), (Bax/Bcl-2, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found induced apoptosis [induced, apoptosis]\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "0 [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 17 0 MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "1 [induced, apoptosis] 2 88 induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "Context Fail <regex.Match object; span=(8, 14), match=' ratio'> regex.Regex('( ratio){e<=0}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(120, 129, ('SIMPLE_CHEMICAL', 'Bax/Bcl-2'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide by decreasing Bax/Bcl-2 ratio and\n",
      "counts between 1 Counter({'NEG': 1})\n",
      "['MiR-21', 'protected', 'human', 'glioblastoma', 'U87MG', 'cells', 'from', 'chemotherapeutic', 'drug', 'temozolomide', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and']\n",
      "Counter({'NEG': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "full sentence <e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing <e2>Bax</e2>/Bcl-2 ratio and caspase-3 activity.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing <e2>Bax</e2>/Bcl-2 ratio and caspase-3 activity.', 'full_sentence': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing <e2>Bax</e2>/Bcl-2 ratio and caspase-3 activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 6), 'entity_text': 'MiR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (120, 123), 'entity_text': 'Bax'}}\n",
      "add special case MiR-21\n",
      "add special case Bcl-2\n",
      "MiR-21 0 Bax/Bcl-2 14\n",
      "(u\"MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\", 0, 14)\n",
      "geneword Bax/Bcl-2 mirword MiR-21\n",
      "Conjuncts [induced, protected]\n",
      "neighbour induced apoptosis\n",
      "neighbour induced apoptosis\n",
      "Conjunction before fc protected\n",
      "fc add children for protected\n",
      "fc add children for protected\n",
      "Conjuncts [protected, induced]\n",
      "neighbour protected human\n",
      "neighbour protected human\n",
      "Conjunction before fc induced\n",
      "fc add children for induced\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity .\n",
      "neighbour activity None\n",
      "Conjunction before fc ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "Conjunctions\n",
      "protected [protected, human, cells, from, chemotherapeutic, temozolomide, induced, apoptosis]\n",
      "ratio [ratio, activity]\n",
      "SDP sent: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "SDP     : [(MiR-21, 'PROPN', 'nsubj', False), (protected, 'VERB', 'ROOT', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', False), (Bax/Bcl-2, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found induced apoptosis [induced, apoptosis]\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "0 [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 17 0 MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "1 [induced, apoptosis] 2 88 induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "Context Fail <regex.Match object; span=(8, 14), match=' ratio'> regex.Regex('( ratio){e<=0}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(120, 129, ('SIMPLE_CHEMICAL', 'Bax/Bcl-2'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide by decreasing Bax/Bcl-2 ratio and\n",
      "counts between 1 Counter({'NEG': 1})\n",
      "['MiR-21', 'protected', 'human', 'glioblastoma', 'U87MG', 'cells', 'from', 'chemotherapeutic', 'drug', 'temozolomide', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and']\n",
      "Counter({'NEG': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "full sentence <e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/<e2>Bcl-2</e2> ratio and caspase-3 activity.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/<e2>Bcl-2</e2> ratio and caspase-3 activity.', 'full_sentence': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/<e2>Bcl-2</e2> ratio and caspase-3 activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 6), 'entity_text': 'MiR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (124, 129), 'entity_text': 'Bcl-2'}}\n",
      "add special case miR-21\n",
      "add special case caspase-3\n",
      "miR-21 8 caspase-3 27\n",
      "(u\"Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\", 8, 27)\n",
      "geneword caspase-3 mirword miR-21\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity ,\n",
      "neighbour activity which\n",
      "Conjunction before fc ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "Conjunctions\n",
      "ratio [ratio, activity]\n",
      "SDP sent: Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "SDP     : [(miR-21, 'PROPN', 'nsubj', False), (inhibit, 'VERB', 'ccomp', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', True), (activity, 'NOUN', 'conj', False), (caspase-3, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity, ,, which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "0 [these, results, suggested, that, over-express, miR-21, could, inhibit, TMZ-induced, apoptosis, in, U87MG, cells, ,, at, least, in, part, ,, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 26 16 these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "1 [Taken, together] 2 0 Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "2 [] 0\n",
      "empty compartment []\n",
      "3 [which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ] 16 189 which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity ,\n",
      "neighbour activity which\n",
      "Conjunction before fc ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "Recognized overlapping entities: {Interval(169, 178, ('GENE_OR_GENE_PRODUCT', 'caspase-3'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2})\n",
      "that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells , at least in part , by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "counts between 1 Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "['that', 'over-express', 'miR-21', 'could', 'inhibit', 'TMZ-induced', 'apoptosis', 'in', 'U87MG', 'cells', ',', 'at', 'least', 'in', 'part', ',', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and', 'caspase-3', 'activity']\n",
      "Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "full sentence Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.', 'full_sentence': 'Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (58, 64), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (169, 178), 'entity_text': 'caspase-3'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Hypoxia induces microRNA miR-210 in vitro and in vivo ephrin-A3 and neuronal pentraxin 1 are potentially regulated by miR-210.  \n",
      "('miR-210', 'Specific_miRNAs', (118, 125)) ('ephrin-A3', 'Genes/Proteins', (54, 63))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Hypoxia induces microRNA miR-210 in vitro and in vivo ephrin-A3 and neuronal pentraxin 1 are potentially regulated by miR-210.  \n",
      "('miR-210', 'Specific_miRNAs', (118, 125)) ('neuronal pentraxin 1', 'Genes/Proteins', (68, 88))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Furthermore, miR-210 was shown to bind to the predicted target sites of ephrin-A3 or neuronal pentraxin 1, causing repression in luciferase reporter activity. \n",
      "('miR-210', 'Specific_miRNAs', (13, 20)) ('ephrin-A3', 'Genes/Proteins', (72, 81))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Furthermore, miR-210 was shown to bind to the predicted target sites of ephrin-A3 or neuronal pentraxin 1, causing repression in luciferase reporter activity. \n",
      "('miR-210', 'Specific_miRNAs', (13, 20)) ('neuronal pentraxin 1', 'Genes/Proteins', (85, 105))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Our data strongly indicate that CDK6 is regulated by microRNA 124 in medulloblastoma and that miR 124 modulates medulloblastoma cell growth.   \n",
      "('microRNA 124', 'Specific_miRNAs', (53, 65)) ('CDK6', 'Genes/Proteins', (32, 36))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays. \n",
      "('miR 124a', 'Specific_miRNAs', (31, 39)) ('CDK6', 'Genes/Proteins', (44, 48))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons. \n",
      "('miR-128a', 'Specific_miRNAs', (73, 81)) ('Tau', 'Genes/Proteins', (122, 125))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.  \n",
      "('miR-29b', 'Specific_miRNAs', (0, 7)) ('podoplanin', 'Genes/Proteins', (30, 40))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.  \n",
      "('miR-125a', 'Specific_miRNAs', (12, 20)) ('podoplanin', 'Genes/Proteins', (30, 40))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "In this study, we focused on miR-29b and miR-125a, which were predicted to regulate PDPN, and demonstrated that these microRNAs directly target the 3' untranslated region of PDPN and inhibit invasion, apoptosis, and proliferation of glioblastomas. \n",
      "('miR-29b', 'Specific_miRNAs', (29, 36)) ('PDPN', 'Genes/Proteins', (84, 88))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "In this study, we focused on miR-29b and miR-125a, which were predicted to regulate PDPN, and demonstrated that these microRNAs directly target the 3' untranslated region of PDPN and inhibit invasion, apoptosis, and proliferation of glioblastomas. \n",
      "('miR-125a', 'Specific_miRNAs', (41, 49)) ('PDPN', 'Genes/Proteins', (84, 88))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case miR-326\n",
      "add special case Ets-1\n",
      "miR-326 4 Ets-1 10\n",
      "(u\"We also found that miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\", 4, 10)\n",
      "geneword Ets-1 mirword miR-326\n",
      "Conjunctions\n",
      "regulator [a, differentiation]\n",
      "SDP sent: We also found that miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\n",
      "SDP     : [(miR-326, 'PROPN', 'nsubj', False), (promoted, 'VERB', 'ccomp', False), (targeting, 'VERB', 'advcl', False), (Ets-1, 'PROPN', 'dobj', False)]\n",
      "deepcheck connection rule noun found promoted miR-326 [that, miR-326, promoted, T(H)-17, differentiation, by, targeting, Ets-1, ,, a, negative, regulator, of, T(H)-17, differentiation]\n",
      "deepcheck connection rule [by, targeting, Ets-1, ,, a, negative, regulator, of, T(H)-17, differentiation]\n",
      "0 [We, also, found] 3 0 We also found that miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\n",
      "1 [miR-326, promoted, T(H)-17, differentiation, by, targeting, Ets-1, ,, a, negative, regulator, of, T(H)-17, differentiation] 14 19 miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\n",
      "Context Fail <regex.Match object; span=(7, 16), match=' negative', fuzzy_counts=(1, 0, 0)> regex.Regex('(-negative){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(73, 78, ('GENE_OR_GENE_PRODUCT', 'Ets-1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence We also found that miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\n",
      "full sentence We also found that <e1>miR-326</e1> promoted T(H)-17 differentiation by targeting <e2>Ets-1</e2>, a negative regulator of T(H)-17 differentiation.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'We also found that <e1>miR-326</e1> promoted T(H)-17 differentiation by targeting <e2>Ets-1</e2>, a negative regulator of T(H)-17 differentiation.', 'full_sentence': 'We also found that <e1>miR-326</e1> promoted T(H)-17 differentiation by targeting <e2>Ets-1</e2>, a negative regulator of T(H)-17 differentiation.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (19, 26), 'entity_text': 'miR-326'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (73, 78), 'entity_text': 'Ets-1'}}\n",
      "add special case microRNA-433\n",
      "add special case FGF20\n",
      "microRNA-433 17 FGF20 22\n",
      "(u\"We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\", 17, 22)\n",
      "geneword FGF20 mirword microRNA-433\n",
      "Conjunctions\n",
      "SDP sent: We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "SDP     : [(microRNA-433, 'PROPN', 'nmod', False), (site, 'NOUN', 'dobj', False), (disrupts, 'VERB', 'ccomp', False), (translation, 'NOUN', 'nmod', False), (FGF20, 'PROPN', 'nmod', False)]\n",
      "0 [We, show, in, several, functional, assays] 6 0 We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "1 [the, risk, allele, for, rs12720208, disrupts, a, binding, site, for, microRNA-433, ,, increasing, translation, of, FGF20, in, vitro, and, in, vivo] 21 42 the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "Recognized overlapping entities: {Interval(141, 146, ('GENE_OR_GENE_PRODUCT', 'FGF20'))}\n",
      "passive? False\n",
      "Counter()\n",
      "site for microRNA-433 , increasing translation of FGF20 in vitro\n",
      "counts between 1 Counter({'POS': 1, 'NEU': 1})\n",
      "['site', 'for', 'microRNA-433', ',', 'increasing', 'translation', 'of', 'FGF20', 'in', 'vitro']\n",
      "Counter({'POS': 1, 'NEU': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "full sentence We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.', 'full_sentence': 'We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (101, 113), 'entity_text': 'microRNA-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (141, 146), 'entity_text': 'FGF20'}}\n",
      "Total:      232\n",
      "Correct:    218 0.9396551724137931\n",
      "Incorrect:  14 0.0603448275862069\n",
      "classes Counter()\n",
      "precision 0.9420289855072463\n",
      "recall 0.9558823529411765\n",
      "specificity 0.9166666666666666\n",
      "f1 0.9489051094890512\n",
      "Correct classified:      128 0.5517241379310345 0.927536231884058\n",
      "Incorrect classified:    10 0.04310344827586207 0.07246376811594203\n",
      "Random classified:  52 0.22413793103448276 0.37681159420289856\n",
      "Counter({'counts between': 7, 'return': 3})\n",
      "Correct interaction&classified:      121 0.521551724137931 0.8768115942028986\n",
      "Classified by:  Counter({'return': 57, 'counts between': 33, 'compartment mir gene': 12, 'counts opp': 9, 'between gene mir': 9, 'counts between equal': 8, 'between mir gene': 4, 'compartment mir gene neg corr': 4, 'counts between alternating': 1, 'compartment gene mir': 1})\n",
      "[[[132   2]\n",
      "  [  0   4]]\n",
      "\n",
      " [[137   0]\n",
      "  [  0   1]]\n",
      "\n",
      " [[ 83   3]\n",
      "  [  5  47]]\n",
      "\n",
      " [[ 55   2]\n",
      "  [  5  76]]\n",
      "\n",
      " [[135   3]\n",
      "  [  0   0]]]\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      " GENE_MIR_NEU       0.67      1.00      0.80         4\n",
      "  GENE_MIR_UP       1.00      1.00      1.00         1\n",
      "MIR_GENE_DOWN       0.94      0.90      0.92        52\n",
      " MIR_GENE_NEU       0.97      0.94      0.96        81\n",
      "  MIR_GENE_UP       0.00      0.00      0.00         0\n",
      "\n",
      "     accuracy                           0.93       138\n",
      "    macro avg       0.72      0.77      0.74       138\n",
      " weighted avg       0.95      0.93      0.94       138\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "test_results = runCheck(-1, \"TEST\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "analyse(\n",
    "#(\"We also found that miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\", 4, 10)\n",
    "    #(\"In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\", 4, 10),\n",
    "    #(\"miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 11, 18),\n",
    "    #(\"Downregulation of Pdcd4 by mir-21 facilitates glioblastoma proliferation in vivo.\", 4,2),\n",
    "    (\"We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \", 3, 15),\n",
    "    100\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "\n",
    "with open(\"scai_test_f1.pickle\", \"wb\") as fout:\n",
    "    pickle.dump(test_results, fout)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "runCheck(-1, \"TRAIN\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def getTokenDist(tk1, tk2):\n",
    "    return abs(tk1.i-tk2.i)\n",
    "\n",
    "def countStems(tokens, geneword, mirword, keepNEU=False):\n",
    "    stemTypes2Count = Counter()\n",
    "    mirgene2stem = defaultdict(lambda: Counter())\n",
    "\n",
    "    for tidx,x in enumerate(tokens):\n",
    "        tknStem, matchStem, wasFound = getstem(x)\n",
    "\n",
    "        if wasFound and (tknStem != \"NEU\" or keepNEU):\n",
    "            \n",
    "            if tidx >= len(tokens)-2 and str(tokens[tidx-1]) == \"to\":\n",
    "                continue\n",
    "            \n",
    "            geneDist = getTokenDist(x, geneword)\n",
    "            mirDist = getTokenDist(x, mirword)\n",
    "                        \n",
    "            if geneDist < mirDist and geneDist < 3:\n",
    "                mirgene2stem[\"gene\"][tknStem] += 1\n",
    "            elif mirDist < geneDist and mirDist < 3:\n",
    "                mirgene2stem[\"mir\"][tknStem] += 1\n",
    "\n",
    "        if wasFound and (tknStem != \"NEU\" or keepNEU):\n",
    "            stemTypes2Count[tknStem] += 1\n",
    "\n",
    "    if len(stemTypes2Count) == 0:\n",
    "        if \"regulated\" in str(geneword):\n",
    "            stemTypes2Count[\"NEU\"] += 1\n",
    "\n",
    "    return stemTypes2Count,mirgene2stem\n",
    "\n",
    "def getRelsForTokens(tokens, desc,verbose):\n",
    "    rels = []\n",
    "    verbSet = set([\"VERB\"])\n",
    "    for x in tokens:\n",
    "        \n",
    "        tokenStemClass, s, _ = getstem(x)\n",
    "        rels.append((s, None, tokenStemClass, x.pos_, x))\n",
    "        if verbose:\n",
    "            print(desc, x, s, tokenStemClass, x.pos_, x)\n",
    "                    \n",
    "    hasVerb = len(list(set([x[3] for x in rels]) & verbSet)) > 0\n",
    "    \n",
    "    if hasVerb:\n",
    "        rels = [x for x in rels if x[3] in verbSet]\n",
    "    \n",
    "    return rels\n",
    "\n",
    "def evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg):\n",
    "\n",
    "    invcor = False\n",
    "    regdir = None\n",
    "\n",
    "    #if dirReg == 0:\n",
    "    if (mirPos > 0 and geneNeg > 0) or (genePos > 0 and mirNeg > 0):\n",
    "        return \"DOWN\", True\n",
    "\n",
    "    #else:\n",
    "    if ((mirPos > 0 and genePos > 0) and(mirNeg == 0 and geneNeg == 0)) or ((mirNeg > 0 and geneNeg >0) and(mirPos == 0 and genePos == 0)):\n",
    "        return \"UP\", False\n",
    "\n",
    "    return None, None\n",
    "\n",
    "def scoreWord(tkn):\n",
    "    \n",
    "    jtkstem, s, _ = getstem(tkn)\n",
    "                        \n",
    "    if jtkstem == None:\n",
    "\n",
    "        if jtk.startswith((\"loss\", \"decreas\", \"low\")):\n",
    "            return 0,1\n",
    "\n",
    "        elif jtk.startswith((\"increas\", \"high\", \"posit\")):\n",
    "            return 1,0\n",
    "\n",
    "    else:\n",
    "        if jtkstem == \"POS\":\n",
    "            return 1,0\n",
    "        elif jtkstem == \"NEG\":\n",
    "            return 0,1\n",
    "        \n",
    "    return 0,0\n",
    "\n",
    "\n",
    "def getTextBetween(tks, lword, rword):\n",
    "    tokenBetween = [x for x in tks if lword.i-3 < x.i < rword.i+3]\n",
    "    \n",
    "    lit = lword\n",
    "    while lit.pos_ in [\"nmod\", \"conj\"]:\n",
    "        if not lit in tokenBetween:\n",
    "            tokenBetween.append(lit)\n",
    "            \n",
    "        lit = lit.head\n",
    "        \n",
    "    rit = rword\n",
    "    while rit.pos_ in [\"nmod\", \"conj\"]:\n",
    "        if not rit in tokenBetween:\n",
    "            tokenBetween.append(rit)\n",
    "            \n",
    "        rit = rit.head\n",
    "\n",
    "\n",
    "    tokenBetween = list(sorted(tokenBetween, key=lambda x: x.i))\n",
    "    tokenBetweenStrict = list(sorted([x for x in tks if lword.i < x.i < rword.i], key=lambda x: x.i))\n",
    "    textBetween = ' '.join(\" \".join([str(x) for x in tokenBetween]).split())\n",
    "    textAfter = ' '.join(\" \".join([str(x) for x in [y for y in rword.doc if rword.i <= y.i < rword.i+5]]).split())\n",
    "    textBefore = ' '.join(\" \".join([str(x) for x in [y for y in rword.doc if lword.i-5 <= y.i < lword.i]]).split())\n",
    "\n",
    "    return tokenBetween, tokenBetweenStrict, textBetween, textBefore, textAfter\n",
    "\n",
    "dir2opp = {\"MIR_GENE\": \"GENE_MIR\", \"GENE_MIR\": \"MIR_GENE\"}\n",
    "idir2opp = {\"POS\": \"NEG\", \"NEG\": \"POS\"}\n",
    "rdir2opp = {\"DOWN\": \"UP\", \"UP\": \"DOWN\"}\n",
    "\n",
    "def testRelClassifier2(doc, mirword, geneword, relchecker, verbose):\n",
    "    isPassive = relchecker.checkPassive(doc, mirword, geneword, verbose=False)\n",
    "    compartment = relchecker.getCompartment(doc, mirword, geneword, verbose=False)\n",
    "    \n",
    "    if verbose:\n",
    "        print(\"passive?\", isPassive)\n",
    "        \n",
    "        \n",
    "    sdpRes = relchecker._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "        \n",
    "    stemTypes2Count = Counter()\n",
    "    \n",
    "    for x in sdpRes:\n",
    "        tknStem, matchStem, _ = getstem(x[0])\n",
    "        \n",
    "        if tknStem != \"NEU\":\n",
    "            stemTypes2Count[tknStem] += 1\n",
    "            \n",
    "    print(stemTypes2Count)\n",
    "        \n",
    "    if len(stemTypes2Count) > 0:\n",
    "        regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "        \n",
    "        if regStem == \"NEG\":\n",
    "            regStem = \"DOWN\"\n",
    "        elif regStem == \"POS\":\n",
    "            regStem = \"UP\"\n",
    "            \n",
    "        if stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"]:\n",
    "            regStem = \"DOWN\"\n",
    "            \n",
    "        idir = \"MIR_GENE\"\n",
    "        if isPassive and mirword.i < geneword.i:\n",
    "            idir = \"GENE_MIR\"\n",
    "            \n",
    "        #return {\"regulation_dir\": regStem, \"interaction_dir\": idir, \"passive\": isPassive, \"reg_detect\": \"sdp counts\"}\n",
    "        \n",
    "    targetWords = [\"targets of\", \"target of\", \"regulator of\", \"targeting\", \"direct regulator\"]\n",
    "        \n",
    "    if compartment != None:\n",
    "        tokenBetween = compartment\n",
    "        textBetween = ' '.join(\" \".join([str(x) for x in tokenBetween]).split())\n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"compartment gene mir pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"compartment gene mir neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment gene mir\"}\n",
    "            \n",
    "        elif mirword.i < geneword.i:\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment mir gene pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment mir gene neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment mir gene\"}\n",
    "        \n",
    "\n",
    "\n",
    "    if True:\n",
    "        \n",
    "        tokSrc = compartment if compartment != None else doc\n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "\n",
    "            tokenBetween, tokenBetweenStrict, textBetween, textBefore, textAfter = getTextBetween(tokSrc, geneword, mirword)\n",
    "\n",
    "            print(textBetween)\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between gene mir pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between gene mir neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"negative regulation of\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between gene mir reg corr\"}\n",
    "      \n",
    "            if any([x in textBefore for x in [\"binding site\", \"binding sites\", \"binding to\", \"binding directly to\"]]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between gene mir sites\"}\n",
    "            \n",
    "            \n",
    "    \n",
    "            if any([x in textBetween for x in targetWords]) or any([x in textBetween for x in [\"target\", \"targeted by\", \"binding efficiency\", \"binding site for\", \"recognizing\"]]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between gene mir\"}\n",
    "            \n",
    "\n",
    "            #stemTypes2Count = Counter()\n",
    "            #for x in tokenBetween:\n",
    "            #    tknStem, matchStem = getstem(x)\n",
    "            #    if tknStem != \"NEU\":\n",
    "            #        stemTypes2Count[tknStem] += 1\n",
    "                    \n",
    "            if False and len(stemTypes2Count) > 0:\n",
    "                regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                if regStem == \"NEG\":\n",
    "                    regStem = \"DOWN\"\n",
    "                elif regStem == \"POS\":\n",
    "                    regStem = \"UP\"\n",
    "                    \n",
    "                if stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"]:\n",
    "                    regStem = \"DOWN\"\n",
    "\n",
    "                idir = \"MIR_GENE\"\n",
    "                if isPassive and geneword.i < mirword.i:\n",
    "                    idir = \"GENE_MIR\"\n",
    "\n",
    "                return {\"regulation_dir\": regStem, \"interaction_dir\": idir, \"passive\": isPassive, \"reg_detect\": \"counts between gene mir\"}\n",
    "            \n",
    "        else:\n",
    "            #mirword.i < geneword.i\n",
    "            tokenBetween, tokenBetweenStrict, textBetween, textBefore, textAfter = getTextBetween(tokSrc, mirword, geneword)\n",
    "                        \n",
    "            print(textBetween)\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]) and stemTypes2Count.get(\"NEG\", 0) == 0:\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"negative regulation of\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene reg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"directly inhibits\", \"directly represses\", \"translational inhibition\" ]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords + [\"directly regulated\", \"its target gene\", \"binding site\"]]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"negatively regulated by\", \"negative regulation of\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between mir gene nrb\"}\n",
    "            \n",
    "            \n",
    "        if len(tokenBetween) > 0:\n",
    "                       \n",
    "            stemTypes2Count,mirgene2stem = countStems(tokenBetween, geneword, mirword)\n",
    "                    \n",
    "            if len(stemTypes2Count) > 0:\n",
    "                regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                \n",
    "                if len(stemTypes2Count) >= 2 and len(mirgene2stem[\"gene\"]) >0 and len(mirgene2stem[\"mir\"])>0 and mirgene2stem[\"gene\"].most_common(1)[0][0] == idir2opp[mirgene2stem[\"mir\"].most_common(1)[0][0]]:\n",
    "                    print(mirgene2stem)\n",
    "                    if geneword.i < mirword.i:\n",
    "                        if mirgene2stem[\"gene\"].most_common(1)[0][0] == \"POS\":\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                        else:\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                    else:\n",
    "                                                \n",
    "                        if mirgene2stem[\"mir\"].most_common(1)[0][0] == \"NEG\":\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                                                        \n",
    "                        else:\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                            \n",
    "                        if \"after\" in textBetween:\n",
    "                            regDir = dir2opp[regDir]\n",
    "                            \n",
    "                    if isPassive:\n",
    "                        regDir = dir2opp[regDir]\n",
    "                        regStem = rdir2opp[regStem]\n",
    "                        \n",
    "                    return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts opp\"}\n",
    "                    \n",
    "                    \n",
    "                else:\n",
    "                    stemTypes2Count,mirgene2stem = countStems(tokenBetween, geneword, mirword, keepNEU=True)\n",
    "                    print(\"counts between 1\", stemTypes2Count)\n",
    "                    regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                    \n",
    "                    if stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"] == 1 and stemTypes2Count[\"NEU\"] <= 3:\n",
    "                        regDir = \"MIR_GENE\"\n",
    "                        regStem = \"DOWN\"\n",
    "                        \n",
    "                        if any(x in textBetween for x in [\"after\"]):\n",
    "                            regDir = dir2opp[regDir]\n",
    "                        \n",
    "                        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts between equal\"}\n",
    "                    \n",
    "                    if stemTypes2Count[\"POS\"] == 2 and stemTypes2Count[\"NEG\"] == 1 and stemTypes2Count[\"NEU\"] <= 2:\n",
    "                        regDir = \"MIR_GENE\"\n",
    "                        regStem = \"DOWN\"\n",
    "                        \n",
    "                        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts between alternating\"}\n",
    "                    \n",
    "                    if regStem == \"NEU\":\n",
    "                        if stemTypes2Count[\"POS\"] > 0 and stemTypes2Count[\"POS\"] > stemTypes2Count[\"NEG\"]:\n",
    "                            regStem = \"POS\"\n",
    "                            \n",
    "                        elif stemTypes2Count[\"NEG\"] > 0 and stemTypes2Count[\"NEG\"] > stemTypes2Count[\"POS\"]:\n",
    "                            regStem = \"NEG\"\n",
    "                        \n",
    "                    \n",
    "                    #regStem = \"NEU\"\n",
    "\n",
    "                    if mirword.i < geneword.i:\n",
    "                        regDir = \"MIR_GENE\"\n",
    "                    else:\n",
    "                        regDir = \"GENE_MIR\"\n",
    "                        \n",
    "                    if mirword.i < geneword.i and \"-stressed\" in textBetween:\n",
    "                        regDir = \"GENE_MIR\"\n",
    "                        \n",
    "                    if mirword.i < geneword.i and str(doc[geneword.i-1]) in [\"by\", \"with\"]:\n",
    "                        regDir = \"GENE_MIR\"\n",
    "\n",
    "                        stemTypes2Count,mirgene2stem = countStems(tokenBetweenStrict, geneword, mirword)\n",
    "\n",
    "                        if len(stemTypes2Count) > 0:\n",
    "                            regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                        #else:\n",
    "                        #    regStem = \"NEU\"\n",
    "                        print(\"counts between 2\", stemTypes2Count, regStem)\n",
    "                            \n",
    "                    elif geneword.i < mirword.i and any([str(doc[mirword.i-j]) in [\"by\", \"with\"] for j in [1,2]]):\n",
    "                        regDir = \"MIR_GENE\"\n",
    "\n",
    "                        stemTypes2Count,mirgene2stem = countStems(tokenBetweenStrict, geneword, mirword)\n",
    "                        if len(stemTypes2Count) > 0:\n",
    "                            regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                        #else:\n",
    "                        #    regStem = \"NEU\"\n",
    "                        print(\"counts between 2\", stemTypes2Count, regStem)\n",
    "                            \n",
    "                    elif isPassive and geneword.i < mirword.i:\n",
    "                        regDir = dir2opp[regDir]\n",
    "\n",
    "                    if stemTypes2Count[\"POS\"] > 0 and stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"]:\n",
    "                        regStem = \"NEU\"                       \n",
    "                        \n",
    "                    \n",
    "                        #regStem = idir2opp[regStem]\n",
    "                    \n",
    "                    if geneword.i < mirword.i:\n",
    "                        if str(geneword).endswith(\"-regulating\"):\n",
    "                            regStem = \"NEU\"\n",
    "                            \n",
    "                    if mirword.i < geneword.i and \"after the addition\" in textBetween:\n",
    "                        regStem = \"DOWN\"\n",
    "                        regDir = \"GENE_MIR\"\n",
    "                            \n",
    "                        \n",
    "                    if regStem == \"NEG\":\n",
    "                        regStem = \"DOWN\"\n",
    "                    elif regStem == \"POS\":\n",
    "                        regStem = \"UP\"\n",
    "\n",
    "                        \n",
    "                    \n",
    "                    return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts between\"}\n",
    "    \n",
    "        \n",
    "    regDir = \"MIR_GENE\"\n",
    "    regStem = \"NEU\"\n",
    "    \n",
    "    regeluateWords = [\"mediated\",]\n",
    "    \n",
    "    if mirword.i < geneword.i and \"mediated\" in str(mirword):\n",
    "        regStem = \"DOWN\"\n",
    "        \n",
    "    if geneword.i < mirword.i and (\"mediated\" in str(mirword) or any([str(doc[mirword.i-j]) in [\"mediated\",] for j in [1]])):\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "        \n",
    "        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"return mediated\"}\n",
    "        \n",
    "    if mirword.i<geneword.i and any([str(doc[geneword.i-j]) in [\"by\", \"with\"] for j in [1,2,3]]):\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "        \n",
    "        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"return m g by\"}\n",
    "        \n",
    "    if geneword.i < mirword.i:\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"GENE_MIR\"\n",
    "    \n",
    "    if geneword.i < mirword.i and any([str(doc[mirword.i-j]) in [\"by\", \"with\"] for j in [1,2,3]]):\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "        \n",
    "    if mirword.i < geneword.i and str(doc[mirword.i+1]) == \"target\":\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "    \n",
    "    #if mirword.i < geneword.i:\n",
    "    #    regDir = \"MIR_GENE\"\n",
    "    #else:\n",
    "    #    regDir = \"GENE_MIR\"\n",
    "    \n",
    "    #if isPassive:\n",
    "    #    if mirword.i < geneword.i:\n",
    "    #        regDir = \"GENE_MIR\"\n",
    "    #    else:\n",
    "    #        regDir = \"MIR_GENE\"\n",
    "        \n",
    "    return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"return\"}\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getstem(tkn):\n",
    "    tknstr = str(tkn).lower()\n",
    "    for sidx, s in enumerate(stem2class):\n",
    "        if tknstr.startswith(s):\n",
    "            \n",
    "            if tknstr in [\"enhancer\",\"inhibitor\"]:\n",
    "                continue\n",
    "\n",
    "            tokenStemClass = stem2class[s]\n",
    "            \n",
    "            if tokenStemClass == \"NEU\":\n",
    "                for i in range(1,2):\n",
    "\n",
    "                    if tkn.i-i<0:\n",
    "                        continue\n",
    "\n",
    "                    #tknPre = str(tkn.doc[tkn.i-i])\n",
    "                    #if tknPre.startswith((\"high\", \"elevat\", \"increas\")):\n",
    "                    #    tokenStemClass = \"POS\"\n",
    "\n",
    "                    #if tknPre.startswith((\"low\", \"decreas\")):\n",
    "                    #    tokenStemClass = \"NEG\"\n",
    "                                                \n",
    "            return tokenStemClass, s, True\n",
    "        \n",
    "        elif tknstr.endswith((\"-stressed\")):\n",
    "            return \"POS\", s, True\n",
    "        \n",
    "    return \"NEU\", str(tkn), False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "allStems, allRels, stem2class = loadStems()\n",
    "\n",
    "print(stem2class[\"underexpress\"])\n",
    "print(\"underexpression\".startswith(\"underexpress\"))\n",
    "\n",
    "doct = nlp(\"expressed , miR-34a , was studied further because sequence analysis suggested a likely interaction with the 3'-untranslated region of bcl2 mRNA\")\n",
    "\n",
    "toks = [x for x in doct]\n",
    "\n",
    "for tok in toks:\n",
    "    print(tok, getstem(tok))\n",
    "\n",
    "countStems(toks, toks[0], toks[-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "runCheck(-1, \"TRAIN\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "runCheck(-1, \"TEST\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "def testRelClassifier(doc, mirword, geneword, relchecker, verbose):\n",
    "    \n",
    "    if verbose:\n",
    "        print()\n",
    "        print()\n",
    "        print()\n",
    "        print(\"Rel Classifier\")\n",
    "    \n",
    "        print(doc.text_with_ws)\n",
    "    \n",
    "    foundStems = set()\n",
    "    sdpRes = relchecker._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "    if verbose:\n",
    "        for x in sdpRes:\n",
    "            print(x)\n",
    "        \n",
    "        \n",
    "    sdpTokens = [x[0] for x in sdpRes]\n",
    "    #print(sdpRes)\n",
    "    \n",
    "    sdpVerbRels = set()\n",
    "\n",
    "    for sidx, x in enumerate(sdpRes):\n",
    "        if x[1] == \"VERB\":\n",
    "            for s in allStems:\n",
    "                if s in str(x[0]):\n",
    "                    sdpVerbRels.add((s, sidx, stem2class[s]))\n",
    "                    if verbose:\n",
    "                        print(\"sdpVerbRels\", x[0], s, sidx, stem2class[s])\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "            \n",
    "    docRels = set()\n",
    "    for tkn in doc:\n",
    "\n",
    "        if tkn in sdpTokens:\n",
    "            continue\n",
    "\n",
    "        for s in allStems:\n",
    "            if str(tkn).lower().startswith(s):\n",
    "                docRels.add((s, sidx, stem2class[s]))\n",
    "                if verbose:    \n",
    "                    print(\"docRels\", tkn, s, sidx, stem2class[s])\n",
    "            \n",
    "            \n",
    "    mirAncests = [x for x in mirword.ancestors]\n",
    "    geneAncests = [x for x in geneword.ancestors if not x in mirAncests]\n",
    "    \n",
    "    if len(mirAncests) > 1:\n",
    "        useMirAncests = mirAncests[0:1]\n",
    "    else:\n",
    "        useMirAncests = mirAncests\n",
    "        \n",
    "    if len(geneAncests) > 1:\n",
    "        useGeneAncests = geneAncests[0:1]\n",
    "    else:\n",
    "        useGeneAncests = geneAncests\n",
    "        \n",
    "    useGeneAncests = [x for x in useGeneAncests if not x in useMirAncests]\n",
    "    \n",
    "    if verbose:\n",
    "        print(\"mirc\", mirAncests)\n",
    "        print(\"genec\", geneAncests)\n",
    "    \n",
    "    invCor = False\n",
    "    regdir = \"NEU\"\n",
    "    intdir = \"MIR_GENE\"\n",
    "    isPassive = relchecker.checkPassive(doc, mirword, geneword, verbose=False)\n",
    "    if verbose:\n",
    "        print(\"passive?\", isPassive)\n",
    "    \n",
    "    ancestHandled = False\n",
    "    #if len(mirAncests) > 0 and len(geneAncests) > 0:\n",
    "    if True:\n",
    "        mirRels = []\n",
    "        geneRels = []\n",
    "        \n",
    "        mirRels = getRelsForTokens(useMirAncests, \"ancestRels mir\", verbose)     \n",
    "        neuMirs = [x for x in mirRels if x[2] in [\"NEU\"]]\n",
    "        #if len(neuMirs) > 0 or len(mirRels) == 0:\n",
    "        #    mirRels = getRelsForTokens(mirAncests, \"ancestRels all mir\", verbose)\n",
    "            \n",
    "        geneRels = getRelsForTokens(useGeneAncests, \"ancestRels gene\", verbose)       \n",
    "        neuGenes = [x for x in geneRels if x[2] in [\"NEU\"]]\n",
    "                \n",
    "        if False and ((len(neuGenes) == len(geneRels)) or len(geneRels) == 0):\n",
    "            geneRels = getRelsForTokens([x for x in geneAncests if not x in useMirAncests], \"ancestRels all gene\", verbose)\n",
    "            neuGenes = [x for x in geneRels if x[2] in [\"NEU\"]]\n",
    "            \n",
    "            if (len(neuGenes) == len(geneRels)) or len(geneRels) == 0:\n",
    "                \n",
    "                if mirword.i < geneword.i:\n",
    "                    tks = [x for x in doc if geneword.i -8 < x.i < geneword.i if not x in useMirAncests]\n",
    "                else:\n",
    "                    tks = [x for x in doc if geneword.i < x.i < geneword.i+5 if not x in useMirAncests]\n",
    "                \n",
    "                geneRels = getRelsForTokens(tks, \"ancestRels tks gene\", verbose)\n",
    "                \n",
    "                                        \n",
    "        posRegs = [x for x in mirRels+geneRels if x[2] in [\"POS\"]]\n",
    "        negRegs = [x for x in mirRels+geneRels if x[2] in [\"NEG\"]]\n",
    "        \n",
    "        mirPos = len([x for x in mirRels if x[2] in [\"POS\"]])\n",
    "        mirNeu = [x for x in mirRels if x[2] in [\"NEU\"]]\n",
    "        mirNeg = len([x for x in mirRels if x[2] in [\"NEG\"]])\n",
    "        \n",
    "        genePos = len([x for x in geneRels if x[2] in [\"POS\"]])\n",
    "        geneNeu = [x for x in geneRels if x[2] in [\"NEU\"]]\n",
    "        geneNeg = len([x for x in geneRels if x[2] in [\"NEG\"]])\n",
    "    \n",
    "        dirReg = (mirPos - geneNeg) + (genePos-mirNeg)\n",
    "        \n",
    "        invCor = False\n",
    "        \n",
    "        if verbose:\n",
    "            print(mirword, geneword)\n",
    "            print(\"ancestRel counts\", mirPos, mirNeg, genePos, geneNeg)\n",
    "                          \n",
    "            \n",
    "        nregdir, ninvcor = evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg)\n",
    "        if verbose:\n",
    "            print(\"evalcount\")\n",
    "            print(nregdir, ninvcor)\n",
    "        \n",
    "        if nregdir != None:\n",
    "            regdir = nregdir\n",
    "            invcor = ninvcor\n",
    "            \n",
    "            ancestHandled = True\n",
    "            \n",
    "            if verbose:\n",
    "                print(\"accepted by evalcount\")\n",
    "            \n",
    "        else:\n",
    "            lookBehind = 8\n",
    "\n",
    "                                \n",
    "            if len(mirNeu) > 0 or len(geneNeu) > 0:\n",
    "                if verbose:\n",
    "                    print(\"ancestRel neu part\")\n",
    "\n",
    "                if mirword.i < geneword.i:\n",
    "                    for x in mirNeu:\n",
    "                        if x[0].startswith((\"correlat\", \"affect\")):\n",
    "\n",
    "                            for j in range(1,lookBehind):\n",
    "                                if str(doc[x[4].i-j]).startswith((\"posit\", \"high\")):\n",
    "                                    mirPos += 1\n",
    "                                    \n",
    "                                elif str(doc[x[4].i-j]).startswith((\"negat\", \"low\")):\n",
    "                                    mirNeg += 1\n",
    "                else:\n",
    "                    for x in geneNeu:\n",
    "                        if x[0].startswith((\"correlat\")):\n",
    "                            \n",
    "                            for j in range(1,lookBehind):\n",
    "                                if str(doc[x[4].i-j]).startswith((\"posit\", \"high\")):\n",
    "                                    genePos += 1\n",
    "                                elif str(doc[x[4].i-j]).startswith((\"negat\", \"low\")):\n",
    "                                    geneNeg += 1    \n",
    "            \n",
    "                                \n",
    "            if not ancestHandled:    \n",
    "                \n",
    "                                \n",
    "                    \n",
    "                if verbose:\n",
    "                    print(\"ancestRel counts stem check\", mirPos, mirNeg, genePos, geneNeg)\n",
    "                    \n",
    "                nregdir, ninvcor = evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg)\n",
    "                if nregdir != None:\n",
    "                    regdir = nregdir\n",
    "                    invcor = ninvcor\n",
    "                    ancestHandled = True\n",
    "                    \n",
    "                    if verbose:\n",
    "                        print(\"accept after before stem check\")\n",
    "                    \n",
    "                if not ancestHandled:\n",
    "                    for j in range(1,lookBehind):\n",
    "                        \n",
    "                        if mirword.i-j < 0:\n",
    "                            continue\n",
    "                        \n",
    "                        jtk = doc[mirword.i-j]\n",
    "                        if verbose:\n",
    "                            print(\"word check mir\", j, jtk)\n",
    "                        pos, neg = scoreWord(jtk)\n",
    "                        \n",
    "                        mirPos += pos\n",
    "                        mirNeg += neg\n",
    "                            \n",
    "                    for j in range(1,lookBehind):\n",
    "                        \n",
    "                        if geneword.i-j < 0:\n",
    "                            continue\n",
    "                        \n",
    "                        jtk = doc[geneword.i-j]\n",
    "                        if verbose:\n",
    "                            print(\"word check gene\", j, jtk)\n",
    "                            \n",
    "                        pos, neg = scoreWord(jtk)\n",
    "                        \n",
    "                        genePos += pos\n",
    "                        geneNeg += neg\n",
    "                            \n",
    "                            \n",
    "                if verbose:\n",
    "                    print(\"ancestRel counts word check\", mirPos, mirNeg, genePos, geneNeg)\n",
    "                \n",
    "                nregdir, ninvcor = evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg)\n",
    "                if nregdir != None:\n",
    "                    regdir = nregdir\n",
    "                    invcor = ninvcor\n",
    "                    ancestHandled = True\n",
    "                    \n",
    "                    if verbose:\n",
    "                        print(\"accept after before word check\")\n",
    "                    \n",
    "           \n",
    "\n",
    "                \n",
    "                \n",
    "        if verbose:\n",
    "            print(\"ancestRels\", dirReg, invCor)\n",
    "            \n",
    "    if ancestHandled:\n",
    "        \n",
    "        if isPassive:\n",
    "            intdir = \"GENE_MIR\"\n",
    "            \n",
    "    \n",
    "    if not ancestHandled:\n",
    "        \n",
    "        if verbose:\n",
    "            print(\"ancestHandled part\")\n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "            lword = geneword\n",
    "            rword = mirword\n",
    "            useTks = [x for x in doc if geneword.i < x.i < mirword.i]\n",
    "        else:\n",
    "            lword = mirword\n",
    "            rword = geneword\n",
    "            useTks = [x for x in doc if mirword.i < x.i < geneword.i]\n",
    "            \n",
    "        if verbose:\n",
    "            print(useTks)\n",
    "        \n",
    "        betRels = getRelsForTokens(useTks, \"ancestRels useTks\", verbose)\n",
    "        \n",
    "        if verbose:\n",
    "            print(betRels)\n",
    "        \n",
    "        betPos = len([x for x in betRels if x[2] in [\"POS\"]])\n",
    "        betNeg = len([x for x in betRels if x[2] in [\"NEG\"]])\n",
    "        \n",
    "        if betPos + betNeg == 1:\n",
    "            ancestHandled = True\n",
    "            \n",
    "            if verbose:\n",
    "                print(\"ancest bet handled\", mirword.i, geneword.i, isPassive)\n",
    "            \n",
    "            if geneword.i < mirword.i:\n",
    "                \n",
    "                if isPassive:\n",
    "                    intdir = \"MIR_GENE\"\n",
    "                    regdir = \"DOWN\" if betNeg > 0 else \"UP\"\n",
    "                    \n",
    "                else:\n",
    "                    intdir = \"GENE_MIR\"\n",
    "                    regdir = \"UP\" if betPos > 0 else \"DOWN\"\n",
    "                    \n",
    "            else:\n",
    "                \n",
    "                if isPassive:\n",
    "                    intdir = \"GENE_MIR\"\n",
    "                    regdir = \"UP\" if betPos > 0 else \"DOWN\"\n",
    "                    \n",
    "                else:\n",
    "                    intdir = \"MIR_GENE\"\n",
    "                    regdir = \"UP\" if betPos > 0 else \"DOWN\"\n",
    "                    \n",
    "        else:\n",
    "            \n",
    "            isMediated = False\n",
    "            \n",
    "            for x in [lword] + useTks[0:2]:\n",
    "                for y in [\"mediated\", \"-mediated\"]:\n",
    "                    \n",
    "                    if y in str(x):\n",
    "                        isMediated = True\n",
    "                        break\n",
    "                if isMediated:\n",
    "                    break\n",
    "                                    \n",
    "            if isMediated:\n",
    "                \n",
    "                ancestHandled = True\n",
    "                \n",
    "                if verbose:\n",
    "                    print(\"ancest mediated handled\")\n",
    "                \n",
    "                if geneword.i < mirword.i:\n",
    "                    intdir = \"GENE_MIR\"\n",
    "                else:\n",
    "                    intdir = \"MIR_GENE\"\n",
    "                    \n",
    "                if betPos > betNeg:\n",
    "                    regdir = \"UP\"\n",
    "                elif betNeg > betPos:\n",
    "                    regdir = \"DOWN\"\n",
    "                        \n",
    "            \n",
    "            \n",
    "            \n",
    "\n",
    "    #classCounter = Counter()\n",
    "    #if len(classCounter) > 0:\n",
    "    #    regdir = classCounter.most_common(1)[0][0]\n",
    "    #else:\n",
    "    #    regdir = \"NEU\"\n",
    "        \n",
    "    #if invCor:\n",
    "    #    regdir = \"DOWN\"       \n",
    "              \n",
    "    \n",
    "    if not ancestHandled:\n",
    "        \n",
    "        if verbose:\n",
    "            print(\"in final resolution\")\n",
    "        \n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "        \n",
    "            textBetween = \" \".join([str(x) for x in doc if geneword.i < x.i < mirword.i])\n",
    "            textBetween = ' '.join(textBetween.split())\n",
    "            \n",
    "            if verbose:\n",
    "                print(textBetween)\n",
    "            \n",
    "\n",
    "            if len(textBetween) < 10 and any([x in textBetween for x in [\"by\"]]):\n",
    "                sdpRels = set()\n",
    "                for sidx,x  in enumerate(sdpRes):\n",
    "                    for s in allStems:\n",
    "                        if str(x[0]).lower().startswith(s):\n",
    "                            sdpRels.add((s, sidx, stem2class[s]))\n",
    "                            if verbose:\n",
    "                                print(\"sdpRels\", x[0], s, sidx, stem2class[s])\n",
    "\n",
    "                posRegs = [x for x in sdpRels if x[2] in [\"POS\"]]\n",
    "                negRegs = [x for x in sdpRels if x[2] in [\"NEG\"]]\n",
    "\n",
    "                dirReg = len(posRegs)-len(negRegs)\n",
    "\n",
    "                if dirReg > 0:\n",
    "                    regdir  = \"UP\"\n",
    "                elif dirReg < 0:\n",
    "                    regdir = \"DOWN\"\n",
    "                else:\n",
    "                    regdir = \"NEU\"\n",
    "                \n",
    "                intdir = \"MIR_GENE\"\n",
    "\n",
    "            elif any([x in textBetween for x in [\"target of\", \"target for\", \"targeting\"]]):\n",
    "                intdir = \"MIR_GENE\"\n",
    "                regdir = \"NEU\"\n",
    "\n",
    "            else:                \n",
    "                intdir = \"GENE_MIR\"\n",
    "    \n",
    "        elif mirword.i < geneword.i:\n",
    "            textBetween = \" \".join([str(x) for x in doc if geneword.i-5 < x.i < geneword.i])\n",
    "\n",
    "            if any([x in textBetween for x in [\"by\"]]):\n",
    "                intdir = \"GENE_MIR\"\n",
    "                regdir = \"UP\"\n",
    "        \n",
    "    \n",
    "    regstems = foundStems\n",
    "    \n",
    "    #displacy.render(doc, style=\"dep\")\n",
    "    \n",
    "    if verbose:\n",
    "        print((intdir, regdir, \"NA\", isPassive))\n",
    "        \n",
    "    return {\"regulation_dir\": regdir, \"interaction_dir\": intdir, \"regulation_keys\": regstems}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
