{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "import os, sys\n",
    "sys.path.insert(0, \"/mnt/f/dev/git/miRExplore/python/\")\n",
    "\n",
    "\n",
    "import scispacy\n",
    "import spacy\n",
    "from spacy import displacy\n",
    "\n",
    "#nlp = spacy.load('/mnt/d/spacy/models/en_core_sci_lg-0.2.4/en_core_sci_lg/en_core_sci_lg-0.2.4/')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp_weblg = spacy.load('/mnt/d/spacy/models/en_core_web_lg-2.2.0/en_core_web_lg/en_core_web_lg-2.2.0/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp = spacy.load('/mnt/f/spacy/en_core_sci_lg-0.2.4/en_core_sci_lg/en_core_sci_lg-0.2.4/')\n",
    "nlp_ent = spacy.load(\"/mnt/f/spacy/en_ner_bionlp13cg_md-0.2.4/en_ner_bionlp13cg_md/en_ner_bionlp13cg_md-0.2.4\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp = spacy.load('/mnt/f/spacy/en_core_sci_lg-0.2.5/en_core_sci_lg/en_core_sci_lg-0.2.5/')\n",
    "nlp_ent = spacy.load(\"/mnt/f/spacy/en_ner_bionlp13cg_md-0.2.5/en_ner_bionlp13cg_md/en_ner_bionlp13cg_md-0.2.5\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!ls /mnt/d/spacy/models/en_core_web_lg-2.2.0/en_core_web_lg/en_core_web_lg-2.2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import textmining.MirGeneRelCheck\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def analyse(testCase, dist=80):\n",
    "    relCheck = textmining.MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "    doc = nlp(testCase[0])\n",
    "    \n",
    "    lWord = doc[testCase[1]]\n",
    "    rWord = doc[testCase[2]]\n",
    "\n",
    "    for t in doc:\n",
    "        print(t.idx, t.i, t, t.pos_, t.dep_, t.head, t.head.idx)\n",
    "\n",
    "    print(lWord, rWord)\n",
    "\n",
    "    if not \"mir\" in str(lWord).lower() and not \"let\" in str(lWord).lower():\n",
    "        tmp = lWord\n",
    "        lWord = rWord\n",
    "        rWord = tmp\n",
    "        \n",
    "    print(lWord, rWord)\n",
    "    \n",
    "    checkResults = relCheck.checkRelation(doc, lWord, rWord, verbose=True)\n",
    "    print(checkResults)\n",
    "    \n",
    "    displacy.render(doc, style=\"dep\", options={\"compact\": True, \"distance\": dist})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testcase = (u\"Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\", 4, 6)\n",
    "\n",
    "analyse(testcase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testcase = (u\"Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\", 6, 10)\n",
    "\n",
    "analyse(testcase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "doc = nlp(testcase[0])\n",
    "\n",
    "allTokens = {t.i: t for t in doc}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "[x for x in allTokens[23].children]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "allRels[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testCase = (u\"We show here that miR-29a, -29b-1, and -9 can regulate BACE1 expression in vitro.\", 4, 12)\n",
    "testCase = (u\"It has been predicted that miR-146a may target Tata Binding Protein (TBP).\",5, 12)\n",
    "testCase = (u\"Our results showed that the expression of miR-106b was inversely correlated with TβRII protein levels and miR-106b can directly inhibit the TβRII translation in vitro.\", 7, 21)\n",
    "\n",
    "relCheck = textmining.MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "doc = nlp(testCase[0])\n",
    "\n",
    "mirword = doc[testCase[1]]\n",
    "geneword = doc[testCase[2]]\n",
    "\n",
    "verbose=False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conjResult, conjs = relCheck.checkCommonConj(doc, mirword, geneword, verbose)\n",
    "sdpPass, passive, negated = relCheck.checkSDP(doc, mirword, geneword, verbose)\n",
    "compPass = relCheck.checkCompartments(doc, mirword, geneword, verbose)\n",
    "sigPathway = relCheck.checkSurContext(doc, mirword, geneword, verbose)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sdpRes = relCheck._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "sdpRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testCase = (u\"We show here that miR-29a, -29b-1, and -9 can regulate BACE1 expression in vitro.\", 4, 12)\n",
    "testCase = (u\"It has been predicted that miR-146a may target Tata Binding Protein (TBP).\",5, 12)\n",
    "testCase = (u\"Our results showed that the expression of miR-106b was inversely correlated with TβRII protein levels and miR-106b can directly inhibit the TβRII translation in vitro.\", 7, 21)\n",
    "testCase = (u\"The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 6, 16)\n",
    "testCase = (u\"These findings identify miR-29b as a novel posttranscriptional regulator of PGRN expression, raising the possibility that miR-29b or other miRNAs might be targeted therapeutically to increase hPGRN levels in some FTD patients.\", 3,10)\n",
    "relCheck = textmining.MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "doc = nlp(testCase[0])\n",
    "\n",
    "mirword = doc[testCase[1]]\n",
    "geneword = doc[testCase[2]]\n",
    "\n",
    "verbose=False\n",
    "\n",
    "from collections import Counter\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def loadStems():\n",
    "    \n",
    "    print(\"Loading stems\")\n",
    "    allRels = []\n",
    "    stem2class = {}\n",
    "\n",
    "    renameClass = {\n",
    "        'INT': 'NEU',\n",
    "        \"BOOST\": \"POS\",\n",
    "        \"COMBINE\": \"NEU\",\n",
    "        \"CHANGE\": \"NEU\",\n",
    "    }\n",
    "\n",
    "    with open(\"/mnt/d/dev/git/miRExplore/python/allrels.csv\") as fin:\n",
    "\n",
    "        for line in fin:\n",
    "            aline = line.strip().split(\",\", 1)\n",
    "            aregSyns = aline[1].split(\":\")\n",
    "            regSyns = []\n",
    "            if len(aregSyns) > 1:\n",
    "                regSyns = aregSyns[1].split(\"|\")\n",
    "            \n",
    "            aline[1] = aline[1].split(\":\")[0]           \n",
    "\n",
    "            regClass = aline[0]\n",
    "            regStem = aline[1]\n",
    "\n",
    "            if regClass in renameClass:\n",
    "                regClass = renameClass[regClass]\n",
    "                \n",
    "            stem2class[regStem] = regClass\n",
    "                \n",
    "            for regSyn in regSyns:\n",
    "                if not regSyn in stem2class:\n",
    "                    stem2class[regSyn] = regClass\n",
    "\n",
    "\n",
    "            allRels.append((regClass, regStem))\n",
    "\n",
    "    allStems = [x for x in stem2class]\n",
    "    \n",
    "    return allStems, allRels, stem2class\n",
    "\n",
    "allStems, allRels, stem2class = loadStems()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "from textmining.MirGeneRelCheck import SentenceRelationChecker, SentenceRelationClassifier\n",
    "from utils.tmutils import normalize_gene_names\n",
    "from collections import Counter\n",
    "from lxml import etree\n",
    "from sklearn.metrics import multilabel_confusion_matrix, classification_report\n",
    "from itertools import chain, combinations\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "scaiBase = \"/mnt/d/owncloud/data/miRExplore/scai_corpus/\"\n",
    "\n",
    "base = \"TRAIN\"\n",
    "\n",
    "if base == \"TRAIN\":\n",
    "    scaiFile = \"miRNA_train_fixed.xml\"\n",
    "    relsFile = \"scai_train_rels.tsv\"\n",
    "elif base == \"TEST\":\n",
    "    scaiFile = \"miRNA_test_fixed.xml\"\n",
    "    relsFile = \"scai_test_rels.tsv\"\n",
    "    \n",
    "normGeneSymbols= normalize_gene_names(path=scaiBase + \"/../obodir/\" + \"/hgnc_no_withdrawn.syn\")\n",
    "\n",
    "relChecker = SentenceRelationChecker(nlp)\n",
    "\n",
    "correctIdentified = 0\n",
    "incorrectIdentified = 0\n",
    "totalChecks = 0\n",
    "incorrectClass = Counter()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "relClasses = loadRelations()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp = spacy.load('/mnt/f/spacy/en_core_sci_lg-0.2.4/en_core_sci_lg/en_core_sci_lg-0.2.4/')\n",
    "nlp_ent = spacy.load(\"/mnt/f/spacy/en_ner_bionlp13cg_md-0.2.4/en_ner_bionlp13cg_md/en_ner_bionlp13cg_md-0.2.4\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "scaiBase = \"/mnt/d/owncloud/data/miRExplore/scai_corpus/\"\n",
    "\n",
    "normGeneSymbols= normalize_gene_names(path=scaiBase + \"/../obodir/\" + \"/hgnc_no_withdrawn.syn\")\n",
    "\n",
    "relChecker = SentenceRelationChecker(nlp, nlp_ent=nlp_ent)\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "\n",
    "def loadRelations(relsFile):\n",
    "    reldir2new = {}\n",
    "    reldir2new[\"TARGET\"] = \"NEU\"\n",
    "    reldir2new[\"INVCORR\"] = \"NEU\"\n",
    "    reldir2new[\"REGULATE\"] = \"NEU\"\n",
    "    relClasses = {}\n",
    "    with open(os.path.join(scaiBase, relsFile)) as fin:\n",
    "        \n",
    "        print(\"Loading relations\", relsFile)\n",
    "        \n",
    "        for lidx, line in enumerate(fin):\n",
    "\n",
    "            if lidx == 0:\n",
    "                continue\n",
    "\n",
    "            line = line.strip().split(\"\\t\")\n",
    "\n",
    "            interactionID = line[0].strip()\n",
    "            relType = line[1].strip()\n",
    "            relRDir = line[2].strip()\n",
    "            relIndirect = line[3].strip()\n",
    "            relPassive = line[4].strip()\n",
    "\n",
    "            if relRDir in reldir2new:\n",
    "                relRDir = reldir2new[relRDir]\n",
    "\n",
    "            relClasses[interactionID] = (relType, relRDir, relIndirect, relPassive)\n",
    "            \n",
    "        print(\"Interactions\", len(relClasses))\n",
    "        print(\"Non NA Interactions\", len([x for x in relClasses if not relClasses[x][0] == \"NA\"]))\n",
    "            \n",
    "    return relClasses\n",
    "\n",
    "def all_subsets(ss):\n",
    "    \n",
    "    return chain(*map(lambda x: combinations(ss, x), range(0, len(ss)+1)))\n",
    "\n",
    "def runCheck(numelems, base):\n",
    "    \n",
    "    if base == \"TRAIN\":\n",
    "        scaiFile = \"miRNA_train_fixed.xml\"\n",
    "        relsFile = \"scai_train_rels.tsv\"\n",
    "    elif base == \"TEST\":\n",
    "        scaiFile = \"miRNA_test_fixed.xml\"\n",
    "        relsFile = \"scai_test_rels.tsv\"\n",
    "    \n",
    "    relClasses = loadRelations(relsFile)\n",
    "    \n",
    "    check2results = {}\n",
    "    checkSubsets = [x for x in all_subsets(relClassifier.major_checks)]\n",
    "    checkSubsets = [checkSubsets[-1]]\n",
    "    \n",
    "    print(\"Will test {} subsets.\".format(len(checkSubsets)))\n",
    "    \n",
    "    for subset_checks in checkSubsets:\n",
    "    \n",
    "        print(\"Testing checks\", subset_checks)\n",
    "        \n",
    "        #relClassifier.active_checks = subset_checks\n",
    "\n",
    "        #allStems, allRels, stem2class = loadStems()\n",
    "\n",
    "        with open(os.path.join(scaiBase, scaiFile), 'r') as fin:\n",
    "            tree = etree.parse(fin)\n",
    "            root = tree.getroot()\n",
    "            scaiPairs = []\n",
    "            totalChecks = 0\n",
    "            correctIdentified = 0\n",
    "            incorrectIdentified = 0\n",
    "\n",
    "            incorrectClassified = 0\n",
    "            correctClassified = 0\n",
    "            totalClassified = 0\n",
    "            correctInteractClassified = 0\n",
    "            totalValidClassified = 0\n",
    "            totalClassifiable = 0\n",
    "            totalMirGeneDownClassified = 0\n",
    "\n",
    "            errorByDetect = Counter()\n",
    "            classifiedByDetect = Counter()\n",
    "            elemCaseCounter = Counter()\n",
    "            incorrectClass = Counter()\n",
    "\n",
    "            classifyTrue = []\n",
    "            classifyPred = []\n",
    "\n",
    "            for elem in root.findall(\".//document\"):\n",
    "\n",
    "                pmid = elem.attrib['origId']\n",
    "\n",
    "                for sentElem in elem:\n",
    "\n",
    "                    allEntities = sentElem.findall(\".//entity\")\n",
    "                    allPairs = sentElem.findall(\".//pair\")\n",
    "\n",
    "                    sentText = sentElem.attrib[\"text\"]\n",
    "\n",
    "                    entId2elem = {}\n",
    "\n",
    "                    for entity in allEntities:\n",
    "                        entId = entity.attrib['id']\n",
    "                        entText = entity.attrib['text']\n",
    "                        entType = entity.attrib['type']\n",
    "                        entOffset = tuple([int(x) for x in entity.attrib['charOffset'].split(\"-\")])\n",
    "\n",
    "                        if entType in [\"Specific_miRNAs\", \"Genes/Proteins\"]:\n",
    "\n",
    "                            if \"Genes\" in entType:\n",
    "                                if entText in normGeneSymbols:\n",
    "                                    entText = normGeneSymbols[entText]\n",
    "                                elif entText.upper() in normGeneSymbols:\n",
    "                                    gene = normGeneSymbols[entText.upper()]\n",
    "                            else:\n",
    "                                try:\n",
    "                                    entText = miRNA(entText).getStringFromParts([miRNAPART.MATURE, miRNAPART.ID, miRNAPART.PRECURSOR])\n",
    "                                except:\n",
    "                                    pass\n",
    "\n",
    "                            entTuple = (entText, entType, (entOffset[0], entOffset[1]+1))\n",
    "                            entId2elem[entId] = entTuple\n",
    "\n",
    "\n",
    "                            sentEntText = sentText[entTuple[2][0]:entTuple[2][1]]\n",
    "\n",
    "                    for pair in allPairs:\n",
    "\n",
    "                        validInteraction = pair.attrib['interaction'].lower() == \"true\"\n",
    "                        pairE1 = pair.attrib['e1']\n",
    "                        pairE2 = pair.attrib['e2']\n",
    "\n",
    "                        #if pairInt == 'true':\n",
    "                        if pairE1 in entId2elem and pairE2 in entId2elem:\n",
    "\n",
    "                            totalChecks += 1\n",
    "\n",
    "                            e1 = entId2elem[pairE1]\n",
    "                            e2 = entId2elem[pairE2]\n",
    "\n",
    "                            if not e1[1] in [\"Specific_miRNAs\"]:\n",
    "\n",
    "                                tmp=e1\n",
    "                                e1=e2\n",
    "                                e2=tmp\n",
    "\n",
    "\n",
    "                            relRes = relChecker.check_sentence(sentText\n",
    "                                                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": e1[2]}\n",
    "                                                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": e2[2]}\n",
    "                                                                , fix_special_chars=False\n",
    "                                                                , relClassifier=relClassifier.classify, verbose=False\n",
    "                                                                )\n",
    "\n",
    "\n",
    "                            fullsentence = relRes['full_sentence']\n",
    "                            acceptInteraction = relRes['accept_relation']\n",
    "\n",
    "                            if not acceptInteraction == validInteraction:\n",
    "\n",
    "                                incorrectIdentified += 1\n",
    "                                \n",
    "                                relResV=relChecker.check_sentence(sentText\n",
    "                                                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": e1[2]}\n",
    "                                                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": e2[2]}\n",
    "                                                                , fix_special_chars=False\n",
    "                                                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                                                )\n",
    "                                \n",
    "                                print(relResV)\n",
    "                            else:\n",
    "                                correctIdentified += 1\n",
    "\n",
    "                            elemCaseCounter[(validInteraction, acceptInteraction)]+=1\n",
    "\n",
    "                            totalClassified += 1\n",
    "\n",
    "                            if validInteraction:# and acceptInteraction:\n",
    "                                #print(relRes[\"check_results\"])   \n",
    "                                totalValidClassified += 1\n",
    "\n",
    "                                foundClasses = relRes[\"check_results\"][\"classification\"] #{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'regulation_keys': set()}\n",
    "                                foundTuple = (foundClasses[\"interaction_dir\"], foundClasses[\"regulation_dir\"])                                \n",
    "                                relationID = pair.attrib[\"id\"]\n",
    "                                testTuple = relClasses[relationID]\n",
    "\n",
    "                                classifyTrue.append((testTuple[0], testTuple[1]))\n",
    "                                classifyPred.append((foundTuple[0], foundTuple[1]))\n",
    "\n",
    "                                if testTuple[0] == \"MIR_GENE\" and testTuple[1] == \"DOWN\":\n",
    "                                    totalMirGeneDownClassified += 1\n",
    "\n",
    "                                classifiedByDetect[foundClasses[\"reg_detect\"]] += 1\n",
    "\n",
    "                                if testTuple[0] != foundTuple[0] or testTuple[1] != foundTuple[1]:\n",
    "\n",
    "\n",
    "                                    #print(sentText)\n",
    "                                    #print(e1, e2)\n",
    "                                    #print(foundTuple, foundClasses[\"reg_detect\"], \"passive?\", foundClasses[\"passive\"])\n",
    "                                    #print(testTuple)\n",
    "\n",
    "                                    #print()\n",
    "                                    #print()\n",
    "                                    #print()\n",
    "\n",
    "                                    errorByDetect[foundClasses[\"reg_detect\"]] += 1\n",
    "\n",
    "                                    incorrectClassified += 1\n",
    "\n",
    "\n",
    "\n",
    "                                    #numelems -= 1\n",
    "                                    #if numelems == 0:\n",
    "                                    #    return\n",
    "                                else:\n",
    "                                    \n",
    "                                    if foundClasses[\"reg_detect_major\"] == \"return\":\n",
    "                                        print(foundClasses)\n",
    "                                        print(sentText)\n",
    "                                        print(e1, e2)\n",
    "                                        print(foundTuple, foundClasses[\"reg_detect\"], \"passive?\", foundClasses[\"passive\"])\n",
    "                                        print(testTuple)\n",
    "\n",
    "                                        print()\n",
    "                                        print()\n",
    "                                        print()\n",
    "                                    correctClassified += 1\n",
    "\n",
    "                                    if acceptInteraction:\n",
    "                                        correctInteractClassified += 1\n",
    "\n",
    "\n",
    "\n",
    "            def printStats(outfile):\n",
    "                print(\"Total:     \", totalChecks, file=outfile)\n",
    "                print(\"Correct:   \", correctIdentified, correctIdentified/totalChecks, file=outfile)\n",
    "                print(\"Incorrect: \", incorrectIdentified, incorrectIdentified/totalChecks, file=outfile)\n",
    "                print(\"classes\", incorrectClass, file=outfile)\n",
    "                \n",
    "                precision = elemCaseCounter[(True, True)] / (elemCaseCounter[(True, True)]+elemCaseCounter[(True, False)])\n",
    "                recall = elemCaseCounter[(True, True)] / (elemCaseCounter[(True, True)]+elemCaseCounter[(False, True)])\n",
    "\n",
    "                f1 = 2* precision * recall / (precision+recall)\n",
    "\n",
    "                specificity = elemCaseCounter[(False, False)] / (elemCaseCounter[(True, False)] + elemCaseCounter[(False, False)])\n",
    "\n",
    "                print(\"precision\", precision, file=outfile)\n",
    "                print(\"recall\", recall, file=outfile)\n",
    "                print(\"specificity\", specificity, file=outfile)\n",
    "                print(\"f1\", f1, file=outfile)\n",
    "\n",
    "                print(\"Correct classified:     \", correctClassified, correctClassified/totalClassified, correctClassified/totalValidClassified,file=outfile)\n",
    "                print(\"Incorrect classified:   \", incorrectClassified, incorrectClassified/totalClassified, incorrectClassified/totalValidClassified,file=outfile)\n",
    "                print(\"Random classified: \", totalMirGeneDownClassified, totalMirGeneDownClassified / totalClassified, totalMirGeneDownClassified/totalValidClassified, file=outfile)\n",
    "                print(errorByDetect)\n",
    "\n",
    "                print(\"Correct interaction&classified:     \", correctInteractClassified, correctInteractClassified/totalClassified, correctInteractClassified/totalValidClassified,file=outfile)\n",
    "\n",
    "                print(\"Classified by: \", classifiedByDetect)\n",
    "\n",
    "                lClassifyTrue = [\"_\".join(x) for x in classifyTrue]\n",
    "                lClassifyPred = [\"_\".join(x) for x in classifyPred]\n",
    "\n",
    "                allLabels = sorted(set(lClassifyTrue+lClassifyPred))\n",
    "                matrix = multilabel_confusion_matrix(lClassifyTrue, lClassifyPred, labels=allLabels)\n",
    "                print(matrix)\n",
    "                print(classification_report(lClassifyTrue,lClassifyPred))\n",
    "                \n",
    "                return classification_report(lClassifyTrue, lClassifyPred, output_dict=True)\n",
    "\n",
    "            rep = printStats(sys.stdout)\n",
    "            check2results[subset_checks] = rep\n",
    "        #printStats(sys.stderr)\n",
    "    return check2results\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading relations scai_test_rels.tsv\n",
      "Interactions 232\n",
      "Non NA Interactions 138\n",
      "Will test 1 subsets.\n",
      "Testing checks ('compartment', 'between', 'counts', 'return')\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('APOBEC3C', 'Genes/Proteins', (99, 103))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('ANGPTL6', 'Genes/Proteins', (105, 112))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('Bmi-1', 'Genes/Proteins', (115, 120))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('angiopoietin-related growth factor protein 5', 'Genes/Proteins', (49, 93))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('APOBEC3C', 'Genes/Proteins', (95, 99))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('ANGPTL6', 'Genes/Proteins', (101, 108))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case miRNA-128\n",
      "add special case Bmi-1\n",
      "miRNA-128 2 Bmi-1 39\n",
      "(u\"Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\", 2, 39)\n",
      "geneword Bmi-1 mirword miRNA-128\n",
      "Conjuncts [inhibits, Bmi-1, found, promotes]\n",
      "neighbour inhibits the\n",
      "neighbour inhibits the\n",
      "neighbour Bmi-1 ,\n",
      "neighbour Bmi-1 and\n",
      "neighbour found to\n",
      "neighbour found to\n",
      "Conjunction before fc promotes\n",
      "fc add children for promotes\n",
      "Conjuncts [promotes, Bmi-1, found, inhibits]\n",
      "neighbour promotes stem\n",
      "neighbour promotes stem\n",
      "neighbour Bmi-1 ,\n",
      "neighbour Bmi-1 and\n",
      "neighbour found to\n",
      "neighbour found to\n",
      "Conjunction before fc inhibits\n",
      "fc add children for inhibits\n",
      "Conjuncts [differentiation, senescence]\n",
      "neighbour differentiation ,\n",
      "neighbour differentiation Bmi-1\n",
      "Conjunction before fc senescence\n",
      "fc add children for senescence\n",
      "Conjuncts [senescence, differentiation]\n",
      "neighbour senescence and\n",
      "neighbour senescence and\n",
      "Conjunction before fc differentiation\n",
      "Conjuncts [promotes, inhibits, found, Bmi-1]\n",
      "neighbour promotes stem\n",
      "neighbour promotes stem\n",
      "neighbour inhibits the\n",
      "neighbour inhibits the\n",
      "neighbour found to\n",
      "neighbour found to\n",
      "Conjunction before fc Bmi-1\n",
      "Conjuncts [promotes, inhibits, Bmi-1, found]\n",
      "neighbour promotes stem\n",
      "neighbour promotes stem\n",
      "neighbour inhibits the\n",
      "neighbour inhibits the\n",
      "neighbour Bmi-1 ,\n",
      "neighbour Bmi-1 and\n",
      "Conjunction before fc found\n",
      "Conjunctions\n",
      "protein [angiopoietin-related, a, up-regulated]\n",
      "ARP5 [(, )]\n",
      "ANGPTL6 [ANGPTL6]\n",
      "promotes [promotes, renewal, inhibits, expression, of, known, genes, involved, in, senescence, differentiation, Bmi-1, found]\n",
      "E2F-3a [E2F-3a]\n",
      "SDP sent: Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\n",
      "SDP     : [(miRNA-128, 'PROPN', 'compound', False), (targets, 'NOUN', 'ROOT', False), (protein, 'NOUN', 'appos', False), (suppressor, 'NOUN', 'appos', False), (promotes, 'VERB', 'acl:relcl', True), (inhibits, 'VERB', 'conj', True), (Bmi-1, 'PROPN', 'conj', False)]\n",
      "deepcheck connection rule noun found found transcription [a, transcription, factor, critical, for, the, control, of, cell-cycle, progression, ,, E2F-3a, ,, were, found, to, be, up-regulated]\n",
      "deepcheck connection rule [to, be, up-regulated]\n",
      "0 [Three, bioinformatics-verified, miRNA-128, targets, ,, angiopoietin-related, growth, factor, protein, 5, (, ARP5, ;, ANGPTL6, ), ,, a, transcription, suppressor, that, promotes, stem, cell, renewal] 24 0 Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\n",
      "1 [inhibits, the, expression, of, known, tumor, suppressor, genes, involved, in, senescence, and, differentiation, ,, Bmi-1] 15 174 inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\n",
      "2 [a, transcription, factor, critical, for, the, control, of, cell-cycle, progression, ,, E2F-3a, ,, were, found, to, be, up-regulated] 18 285 a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\n",
      "Conjuncts [inhibits, Bmi-1, found, promotes]\n",
      "neighbour inhibits the\n",
      "neighbour inhibits the\n",
      "neighbour Bmi-1 ,\n",
      "neighbour Bmi-1 and\n",
      "neighbour found to\n",
      "neighbour found to\n",
      "Conjunction before fc promotes\n",
      "fc add children for promotes\n",
      "Conjuncts [promotes, Bmi-1, found, inhibits]\n",
      "neighbour promotes stem\n",
      "neighbour promotes stem\n",
      "neighbour Bmi-1 ,\n",
      "neighbour Bmi-1 and\n",
      "neighbour found to\n",
      "neighbour found to\n",
      "Conjunction before fc inhibits\n",
      "fc add children for inhibits\n",
      "Conjuncts [differentiation, senescence]\n",
      "neighbour differentiation ,\n",
      "neighbour differentiation Bmi-1\n",
      "Conjunction before fc senescence\n",
      "fc add children for senescence\n",
      "Conjuncts [senescence, differentiation]\n",
      "neighbour senescence and\n",
      "neighbour senescence and\n",
      "Conjunction before fc differentiation\n",
      "Conjuncts [promotes, inhibits, found, Bmi-1]\n",
      "neighbour promotes stem\n",
      "neighbour promotes stem\n",
      "neighbour inhibits the\n",
      "neighbour inhibits the\n",
      "neighbour found to\n",
      "neighbour found to\n",
      "Conjunction before fc Bmi-1\n",
      "Conjuncts [promotes, inhibits, Bmi-1, found]\n",
      "neighbour promotes stem\n",
      "neighbour promotes stem\n",
      "neighbour inhibits the\n",
      "neighbour inhibits the\n",
      "neighbour Bmi-1 ,\n",
      "neighbour Bmi-1 and\n",
      "Conjunction before fc found\n",
      "Recognized overlapping entities: {Interval(274, 279, ('GENE_OR_GENE_PRODUCT', 'Bmi-1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "Three bioinformatics-verified miRNA-128 targets , angiopoietin-related growth factor protein 5 ( ARP5 ; ANGPTL6 ) , a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation , Bmi-1 , and\n",
      "counts between 1 Counter({'NEU': 3, 'NEG': 3, 'POS': 1})\n",
      "['Three', 'bioinformatics-verified', 'miRNA-128', 'targets', ',', 'angiopoietin-related', 'growth', 'factor', 'protein', '5', '(', 'ARP5', ';', 'ANGPTL6', ')', ',', 'a', 'transcription', 'suppressor', 'that', 'promotes', 'stem', 'cell', 'renewal', 'and', 'inhibits', 'the', 'expression', 'of', 'known', 'tumor', 'suppressor', 'genes', 'involved', 'in', 'senescence', 'and', 'differentiation', ',', 'Bmi-1', ',', 'and']\n",
      "Counter({'NEU': 3, 'NEG': 3, 'POS': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\n",
      "full sentence Three bioinformatics-verified <e1>miRNA-128</e1> targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, <e2>Bmi-1</e2>, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Three bioinformatics-verified <e1>miRNA-128</e1> targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, <e2>Bmi-1</e2>, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.', 'full_sentence': 'Three bioinformatics-verified <e1>miRNA-128</e1> targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, <e2>Bmi-1</e2>, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (30, 39), 'entity_text': 'miRNA-128'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (274, 279), 'entity_text': 'Bmi-1'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (100, 107)) ('SRF', 'Genes/Proteins', (13, 16))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (100, 107)) ('myocardin', 'Genes/Proteins', (21, 30))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (112, 119)) ('SRF', 'Genes/Proteins', (13, 16))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (112, 119)) ('myocardin', 'Genes/Proteins', (21, 30))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These abnormalities reflect the regulation of a cadre of modulators of SRF activity and actin dynamics by miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (106, 113)) ('SRF', 'Genes/Proteins', (71, 74))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These abnormalities reflect the regulation of a cadre of modulators of SRF activity and actin dynamics by miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (118, 125)) ('SRF', 'Genes/Proteins', (71, 74))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case microRNA-30\n",
      "add special case Oas1\n",
      "microRNA-30 12 Oas1 18\n",
      "(u\"Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\", 12, 18)\n",
      "geneword Oas1 mirword microRNA-30\n",
      "Conjunctions\n",
      "SDP sent: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "SDP     : [(microRNA-30, 'PROPN', 'compound', False), (precursor, 'NOUN', 'nmod', False), (features, 'NOUN', 'dobj', False), (implementing, 'VERB', 'advcl', False), (Modification, 'NOUN', 'nsubj', False), (avoided, 'VERB', 'ROOT', False), (induction, 'NOUN', 'dobj', False), (Oas1, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "Recognized overlapping entities: {Interval(127, 131, ('GENE_OR_GENE_PRODUCT', 'Oas1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "naturally occurring microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in\n",
      "counts between 1 Counter({'POS': 1})\n",
      "['naturally', 'occurring', 'microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "Counter({'POS': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "full sentence Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'full_sentence': 'Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (88, 99), 'entity_text': 'microRNA-30'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (127, 131), 'entity_text': 'Oas1'}}\n",
      "add special case miR-30\n",
      "add special case Oas1\n",
      "miR-30 14 Oas1 18\n",
      "(u\"Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\", 14, 18)\n",
      "geneword Oas1 mirword miR-30\n",
      "Conjunctions\n",
      "SDP sent: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "SDP     : [(miR-30, 'PROPN', 'compound', False), (precursor, 'NOUN', 'nmod', False), (features, 'NOUN', 'dobj', False), (implementing, 'VERB', 'advcl', False), (Modification, 'NOUN', 'nsubj', False), (avoided, 'VERB', 'ROOT', False), (induction, 'NOUN', 'dobj', False), (Oas1, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "Recognized overlapping entities: {Interval(127, 131, ('GENE_OR_GENE_PRODUCT', 'Oas1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in\n",
      "counts between 1 Counter({'POS': 1})\n",
      "['microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "Counter({'POS': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "full sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'full_sentence': 'Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (101, 107), 'entity_text': 'miR-30'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (127, 131), 'entity_text': 'Oas1'}}\n",
      "add special case mir-21\n",
      "add special case Pdcd4\n",
      "mir-21 10 Pdcd4 17\n",
      "(u\"In addition, our data indicate that either downregulation of mir-21 or overexpression of its target, Pdcd4, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.\", 10, 17)\n",
      "geneword Pdcd4 mirword mir-21\n",
      "Conjuncts [overexpression, mir-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc mir-21\n",
      "fc add children for mir-21\n",
      "Conjuncts [mir-21, overexpression]\n",
      "neighbour mir-21 or\n",
      "neighbour mir-21 or\n",
      "Conjunction before fc overexpression\n",
      "fc add children for overexpression\n",
      "Conjuncts [decreased, leads]\n",
      "neighbour decreased colony\n",
      "neighbour decreased colony\n",
      "Conjunction before fc leads\n",
      "fc add children for leads\n",
      "Conjuncts [apoptosis, proliferation]\n",
      "neighbour apoptosis ,\n",
      "neighbour apoptosis and\n",
      "Conjunction before fc proliferation\n",
      "fc add children for proliferation\n",
      "fc add children for proliferation\n",
      "Conjuncts [proliferation, apoptosis]\n",
      "neighbour proliferation ,\n",
      "neighbour proliferation increased\n",
      "Conjunction before fc apoptosis\n",
      "fc add children for apoptosis\n",
      "Conjuncts [leads, decreased]\n",
      "neighbour leads to\n",
      "neighbour leads to\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "fc add children for decreased\n",
      "Conjunctions\n",
      "mir-21 [of, mir-21, overexpression, of, target, Pdcd4]\n",
      "leads [leads, to, decreased, proliferation, increased, apoptosis, decreased, formation, in, soft, agar]\n",
      "achildren {mir-21}\n",
      "achildren False\n",
      "SDP sent: In addition, our data indicate that either downregulation of mir-21 or overexpression of its target, Pdcd4, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.\n",
      "SDP     : [(mir-21, 'PROPN', 'nmod', False), (indicate, 'VERB', 'ROOT', False), (leads, 'VERB', 'ccomp', False), (downregulation, 'NOUN', 'nsubj', False), (mir-21, 'PROPN', 'nmod', False)]\n",
      "deepcheck connection rule noun found decreased colony [decreased, colony, formation, in, soft, agar]\n",
      "0 [In, addition, ,, our, data, indicate] 6 0 In addition, our data indicate that either downregulation of mir-21 or overexpression of its target, Pdcd4, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.\n",
      "1 [downregulation, of, mir-21, or, overexpression, of, its, target, ,, Pdcd4, ,, in, glioblastoma-derived, cell, lines, leads, to, decreased, proliferation, ,, increased, apoptosis] 22 43 downregulation of mir-21 or overexpression of its target, Pdcd4, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.\n",
      "2 [decreased, colony, formation, in, soft, agar] 6 202 decreased colony formation in soft agar.\n",
      "Conjuncts [overexpression, mir-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc mir-21\n",
      "fc add children for mir-21\n",
      "Conjuncts [mir-21, overexpression]\n",
      "neighbour mir-21 or\n",
      "neighbour mir-21 or\n",
      "Conjunction before fc overexpression\n",
      "fc add children for overexpression\n",
      "Conjuncts [decreased, leads]\n",
      "neighbour decreased colony\n",
      "neighbour decreased colony\n",
      "Conjunction before fc leads\n",
      "fc add children for leads\n",
      "Conjuncts [apoptosis, proliferation]\n",
      "neighbour apoptosis ,\n",
      "neighbour apoptosis and\n",
      "Conjunction before fc proliferation\n",
      "fc add children for proliferation\n",
      "fc add children for proliferation\n",
      "Conjuncts [proliferation, apoptosis]\n",
      "neighbour proliferation ,\n",
      "neighbour proliferation increased\n",
      "Conjunction before fc apoptosis\n",
      "fc add children for apoptosis\n",
      "Conjuncts [leads, decreased]\n",
      "neighbour leads to\n",
      "neighbour leads to\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "fc add children for decreased\n",
      "Recognized overlapping entities: {Interval(101, 106, ('GENE_OR_GENE_PRODUCT', 'Pdcd4'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "downregulation of mir-21 or overexpression of its target , Pdcd4 , in\n",
      "counts between 1 Counter({'NEG': 1, 'POS': 1, 'NEU': 1})\n",
      "(False, {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence In addition, our data indicate that either downregulation of mir-21 or overexpression of its target, Pdcd4, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.\n",
      "full sentence In addition, our data indicate that either downregulation of <e1>mir-21</e1> or overexpression of its target, <e2>Pdcd4</e2>, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'In addition, our data indicate that either downregulation of <e1>mir-21</e1> or overexpression of its target, <e2>Pdcd4</e2>, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.', 'full_sentence': 'In addition, our data indicate that either downregulation of <e1>mir-21</e1> or overexpression of its target, <e2>Pdcd4</e2>, in glioblastoma-derived cell lines leads to decreased proliferation, increased apoptosis, and decreased colony formation in soft agar.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (61, 67), 'entity_text': 'mir-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (101, 106), 'entity_text': 'Pdcd4'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('IQGAP1', 'Genes/Proteins', (132, 138))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('IQ motif containing GTPase activating protein 1', 'Genes/Proteins', (83, 130))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('laminin γ1', 'Genes/Proteins', (141, 151))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('LAMC1', 'Genes/Proteins', (153, 158))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('integrin β1', 'Genes/Proteins', (164, 175))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin γ1 (LAMC1) and integrin β1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('ITGB1', 'Genes/Proteins', (177, 182))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('chemokine CCL8', 'Genes/Proteins', (51, 65))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CCL8', 'Genes/Proteins', (66, 71))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CCR5 chemokine receptor', 'Genes/Proteins', (92, 115))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CD4', 'Genes/Proteins', (142, 145))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CCR5', 'Genes/Proteins', (146, 150))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case miR-9\n",
      "add special case STAT3\n",
      "miR-9 4 STAT3 22\n",
      "(u\"Foremost among these is miR-9, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of STAT3.\", 4, 22)\n",
      "geneword STAT3 mirword miR-9\n",
      "Conjuncts [inhibiting, downregulating]\n",
      "neighbour inhibiting activation\n",
      "neighbour inhibiting activation\n",
      "Conjunction before fc downregulating\n",
      "fc add children for downregulating\n",
      "Conjuncts [downregulating, inhibiting]\n",
      "neighbour downregulating expression\n",
      "neighbour downregulating expression\n",
      "Conjunction before fc inhibiting\n",
      "fc add children for inhibiting\n",
      "Conjunctions\n",
      "downregulating [downregulating, expression, of, kinases, inhibiting, activation, of, STAT3]\n",
      "SDP sent: Foremost among these is miR-9, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of STAT3.\n",
      "SDP     : [(miR-9, 'PROPN', 'ROOT', False), (miR-9, 'PROPN', 'ROOT', False)]\n",
      "deepcheck connection rule [by, downregulating, expression, of, JAK, kinases, and, inhibiting, activation, of, STAT3]\n",
      "deepcheck connection rule noun found inhibiting activation [inhibiting, activation, of, STAT3]\n",
      "0 [Foremost, among, these, is, miR-9, ,, which, suppresses, mesenchymal, differentiation, in, glioblastoma, by, downregulating, expression, of, JAK, kinases] 18 0 Foremost among these is miR-9, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of STAT3.\n",
      "1 [inhibiting, activation, of, STAT3] 4 140 inhibiting activation of STAT3.\n",
      "Conjuncts [inhibiting, downregulating]\n",
      "neighbour inhibiting activation\n",
      "neighbour inhibiting activation\n",
      "Conjunction before fc downregulating\n",
      "fc add children for downregulating\n",
      "Conjuncts [downregulating, inhibiting]\n",
      "neighbour downregulating expression\n",
      "neighbour downregulating expression\n",
      "Conjunction before fc inhibiting\n",
      "fc add children for inhibiting\n",
      "Recognized overlapping entities: {Interval(165, 170, ('GENE_OR_GENE_PRODUCT', 'STAT3'))}\n",
      "passive? False\n",
      "Counter()\n",
      "these is miR-9 , which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of STAT3 .\n",
      "counts between 1 Counter({'NEG': 3, 'NEU': 2})\n",
      "['these', 'is', 'miR-9', ',', 'which', 'suppresses', 'mesenchymal', 'differentiation', 'in', 'glioblastoma', 'by', 'downregulating', 'expression', 'of', 'JAK', 'kinases', 'and', 'inhibiting', 'activation', 'of', 'STAT3', '.']\n",
      "Counter({'NEG': 3, 'NEU': 2})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Foremost among these is miR-9, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of STAT3.\n",
      "full sentence Foremost among these is <e1>miR-9</e1>, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of <e2>STAT3</e2>.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Foremost among these is <e1>miR-9</e1>, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of <e2>STAT3</e2>.', 'full_sentence': 'Foremost among these is <e1>miR-9</e1>, which suppresses mesenchymal differentiation in glioblastoma by downregulating expression of JAK kinases and inhibiting activation of <e2>STAT3</e2>.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (24, 29), 'entity_text': 'miR-9'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (165, 170), 'entity_text': 'STAT3'}}\n",
      "add special case miR-221/222\n",
      "add special case p27Kip1\n",
      "miR-221/222 2 p27Kip1 25\n",
      "(u\"Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\", 2, 25)\n",
      "geneword p27Kip1 mirword miR-221/222\n",
      "Conjuncts [reduced, p27Kip1]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conjuncts [p27Kip1, reduced]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "fc add children for reduced\n",
      "Conjunctions\n",
      "p27Kip1 [increased, p27Kip1, in, subcutaneous, mice, reduced, growth, through, up, regulation, of, p27Kip1]\n",
      "SDP sent: Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "SDP     : [(miR-221/222, 'PROPN', 'amod', False), (knocked-down, 'VERB', 'nsubj', False), (increased, 'VERB', 'amod', False), (p27Kip1, 'PROPN', 'ROOT', True), (reduced, 'VERB', 'conj', False), (regulation, 'NOUN', 'nmod', False), (p27Kip1, 'PROPN', 'nmod', False)]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "Conjuncts [reduced, p27Kip1]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conjuncts [p27Kip1, reduced]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "fc add children for reduced\n",
      "Recognized overlapping entities: {Interval(199, 206, ('GENE_OR_GENE_PRODUCT', 'p27Kip1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "Consistently , miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1 .\n",
      "counts between 1 Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "['Consistently', ',', 'miR-221/222', 'knocked-down', 'through', 'antisense', \"2'-OME-oligonucleotides\", 'increased', 'p27Kip1', 'in', 'U251', 'glioma', 'subcutaneous', 'mice', 'and', 'strongly', 'reduced', 'tumor', 'growth', 'in', 'vivo', 'through', 'up', 'regulation', 'of', 'p27Kip1', '.']\n",
      "Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "full sentence Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'full_sentence': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 25), 'entity_text': 'miR-221/222'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (199, 206), 'entity_text': 'p27Kip1'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "The data strongly suggested that a regulatory loop between miR-21 and STAT3 might provide an insight into the mechanism of modulating EGFR/STAT3 signaling. \n",
      "('miR-21', 'Specific_miRNAs', (59, 65)) ('STAT3', 'Genes/Proteins', (70, 75))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('microRNA-29b', 'Specific_miRNAs', (59, 71)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('miR-29b', 'Specific_miRNAs', (73, 80)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "MicroRNA-29b regulates the expression level of human progranulin, a secreted glycoprotein implicated in frontotemporal dementia.  \n",
      "('MicroRNA-29b', 'Specific_miRNAs', (0, 12)) ('progranulin', 'Genes/Proteins', (53, 64))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites. \n",
      "('microRNA-34a', 'Specific_miRNAs', (65, 77)) ('CD47', 'Genes/Proteins', (149, 153))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites. \n",
      "('microRNA-155', 'Specific_miRNAs', (79, 91)) ('CD47', 'Genes/Proteins', (149, 153))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites. \n",
      "('microRNA-326', 'Specific_miRNAs', (96, 108)) ('CD47', 'Genes/Proteins', (149, 153))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case CD47\n",
      "add special case microRNA-155\n",
      "CD47 20 microRNA-155 26\n",
      "(u\"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\", 20, 26)\n",
      "geneword CD47 mirword microRNA-155\n",
      "Conjuncts [microRNA-155, microRNA-326, microRNA-34a]\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-34a\n",
      "Conjuncts [microRNA-34a, microRNA-326, microRNA-155]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-155\n",
      "Conjuncts [microRNA-34a, microRNA-155, microRNA-326]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "Conjunction before fc microRNA-326\n",
      "Conjunctions\n",
      "microRNA-34a [microRNA-34a, microRNA-155, microRNA-326]\n",
      "SDP sent: Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "SDP     : [(CD47, 'PROPN', 'nmod', False), (region, 'NOUN', 'dobj', False), (targeted, 'VERB', 'ROOT', False), (microRNA-155, 'PROPN', 'nmod', False)]\n",
      "0 [Three, microRNA, upregulated, in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, ), targeted, the, 3'-untranslated, region, of, CD47, in, reporter, assays, ,, with, microRNA-155, even, at, two, distinct, sites] 32 0 Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "Conjuncts [microRNA-155, microRNA-326, microRNA-34a]\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-34a\n",
      "Conjuncts [microRNA-34a, microRNA-326, microRNA-155]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-155\n",
      "Conjuncts [microRNA-34a, microRNA-155, microRNA-326]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "Conjunction before fc microRNA-326\n",
      "Recognized overlapping entities: {Interval(149, 153, ('GENE_OR_GENE_PRODUCT', 'CD47'))}\n",
      "passive? False\n",
      "Counter()\n",
      "region of CD47 in reporter assays , with microRNA-155 even at\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "full sentence Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}}, 'tagged_sent': \"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\", 'full_sentence': \"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (149, 153), 'entity_text': 'CD47'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (179, 191), 'entity_text': 'microRNA-155'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "The miRNA miR-23a was predicted to bind XIAP mRNA. \n",
      "('miR-23a', 'Specific_miRNAs', (10, 17)) ('XIAP', 'Genes/Proteins', (40, 44))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a. \n",
      "('miR-23a', 'Specific_miRNAs', (0, 7)) ('XIAP', 'Genes/Proteins', (37, 41))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case miR-23a\n",
      "add special case XIAP\n",
      "miR-23a 11 XIAP 18\n",
      "(u\"miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 11, 18)\n",
      "geneword XIAP mirword miR-23a\n",
      "Conjuncts [led, bound]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc bound\n",
      "fc add children for bound\n",
      "Conjuncts [miR-23a, UTR]\n",
      "neighbour miR-23a inhibition\n",
      "neighbour miR-23a inhibition\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [UTR, miR-23a]\n",
      "neighbour UTR of\n",
      "neighbour UTR of\n",
      "Conjunction before fc miR-23a\n",
      "Conjuncts [bound, led]\n",
      "neighbour bound the\n",
      "neighbour bound the\n",
      "Conjunction before fc led\n",
      "fc add children for led\n",
      "Conjunctions\n",
      "bound [bound, 3, UTR, of, XIAP, miR-23a, led, to, increase, in, mRNA]\n",
      "SDP sent: miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "SDP     : [(miR-23a, 'PROPN', 'conj', False), (UTR, 'NOUN', 'dobj', True), (bound, 'VERB', 'ROOT', True), (led, 'VERB', 'conj', False), (increase, 'NOUN', 'nmod', False), (mRNA, 'NOUN', 'nmod', False), (XIAP, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule noun found led inhibition [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro]\n",
      "0 [miR-23a, directly, bound, the, 3, ', UTR, of, XIAP, ,, and, miR-23a] 12 0 miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "1 [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro] 10 55 inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "2 [demonstrating, that, XIAP, is, a, previously, uncharacterized, target, for, miR-23a] 10 108 demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "Conjuncts [led, bound]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc bound\n",
      "fc add children for bound\n",
      "Conjuncts [miR-23a, UTR]\n",
      "neighbour miR-23a inhibition\n",
      "neighbour miR-23a inhibition\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [UTR, miR-23a]\n",
      "neighbour UTR of\n",
      "neighbour UTR of\n",
      "Conjunction before fc miR-23a\n",
      "Conjuncts [bound, led]\n",
      "neighbour bound the\n",
      "neighbour bound the\n",
      "Conjunction before fc led\n",
      "fc add children for led\n",
      "Recognized overlapping entities: {Interval(88, 92, ('GENE_OR_GENE_PRODUCT', 'XIAP'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      ", and miR-23a inhibition led to an increase in XIAP mRNA in\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7ff758df31f0>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "full sentence miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 'full_sentence': \"miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (47, 54), 'entity_text': 'miR-23a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (88, 92), 'entity_text': 'XIAP'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-23a regulation of X-linked inhibitor of apoptosis (XIAP) contributes to sex differences in the response to cerebral ischemia.  \n",
      "('miR-23a', 'Specific_miRNAs', (0, 7)) ('X-linked inhibitor of apoptosis', 'Genes/Proteins', (22, 53))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-23a regulation of X-linked inhibitor of apoptosis (XIAP) contributes to sex differences in the response to cerebral ischemia.  \n",
      "('miR-23a', 'Specific_miRNAs', (0, 7)) ('XIAP', 'Genes/Proteins', (55, 59))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Here, two microRNAs, namely mir-7 and mir-153, have been identified to regulate alpha-synuclein levels post-transcriptionally. \n",
      "('mir-7', 'Specific_miRNAs', (28, 33)) ('alpha-synuclein', 'Genes/Proteins', (80, 95))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Here, two microRNAs, namely mir-7 and mir-153, have been identified to regulate alpha-synuclein levels post-transcriptionally. \n",
      "('mir-153', 'Specific_miRNAs', (38, 45)) ('alpha-synuclein', 'Genes/Proteins', (80, 95))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Post-transcriptional regulation of alpha-synuclein expression by mir-7 and mir-153.  \n",
      "('mir-7', 'Specific_miRNAs', (65, 70)) ('alpha-synuclein', 'Genes/Proteins', (35, 50))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Post-transcriptional regulation of alpha-synuclein expression by mir-7 and mir-153.  \n",
      "('mir-153', 'Specific_miRNAs', (75, 82)) ('alpha-synuclein', 'Genes/Proteins', (35, 50))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These findings indicate that miR-107 contributes to GRN expression regulation with implications for brain disorders.   \n",
      "('miR-107', 'Specific_miRNAs', (29, 36)) ('GRN', 'Genes/Proteins', (52, 55))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case miR-107\n",
      "add special case GRN\n",
      "miR-107 17 GRN 22\n",
      "(u\"In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\", 17, 22)\n",
      "geneword GRN mirword miR-107\n",
      "Conjuncts [decreased, subjacent]\n",
      "neighbour decreased miR-107\n",
      "neighbour decreased miR-107\n",
      "Conjunction before fc subjacent\n",
      "fc add children for subjacent\n",
      "Conjuncts [subjacent, decreased]\n",
      "neighbour subjacent to\n",
      "neighbour subjacent to\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "fc add children for decreased\n",
      "Conjunctions\n",
      "subjacent [subjacent, to, depth, controlled, cortical, impact, decreased, miR-107, with, augmentation, of, neuronal, expression]\n",
      "SDP sent: In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "SDP     : [(miR-107, 'PROPN', 'dobj', False), (decreased, 'VERB', 'conj', False), (augmentation, 'NOUN', 'nmod', False), (expression, 'NOUN', 'nmod', False), (GRN, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule noun found decreased miR-107 [decreased, miR-107, with, augmentation, of, neuronal, GRN, expression]\n",
      "0 [In, brain, tissue, subjacent, to, 1.0, mm, depth, controlled, cortical, impact, ,, show] 13 0 In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "1 [surviving, hippocampal, neurons] 3 70 surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "2 [] 0\n",
      "empty compartment []\n",
      "3 [decreased, miR-107, with, augmentation, of, neuronal, GRN, expression] 8 105 decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "Conjuncts [decreased, subjacent]\n",
      "neighbour decreased miR-107\n",
      "neighbour decreased miR-107\n",
      "Conjunction before fc subjacent\n",
      "fc add children for subjacent\n",
      "Conjuncts [subjacent, decreased]\n",
      "neighbour subjacent to\n",
      "neighbour subjacent to\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "fc add children for decreased\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "decreased miR-107 with augmentation of neuronal GRN expression\n",
      "counts between 1 Counter({'NEG': 1, 'POS': 1, 'NEU': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased miR-107 with augmentation of neuronal GRN expression.\n",
      "full sentence In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased <e1>miR-107</e1> with augmentation of neuronal <e2>GRN</e2> expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased <e1>miR-107</e1> with augmentation of neuronal <e2>GRN</e2> expression.', 'full_sentence': 'In brain tissue subjacent to 1.0 mm depth controlled cortical impact, surviving hippocampal neurons show decreased <e1>miR-107</e1> with augmentation of neuronal <e2>GRN</e2> expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (115, 122), 'entity_text': 'miR-107'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (153, 156), 'entity_text': 'GRN'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of GRN mRNA are recognized by miR-107 and are highly conserved among vertebrate species. \n",
      "('miR-107', 'Specific_miRNAs', (182, 189)) ('GRN', 'Genes/Proteins', (155, 158))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-107 regulates granulin/progranulin with implications for traumatic brain injury and neurodegenerative disease.  \n",
      "('miR-107', 'Specific_miRNAs', (0, 7)) ('granulin', 'Genes/Proteins', (18, 26))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-107 regulates granulin/progranulin with implications for traumatic brain injury and neurodegenerative disease.  \n",
      "('miR-107', 'Specific_miRNAs', (0, 7)) ('progranulin', 'Genes/Proteins', (27, 38))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "GRN mRNA interacts preferentially with Argonaute 2. In vitro and in vivo studies indicate that regulation of GRN by miR-107 may be functionally important. \n",
      "('miR-107', 'Specific_miRNAs', (116, 123)) ('GRN', 'Genes/Proteins', (109, 112))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case MiR-21\n",
      "add special case Bax\n",
      "MiR-21 0 Bax/Bcl-2 14\n",
      "(u\"MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\", 0, 14)\n",
      "geneword Bax/Bcl-2 mirword MiR-21\n",
      "Conjuncts [induced, protected]\n",
      "neighbour induced apoptosis\n",
      "neighbour induced apoptosis\n",
      "Conjunction before fc protected\n",
      "fc add children for protected\n",
      "fc add children for protected\n",
      "Conjuncts [protected, induced]\n",
      "neighbour protected human\n",
      "neighbour protected human\n",
      "Conjunction before fc induced\n",
      "fc add children for induced\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity .\n",
      "neighbour activity None\n",
      "Conjunction before fc ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "Conjunctions\n",
      "protected [protected, human, cells, from, chemotherapeutic, temozolomide, induced, apoptosis]\n",
      "ratio [ratio, activity]\n",
      "SDP sent: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "SDP     : [(MiR-21, 'PROPN', 'nsubj', False), (protected, 'VERB', 'ROOT', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', False), (Bax/Bcl-2, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found induced apoptosis [induced, apoptosis]\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "0 [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 17 0 MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "1 [induced, apoptosis] 2 88 induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "Context Fail <regex.Match object; span=(8, 14), match=' ratio'> regex.Regex('( ratio){e<=0}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(120, 129, ('SIMPLE_CHEMICAL', 'Bax/Bcl-2'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide by decreasing Bax/Bcl-2 ratio and\n",
      "counts between 1 Counter({'NEG': 1})\n",
      "['MiR-21', 'protected', 'human', 'glioblastoma', 'U87MG', 'cells', 'from', 'chemotherapeutic', 'drug', 'temozolomide', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and']\n",
      "Counter({'NEG': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "full sentence <e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing <e2>Bax</e2>/Bcl-2 ratio and caspase-3 activity.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing <e2>Bax</e2>/Bcl-2 ratio and caspase-3 activity.', 'full_sentence': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing <e2>Bax</e2>/Bcl-2 ratio and caspase-3 activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 6), 'entity_text': 'MiR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (120, 123), 'entity_text': 'Bax'}}\n",
      "add special case MiR-21\n",
      "add special case Bcl-2\n",
      "MiR-21 0 Bax/Bcl-2 14\n",
      "(u\"MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\", 0, 14)\n",
      "geneword Bax/Bcl-2 mirword MiR-21\n",
      "Conjuncts [induced, protected]\n",
      "neighbour induced apoptosis\n",
      "neighbour induced apoptosis\n",
      "Conjunction before fc protected\n",
      "fc add children for protected\n",
      "fc add children for protected\n",
      "Conjuncts [protected, induced]\n",
      "neighbour protected human\n",
      "neighbour protected human\n",
      "Conjunction before fc induced\n",
      "fc add children for induced\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity .\n",
      "neighbour activity None\n",
      "Conjunction before fc ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "Conjunctions\n",
      "protected [protected, human, cells, from, chemotherapeutic, temozolomide, induced, apoptosis]\n",
      "ratio [ratio, activity]\n",
      "SDP sent: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "SDP     : [(MiR-21, 'PROPN', 'nsubj', False), (protected, 'VERB', 'ROOT', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', False), (Bax/Bcl-2, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found induced apoptosis [induced, apoptosis]\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "0 [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 17 0 MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "1 [induced, apoptosis] 2 88 induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "Context Fail <regex.Match object; span=(8, 14), match=' ratio'> regex.Regex('( ratio){e<=0}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(120, 129, ('SIMPLE_CHEMICAL', 'Bax/Bcl-2'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide by decreasing Bax/Bcl-2 ratio and\n",
      "counts between 1 Counter({'NEG': 1})\n",
      "['MiR-21', 'protected', 'human', 'glioblastoma', 'U87MG', 'cells', 'from', 'chemotherapeutic', 'drug', 'temozolomide', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and']\n",
      "Counter({'NEG': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "full sentence <e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/<e2>Bcl-2</e2> ratio and caspase-3 activity.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/<e2>Bcl-2</e2> ratio and caspase-3 activity.', 'full_sentence': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/<e2>Bcl-2</e2> ratio and caspase-3 activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 6), 'entity_text': 'MiR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (124, 129), 'entity_text': 'Bcl-2'}}\n",
      "add special case miR-21\n",
      "add special case caspase-3\n",
      "miR-21 8 caspase-3 27\n",
      "(u\"Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\", 8, 27)\n",
      "geneword caspase-3 mirword miR-21\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity ,\n",
      "neighbour activity which\n",
      "Conjunction before fc ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "Conjunctions\n",
      "ratio [ratio, activity]\n",
      "SDP sent: Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "SDP     : [(miR-21, 'PROPN', 'nsubj', False), (inhibit, 'VERB', 'ccomp', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', True), (activity, 'NOUN', 'conj', False), (caspase-3, 'PROPN', 'compound', False)]\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity, ,, which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "0 [these, results, suggested, that, over-express, miR-21, could, inhibit, TMZ-induced, apoptosis, in, U87MG, cells, ,, at, least, in, part, ,, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 26 16 these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "1 [Taken, together] 2 0 Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "2 [] 0\n",
      "empty compartment []\n",
      "3 [which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ] 16 189 which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity ,\n",
      "neighbour activity which\n",
      "Conjunction before fc ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "Recognized overlapping entities: {Interval(169, 178, ('GENE_OR_GENE_PRODUCT', 'caspase-3'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2})\n",
      "that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells , at least in part , by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "counts between 1 Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "['that', 'over-express', 'miR-21', 'could', 'inhibit', 'TMZ-induced', 'apoptosis', 'in', 'U87MG', 'cells', ',', 'at', 'least', 'in', 'part', ',', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and', 'caspase-3', 'activity']\n",
      "Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "full sentence Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.', 'full_sentence': 'Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (58, 64), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (169, 178), 'entity_text': 'caspase-3'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Hypoxia induces microRNA miR-210 in vitro and in vivo ephrin-A3 and neuronal pentraxin 1 are potentially regulated by miR-210.  \n",
      "('miR-210', 'Specific_miRNAs', (118, 125)) ('ephrin-A3', 'Genes/Proteins', (54, 63))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Hypoxia induces microRNA miR-210 in vitro and in vivo ephrin-A3 and neuronal pentraxin 1 are potentially regulated by miR-210.  \n",
      "('miR-210', 'Specific_miRNAs', (118, 125)) ('neuronal pentraxin 1', 'Genes/Proteins', (68, 88))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Furthermore, miR-210 was shown to bind to the predicted target sites of ephrin-A3 or neuronal pentraxin 1, causing repression in luciferase reporter activity. \n",
      "('miR-210', 'Specific_miRNAs', (13, 20)) ('ephrin-A3', 'Genes/Proteins', (72, 81))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Furthermore, miR-210 was shown to bind to the predicted target sites of ephrin-A3 or neuronal pentraxin 1, causing repression in luciferase reporter activity. \n",
      "('miR-210', 'Specific_miRNAs', (13, 20)) ('neuronal pentraxin 1', 'Genes/Proteins', (85, 105))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Our data strongly indicate that CDK6 is regulated by microRNA 124 in medulloblastoma and that miR 124 modulates medulloblastoma cell growth.   \n",
      "('microRNA 124', 'Specific_miRNAs', (53, 65)) ('CDK6', 'Genes/Proteins', (32, 36))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays. \n",
      "('miR 124a', 'Specific_miRNAs', (31, 39)) ('CDK6', 'Genes/Proteins', (44, 48))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons. \n",
      "('miR-128a', 'Specific_miRNAs', (73, 81)) ('Tau', 'Genes/Proteins', (122, 125))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.  \n",
      "('miR-29b', 'Specific_miRNAs', (0, 7)) ('podoplanin', 'Genes/Proteins', (30, 40))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.  \n",
      "('miR-125a', 'Specific_miRNAs', (12, 20)) ('podoplanin', 'Genes/Proteins', (30, 40))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "In this study, we focused on miR-29b and miR-125a, which were predicted to regulate PDPN, and demonstrated that these microRNAs directly target the 3' untranslated region of PDPN and inhibit invasion, apoptosis, and proliferation of glioblastomas. \n",
      "('miR-29b', 'Specific_miRNAs', (29, 36)) ('PDPN', 'Genes/Proteins', (84, 88))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "In this study, we focused on miR-29b and miR-125a, which were predicted to regulate PDPN, and demonstrated that these microRNAs directly target the 3' untranslated region of PDPN and inhibit invasion, apoptosis, and proliferation of glioblastomas. \n",
      "('miR-125a', 'Specific_miRNAs', (41, 49)) ('PDPN', 'Genes/Proteins', (84, 88))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "add special case miR-326\n",
      "add special case Ets-1\n",
      "miR-326 4 Ets-1 10\n",
      "(u\"We also found that miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\", 4, 10)\n",
      "geneword Ets-1 mirword miR-326\n",
      "Conjunctions\n",
      "regulator [a, differentiation]\n",
      "SDP sent: We also found that miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\n",
      "SDP     : [(miR-326, 'PROPN', 'nsubj', False), (promoted, 'VERB', 'ccomp', False), (targeting, 'VERB', 'advcl', False), (Ets-1, 'PROPN', 'dobj', False)]\n",
      "deepcheck connection rule noun found promoted miR-326 [that, miR-326, promoted, T(H)-17, differentiation, by, targeting, Ets-1, ,, a, negative, regulator, of, T(H)-17, differentiation]\n",
      "deepcheck connection rule [by, targeting, Ets-1, ,, a, negative, regulator, of, T(H)-17, differentiation]\n",
      "0 [We, also, found] 3 0 We also found that miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\n",
      "1 [miR-326, promoted, T(H)-17, differentiation, by, targeting, Ets-1, ,, a, negative, regulator, of, T(H)-17, differentiation] 14 19 miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\n",
      "Context Fail <regex.Match object; span=(7, 16), match=' negative', fuzzy_counts=(1, 0, 0)> regex.Regex('(-negative){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(73, 78, ('GENE_OR_GENE_PRODUCT', 'Ets-1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "(False, {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence We also found that miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\n",
      "full sentence We also found that <e1>miR-326</e1> promoted T(H)-17 differentiation by targeting <e2>Ets-1</e2>, a negative regulator of T(H)-17 differentiation.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'We also found that <e1>miR-326</e1> promoted T(H)-17 differentiation by targeting <e2>Ets-1</e2>, a negative regulator of T(H)-17 differentiation.', 'full_sentence': 'We also found that <e1>miR-326</e1> promoted T(H)-17 differentiation by targeting <e2>Ets-1</e2>, a negative regulator of T(H)-17 differentiation.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (19, 26), 'entity_text': 'miR-326'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (73, 78), 'entity_text': 'Ets-1'}}\n",
      "add special case microRNA-433\n",
      "add special case FGF20\n",
      "microRNA-433 17 FGF20 22\n",
      "(u\"We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\", 17, 22)\n",
      "geneword FGF20 mirword microRNA-433\n",
      "Conjunctions\n",
      "SDP sent: We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "SDP     : [(microRNA-433, 'PROPN', 'nmod', False), (site, 'NOUN', 'dobj', False), (disrupts, 'VERB', 'ccomp', False), (translation, 'NOUN', 'nmod', False), (FGF20, 'PROPN', 'nmod', False)]\n",
      "0 [We, show, in, several, functional, assays] 6 0 We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "1 [the, risk, allele, for, rs12720208, disrupts, a, binding, site, for, microRNA-433, ,, increasing, translation, of, FGF20, in, vitro, and, in, vivo] 21 42 the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "Recognized overlapping entities: {Interval(141, 146, ('GENE_OR_GENE_PRODUCT', 'FGF20'))}\n",
      "passive? False\n",
      "Counter()\n",
      "site for microRNA-433 , increasing translation of FGF20 in vitro\n",
      "counts between 1 Counter({'POS': 1, 'NEU': 1})\n",
      "['site', 'for', 'microRNA-433', ',', 'increasing', 'translation', 'of', 'FGF20', 'in', 'vitro']\n",
      "Counter({'POS': 1, 'NEU': 1})\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "full sentence We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.', 'full_sentence': 'We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (101, 113), 'entity_text': 'microRNA-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (141, 146), 'entity_text': 'FGF20'}}\n",
      "Total:      232\n",
      "Correct:    218 0.9396551724137931\n",
      "Incorrect:  14 0.0603448275862069\n",
      "classes Counter()\n",
      "precision 0.9420289855072463\n",
      "recall 0.9558823529411765\n",
      "specificity 0.9166666666666666\n",
      "f1 0.9489051094890512\n",
      "Correct classified:      128 0.5517241379310345 0.927536231884058\n",
      "Incorrect classified:    10 0.04310344827586207 0.07246376811594203\n",
      "Random classified:  52 0.22413793103448276 0.37681159420289856\n",
      "Counter({'counts between': 7, 'return': 3})\n",
      "Correct interaction&classified:      121 0.521551724137931 0.8768115942028986\n",
      "Classified by:  Counter({'return': 57, 'counts between': 33, 'compartment mir gene': 12, 'counts opp': 9, 'between gene mir': 9, 'counts between equal': 8, 'between mir gene': 4, 'compartment mir gene neg corr': 4, 'counts between alternating': 1, 'compartment gene mir': 1})\n",
      "[[[132   2]\n",
      "  [  0   4]]\n",
      "\n",
      " [[137   0]\n",
      "  [  0   1]]\n",
      "\n",
      " [[ 83   3]\n",
      "  [  5  47]]\n",
      "\n",
      " [[ 55   2]\n",
      "  [  5  76]]\n",
      "\n",
      " [[135   3]\n",
      "  [  0   0]]]\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      " GENE_MIR_NEU       0.67      1.00      0.80         4\n",
      "  GENE_MIR_UP       1.00      1.00      1.00         1\n",
      "MIR_GENE_DOWN       0.94      0.90      0.92        52\n",
      " MIR_GENE_NEU       0.97      0.94      0.96        81\n",
      "  MIR_GENE_UP       0.00      0.00      0.00         0\n",
      "\n",
      "     accuracy                           0.93       138\n",
      "    macro avg       0.72      0.77      0.74       138\n",
      " weighted avg       0.95      0.93      0.94       138\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "test_results = runCheck(-1, \"TEST\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "analyse(\n",
    "#(\"We also found that miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\", 4, 10)\n",
    "    #(\"In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\", 4, 10),\n",
    "    #(\"miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 11, 18),\n",
    "    #(\"Downregulation of Pdcd4 by mir-21 facilitates glioblastoma proliferation in vivo.\", 4,2),\n",
    "    (\"We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \", 3, 15),\n",
    "    100\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "\n",
    "with open(\"scai_test_f1.pickle\", \"wb\") as fout:\n",
    "    pickle.dump(test_results, fout)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "runCheck(-1, \"TRAIN\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def getTokenDist(tk1, tk2):\n",
    "    return abs(tk1.i-tk2.i)\n",
    "\n",
    "def countStems(tokens, geneword, mirword, keepNEU=False):\n",
    "    stemTypes2Count = Counter()\n",
    "    mirgene2stem = defaultdict(lambda: Counter())\n",
    "\n",
    "    for tidx,x in enumerate(tokens):\n",
    "        tknStem, matchStem, wasFound = getstem(x)\n",
    "\n",
    "        if wasFound and (tknStem != \"NEU\" or keepNEU):\n",
    "            \n",
    "            if tidx >= len(tokens)-2 and str(tokens[tidx-1]) == \"to\":\n",
    "                continue\n",
    "            \n",
    "            geneDist = getTokenDist(x, geneword)\n",
    "            mirDist = getTokenDist(x, mirword)\n",
    "                        \n",
    "            if geneDist < mirDist and geneDist < 3:\n",
    "                mirgene2stem[\"gene\"][tknStem] += 1\n",
    "            elif mirDist < geneDist and mirDist < 3:\n",
    "                mirgene2stem[\"mir\"][tknStem] += 1\n",
    "\n",
    "        if wasFound and (tknStem != \"NEU\" or keepNEU):\n",
    "            stemTypes2Count[tknStem] += 1\n",
    "\n",
    "    if len(stemTypes2Count) == 0:\n",
    "        if \"regulated\" in str(geneword):\n",
    "            stemTypes2Count[\"NEU\"] += 1\n",
    "\n",
    "    return stemTypes2Count,mirgene2stem\n",
    "\n",
    "def getRelsForTokens(tokens, desc,verbose):\n",
    "    rels = []\n",
    "    verbSet = set([\"VERB\"])\n",
    "    for x in tokens:\n",
    "        \n",
    "        tokenStemClass, s, _ = getstem(x)\n",
    "        rels.append((s, None, tokenStemClass, x.pos_, x))\n",
    "        if verbose:\n",
    "            print(desc, x, s, tokenStemClass, x.pos_, x)\n",
    "                    \n",
    "    hasVerb = len(list(set([x[3] for x in rels]) & verbSet)) > 0\n",
    "    \n",
    "    if hasVerb:\n",
    "        rels = [x for x in rels if x[3] in verbSet]\n",
    "    \n",
    "    return rels\n",
    "\n",
    "def evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg):\n",
    "\n",
    "    invcor = False\n",
    "    regdir = None\n",
    "\n",
    "    #if dirReg == 0:\n",
    "    if (mirPos > 0 and geneNeg > 0) or (genePos > 0 and mirNeg > 0):\n",
    "        return \"DOWN\", True\n",
    "\n",
    "    #else:\n",
    "    if ((mirPos > 0 and genePos > 0) and(mirNeg == 0 and geneNeg == 0)) or ((mirNeg > 0 and geneNeg >0) and(mirPos == 0 and genePos == 0)):\n",
    "        return \"UP\", False\n",
    "\n",
    "    return None, None\n",
    "\n",
    "def scoreWord(tkn):\n",
    "    \n",
    "    jtkstem, s, _ = getstem(tkn)\n",
    "                        \n",
    "    if jtkstem == None:\n",
    "\n",
    "        if jtk.startswith((\"loss\", \"decreas\", \"low\")):\n",
    "            return 0,1\n",
    "\n",
    "        elif jtk.startswith((\"increas\", \"high\", \"posit\")):\n",
    "            return 1,0\n",
    "\n",
    "    else:\n",
    "        if jtkstem == \"POS\":\n",
    "            return 1,0\n",
    "        elif jtkstem == \"NEG\":\n",
    "            return 0,1\n",
    "        \n",
    "    return 0,0\n",
    "\n",
    "\n",
    "def getTextBetween(tks, lword, rword):\n",
    "    tokenBetween = [x for x in tks if lword.i-3 < x.i < rword.i+3]\n",
    "    \n",
    "    lit = lword\n",
    "    while lit.pos_ in [\"nmod\", \"conj\"]:\n",
    "        if not lit in tokenBetween:\n",
    "            tokenBetween.append(lit)\n",
    "            \n",
    "        lit = lit.head\n",
    "        \n",
    "    rit = rword\n",
    "    while rit.pos_ in [\"nmod\", \"conj\"]:\n",
    "        if not rit in tokenBetween:\n",
    "            tokenBetween.append(rit)\n",
    "            \n",
    "        rit = rit.head\n",
    "\n",
    "\n",
    "    tokenBetween = list(sorted(tokenBetween, key=lambda x: x.i))\n",
    "    tokenBetweenStrict = list(sorted([x for x in tks if lword.i < x.i < rword.i], key=lambda x: x.i))\n",
    "    textBetween = ' '.join(\" \".join([str(x) for x in tokenBetween]).split())\n",
    "    textAfter = ' '.join(\" \".join([str(x) for x in [y for y in rword.doc if rword.i <= y.i < rword.i+5]]).split())\n",
    "    textBefore = ' '.join(\" \".join([str(x) for x in [y for y in rword.doc if lword.i-5 <= y.i < lword.i]]).split())\n",
    "\n",
    "    return tokenBetween, tokenBetweenStrict, textBetween, textBefore, textAfter\n",
    "\n",
    "dir2opp = {\"MIR_GENE\": \"GENE_MIR\", \"GENE_MIR\": \"MIR_GENE\"}\n",
    "idir2opp = {\"POS\": \"NEG\", \"NEG\": \"POS\"}\n",
    "rdir2opp = {\"DOWN\": \"UP\", \"UP\": \"DOWN\"}\n",
    "\n",
    "def testRelClassifier2(doc, mirword, geneword, relchecker, verbose):\n",
    "    isPassive = relchecker.checkPassive(doc, mirword, geneword, verbose=False)\n",
    "    compartment = relchecker.getCompartment(doc, mirword, geneword, verbose=False)\n",
    "    \n",
    "    if verbose:\n",
    "        print(\"passive?\", isPassive)\n",
    "        \n",
    "        \n",
    "    sdpRes = relchecker._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "        \n",
    "    stemTypes2Count = Counter()\n",
    "    \n",
    "    for x in sdpRes:\n",
    "        tknStem, matchStem, _ = getstem(x[0])\n",
    "        \n",
    "        if tknStem != \"NEU\":\n",
    "            stemTypes2Count[tknStem] += 1\n",
    "            \n",
    "    print(stemTypes2Count)\n",
    "        \n",
    "    if len(stemTypes2Count) > 0:\n",
    "        regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "        \n",
    "        if regStem == \"NEG\":\n",
    "            regStem = \"DOWN\"\n",
    "        elif regStem == \"POS\":\n",
    "            regStem = \"UP\"\n",
    "            \n",
    "        if stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"]:\n",
    "            regStem = \"DOWN\"\n",
    "            \n",
    "        idir = \"MIR_GENE\"\n",
    "        if isPassive and mirword.i < geneword.i:\n",
    "            idir = \"GENE_MIR\"\n",
    "            \n",
    "        #return {\"regulation_dir\": regStem, \"interaction_dir\": idir, \"passive\": isPassive, \"reg_detect\": \"sdp counts\"}\n",
    "        \n",
    "    targetWords = [\"targets of\", \"target of\", \"regulator of\", \"targeting\", \"direct regulator\"]\n",
    "        \n",
    "    if compartment != None:\n",
    "        tokenBetween = compartment\n",
    "        textBetween = ' '.join(\" \".join([str(x) for x in tokenBetween]).split())\n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"compartment gene mir pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"compartment gene mir neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment gene mir\"}\n",
    "            \n",
    "        elif mirword.i < geneword.i:\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment mir gene pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment mir gene neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment mir gene\"}\n",
    "        \n",
    "\n",
    "\n",
    "    if True:\n",
    "        \n",
    "        tokSrc = compartment if compartment != None else doc\n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "\n",
    "            tokenBetween, tokenBetweenStrict, textBetween, textBefore, textAfter = getTextBetween(tokSrc, geneword, mirword)\n",
    "\n",
    "            print(textBetween)\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between gene mir pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between gene mir neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"negative regulation of\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between gene mir reg corr\"}\n",
    "      \n",
    "            if any([x in textBefore for x in [\"binding site\", \"binding sites\", \"binding to\", \"binding directly to\"]]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between gene mir sites\"}\n",
    "            \n",
    "            \n",
    "    \n",
    "            if any([x in textBetween for x in targetWords]) or any([x in textBetween for x in [\"target\", \"targeted by\", \"binding efficiency\", \"binding site for\", \"recognizing\"]]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between gene mir\"}\n",
    "            \n",
    "\n",
    "            #stemTypes2Count = Counter()\n",
    "            #for x in tokenBetween:\n",
    "            #    tknStem, matchStem = getstem(x)\n",
    "            #    if tknStem != \"NEU\":\n",
    "            #        stemTypes2Count[tknStem] += 1\n",
    "                    \n",
    "            if False and len(stemTypes2Count) > 0:\n",
    "                regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                if regStem == \"NEG\":\n",
    "                    regStem = \"DOWN\"\n",
    "                elif regStem == \"POS\":\n",
    "                    regStem = \"UP\"\n",
    "                    \n",
    "                if stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"]:\n",
    "                    regStem = \"DOWN\"\n",
    "\n",
    "                idir = \"MIR_GENE\"\n",
    "                if isPassive and geneword.i < mirword.i:\n",
    "                    idir = \"GENE_MIR\"\n",
    "\n",
    "                return {\"regulation_dir\": regStem, \"interaction_dir\": idir, \"passive\": isPassive, \"reg_detect\": \"counts between gene mir\"}\n",
    "            \n",
    "        else:\n",
    "            #mirword.i < geneword.i\n",
    "            tokenBetween, tokenBetweenStrict, textBetween, textBefore, textAfter = getTextBetween(tokSrc, mirword, geneword)\n",
    "                        \n",
    "            print(textBetween)\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]) and stemTypes2Count.get(\"NEG\", 0) == 0:\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"negative regulation of\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene reg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"directly inhibits\", \"directly represses\", \"translational inhibition\" ]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords + [\"directly regulated\", \"its target gene\", \"binding site\"]]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"negatively regulated by\", \"negative regulation of\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between mir gene nrb\"}\n",
    "            \n",
    "            \n",
    "        if len(tokenBetween) > 0:\n",
    "                       \n",
    "            stemTypes2Count,mirgene2stem = countStems(tokenBetween, geneword, mirword)\n",
    "                    \n",
    "            if len(stemTypes2Count) > 0:\n",
    "                regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                \n",
    "                if len(stemTypes2Count) >= 2 and len(mirgene2stem[\"gene\"]) >0 and len(mirgene2stem[\"mir\"])>0 and mirgene2stem[\"gene\"].most_common(1)[0][0] == idir2opp[mirgene2stem[\"mir\"].most_common(1)[0][0]]:\n",
    "                    print(mirgene2stem)\n",
    "                    if geneword.i < mirword.i:\n",
    "                        if mirgene2stem[\"gene\"].most_common(1)[0][0] == \"POS\":\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                        else:\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                    else:\n",
    "                                                \n",
    "                        if mirgene2stem[\"mir\"].most_common(1)[0][0] == \"NEG\":\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                                                        \n",
    "                        else:\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                            \n",
    "                        if \"after\" in textBetween:\n",
    "                            regDir = dir2opp[regDir]\n",
    "                            \n",
    "                    if isPassive:\n",
    "                        regDir = dir2opp[regDir]\n",
    "                        regStem = rdir2opp[regStem]\n",
    "                        \n",
    "                    return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts opp\"}\n",
    "                    \n",
    "                    \n",
    "                else:\n",
    "                    stemTypes2Count,mirgene2stem = countStems(tokenBetween, geneword, mirword, keepNEU=True)\n",
    "                    print(\"counts between 1\", stemTypes2Count)\n",
    "                    regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                    \n",
    "                    if stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"] == 1 and stemTypes2Count[\"NEU\"] <= 3:\n",
    "                        regDir = \"MIR_GENE\"\n",
    "                        regStem = \"DOWN\"\n",
    "                        \n",
    "                        if any(x in textBetween for x in [\"after\"]):\n",
    "                            regDir = dir2opp[regDir]\n",
    "                        \n",
    "                        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts between equal\"}\n",
    "                    \n",
    "                    if stemTypes2Count[\"POS\"] == 2 and stemTypes2Count[\"NEG\"] == 1 and stemTypes2Count[\"NEU\"] <= 2:\n",
    "                        regDir = \"MIR_GENE\"\n",
    "                        regStem = \"DOWN\"\n",
    "                        \n",
    "                        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts between alternating\"}\n",
    "                    \n",
    "                    if regStem == \"NEU\":\n",
    "                        if stemTypes2Count[\"POS\"] > 0 and stemTypes2Count[\"POS\"] > stemTypes2Count[\"NEG\"]:\n",
    "                            regStem = \"POS\"\n",
    "                            \n",
    "                        elif stemTypes2Count[\"NEG\"] > 0 and stemTypes2Count[\"NEG\"] > stemTypes2Count[\"POS\"]:\n",
    "                            regStem = \"NEG\"\n",
    "                        \n",
    "                    \n",
    "                    #regStem = \"NEU\"\n",
    "\n",
    "                    if mirword.i < geneword.i:\n",
    "                        regDir = \"MIR_GENE\"\n",
    "                    else:\n",
    "                        regDir = \"GENE_MIR\"\n",
    "                        \n",
    "                    if mirword.i < geneword.i and \"-stressed\" in textBetween:\n",
    "                        regDir = \"GENE_MIR\"\n",
    "                        \n",
    "                    if mirword.i < geneword.i and str(doc[geneword.i-1]) in [\"by\", \"with\"]:\n",
    "                        regDir = \"GENE_MIR\"\n",
    "\n",
    "                        stemTypes2Count,mirgene2stem = countStems(tokenBetweenStrict, geneword, mirword)\n",
    "\n",
    "                        if len(stemTypes2Count) > 0:\n",
    "                            regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                        #else:\n",
    "                        #    regStem = \"NEU\"\n",
    "                        print(\"counts between 2\", stemTypes2Count, regStem)\n",
    "                            \n",
    "                    elif geneword.i < mirword.i and any([str(doc[mirword.i-j]) in [\"by\", \"with\"] for j in [1,2]]):\n",
    "                        regDir = \"MIR_GENE\"\n",
    "\n",
    "                        stemTypes2Count,mirgene2stem = countStems(tokenBetweenStrict, geneword, mirword)\n",
    "                        if len(stemTypes2Count) > 0:\n",
    "                            regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                        #else:\n",
    "                        #    regStem = \"NEU\"\n",
    "                        print(\"counts between 2\", stemTypes2Count, regStem)\n",
    "                            \n",
    "                    elif isPassive and geneword.i < mirword.i:\n",
    "                        regDir = dir2opp[regDir]\n",
    "\n",
    "                    if stemTypes2Count[\"POS\"] > 0 and stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"]:\n",
    "                        regStem = \"NEU\"                       \n",
    "                        \n",
    "                    \n",
    "                        #regStem = idir2opp[regStem]\n",
    "                    \n",
    "                    if geneword.i < mirword.i:\n",
    "                        if str(geneword).endswith(\"-regulating\"):\n",
    "                            regStem = \"NEU\"\n",
    "                            \n",
    "                    if mirword.i < geneword.i and \"after the addition\" in textBetween:\n",
    "                        regStem = \"DOWN\"\n",
    "                        regDir = \"GENE_MIR\"\n",
    "                            \n",
    "                        \n",
    "                    if regStem == \"NEG\":\n",
    "                        regStem = \"DOWN\"\n",
    "                    elif regStem == \"POS\":\n",
    "                        regStem = \"UP\"\n",
    "\n",
    "                        \n",
    "                    \n",
    "                    return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts between\"}\n",
    "    \n",
    "        \n",
    "    regDir = \"MIR_GENE\"\n",
    "    regStem = \"NEU\"\n",
    "    \n",
    "    regeluateWords = [\"mediated\",]\n",
    "    \n",
    "    if mirword.i < geneword.i and \"mediated\" in str(mirword):\n",
    "        regStem = \"DOWN\"\n",
    "        \n",
    "    if geneword.i < mirword.i and (\"mediated\" in str(mirword) or any([str(doc[mirword.i-j]) in [\"mediated\",] for j in [1]])):\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "        \n",
    "        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"return mediated\"}\n",
    "        \n",
    "    if mirword.i<geneword.i and any([str(doc[geneword.i-j]) in [\"by\", \"with\"] for j in [1,2,3]]):\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "        \n",
    "        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"return m g by\"}\n",
    "        \n",
    "    if geneword.i < mirword.i:\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"GENE_MIR\"\n",
    "    \n",
    "    if geneword.i < mirword.i and any([str(doc[mirword.i-j]) in [\"by\", \"with\"] for j in [1,2,3]]):\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "        \n",
    "    if mirword.i < geneword.i and str(doc[mirword.i+1]) == \"target\":\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "    \n",
    "    #if mirword.i < geneword.i:\n",
    "    #    regDir = \"MIR_GENE\"\n",
    "    #else:\n",
    "    #    regDir = \"GENE_MIR\"\n",
    "    \n",
    "    #if isPassive:\n",
    "    #    if mirword.i < geneword.i:\n",
    "    #        regDir = \"GENE_MIR\"\n",
    "    #    else:\n",
    "    #        regDir = \"MIR_GENE\"\n",
    "        \n",
    "    return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"return\"}\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getstem(tkn):\n",
    "    tknstr = str(tkn).lower()\n",
    "    for sidx, s in enumerate(stem2class):\n",
    "        if tknstr.startswith(s):\n",
    "            \n",
    "            if tknstr in [\"enhancer\",\"inhibitor\"]:\n",
    "                continue\n",
    "\n",
    "            tokenStemClass = stem2class[s]\n",
    "            \n",
    "            if tokenStemClass == \"NEU\":\n",
    "                for i in range(1,2):\n",
    "\n",
    "                    if tkn.i-i<0:\n",
    "                        continue\n",
    "\n",
    "                    #tknPre = str(tkn.doc[tkn.i-i])\n",
    "                    #if tknPre.startswith((\"high\", \"elevat\", \"increas\")):\n",
    "                    #    tokenStemClass = \"POS\"\n",
    "\n",
    "                    #if tknPre.startswith((\"low\", \"decreas\")):\n",
    "                    #    tokenStemClass = \"NEG\"\n",
    "                                                \n",
    "            return tokenStemClass, s, True\n",
    "        \n",
    "        elif tknstr.endswith((\"-stressed\")):\n",
    "            return \"POS\", s, True\n",
    "        \n",
    "    return \"NEU\", str(tkn), False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "allStems, allRels, stem2class = loadStems()\n",
    "\n",
    "print(stem2class[\"underexpress\"])\n",
    "print(\"underexpression\".startswith(\"underexpress\"))\n",
    "\n",
    "doct = nlp(\"expressed , miR-34a , was studied further because sequence analysis suggested a likely interaction with the 3'-untranslated region of bcl2 mRNA\")\n",
    "\n",
    "toks = [x for x in doct]\n",
    "\n",
    "for tok in toks:\n",
    "    print(tok, getstem(tok))\n",
    "\n",
    "countStems(toks, toks[0], toks[-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "runCheck(-1, \"TRAIN\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "runCheck(-1, \"TEST\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "def testRelClassifier(doc, mirword, geneword, relchecker, verbose):\n",
    "    \n",
    "    if verbose:\n",
    "        print()\n",
    "        print()\n",
    "        print()\n",
    "        print(\"Rel Classifier\")\n",
    "    \n",
    "        print(doc.text_with_ws)\n",
    "    \n",
    "    foundStems = set()\n",
    "    sdpRes = relchecker._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "    if verbose:\n",
    "        for x in sdpRes:\n",
    "            print(x)\n",
    "        \n",
    "        \n",
    "    sdpTokens = [x[0] for x in sdpRes]\n",
    "    #print(sdpRes)\n",
    "    \n",
    "    sdpVerbRels = set()\n",
    "\n",
    "    for sidx, x in enumerate(sdpRes):\n",
    "        if x[1] == \"VERB\":\n",
    "            for s in allStems:\n",
    "                if s in str(x[0]):\n",
    "                    sdpVerbRels.add((s, sidx, stem2class[s]))\n",
    "                    if verbose:\n",
    "                        print(\"sdpVerbRels\", x[0], s, sidx, stem2class[s])\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "            \n",
    "    docRels = set()\n",
    "    for tkn in doc:\n",
    "\n",
    "        if tkn in sdpTokens:\n",
    "            continue\n",
    "\n",
    "        for s in allStems:\n",
    "            if str(tkn).lower().startswith(s):\n",
    "                docRels.add((s, sidx, stem2class[s]))\n",
    "                if verbose:    \n",
    "                    print(\"docRels\", tkn, s, sidx, stem2class[s])\n",
    "            \n",
    "            \n",
    "    mirAncests = [x for x in mirword.ancestors]\n",
    "    geneAncests = [x for x in geneword.ancestors if not x in mirAncests]\n",
    "    \n",
    "    if len(mirAncests) > 1:\n",
    "        useMirAncests = mirAncests[0:1]\n",
    "    else:\n",
    "        useMirAncests = mirAncests\n",
    "        \n",
    "    if len(geneAncests) > 1:\n",
    "        useGeneAncests = geneAncests[0:1]\n",
    "    else:\n",
    "        useGeneAncests = geneAncests\n",
    "        \n",
    "    useGeneAncests = [x for x in useGeneAncests if not x in useMirAncests]\n",
    "    \n",
    "    if verbose:\n",
    "        print(\"mirc\", mirAncests)\n",
    "        print(\"genec\", geneAncests)\n",
    "    \n",
    "    invCor = False\n",
    "    regdir = \"NEU\"\n",
    "    intdir = \"MIR_GENE\"\n",
    "    isPassive = relchecker.checkPassive(doc, mirword, geneword, verbose=False)\n",
    "    if verbose:\n",
    "        print(\"passive?\", isPassive)\n",
    "    \n",
    "    ancestHandled = False\n",
    "    #if len(mirAncests) > 0 and len(geneAncests) > 0:\n",
    "    if True:\n",
    "        mirRels = []\n",
    "        geneRels = []\n",
    "        \n",
    "        mirRels = getRelsForTokens(useMirAncests, \"ancestRels mir\", verbose)     \n",
    "        neuMirs = [x for x in mirRels if x[2] in [\"NEU\"]]\n",
    "        #if len(neuMirs) > 0 or len(mirRels) == 0:\n",
    "        #    mirRels = getRelsForTokens(mirAncests, \"ancestRels all mir\", verbose)\n",
    "            \n",
    "        geneRels = getRelsForTokens(useGeneAncests, \"ancestRels gene\", verbose)       \n",
    "        neuGenes = [x for x in geneRels if x[2] in [\"NEU\"]]\n",
    "                \n",
    "        if False and ((len(neuGenes) == len(geneRels)) or len(geneRels) == 0):\n",
    "            geneRels = getRelsForTokens([x for x in geneAncests if not x in useMirAncests], \"ancestRels all gene\", verbose)\n",
    "            neuGenes = [x for x in geneRels if x[2] in [\"NEU\"]]\n",
    "            \n",
    "            if (len(neuGenes) == len(geneRels)) or len(geneRels) == 0:\n",
    "                \n",
    "                if mirword.i < geneword.i:\n",
    "                    tks = [x for x in doc if geneword.i -8 < x.i < geneword.i if not x in useMirAncests]\n",
    "                else:\n",
    "                    tks = [x for x in doc if geneword.i < x.i < geneword.i+5 if not x in useMirAncests]\n",
    "                \n",
    "                geneRels = getRelsForTokens(tks, \"ancestRels tks gene\", verbose)\n",
    "                \n",
    "                                        \n",
    "        posRegs = [x for x in mirRels+geneRels if x[2] in [\"POS\"]]\n",
    "        negRegs = [x for x in mirRels+geneRels if x[2] in [\"NEG\"]]\n",
    "        \n",
    "        mirPos = len([x for x in mirRels if x[2] in [\"POS\"]])\n",
    "        mirNeu = [x for x in mirRels if x[2] in [\"NEU\"]]\n",
    "        mirNeg = len([x for x in mirRels if x[2] in [\"NEG\"]])\n",
    "        \n",
    "        genePos = len([x for x in geneRels if x[2] in [\"POS\"]])\n",
    "        geneNeu = [x for x in geneRels if x[2] in [\"NEU\"]]\n",
    "        geneNeg = len([x for x in geneRels if x[2] in [\"NEG\"]])\n",
    "    \n",
    "        dirReg = (mirPos - geneNeg) + (genePos-mirNeg)\n",
    "        \n",
    "        invCor = False\n",
    "        \n",
    "        if verbose:\n",
    "            print(mirword, geneword)\n",
    "            print(\"ancestRel counts\", mirPos, mirNeg, genePos, geneNeg)\n",
    "                          \n",
    "            \n",
    "        nregdir, ninvcor = evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg)\n",
    "        if verbose:\n",
    "            print(\"evalcount\")\n",
    "            print(nregdir, ninvcor)\n",
    "        \n",
    "        if nregdir != None:\n",
    "            regdir = nregdir\n",
    "            invcor = ninvcor\n",
    "            \n",
    "            ancestHandled = True\n",
    "            \n",
    "            if verbose:\n",
    "                print(\"accepted by evalcount\")\n",
    "            \n",
    "        else:\n",
    "            lookBehind = 8\n",
    "\n",
    "                                \n",
    "            if len(mirNeu) > 0 or len(geneNeu) > 0:\n",
    "                if verbose:\n",
    "                    print(\"ancestRel neu part\")\n",
    "\n",
    "                if mirword.i < geneword.i:\n",
    "                    for x in mirNeu:\n",
    "                        if x[0].startswith((\"correlat\", \"affect\")):\n",
    "\n",
    "                            for j in range(1,lookBehind):\n",
    "                                if str(doc[x[4].i-j]).startswith((\"posit\", \"high\")):\n",
    "                                    mirPos += 1\n",
    "                                    \n",
    "                                elif str(doc[x[4].i-j]).startswith((\"negat\", \"low\")):\n",
    "                                    mirNeg += 1\n",
    "                else:\n",
    "                    for x in geneNeu:\n",
    "                        if x[0].startswith((\"correlat\")):\n",
    "                            \n",
    "                            for j in range(1,lookBehind):\n",
    "                                if str(doc[x[4].i-j]).startswith((\"posit\", \"high\")):\n",
    "                                    genePos += 1\n",
    "                                elif str(doc[x[4].i-j]).startswith((\"negat\", \"low\")):\n",
    "                                    geneNeg += 1    \n",
    "            \n",
    "                                \n",
    "            if not ancestHandled:    \n",
    "                \n",
    "                                \n",
    "                    \n",
    "                if verbose:\n",
    "                    print(\"ancestRel counts stem check\", mirPos, mirNeg, genePos, geneNeg)\n",
    "                    \n",
    "                nregdir, ninvcor = evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg)\n",
    "                if nregdir != None:\n",
    "                    regdir = nregdir\n",
    "                    invcor = ninvcor\n",
    "                    ancestHandled = True\n",
    "                    \n",
    "                    if verbose:\n",
    "                        print(\"accept after before stem check\")\n",
    "                    \n",
    "                if not ancestHandled:\n",
    "                    for j in range(1,lookBehind):\n",
    "                        \n",
    "                        if mirword.i-j < 0:\n",
    "                            continue\n",
    "                        \n",
    "                        jtk = doc[mirword.i-j]\n",
    "                        if verbose:\n",
    "                            print(\"word check mir\", j, jtk)\n",
    "                        pos, neg = scoreWord(jtk)\n",
    "                        \n",
    "                        mirPos += pos\n",
    "                        mirNeg += neg\n",
    "                            \n",
    "                    for j in range(1,lookBehind):\n",
    "                        \n",
    "                        if geneword.i-j < 0:\n",
    "                            continue\n",
    "                        \n",
    "                        jtk = doc[geneword.i-j]\n",
    "                        if verbose:\n",
    "                            print(\"word check gene\", j, jtk)\n",
    "                            \n",
    "                        pos, neg = scoreWord(jtk)\n",
    "                        \n",
    "                        genePos += pos\n",
    "                        geneNeg += neg\n",
    "                            \n",
    "                            \n",
    "                if verbose:\n",
    "                    print(\"ancestRel counts word check\", mirPos, mirNeg, genePos, geneNeg)\n",
    "                \n",
    "                nregdir, ninvcor = evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg)\n",
    "                if nregdir != None:\n",
    "                    regdir = nregdir\n",
    "                    invcor = ninvcor\n",
    "                    ancestHandled = True\n",
    "                    \n",
    "                    if verbose:\n",
    "                        print(\"accept after before word check\")\n",
    "                    \n",
    "           \n",
    "\n",
    "                \n",
    "                \n",
    "        if verbose:\n",
    "            print(\"ancestRels\", dirReg, invCor)\n",
    "            \n",
    "    if ancestHandled:\n",
    "        \n",
    "        if isPassive:\n",
    "            intdir = \"GENE_MIR\"\n",
    "            \n",
    "    \n",
    "    if not ancestHandled:\n",
    "        \n",
    "        if verbose:\n",
    "            print(\"ancestHandled part\")\n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "            lword = geneword\n",
    "            rword = mirword\n",
    "            useTks = [x for x in doc if geneword.i < x.i < mirword.i]\n",
    "        else:\n",
    "            lword = mirword\n",
    "            rword = geneword\n",
    "            useTks = [x for x in doc if mirword.i < x.i < geneword.i]\n",
    "            \n",
    "        if verbose:\n",
    "            print(useTks)\n",
    "        \n",
    "        betRels = getRelsForTokens(useTks, \"ancestRels useTks\", verbose)\n",
    "        \n",
    "        if verbose:\n",
    "            print(betRels)\n",
    "        \n",
    "        betPos = len([x for x in betRels if x[2] in [\"POS\"]])\n",
    "        betNeg = len([x for x in betRels if x[2] in [\"NEG\"]])\n",
    "        \n",
    "        if betPos + betNeg == 1:\n",
    "            ancestHandled = True\n",
    "            \n",
    "            if verbose:\n",
    "                print(\"ancest bet handled\", mirword.i, geneword.i, isPassive)\n",
    "            \n",
    "            if geneword.i < mirword.i:\n",
    "                \n",
    "                if isPassive:\n",
    "                    intdir = \"MIR_GENE\"\n",
    "                    regdir = \"DOWN\" if betNeg > 0 else \"UP\"\n",
    "                    \n",
    "                else:\n",
    "                    intdir = \"GENE_MIR\"\n",
    "                    regdir = \"UP\" if betPos > 0 else \"DOWN\"\n",
    "                    \n",
    "            else:\n",
    "                \n",
    "                if isPassive:\n",
    "                    intdir = \"GENE_MIR\"\n",
    "                    regdir = \"UP\" if betPos > 0 else \"DOWN\"\n",
    "                    \n",
    "                else:\n",
    "                    intdir = \"MIR_GENE\"\n",
    "                    regdir = \"UP\" if betPos > 0 else \"DOWN\"\n",
    "                    \n",
    "        else:\n",
    "            \n",
    "            isMediated = False\n",
    "            \n",
    "            for x in [lword] + useTks[0:2]:\n",
    "                for y in [\"mediated\", \"-mediated\"]:\n",
    "                    \n",
    "                    if y in str(x):\n",
    "                        isMediated = True\n",
    "                        break\n",
    "                if isMediated:\n",
    "                    break\n",
    "                                    \n",
    "            if isMediated:\n",
    "                \n",
    "                ancestHandled = True\n",
    "                \n",
    "                if verbose:\n",
    "                    print(\"ancest mediated handled\")\n",
    "                \n",
    "                if geneword.i < mirword.i:\n",
    "                    intdir = \"GENE_MIR\"\n",
    "                else:\n",
    "                    intdir = \"MIR_GENE\"\n",
    "                    \n",
    "                if betPos > betNeg:\n",
    "                    regdir = \"UP\"\n",
    "                elif betNeg > betPos:\n",
    "                    regdir = \"DOWN\"\n",
    "                        \n",
    "            \n",
    "            \n",
    "            \n",
    "\n",
    "    #classCounter = Counter()\n",
    "    #if len(classCounter) > 0:\n",
    "    #    regdir = classCounter.most_common(1)[0][0]\n",
    "    #else:\n",
    "    #    regdir = \"NEU\"\n",
    "        \n",
    "    #if invCor:\n",
    "    #    regdir = \"DOWN\"       \n",
    "              \n",
    "    \n",
    "    if not ancestHandled:\n",
    "        \n",
    "        if verbose:\n",
    "            print(\"in final resolution\")\n",
    "        \n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "        \n",
    "            textBetween = \" \".join([str(x) for x in doc if geneword.i < x.i < mirword.i])\n",
    "            textBetween = ' '.join(textBetween.split())\n",
    "            \n",
    "            if verbose:\n",
    "                print(textBetween)\n",
    "            \n",
    "\n",
    "            if len(textBetween) < 10 and any([x in textBetween for x in [\"by\"]]):\n",
    "                sdpRels = set()\n",
    "                for sidx,x  in enumerate(sdpRes):\n",
    "                    for s in allStems:\n",
    "                        if str(x[0]).lower().startswith(s):\n",
    "                            sdpRels.add((s, sidx, stem2class[s]))\n",
    "                            if verbose:\n",
    "                                print(\"sdpRels\", x[0], s, sidx, stem2class[s])\n",
    "\n",
    "                posRegs = [x for x in sdpRels if x[2] in [\"POS\"]]\n",
    "                negRegs = [x for x in sdpRels if x[2] in [\"NEG\"]]\n",
    "\n",
    "                dirReg = len(posRegs)-len(negRegs)\n",
    "\n",
    "                if dirReg > 0:\n",
    "                    regdir  = \"UP\"\n",
    "                elif dirReg < 0:\n",
    "                    regdir = \"DOWN\"\n",
    "                else:\n",
    "                    regdir = \"NEU\"\n",
    "                \n",
    "                intdir = \"MIR_GENE\"\n",
    "\n",
    "            elif any([x in textBetween for x in [\"target of\", \"target for\", \"targeting\"]]):\n",
    "                intdir = \"MIR_GENE\"\n",
    "                regdir = \"NEU\"\n",
    "\n",
    "            else:                \n",
    "                intdir = \"GENE_MIR\"\n",
    "    \n",
    "        elif mirword.i < geneword.i:\n",
    "            textBetween = \" \".join([str(x) for x in doc if geneword.i-5 < x.i < geneword.i])\n",
    "\n",
    "            if any([x in textBetween for x in [\"by\"]]):\n",
    "                intdir = \"GENE_MIR\"\n",
    "                regdir = \"UP\"\n",
    "        \n",
    "    \n",
    "    regstems = foundStems\n",
    "    \n",
    "    #displacy.render(doc, style=\"dep\")\n",
    "    \n",
    "    if verbose:\n",
    "        print((intdir, regdir, \"NA\", isPassive))\n",
    "        \n",
    "    return {\"regulation_dir\": regdir, \"interaction_dir\": intdir, \"regulation_keys\": regstems}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
