{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "import os, sys\n",
    "sys.path.insert(0, \"/mnt/d/dev/git/miRExplore/python/\")\n",
    "\n",
    "\n",
    "import scispacy\n",
    "import spacy\n",
    "from spacy import displacy\n",
    "\n",
    "nlp = spacy.load('/mnt/d/spacy/models/en_core_sci_lg-0.2.4/en_core_sci_lg/en_core_sci_lg-0.2.4/')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp_weblg = spacy.load('/mnt/d/spacy/models/en_core_web_lg-2.2.0/en_core_web_lg/en_core_web_lg-2.2.0/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "accuracy.json  meta.json  ner  parser  tagger  tokenizer  vocab\n"
     ]
    }
   ],
   "source": [
    "!ls /mnt/d/spacy/models/en_core_web_lg-2.2.0/en_core_web_lg/en_core_web_lg-2.2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import textmining.MirGeneRelCheck\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def analyse(testCase, dist=80):\n",
    "    relCheck = textmining.MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "    doc = nlp(testCase[0])\n",
    "    \n",
    "    lWord = doc[testCase[1]]\n",
    "    rWord = doc[testCase[2]]\n",
    "\n",
    "    for t in doc:\n",
    "        print(t.idx, t.i, t, t.pos_, t.dep_, t.head, t.head.idx)\n",
    "\n",
    "    print(lWord, rWord)\n",
    "\n",
    "    if not \"mir\" in str(lWord).lower() and not \"let\" in str(lWord).lower():\n",
    "        tmp = lWord\n",
    "        lWord = rWord\n",
    "        rWord = tmp\n",
    "        \n",
    "    print(lWord, rWord)\n",
    "    \n",
    "    checkResults = relCheck.checkRelation(doc, lWord, rWord, verbose=True)\n",
    "    print(checkResults)\n",
    "    \n",
    "    displacy.render(doc, style=\"dep\", options={\"compact\": True, \"distance\": dist})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 Functional ADJ amod association 11\n",
      "11 1 association NOUN nsubjpass established 72\n",
      "23 2 between ADP case miR 31\n",
      "31 3 miR PROPN nmod association 11\n",
      "35 4 124a NOUN dep miR 31\n",
      "40 5 and CCONJ cc 124a 35\n",
      "44 6 CDK6 PROPN conj 124a 35\n",
      "49 7 in ADP case medulloblastoma 52\n",
      "52 8 medulloblastoma NOUN nmod association 11\n",
      "68 9 was AUX auxpass established 72\n",
      "72 10 established VERB ROOT established 72\n",
      "84 11 using VERB xcomp established 72\n",
      "90 12 luciferase NOUN compound assays 101\n",
      "101 13 assays NOUN dobj using 84\n",
      "107 14 . PUNCT punct established 72\n",
      "124a CDK6\n",
      "CDK6 124a\n",
      "Conjuncts [CDK6, 124a]\n",
      "neighbour CDK6 in\n",
      "neighbour CDK6 in\n",
      "Conjunction before fc 124a\n",
      "Conjuncts [124a, CDK6]\n",
      "neighbour 124a and\n",
      "neighbour 124a and\n",
      "Conjunction before fc CDK6\n",
      "Conjunctions\n",
      "124a [124a, CDK6]\n",
      "achildren {CDK6}\n",
      "achildren False\n",
      "SDP sent: Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\n",
      "SDP     : [(124a, 'NOUN', 'dep', False), (established, 'VERB', 'ROOT', False), (association, 'NOUN', 'nsubjpass', False), (miR, 'PROPN', 'nmod', False), (124a, 'NOUN', 'dep', False)]\n",
      "deepcheck connection rule noun found using luciferase [using, luciferase, assays]\n",
      "0 [Functional, association, between, miR, 124a, and, CDK6, in, medulloblastoma, was, established] 11 0 Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\n",
      "1 [using, luciferase, assays] 3 84 using luciferase assays.\n",
      "Conjuncts [CDK6, 124a]\n",
      "neighbour CDK6 in\n",
      "neighbour CDK6 in\n",
      "Conjunction before fc 124a\n",
      "Conjuncts [124a, CDK6]\n",
      "neighbour 124a and\n",
      "neighbour 124a and\n",
      "Conjunction before fc CDK6\n",
      "(False, {'conj': False, 'sdp': True, 'compartment': True, 'context': True, 'passive': True, 'negation': False})\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"449c8ba6604048b7b084898c1e3ec2c3-0\" class=\"displacy\" width=\"1170\" height=\"297.0\" direction=\"ltr\" style=\"max-width: none; height: 297.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">Functional</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"130\">association</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"130\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"210\">between</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"210\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"290\">miR</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"290\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"370\">124a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"370\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"450\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"450\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"530\">CDK6</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"530\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"610\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"610\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"690\">medulloblastoma</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"690\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"770\">was</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"770\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"850\">established</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"850\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"930\">using</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"930\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1010\">luciferase</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1010\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1090\">assays.</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1090\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-0\" stroke-width=\"2px\" d=\"M62,162.0 62,148.66666666666666 121.0,148.66666666666666 121.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M62,164.0 L58,156.0 66,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-1\" stroke-width=\"2px\" d=\"M142,162.0 142,108.66666666666666 850.0,108.66666666666666 850.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubjpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M142,164.0 L138,156.0 146,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-2\" stroke-width=\"2px\" d=\"M222,162.0 222,148.66666666666666 281.0,148.66666666666666 281.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M222,164.0 L218,156.0 226,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-3\" stroke-width=\"2px\" d=\"M142,162.0 142,135.33333333333334 284.0,135.33333333333334 284.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M284.0,164.0 L288.0,156.0 280.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-4\" stroke-width=\"2px\" d=\"M302,162.0 302,148.66666666666666 361.0,148.66666666666666 361.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dep</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M361.0,164.0 L365.0,156.0 357.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-5\" stroke-width=\"2px\" d=\"M382,162.0 382,148.66666666666666 441.0,148.66666666666666 441.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M441.0,164.0 L445.0,156.0 437.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-6\" stroke-width=\"2px\" d=\"M382,162.0 382,135.33333333333334 524.0,135.33333333333334 524.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M524.0,164.0 L528.0,156.0 520.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-7\" stroke-width=\"2px\" d=\"M622,162.0 622,148.66666666666666 681.0,148.66666666666666 681.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M622,164.0 L618,156.0 626,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-8\" stroke-width=\"2px\" d=\"M142,162.0 142,122.0 687.0,122.0 687.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M687.0,164.0 L691.0,156.0 683.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-9\" stroke-width=\"2px\" d=\"M782,162.0 782,148.66666666666666 841.0,148.66666666666666 841.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">auxpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M782,164.0 L778,156.0 786,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-10\" stroke-width=\"2px\" d=\"M862,162.0 862,148.66666666666666 921.0,148.66666666666666 921.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">xcomp</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M921.0,164.0 L925.0,156.0 917.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-11\" stroke-width=\"2px\" d=\"M1022,162.0 1022,148.66666666666666 1081.0,148.66666666666666 1081.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1022,164.0 L1018,156.0 1026,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-449c8ba6604048b7b084898c1e3ec2c3-0-12\" stroke-width=\"2px\" d=\"M942,162.0 942,135.33333333333334 1084.0,135.33333333333334 1084.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-449c8ba6604048b7b084898c1e3ec2c3-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1084.0,164.0 L1088.0,156.0 1080.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testcase = (u\"Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\", 4, 6)\n",
    "\n",
    "analyse(testcase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 Overexpression PROPN nsubj reduces 50\n",
      "15 1 of ADP prep Overexpression 0\n",
      "18 2 mir-7 PROPN pobj of 15\n",
      "24 3 and CCONJ cc mir-7 18\n",
      "28 4 mir-153 NOUN conj mir-7 18\n",
      "36 5 significantly ADV advmod reduces 50\n",
      "50 6 reduces VERB ROOT reduces 50\n",
      "58 7 endogenous ADJ amod levels 85\n",
      "69 8 alpha NOUN compound synuclein 75\n",
      "74 9 - PUNCT punct synuclein 75\n",
      "75 10 synuclein NOUN compound levels 85\n",
      "85 11 levels NOUN dobj reduces 50\n",
      "91 12 , PUNCT punct reduces 50\n",
      "93 13 whereas SCONJ mark enhances 133\n",
      "101 14 inhibition NOUN nsubj enhances 133\n",
      "112 15 of ADP prep inhibition 101\n",
      "115 16 mir-7 PROPN dep and 121\n",
      "121 17 and CCONJ cc mir-153 125\n",
      "125 18 mir-153 PROPN pobj of 112\n",
      "133 19 enhances VERB advcl reduces 50\n",
      "142 20 translation NOUN dobj enhances 133\n",
      "154 21 of ADP prep translation 142\n",
      "157 22 a DET det luciferase 159\n",
      "159 23 luciferase NOUN pobj of 154\n",
      "170 24 construct NOUN advcl enhances 133\n",
      "180 25 bearing VERB xcomp construct 170\n",
      "188 26 the DET det s 198\n",
      "192 27 alpha NOUN compound s 198\n",
      "197 28 - PUNCT punct s 198\n",
      "198 29 s PART dobj bearing 180\n",
      "reduces synuclein\n",
      "synuclein reduces\n",
      "Conjuncts [mir-153, mir-7]\n",
      "neighbour mir-153 significantly\n",
      "neighbour mir-153 significantly\n",
      "Conjunction before fc mir-7\n",
      "Conjuncts [mir-7, mir-153]\n",
      "neighbour mir-7 and\n",
      "neighbour mir-7 and\n",
      "Conjunction before fc mir-153\n",
      "Conjunctions\n",
      "mir-7 [mir-7, mir-153]\n",
      "SDP sent: Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\n",
      "SDP     : [(reduces, 'VERB', 'ROOT', False), (reduces, 'VERB', 'ROOT', False)]\n",
      "deepcheck connection rule noun found enhances inhibition [whereas, inhibition, of, mir-7, and, mir-153, enhances, translation, of, a, luciferase, construct, bearing, the, alpha, -, s]\n",
      "0 [Overexpression, of, mir-7, and, mir-153, significantly, reduces, endogenous, alpha, -, synuclein, levels] 12 0 Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\n",
      "1 [inhibition, of, mir-7, and, mir-153, enhances, translation, of, a, luciferase, construct] 11 101 inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\n",
      "2 [bearing, the, alpha, -, s] 5 180 bearing the alpha-s\n",
      "Conjuncts [mir-153, mir-7]\n",
      "neighbour mir-153 significantly\n",
      "neighbour mir-153 significantly\n",
      "Conjunction before fc mir-7\n",
      "Conjuncts [mir-7, mir-153]\n",
      "neighbour mir-7 and\n",
      "neighbour mir-7 and\n",
      "Conjunction before fc mir-153\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'passive': False, 'negation': False})\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"cff33efab0ff426589071bcad47aaf58-0\" class=\"displacy\" width=\"4775\" height=\"749.5\" direction=\"ltr\" style=\"max-width: none; height: 749.5px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">Overexpression</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"225\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"225\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"400\">mir-7</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"400\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"575\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"575\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"750\">mir-153</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"750\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"925\">significantly</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"925\">ADV</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1100\">reduces</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1100\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1275\">endogenous</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1275\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1450\">alpha-</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1450\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1625\">synuclein</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1625\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1800\">levels,</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1800\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1975\">whereas</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1975\">SCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2150\">inhibition</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2150\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2325\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2325\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2500\">mir-7</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2500\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2675\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2675\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2850\">mir-153</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2850\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3025\">enhances</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3025\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3200\">translation</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3200\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3375\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3375\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3550\">a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3550\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3725\">luciferase</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3725\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3900\">construct</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3900\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"4075\">bearing</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"4075\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"4250\">the</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"4250\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"4425\">alpha-</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"4425\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"659.5\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"4600\">s</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"4600\">PART</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-0\" stroke-width=\"2px\" d=\"M70,614.5 C70,89.5 1095.0,89.5 1095.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M70,616.5 L62,604.5 78,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-1\" stroke-width=\"2px\" d=\"M70,614.5 C70,527.0 195.0,527.0 195.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">prep</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M195.0,616.5 L203.0,604.5 187.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-2\" stroke-width=\"2px\" d=\"M245,614.5 C245,527.0 370.0,527.0 370.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">pobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M370.0,616.5 L378.0,604.5 362.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-3\" stroke-width=\"2px\" d=\"M420,614.5 C420,527.0 545.0,527.0 545.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M545.0,616.5 L553.0,604.5 537.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-4\" stroke-width=\"2px\" d=\"M420,614.5 C420,439.5 725.0,439.5 725.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M725.0,616.5 L733.0,604.5 717.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-5\" stroke-width=\"2px\" d=\"M945,614.5 C945,527.0 1070.0,527.0 1070.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">advmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M945,616.5 L937,604.5 953,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-6\" stroke-width=\"2px\" d=\"M1295,614.5 C1295,352.0 1780.0,352.0 1780.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1295,616.5 L1287,604.5 1303,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-7\" stroke-width=\"2px\" d=\"M1470,614.5 C1470,527.0 1595.0,527.0 1595.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1470,616.5 L1462,604.5 1478,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-8\" stroke-width=\"2px\" d=\"M1645,614.5 C1645,527.0 1770.0,527.0 1770.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1645,616.5 L1637,604.5 1653,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-9\" stroke-width=\"2px\" d=\"M1120,614.5 C1120,264.5 1785.0,264.5 1785.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1785.0,616.5 L1793.0,604.5 1777.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-10\" stroke-width=\"2px\" d=\"M1995,614.5 C1995,89.5 3020.0,89.5 3020.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">mark</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1995,616.5 L1987,604.5 2003,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-11\" stroke-width=\"2px\" d=\"M2170,614.5 C2170,177.0 3015.0,177.0 3015.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2170,616.5 L2162,604.5 2178,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-12\" stroke-width=\"2px\" d=\"M2170,614.5 C2170,527.0 2295.0,527.0 2295.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">prep</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2295.0,616.5 L2303.0,604.5 2287.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-13\" stroke-width=\"2px\" d=\"M2520,614.5 C2520,527.0 2645.0,527.0 2645.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-13\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dep</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2520,616.5 L2512,604.5 2528,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-14\" stroke-width=\"2px\" d=\"M2695,614.5 C2695,527.0 2820.0,527.0 2820.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-14\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2695,616.5 L2687,604.5 2703,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-15\" stroke-width=\"2px\" d=\"M2345,614.5 C2345,352.0 2830.0,352.0 2830.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-15\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">pobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2830.0,616.5 L2838.0,604.5 2822.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-16\" stroke-width=\"2px\" d=\"M1120,614.5 C1120,2.0 3025.0,2.0 3025.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-16\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">advcl</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3025.0,616.5 L3033.0,604.5 3017.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-17\" stroke-width=\"2px\" d=\"M3045,614.5 C3045,527.0 3170.0,527.0 3170.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-17\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3170.0,616.5 L3178.0,604.5 3162.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-18\" stroke-width=\"2px\" d=\"M3220,614.5 C3220,527.0 3345.0,527.0 3345.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-18\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">prep</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3345.0,616.5 L3353.0,604.5 3337.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-19\" stroke-width=\"2px\" d=\"M3570,614.5 C3570,527.0 3695.0,527.0 3695.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-19\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3570,616.5 L3562,604.5 3578,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-20\" stroke-width=\"2px\" d=\"M3395,614.5 C3395,439.5 3700.0,439.5 3700.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-20\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">pobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3700.0,616.5 L3708.0,604.5 3692.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-21\" stroke-width=\"2px\" d=\"M3045,614.5 C3045,177.0 3890.0,177.0 3890.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-21\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">advcl</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3890.0,616.5 L3898.0,604.5 3882.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-22\" stroke-width=\"2px\" d=\"M3920,614.5 C3920,527.0 4045.0,527.0 4045.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-22\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">xcomp</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M4045.0,616.5 L4053.0,604.5 4037.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-23\" stroke-width=\"2px\" d=\"M4270,614.5 C4270,439.5 4575.0,439.5 4575.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-23\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M4270,616.5 L4262,604.5 4278,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-24\" stroke-width=\"2px\" d=\"M4445,614.5 C4445,527.0 4570.0,527.0 4570.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-24\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M4445,616.5 L4437,604.5 4453,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-cff33efab0ff426589071bcad47aaf58-0-25\" stroke-width=\"2px\" d=\"M4095,614.5 C4095,352.0 4580.0,352.0 4580.0,614.5\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-cff33efab0ff426589071bcad47aaf58-0-25\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M4580.0,616.5 L4588.0,604.5 4572.0,604.5\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testcase = (u\"Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\", 6, 10)\n",
    "\n",
    "analyse(testcase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "doc = nlp(testcase[0])\n",
    "\n",
    "allTokens = {t.i: t for t in doc}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[alpha-s]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[x for x in allTokens[23].children]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'allRels' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-9-58076f1c53e7>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mallRels\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'allRels' is not defined"
     ]
    }
   ],
   "source": [
    "allRels[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testCase = (u\"We show here that miR-29a, -29b-1, and -9 can regulate BACE1 expression in vitro.\", 4, 12)\n",
    "testCase = (u\"It has been predicted that miR-146a may target Tata Binding Protein (TBP).\",5, 12)\n",
    "testCase = (u\"Our results showed that the expression of miR-106b was inversely correlated with TRII protein levels and miR-106b can directly inhibit the TRII translation in vitro.\", 7, 21)\n",
    "\n",
    "relCheck = textmining.MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "doc = nlp(testCase[0])\n",
    "\n",
    "mirword = doc[testCase[1]]\n",
    "geneword = doc[testCase[2]]\n",
    "\n",
    "verbose=False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "conjResult, conjs = relCheck.checkCommonConj(doc, mirword, geneword, verbose)\n",
    "sdpPass, passive, negated = relCheck.checkSDP(doc, mirword, geneword, verbose)\n",
    "compPass = relCheck.checkCompartments(doc, mirword, geneword, verbose)\n",
    "sigPathway = relCheck.checkSurContext(doc, mirword, geneword, verbose)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(miR-106b, 'NOUN', 'nmod', False),\n",
       " (expression, 'NOUN', 'nsubjpass', False),\n",
       " (correlated, 'VERB', 'ccomp', True),\n",
       " (inhibit, 'VERB', 'conj', False),\n",
       " (translation, 'NOUN', 'dobj', False),\n",
       " (TRII, 'NOUN', 'compound', False)]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdpRes = relCheck._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "sdpRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "testCase = (u\"We show here that miR-29a, -29b-1, and -9 can regulate BACE1 expression in vitro.\", 4, 12)\n",
    "testCase = (u\"It has been predicted that miR-146a may target Tata Binding Protein (TBP).\",5, 12)\n",
    "testCase = (u\"Our results showed that the expression of miR-106b was inversely correlated with TRII protein levels and miR-106b can directly inhibit the TRII translation in vitro.\", 7, 21)\n",
    "testCase = (u\"The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 6, 16)\n",
    "testCase = (u\"These findings identify miR-29b as a novel posttranscriptional regulator of PGRN expression, raising the possibility that miR-29b or other miRNAs might be targeted therapeutically to increase hPGRN levels in some FTD patients.\", 3,10)\n",
    "relCheck = textmining.MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "doc = nlp(testCase[0])\n",
    "\n",
    "mirword = doc[testCase[1]]\n",
    "geneword = doc[testCase[2]]\n",
    "\n",
    "verbose=False\n",
    "\n",
    "from collections import Counter\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading stems\n"
     ]
    }
   ],
   "source": [
    "def loadStems():\n",
    "    \n",
    "    print(\"Loading stems\")\n",
    "    allRels = []\n",
    "    stem2class = {}\n",
    "\n",
    "    renameClass = {\n",
    "        'INT': 'NEU',\n",
    "        \"BOOST\": \"POS\",\n",
    "        \"COMBINE\": \"NEU\",\n",
    "        \"CHANGE\": \"NEU\",\n",
    "    }\n",
    "\n",
    "    with open(\"/mnt/d/dev/git/miRExplore/python/allrels.csv\") as fin:\n",
    "\n",
    "        for line in fin:\n",
    "            aline = line.strip().split(\",\", 1)\n",
    "            aregSyns = aline[1].split(\":\")\n",
    "            regSyns = []\n",
    "            if len(aregSyns) > 1:\n",
    "                regSyns = aregSyns[1].split(\"|\")\n",
    "            \n",
    "            aline[1] = aline[1].split(\":\")[0]           \n",
    "\n",
    "            regClass = aline[0]\n",
    "            regStem = aline[1]\n",
    "\n",
    "            if regClass in renameClass:\n",
    "                regClass = renameClass[regClass]\n",
    "                \n",
    "            stem2class[regStem] = regClass\n",
    "                \n",
    "            for regSyn in regSyns:\n",
    "                if not regSyn in stem2class:\n",
    "                    stem2class[regSyn] = regClass\n",
    "\n",
    "\n",
    "            allRels.append((regClass, regStem))\n",
    "\n",
    "    allStems = [x for x in stem2class]\n",
    "    \n",
    "    return allStems, allRels, stem2class\n",
    "\n",
    "allStems, allRels, stem2class = loadStems()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "from textmining.MirGeneRelCheck import SentenceRelationChecker, SentenceRelationClassifier\n",
    "from utils.tmutils import normalize_gene_names\n",
    "from collections import Counter\n",
    "from lxml import etree\n",
    "from sklearn.metrics import multilabel_confusion_matrix, classification_report\n",
    "from itertools import chain, combinations\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MAP3K7 TAB1\n"
     ]
    }
   ],
   "source": [
    "scaiBase = \"/mnt/d/owncloud/data/miRExplore/scai_corpus/\"\n",
    "\n",
    "base = \"TRAIN\"\n",
    "\n",
    "if base == \"TRAIN\":\n",
    "    scaiFile = \"miRNA_train_fixed.xml\"\n",
    "    relsFile = \"scai_train_rels.tsv\"\n",
    "elif base == \"TEST\":\n",
    "    scaiFile = \"miRNA_test_fixed.xml\"\n",
    "    relsFile = \"scai_test_rels.tsv\"\n",
    "    \n",
    "normGeneSymbols= normalize_gene_names(path=scaiBase + \"/../obodir/\" + \"/hgnc_no_withdrawn.syn\")\n",
    "\n",
    "relChecker = SentenceRelationChecker(nlp)\n",
    "\n",
    "correctIdentified = 0\n",
    "incorrectIdentified = 0\n",
    "totalChecks = 0\n",
    "incorrectClass = Counter()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading relations scai_test_rels.tsv\n",
      "Interactions 232\n",
      "Non NA Interactions 138\n"
     ]
    }
   ],
   "source": [
    "relClasses = loadRelations()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MAP3K7 TAB1\n",
      "Loading stems\n"
     ]
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "scaiBase = \"/mnt/d/owncloud/data/miRExplore/scai_corpus/\"\n",
    "\n",
    "normGeneSymbols= normalize_gene_names(path=scaiBase + \"/../obodir/\" + \"/hgnc_no_withdrawn.syn\")\n",
    "\n",
    "relChecker = SentenceRelationChecker(nlp)\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "\n",
    "def loadRelations(relsFile):\n",
    "    reldir2new = {}\n",
    "    reldir2new[\"TARGET\"] = \"NEU\"\n",
    "    reldir2new[\"INVCORR\"] = \"NEU\"\n",
    "    reldir2new[\"REGULATE\"] = \"NEU\"\n",
    "    relClasses = {}\n",
    "    with open(os.path.join(scaiBase, relsFile)) as fin:\n",
    "        \n",
    "        print(\"Loading relations\", relsFile)\n",
    "        \n",
    "        for lidx, line in enumerate(fin):\n",
    "\n",
    "            if lidx == 0:\n",
    "                continue\n",
    "\n",
    "            line = line.strip().split(\"\\t\")\n",
    "\n",
    "            interactionID = line[0].strip()\n",
    "            relType = line[1].strip()\n",
    "            relRDir = line[2].strip()\n",
    "            relIndirect = line[3].strip()\n",
    "            relPassive = line[4].strip()\n",
    "\n",
    "            if relRDir in reldir2new:\n",
    "                relRDir = reldir2new[relRDir]\n",
    "\n",
    "            relClasses[interactionID] = (relType, relRDir, relIndirect, relPassive)\n",
    "            \n",
    "        print(\"Interactions\", len(relClasses))\n",
    "        print(\"Non NA Interactions\", len([x for x in relClasses if not relClasses[x][0] == \"NA\"]))\n",
    "            \n",
    "    return relClasses\n",
    "\n",
    "def all_subsets(ss):\n",
    "    \n",
    "    return chain(*map(lambda x: combinations(ss, x), range(0, len(ss)+1)))\n",
    "\n",
    "def runCheck(numelems, base):\n",
    "    \n",
    "    if base == \"TRAIN\":\n",
    "        scaiFile = \"miRNA_train_fixed.xml\"\n",
    "        relsFile = \"scai_train_rels.tsv\"\n",
    "    elif base == \"TEST\":\n",
    "        scaiFile = \"miRNA_test_fixed.xml\"\n",
    "        relsFile = \"scai_test_rels.tsv\"\n",
    "    \n",
    "    relClasses = loadRelations(relsFile)\n",
    "    \n",
    "    check2results = {}\n",
    "    checkSubsets = [x for x in all_subsets(relClassifier.major_checks)]\n",
    "    checkSubsets = [checkSubsets[-1]]\n",
    "    \n",
    "    print(\"Will test {} subsets.\".format(len(checkSubsets)))\n",
    "    \n",
    "    for subset_checks in checkSubsets:\n",
    "    \n",
    "        print(\"Testing checks\", subset_checks)\n",
    "        \n",
    "        #relClassifier.active_checks = subset_checks\n",
    "\n",
    "        #allStems, allRels, stem2class = loadStems()\n",
    "\n",
    "        with open(os.path.join(scaiBase, scaiFile), 'r') as fin:\n",
    "            tree = etree.parse(fin)\n",
    "            root = tree.getroot()\n",
    "            scaiPairs = []\n",
    "            totalChecks = 0\n",
    "            correctIdentified = 0\n",
    "            incorrectIdentified = 0\n",
    "\n",
    "            incorrectClassified = 0\n",
    "            correctClassified = 0\n",
    "            totalClassified = 0\n",
    "            correctInteractClassified = 0\n",
    "            totalValidClassified = 0\n",
    "            totalClassifiable = 0\n",
    "            totalMirGeneDownClassified = 0\n",
    "\n",
    "            errorByDetect = Counter()\n",
    "            classifiedByDetect = Counter()\n",
    "            elemCaseCounter = Counter()\n",
    "\n",
    "            classifyTrue = []\n",
    "            classifyPred = []\n",
    "\n",
    "            for elem in root.findall(\".//document\"):\n",
    "\n",
    "                pmid = elem.attrib['origId']\n",
    "\n",
    "                for sentElem in elem:\n",
    "\n",
    "                    allEntities = sentElem.findall(\".//entity\")\n",
    "                    allPairs = sentElem.findall(\".//pair\")\n",
    "\n",
    "                    sentText = sentElem.attrib[\"text\"]\n",
    "\n",
    "                    entId2elem = {}\n",
    "\n",
    "                    for entity in allEntities:\n",
    "                        entId = entity.attrib['id']\n",
    "                        entText = entity.attrib['text']\n",
    "                        entType = entity.attrib['type']\n",
    "                        entOffset = tuple([int(x) for x in entity.attrib['charOffset'].split(\"-\")])\n",
    "\n",
    "                        if entType in [\"Specific_miRNAs\", \"Genes/Proteins\"]:\n",
    "\n",
    "                            if \"Genes\" in entType:\n",
    "                                if entText in normGeneSymbols:\n",
    "                                    entText = normGeneSymbols[entText]\n",
    "                                elif entText.upper() in normGeneSymbols:\n",
    "                                    gene = normGeneSymbols[entText.upper()]\n",
    "                            else:\n",
    "                                try:\n",
    "                                    entText = miRNA(entText).getStringFromParts([miRNAPART.MATURE, miRNAPART.ID, miRNAPART.PRECURSOR])\n",
    "                                except:\n",
    "                                    pass\n",
    "\n",
    "                            entTuple = (entText, entType, (entOffset[0], entOffset[1]+1))\n",
    "                            entId2elem[entId] = entTuple\n",
    "\n",
    "\n",
    "                            sentEntText = sentText[entTuple[2][0]:entTuple[2][1]]\n",
    "\n",
    "                    for pair in allPairs:\n",
    "\n",
    "                        validInteraction = pair.attrib['interaction'].lower() == \"true\"\n",
    "                        pairE1 = pair.attrib['e1']\n",
    "                        pairE2 = pair.attrib['e2']\n",
    "\n",
    "                        #if pairInt == 'true':\n",
    "                        if pairE1 in entId2elem and pairE2 in entId2elem:\n",
    "\n",
    "                            totalChecks += 1\n",
    "\n",
    "                            e1 = entId2elem[pairE1]\n",
    "                            e2 = entId2elem[pairE2]\n",
    "\n",
    "                            if not e1[1] in [\"Specific_miRNAs\"]:\n",
    "\n",
    "                                tmp=e1\n",
    "                                e1=e2\n",
    "                                e2=tmp\n",
    "\n",
    "\n",
    "                            relRes = relChecker.check_sentence(sentText\n",
    "                                                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": e1[2]}\n",
    "                                                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": e2[2]}\n",
    "                                                                , fix_special_chars=False\n",
    "                                                                , relClassifier=relClassifier.classify, verbose=False\n",
    "                                                                )\n",
    "\n",
    "\n",
    "                            fullsentence = relRes['full_sentence']\n",
    "                            acceptInteraction = relRes['accept_relation']\n",
    "\n",
    "                            if not acceptInteraction == validInteraction:\n",
    "\n",
    "                                incorrectClass[(validInteraction, acceptInteraction)]+=1\n",
    "                                incorrectIdentified += 1\n",
    "                            else:\n",
    "                                correctIdentified += 1\n",
    "\n",
    "\n",
    "                            totalClassified += 1\n",
    "\n",
    "                            if validInteraction:# and acceptInteraction:\n",
    "                                #print(relRes[\"check_results\"])   \n",
    "                                totalValidClassified += 1\n",
    "\n",
    "                                foundClasses = relRes[\"check_results\"][\"classification\"] #{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'regulation_keys': set()}\n",
    "                                foundTuple = (foundClasses[\"interaction_dir\"], foundClasses[\"regulation_dir\"])                                \n",
    "                                relationID = pair.attrib[\"id\"]\n",
    "                                testTuple = relClasses[relationID]\n",
    "\n",
    "                                classifyTrue.append((testTuple[0], testTuple[1]))\n",
    "                                classifyPred.append((foundTuple[0], foundTuple[1]))\n",
    "\n",
    "                                if testTuple[0] == \"MIR_GENE\" and testTuple[1] == \"DOWN\":\n",
    "                                    totalMirGeneDownClassified += 1\n",
    "\n",
    "                                classifiedByDetect[foundClasses[\"reg_detect\"]] += 1\n",
    "\n",
    "                                if testTuple[0] != foundTuple[0] or testTuple[1] != foundTuple[1]:\n",
    "\n",
    "\n",
    "                                    #print(sentText)\n",
    "                                    #print(e1, e2)\n",
    "                                    #print(foundTuple, foundClasses[\"reg_detect\"], \"passive?\", foundClasses[\"passive\"])\n",
    "                                    #print(testTuple)\n",
    "\n",
    "                                    #print()\n",
    "                                    #print()\n",
    "                                    #print()\n",
    "\n",
    "                                    errorByDetect[foundClasses[\"reg_detect\"]] += 1\n",
    "\n",
    "                                    incorrectClassified += 1\n",
    "\n",
    "\n",
    "\n",
    "                                    #numelems -= 1\n",
    "                                    #if numelems == 0:\n",
    "                                    #    return\n",
    "                                else:\n",
    "                                    \n",
    "                                    if foundClasses[\"reg_detect_major\"] == \"return\":\n",
    "                                        print(foundClasses)\n",
    "                                        print(sentText)\n",
    "                                        print(e1, e2)\n",
    "                                        print(foundTuple, foundClasses[\"reg_detect\"], \"passive?\", foundClasses[\"passive\"])\n",
    "                                        print(testTuple)\n",
    "\n",
    "                                        print()\n",
    "                                        print()\n",
    "                                        print()\n",
    "                                    correctClassified += 1\n",
    "\n",
    "                                    if acceptInteraction:\n",
    "                                        correctInteractClassified += 1\n",
    "\n",
    "\n",
    "\n",
    "            def printStats(outfile):\n",
    "                print(\"Total:     \", totalChecks, file=outfile)\n",
    "                print(\"Correct:   \", correctIdentified, correctIdentified/totalChecks, file=outfile)\n",
    "                print(\"Incorrect: \", incorrectIdentified, incorrectIdentified/totalChecks, file=outfile)\n",
    "                print(\"classes\", incorrectClass, file=outfile)\n",
    "\n",
    "                print(\"Correct classified:     \", correctClassified, correctClassified/totalClassified, correctClassified/totalValidClassified,file=outfile)\n",
    "                print(\"Incorrect classified:   \", incorrectClassified, incorrectClassified/totalClassified, incorrectClassified/totalValidClassified,file=outfile)\n",
    "                print(\"Random classified: \", totalMirGeneDownClassified, totalMirGeneDownClassified / totalClassified, totalMirGeneDownClassified/totalValidClassified, file=outfile)\n",
    "                print(errorByDetect)\n",
    "\n",
    "                print(\"Correct interaction&classified:     \", correctInteractClassified, correctInteractClassified/totalClassified, correctInteractClassified/totalValidClassified,file=outfile)\n",
    "\n",
    "                print(\"Classified by: \", classifiedByDetect)\n",
    "\n",
    "                lClassifyTrue = [\"_\".join(x) for x in classifyTrue]\n",
    "                lClassifyPred = [\"_\".join(x) for x in classifyPred]\n",
    "\n",
    "                allLabels = sorted(set(lClassifyTrue+lClassifyPred))\n",
    "                matrix = multilabel_confusion_matrix(lClassifyTrue, lClassifyPred, labels=allLabels)\n",
    "                print(matrix)\n",
    "                print(classification_report(lClassifyTrue,lClassifyPred))\n",
    "                \n",
    "                return classification_report(lClassifyTrue, lClassifyPred, output_dict=True)\n",
    "\n",
    "            rep = printStats(sys.stdout)\n",
    "            check2results[subset_checks] = rep\n",
    "        #printStats(sys.stderr)\n",
    "    return check2results\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading stems\n",
      "Loading relations scai_test_rels.tsv\n",
      "Interactions 232\n",
      "Non NA Interactions 138\n",
      "Will test 1 subsets.\n",
      "Testing checks ('compartment', 'between', 'counts', 'return')\n",
      "Counter({'NEG': 1, 'POS': 1}) ['down-regulation', 'of', 'miRNA-128', 'may', 'contribute', 'to', 'glioma', 'and', 'GBM', ',', 'in', 'part', ',', 'by', 'coordinately', 'up-regulating', 'ARP5', '(', 'ANGPTL6']\n",
      "Counter({'NEG': 1, 'POS': 1}) ['down-regulation', 'of', 'miRNA-128', 'may', 'contribute', 'to', 'glioma', 'and', 'GBM', ',', 'in', 'part', ',', 'by', 'coordinately', 'up-regulating', 'ARP5', '(', 'ANGPTL6', ')', ',']\n",
      "Counter({'NEG': 1, 'POS': 1}) ['down-regulation', 'of', 'miRNA-128', 'may', 'contribute', 'to', 'glioma', 'and', 'GBM', ',', 'in', 'part', ',', 'by', 'coordinately', 'up-regulating', 'ARP5', '(', 'ANGPTL6', ')', ',', 'Bmi-1', 'and', 'E2F-3a']\n",
      "Counter() ['of', 'exogenous', 'miRNA-128', 'to', 'CRL-1690', 'and', 'CRL-2610', 'GBM', 'cell', 'lines', '(', 'a', ')', 'restored', \"'\", 'homeostatic', \"'\", 'ARP5', '(', 'ANGPTL6']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('APOBEC3C', 'Genes/Proteins', (99, 103))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['of', 'exogenous', 'miRNA-128', 'to', 'CRL-1690', 'and', 'CRL-2610', 'GBM', 'cell', 'lines', '(', 'a', ')', 'restored', \"'\", 'homeostatic', \"'\", 'ARP5', '(', 'ANGPTL6', ')', ',']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('ANGPTL6', 'Genes/Proteins', (105, 112))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['of', 'exogenous', 'miRNA-128', 'to', 'CRL-1690', 'and', 'CRL-2610', 'GBM', 'cell', 'lines', '(', 'a', ')', 'restored', \"'\", 'homeostatic', \"'\", 'ARP5', '(', 'ANGPTL6', ')', ',', 'Bmi-1', 'and', 'E2F-3a']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('Bmi-1', 'Genes/Proteins', (115, 120))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['Three', 'bioinformatics-verified', 'miRNA-128', 'targets', ',', 'angiopoietin-related', 'growth', 'factor', 'protein', '5', '(', 'ARP5']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('angiopoietin-related growth factor protein 5', 'Genes/Proteins', (49, 93))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['Three', 'bioinformatics-verified', 'miRNA-128', 'targets', ',', 'angiopoietin-related', 'growth', 'factor', 'protein', '5', '(', 'ARP5', ';', 'ANGPTL6']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('APOBEC3C', 'Genes/Proteins', (95, 99))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['Three', 'bioinformatics-verified', 'miRNA-128', 'targets', ',', 'angiopoietin-related', 'growth', 'factor', 'protein', '5', '(', 'ARP5', ';', 'ANGPTL6', ')', ',']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('ANGPTL6', 'Genes/Proteins', (101, 108))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 3, 'POS': 1}) ['Three', 'bioinformatics-verified', 'miRNA-128', 'targets', ',', 'angiopoietin-related', 'growth', 'factor', 'protein', '5', '(', 'ARP5', ';', 'ANGPTL6', ')', ',', 'a', 'transcription', 'suppressor', 'that', 'promotes', 'stem', 'cell', 'renewal', 'and', 'inhibits', 'the', 'expression', 'of', 'known', 'tumor', 'suppressor', 'genes', 'involved', 'in', 'senescence', 'and', 'differentiation', ',', 'Bmi-1', ',', 'and']\n",
      "['Three', 'bioinformatics-verified', 'miRNA-128', 'targets', ',', 'angiopoietin-related', 'growth', 'factor', 'protein', '5', '(', 'ARP5', ';', 'ANGPTL6', ')', ',', 'a', 'transcription', 'suppressor', 'that', 'promotes', 'stem', 'cell', 'renewal', 'and', 'inhibits', 'the', 'expression', 'of', 'known', 'tumor', 'suppressor', 'genes', 'involved', 'in', 'senescence', 'and', 'differentiation', ',', 'Bmi-1', ',', 'and']\n",
      "Counter({'NEU': 3, 'NEG': 3, 'POS': 1})\n",
      "Counter({'NEG': 1}) ['Micro-RNA-128', '(', 'miRNA-128', ')', 'down-regulation', 'in', 'glioblastoma', 'targets', 'ARP5', '(', 'ANGPTL6']\n",
      "['Micro-RNA-128', '(', 'miRNA-128', ')', 'down-regulation', 'in', 'glioblastoma', 'targets', 'ARP5', '(', 'ANGPTL6']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1}) ['Micro-RNA-128', '(', 'miRNA-128', ')', 'down-regulation', 'in', 'glioblastoma', 'targets', 'ARP5', '(', 'ANGPTL6', ')', ',']\n",
      "['Micro-RNA-128', '(', 'miRNA-128', ')', 'down-regulation', 'in', 'glioblastoma', 'targets', 'ARP5', '(', 'ANGPTL6', ')', ',']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1}) ['Micro-RNA-128', '(', 'miRNA-128', ')', 'down-regulation', 'in', 'glioblastoma', 'targets', 'ARP5', '(', 'ANGPTL6', ')', ',', 'Bmi-1', 'and', 'E2F-3a']\n",
      "['Micro-RNA-128', '(', 'miRNA-128', ')', 'down-regulation', 'in', 'glioblastoma', 'targets', 'ARP5', '(', 'ANGPTL6', ')', ',', 'Bmi-1', 'and', 'E2F-3a']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1}) ['Micro-RNA-128', '(', 'miRNA-128', ')', 'down-regulation', 'in', 'glioblastoma', 'targets', 'ARP5', '(', 'ANGPTL6']\n",
      "['Micro-RNA-128', '(', 'miRNA-128', ')', 'down-regulation', 'in', 'glioblastoma', 'targets', 'ARP5', '(', 'ANGPTL6']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1}) ['Micro-RNA-128', '(', 'miRNA-128', ')', 'down-regulation', 'in', 'glioblastoma', 'targets', 'ARP5', '(', 'ANGPTL6', ')', ',']\n",
      "['Micro-RNA-128', '(', 'miRNA-128', ')', 'down-regulation', 'in', 'glioblastoma', 'targets', 'ARP5', '(', 'ANGPTL6', ')', ',']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1}) ['Micro-RNA-128', '(', 'miRNA-128', ')', 'down-regulation', 'in', 'glioblastoma', 'targets', 'ARP5', '(', 'ANGPTL6', ')', ',', 'Bmi-1', 'and', 'E2F-3a']\n",
      "['Micro-RNA-128', '(', 'miRNA-128', ')', 'down-regulation', 'in', 'glioblastoma', 'targets', 'ARP5', '(', 'ANGPTL6', ')', ',', 'Bmi-1', 'and', 'E2F-3a']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1, 'POS': 1}) ['Reduction', 'of', 'miR-21', 'induces', 'glioma', 'cell', 'apoptosis', 'via', 'activating', 'caspase', '9', 'and', '3']\n",
      "Counter() ['activating', 'caspase', '9', 'and', '3', '.', ' ', 'Extensive', 'data', 'indicate', 'that', 'miR-21', 'plays', 'a']\n",
      "Counter() ['activating', 'caspase', '9', 'and', '3', '.', ' ', 'Extensive', 'data', 'indicate', 'that', 'miR-21', 'plays', 'a', 'critical', 'role', 'in', 'gliomagenesis', ',', 'however', ',', 'knowledge', 'is', 'limited', 'on', 'the', 'mechanism', 'of', 'action', 'of', 'miR-21', ',', 'including']\n",
      "Counter({'NEG': 1}) ['reduction', 'of', 'miR-21', 'activated', 'caspase', '9', 'and', '3']\n",
      "['reduction', 'of', 'miR-21', 'activated', 'caspase', '9', 'and', '3']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1}) ['reduction', 'of', 'miR-21', 'activated', 'caspase', '9', 'and', '3', ',', 'which', 'may', 'be', 'mediated', 'by', 'modulating', 'multiple', 'potential', 'target', 'genes', ',', 'such', 'as', 'TIMP3']\n",
      "['reduction', 'of', 'miR-21', 'activated', 'caspase', '9', 'and', '3', ',', 'which', 'may', 'be', 'mediated', 'by', 'modulating', 'multiple', 'potential', 'target', 'genes', ',', 'such', 'as', 'TIMP3']\n",
      "Counter({'NEU': 4, 'NEG': 1})\n",
      "Counter({'POS': 1}) ['affinity', 'of', 'miR-433', 'to', 'the', '3', \"'\", 'UTR', 'would', 'result', 'in', 'increased', 'FGF20', 'expression', 'and']\n",
      "['affinity', 'of', 'miR-433', 'to', 'the', '3', \"'\", 'UTR', 'would', 'result', 'in', 'increased', 'FGF20', 'expression', 'and']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "Counter({'POS': 2}) ['affinity', 'of', 'miR-433', 'to', 'the', '3', \"'\", 'UTR', 'would', 'result', 'in', 'increased', 'FGF20', 'expression', 'and', 'upregulation', 'of', 'alpha-synuclein', ',', 'which']\n",
      "['affinity', 'of', 'miR-433', 'to', 'the', '3', \"'\", 'UTR', 'would', 'result', 'in', 'increased', 'FGF20', 'expression', 'and', 'upregulation', 'of', 'alpha-synuclein', ',', 'which']\n",
      "Counter({'NEU': 2, 'POS': 2})\n",
      "Counter() ['FGF20', 'rs12720208', 'SNP', 'and', 'microRNA-433', 'variation']\n",
      "Counter() ['of', 'the', 'apl-1', 'transcription', 'are', 'modulated', 'by', 'the', 'activity', 'of', 'let-7', 'family', 'microRNAs']\n",
      "Counter() ['C.', 'elegans', 'apl-1', 'shows', 'significant', 'genetic', 'interactions', 'with', 'let-7', 'family', 'microRNAs']\n",
      "Counter() ['interactions', 'with', 'let-7', 'family', 'microRNAs', 'and', 'let-7-targeted', 'heterochronic', 'genes', ',', 'hbl-1', ',', 'lin-41']\n",
      "Counter() ['interactions', 'with', 'let-7', 'family', 'microRNAs', 'and', 'let-7-targeted', 'heterochronic', 'genes', ',', 'hbl-1', ',', 'lin-41', 'and', 'lin-42']\n",
      "Counter() ['microRNAs', 'and', 'let-7-targeted', 'heterochronic', 'genes', ',', 'hbl-1', ',', 'lin-41']\n",
      "Counter() ['microRNAs', 'and', 'let-7-targeted', 'heterochronic', 'genes', ',', 'hbl-1', ',', 'lin-41', 'and', 'lin-42']\n",
      "Counter() ['SRF', 'and', 'myocardin', 'regulate', 'a', 'cardiovascular-specific', 'microRNA', '(', 'miRNA', ')', 'cluster', 'encoding', 'miR-143', 'and', 'miR-145']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (100, 107)) ('SRF', 'Genes/Proteins', (13, 16))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['SRF', 'and', 'myocardin', 'regulate', 'a', 'cardiovascular-specific', 'microRNA', '(', 'miRNA', ')', 'cluster', 'encoding', 'miR-143', 'and', 'miR-145']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (100, 107)) ('myocardin', 'Genes/Proteins', (21, 30))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['SRF', 'and', 'myocardin', 'regulate', 'a', 'cardiovascular-specific', 'microRNA', '(', 'miRNA', ')', 'cluster', 'encoding', 'miR-143', 'and', 'miR-145']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (112, 119)) ('SRF', 'Genes/Proteins', (13, 16))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['SRF', 'and', 'myocardin', 'regulate', 'a', 'cardiovascular-specific', 'microRNA', '(', 'miRNA', ')', 'cluster', 'encoding', 'miR-143', 'and', 'miR-145']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (112, 119)) ('myocardin', 'Genes/Proteins', (21, 30))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['modulators', 'of', 'SRF', 'activity', 'and', 'actin', 'dynamics', 'by', 'miR-143', 'and', 'miR-145']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These abnormalities reflect the regulation of a cadre of modulators of SRF activity and actin dynamics by miR-143 and miR-145. \n",
      "('miR-143', 'Specific_miRNAs', (106, 113)) ('SRF', 'Genes/Proteins', (71, 74))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['modulators', 'of', 'SRF', 'activity', 'and', 'actin', 'dynamics', 'by', 'miR-143', 'and', 'miR-145']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These abnormalities reflect the regulation of a cadre of modulators of SRF activity and actin dynamics by miR-143 and miR-145. \n",
      "('miR-145', 'Specific_miRNAs', (118, 125)) ('SRF', 'Genes/Proteins', (71, 74))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['Thus', ',', 'miR-143', 'and', 'miR-145', 'act', 'as', 'integral', 'components', 'of', 'the', 'regulatory', 'network', 'whereby', 'SRF', 'controls', 'cytoskeletal']\n",
      "Counter() ['miR-143', 'and', 'miR-145', 'act', 'as', 'integral', 'components', 'of', 'the', 'regulatory', 'network', 'whereby', 'SRF', 'controls', 'cytoskeletal']\n",
      "Counter({'NEG': 1}) ['an', 'miR-30-based', 'shRNA', 'construct', 'prevents', 'Oas1', 'pathway', 'associated']\n",
      "['an', 'miR-30-based', 'shRNA', 'construct', 'prevents', 'Oas1', 'pathway', 'associated']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'POS': 1}) ['naturally', 'occurring', 'microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "['naturally', 'occurring', 'microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "Counter({'POS': 1})\n",
      "Counter({'POS': 1}) ['microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "['microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "Counter({'POS': 1})\n",
      "Counter({'NEG': 1}) ['Downregulation', 'of', 'Pdcd4', 'by', 'mir-21', 'facilitates', 'glioblastoma']\n",
      "counts between 3 Counter() NEG\n",
      "['Downregulation', 'of', 'Pdcd4', 'by', 'mir-21', 'facilitates', 'glioblastoma']\n",
      "Counter()\n",
      "Counter({'NEG': 1, 'POS': 1}) ['downregulation', 'of', 'mir-21', 'in', 'glioblastoma-derived', 'cell', 'lines', 'results', 'in', 'increased', 'expression', 'of', 'its', 'target', ',', 'programmed', 'cell', 'death', '4', '(', 'Pdcd4']\n",
      "Counter({'NEG': 1, 'POS': 1}) ['downregulation', 'of', 'mir-21', 'in', 'glioblastoma-derived', 'cell', 'lines', 'results', 'in', 'increased', 'expression', 'of', 'its', 'target', ',', 'programmed', 'cell', 'death', '4', '(', 'Pdcd4', ')', ',']\n",
      "Counter({'NEG': 1, 'POS': 1}) ['in', 'which', 'mir-21', 'levels', 'are', 'downregulated', 'or', 'Pdcd4', 'is', 'over-expressed']\n",
      "Counter({'NEG': 1, 'POS': 1}) ['downregulation', 'of', 'mir-21', 'or', 'overexpression', 'of', 'its', 'target', ',', 'Pdcd4', ',', 'in']\n",
      "Counter() ['linkage', 'between', 'mir-21', 'and', 'Pdcd4']\n",
      "Counter({'NEG': 1}) ['linkage', 'between', 'mir-21', 'and', 'Pdcd4', 'and', 'further', 'elucidate', 'the', 'molecular', 'mechanisms', 'by', 'which', 'the', 'known', 'high', 'level', 'of', 'mir-21', 'expression', 'in', 'glioblastoma', 'can', 'attribute', 'to', 'tumorigenesis', '--', 'namely', ',', 'inhibition', 'of', 'Pdcd4', 'and', 'its']\n",
      "['linkage', 'between', 'mir-21', 'and', 'Pdcd4', 'and', 'further', 'elucidate', 'the', 'molecular', 'mechanisms', 'by', 'which', 'the', 'known', 'high', 'level', 'of', 'mir-21', 'expression', 'in', 'glioblastoma', 'can', 'attribute', 'to', 'tumorigenesis', '--', 'namely', ',', 'inhibition', 'of', 'Pdcd4', 'and', 'its']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter() ['mir-21', 'and', 'Pdcd4', 'and', 'further', 'elucidate', 'the', 'molecular', 'mechanisms', 'by', 'which', 'the', 'known', 'high', 'level', 'of', 'mir-21', 'expression', 'in']\n",
      "Counter({'NEG': 1}) ['level', 'of', 'mir-21', 'expression', 'in', 'glioblastoma', 'can', 'attribute', 'to', 'tumorigenesis', '--', 'namely', ',', 'inhibition', 'of', 'Pdcd4', 'and', 'its']\n",
      "['level', 'of', 'mir-21', 'expression', 'in', 'glioblastoma', 'can', 'attribute', 'to', 'tumorigenesis', '--', 'namely', ',', 'inhibition', 'of', 'Pdcd4', 'and', 'its']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter() ['Endogenous', 'miR-124a', 'expression', 'and', 'the', 'protein', 'expression', 'of', 'three', 'of', 'its', 'targets', ';', 'IQ', 'motif', 'containing', 'GTPase', 'activating', 'protein', '1', '(', 'IQGAP1', ')', ',']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin 1 (LAMC1) and integrin 1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('IQGAP1', 'Genes/Proteins', (132, 138))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['Endogenous', 'miR-124a', 'expression', 'and', 'the', 'protein', 'expression', 'of', 'three', 'of', 'its', 'targets', ';', 'IQ', 'motif', 'containing', 'GTPase', 'activating', 'protein', '1', '(', 'IQGAP1']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin 1 (LAMC1) and integrin 1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('IQ motif containing GTPase activating protein 1', 'Genes/Proteins', (83, 130))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['Endogenous', 'miR-124a', 'expression', 'and', 'the', 'protein', 'expression', 'of', 'three', 'of', 'its', 'targets', ';', 'IQ', 'motif', 'containing', 'GTPase', 'activating', 'protein', '1', '(', 'IQGAP1', ')', ',', 'laminin', '1', '(', 'LAMC1']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin 1 (LAMC1) and integrin 1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('laminin 1', 'Genes/Proteins', (141, 151))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['Endogenous', 'miR-124a', 'expression', 'and', 'the', 'protein', 'expression', 'of', 'three', 'of', 'its', 'targets', ';', 'IQ', 'motif', 'containing', 'GTPase', 'activating', 'protein', '1', '(', 'IQGAP1', ')', ',', 'laminin', '1', '(', 'LAMC1', ')', 'and']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin 1 (LAMC1) and integrin 1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('LAMC1', 'Genes/Proteins', (153, 158))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['Endogenous', 'miR-124a', 'expression', 'and', 'the', 'protein', 'expression', 'of', 'three', 'of', 'its', 'targets', ';', 'IQ', 'motif', 'containing', 'GTPase', 'activating', 'protein', '1', '(', 'IQGAP1', ')', ',', 'laminin', '1', '(', 'LAMC1', ')', 'and', 'integrin', '1', '(', 'ITGB1']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin 1 (LAMC1) and integrin 1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('integrin 1', 'Genes/Proteins', (164, 175))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['Endogenous', 'miR-124a', 'expression', 'and', 'the', 'protein', 'expression', 'of', 'three', 'of', 'its', 'targets', ';', 'IQ', 'motif', 'containing', 'GTPase', 'activating', 'protein', '1', '(', 'IQGAP1', ')', ',', 'laminin', '1', '(', 'LAMC1', ')', 'and', 'integrin', '1', '(', 'ITGB1', ')', 'were']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Endogenous miR-124a expression and the protein expression of three of its targets; IQ motif containing GTPase activating protein 1 (IQGAP1), laminin 1 (LAMC1) and integrin 1 (ITGB1) were significantly reciprocally associated in the majority of the clinical cases. \n",
      "('miR-124a', 'Specific_miRNAs', (11, 19)) ('ITGB1', 'Genes/Proteins', (177, 182))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'POS': 1}) ['levels', 'of', 'MCP-2', 'and', 'increased', 'levels', 'of', 'mir-146a', 'were', 'observed']\n",
      "['levels', 'of', 'MCP-2', 'and', 'increased', 'levels', 'of', 'mir-146a', 'were', 'observed']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "Counter({'POS': 1}) ['levels', 'of', 'MCP-2', 'and', 'increased', 'levels', 'of', 'mir-146a', 'were', 'observed', ',', 'suggesting', 'a', 'role', 'for', 'mir-146a', 'in', 'the']\n",
      "['levels', 'of', 'MCP-2', 'and', 'increased', 'levels', 'of', 'mir-146a', 'were', 'observed', ',', 'suggesting', 'a', 'role', 'for', 'mir-146a', 'in', 'the']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "Counter({'POS': 2, 'NEG': 1}) ['target', 'for', 'mir-146a', 'in', 'HIV-1', 'infected', 'microglia', ',', 'as', 'overexpression', 'of', 'mir-146a', 'prevented', 'HIV-induced', 'secretion', 'of', 'MCP-2', 'chemokine', '.']\n",
      "Counter({'POS': 2, 'NEG': 1}) ['overexpression', 'of', 'mir-146a', 'prevented', 'HIV-induced', 'secretion', 'of', 'MCP-2', 'chemokine']\n",
      "Counter() ['targets', 'for', 'mir-146a', ',', 'the', 'chemokine', 'CCL8/MCP-2', 'is', 'a']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('chemokine CCL8', 'Genes/Proteins', (51, 65))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['targets', 'for', 'mir-146a', ',', 'the', 'chemokine', 'CCL8/MCP-2', 'is', 'a']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CCL8', 'Genes/Proteins', (66, 71))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['targets', 'for', 'mir-146a', ',', 'the', 'chemokine', 'CCL8/MCP-2', 'is', 'a', 'ligand', 'for', 'the', 'CCR5', 'chemokine', 'receptor', 'and', 'a']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CCR5 chemokine receptor', 'Genes/Proteins', (92, 115))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['targets', 'for', 'mir-146a', ',', 'the', 'chemokine', 'CCL8/MCP-2', 'is', 'a', 'ligand', 'for', 'the', 'CCR5', 'chemokine', 'receptor', 'and', 'a', 'potent', 'inhibitor', 'of', 'CD4/CCR5-mediated', 'HIV-1', 'entry']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CD4', 'Genes/Proteins', (142, 145))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['targets', 'for', 'mir-146a', ',', 'the', 'chemokine', 'CCL8/MCP-2', 'is', 'a', 'ligand', 'for', 'the', 'CCR5', 'chemokine', 'receptor', 'and', 'a', 'potent', 'inhibitor', 'of', 'CD4/CCR5-mediated', 'HIV-1', 'entry']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication. \n",
      "('mir-146a', 'Specific_miRNAs', (37, 45)) ('CCR5', 'Genes/Proteins', (146, 150))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 2}) ['Downregulation', 'of', 'ABCG2', 'expression', 'in', 'glioblastoma', 'cancer', 'stem', 'cells', 'with', 'miRNA-328', 'may', 'decrease']\n",
      "counts between 3 Counter() NEG\n",
      "['Downregulation', 'of', 'ABCG2', 'expression', 'in', 'glioblastoma', 'cancer', 'stem', 'cells', 'with', 'miRNA-328', 'may', 'decrease']\n",
      "Counter()\n",
      "Counter({'NEG': 3}) ['these', 'is', 'miR-9', ',', 'which', 'suppresses', 'mesenchymal', 'differentiation', 'in', 'glioblastoma', 'by', 'downregulating', 'expression', 'of', 'JAK', 'kinases', 'and', 'inhibiting', 'activation', 'of', 'STAT3', '.']\n",
      "['these', 'is', 'miR-9', ',', 'which', 'suppresses', 'mesenchymal', 'differentiation', 'in', 'glioblastoma', 'by', 'downregulating', 'expression', 'of', 'JAK', 'kinases', 'and', 'inhibiting', 'activation', 'of', 'STAT3', '.']\n",
      "Counter({'NEG': 3, 'NEU': 2})\n",
      "Counter({'NEG': 1, 'POS': 1}) ['Co-suppression', 'of', 'miR-221/222', 'directly', 'resulted', 'in', 'the', 'up-regulation', 'of', 'p27Kip1', 'in', 'the']\n",
      "Counter({'NEG': 1, 'POS': 1}) ['Consistently', ',', 'miR-221/222', 'knocked-down', 'through', 'antisense', \"2'-OME-oligonucleotides\", 'increased', 'p27Kip1', 'in', 'U251']\n",
      "Counter({'NEG': 2, 'POS': 1}) ['Consistently', ',', 'miR-221/222', 'knocked-down', 'through', 'antisense', \"2'-OME-oligonucleotides\", 'increased', 'p27Kip1', 'in', 'U251', 'glioma', 'subcutaneous', 'mice', 'and', 'strongly', 'reduced', 'tumor', 'growth', 'in', 'vivo', 'through', 'up', 'regulation', 'of', 'p27Kip1', '.']\n",
      "['Consistently', ',', 'miR-221/222', 'knocked-down', 'through', 'antisense', \"2'-OME-oligonucleotides\", 'increased', 'p27Kip1', 'in', 'U251', 'glioma', 'subcutaneous', 'mice', 'and', 'strongly', 'reduced', 'tumor', 'growth', 'in', 'vivo', 'through', 'up', 'regulation', 'of', 'p27Kip1', '.']\n",
      "Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "Counter({'NEG': 2}) ['suppressor', 'gene', 'p27Kip1', ',', 'and', 'co-suppression', 'of', 'miR-221/222', 'expression', 'in']\n",
      "['suppressor', 'gene', 'p27Kip1', ',', 'and', 'co-suppression', 'of', 'miR-221/222', 'expression', 'in']\n",
      "Counter({'NEG': 2, 'NEU': 1})\n",
      "Counter({'NEG': 2, 'POS': 1}) ['co-suppression', 'of', 'miR-221/222', 'expression', 'in', 'advanced', 'gliomas', 'may', 'inhibit', 'glioma', 'cell', 'proliferation', 'by', 'a', 'mechanism', 'involving', 'the', 'up-regulation', 'of', 'p27Kip1', 'in', 'vitro']\n",
      "Counter() ['expression', 'of', 'miR-221/miR-222', 'and', 'the', 'cell', 'cycle', 'inhibitor', 'p27Kip1', 'was', 'identified']\n",
      "Counter() ['expression', 'of', 'miR-221/miR-222', 'and', 'the', 'cell', 'cycle', 'inhibitor', 'p27Kip1', 'was', 'identified']\n",
      "Counter() ['loop', 'between', 'miR-21', 'and', 'STAT3', 'might', 'provide']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "The data strongly suggested that a regulatory loop between miR-21 and STAT3 might provide an insight into the mechanism of modulating EGFR/STAT3 signaling. \n",
      "('miR-21', 'Specific_miRNAs', (59, 65)) ('STAT3', 'Genes/Proteins', (70, 75))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['loop', 'between', 'miR-21', 'and', 'STAT3', 'might', 'provide', 'an', 'insight', 'into', 'the', 'mechanism', 'of', 'modulating', 'EGFR/STAT3', 'signaling']\n",
      "Counter() ['loop', 'between', 'miR-21', 'and', 'STAT3', 'might', 'provide', 'an', 'insight', 'into', 'the', 'mechanism', 'of', 'modulating', 'EGFR/STAT3', 'signaling']\n",
      "Counter({'NEG': 1}) ['miR-29b', 'downregulates', 'the', 'expression', 'of', 'luciferase', 'through', 'hPGRN', 'or', 'mouse', 'PGRN', '(', 'mPGRN']\n",
      "['miR-29b', 'downregulates', 'the', 'expression', 'of', 'luciferase', 'through', 'hPGRN', 'or', 'mouse', 'PGRN', '(', 'mPGRN']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1}) ['or', 'mouse', 'PGRN', '(', 'mPGRN', ')', \"3'UTRs\", ',', 'and', 'the', 'regulation', 'was', 'abolished', 'by', 'mutations', 'in', 'the', 'miR-29b', 'binding', 'site']\n",
      "['or', 'mouse', 'PGRN', '(', 'mPGRN', ')', \"3'UTRs\", ',', 'and', 'the', 'regulation', 'was', 'abolished', 'by', 'mutations', 'in', 'the', 'miR-29b', 'binding', 'site']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter() ['site', 'for', 'microRNA-29b', '(', 'miR-29b', ')', 'in', 'the', '3', \"'\", 'untranslated', 'region', '(', \"3'UTR\", ')', 'of', 'the', 'human', 'PGRN', '(', 'hPGRN']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('microRNA-29b', 'Specific_miRNAs', (59, 71)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['microRNA-29b', '(', 'miR-29b', ')', 'in', 'the', '3', \"'\", 'untranslated', 'region', '(', \"3'UTR\", ')', 'of', 'the', 'human', 'PGRN', '(', 'hPGRN']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('miR-29b', 'Specific_miRNAs', (73, 80)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['MicroRNA-29b', 'regulates', 'the', 'expression', 'level', 'of', 'human', 'progranulin', ',', 'a']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "MicroRNA-29b regulates the expression level of human progranulin, a secreted glycoprotein implicated in frontotemporal dementia.  \n",
      "('MicroRNA-29b', 'Specific_miRNAs', (0, 12)) ('progranulin', 'Genes/Proteins', (53, 64))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'POS': 1, 'NEG': 1}) ['overexpression', 'of', 'miR-21', 'protects', 'against', 'ischemic', 'neuronal', 'death', ',', 'and', 'that', 'downregulation', 'of', 'FASLG', ',', 'a']\n",
      "Counter({'POS': 1, 'NEG': 1}) ['overexpression', 'of', 'miR-21', 'protects', 'against', 'ischemic', 'neuronal', 'death', ',', 'and', 'that', 'downregulation', 'of', 'FASLG', ',', 'a', 'tumor', 'necrosis', 'factor-', 'family', 'member']\n",
      "Counter({'NEG': 2, 'POS': 1}) ['overexpression', 'of', 'miR-21', 'in', 'neurons', 'significantly', 'reduced', 'FASLG', 'levels', ',', 'and', 'introduction', 'of', 'an', 'miR-21', 'mimic', 'into', '293-HEK', 'cells', 'substantially', 'reduced', 'luciferase', 'activity', 'in', 'a', 'reporter', 'system', 'containing', 'the', \"3'-UTR\", 'of', 'Faslg', '.']\n",
      "['overexpression', 'of', 'miR-21', 'in', 'neurons', 'significantly', 'reduced', 'FASLG', 'levels', ',', 'and', 'introduction', 'of', 'an', 'miR-21', 'mimic', 'into', '293-HEK', 'cells', 'substantially', 'reduced', 'luciferase', 'activity', 'in', 'a', 'reporter', 'system', 'containing', 'the', \"3'-UTR\", 'of', 'Faslg', '.']\n",
      "Counter({'NEU': 3, 'NEG': 2, 'POS': 1})\n",
      "Counter({'POS': 1, 'NEG': 1}) ['overexpression', 'of', 'miR-21', 'in', 'neurons', 'significantly', 'reduced', 'FASLG', 'levels']\n",
      "Counter({'NEG': 1}) ['of', 'an', 'miR-21', 'mimic', 'into', '293-HEK', 'cells', 'substantially', 'reduced', 'luciferase', 'activity', 'in', 'a', 'reporter', 'system', 'containing', 'the', \"3'-UTR\", 'of', 'Faslg']\n",
      "['of', 'an', 'miR-21', 'mimic', 'into', '293-HEK', 'cells', 'substantially', 'reduced', 'luciferase', 'activity', 'in', 'a', 'reporter', 'system', 'containing', 'the', \"3'-UTR\", 'of', 'Faslg']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter({'NEG': 1}) ['significantly', 'reduced', 'FASLG', 'levels', ',', 'and', 'introduction', 'of', 'an', 'miR-21', 'mimic', 'into']\n",
      "['significantly', 'reduced', 'FASLG', 'levels', ',', 'and', 'introduction', 'of', 'an', 'miR-21', 'mimic', 'into']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter() ['lesions', '(', 'microRNA-34a', ',', 'microRNA-155', 'and', 'microRNA-326', ')', 'targeted', 'the', \"3'-untranslated\", 'region', 'of', 'CD47', 'in', 'reporter']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites. \n",
      "('microRNA-34a', 'Specific_miRNAs', (65, 77)) ('CD47', 'Genes/Proteins', (149, 153))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['microRNA-34a', ',', 'microRNA-155', 'and', 'microRNA-326', ')', 'targeted', 'the', \"3'-untranslated\", 'region', 'of', 'CD47', 'in', 'reporter']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites. \n",
      "('microRNA-155', 'Specific_miRNAs', (79, 91)) ('CD47', 'Genes/Proteins', (149, 153))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['microRNA-155', 'and', 'microRNA-326', ')', 'targeted', 'the', \"3'-untranslated\", 'region', 'of', 'CD47', 'in', 'reporter']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites. \n",
      "('microRNA-326', 'Specific_miRNAs', (96, 108)) ('CD47', 'Genes/Proteins', (149, 153))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['region', 'of', 'CD47', 'in', 'reporter', 'assays', ',', 'with', 'microRNA-155', 'even', 'at']\n",
      "Counter() ['The', 'miRNA', 'miR-23a', 'was', 'predicted', 'to', 'bind', 'XIAP', 'mRNA']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "The miRNA miR-23a was predicted to bind XIAP mRNA. \n",
      "('miR-23a', 'Specific_miRNAs', (10, 17)) ('XIAP', 'Genes/Proteins', (40, 44))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['miR-23a', 'directly', 'bound', 'the', '3', \"'\", 'UTR', 'of', 'XIAP', ',', 'and']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a. \n",
      "('miR-23a', 'Specific_miRNAs', (0, 7)) ('XIAP', 'Genes/Proteins', (37, 41))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1, 'POS': 1}) ['miR-23a', 'directly', 'bound', 'the', '3', \"'\", 'UTR', 'of', 'XIAP', ',', 'and', 'miR-23a', 'inhibition', 'led', 'to', 'an', 'increase', 'in', 'XIAP', 'mRNA', 'in']\n",
      "Counter({'NEG': 1, 'POS': 1}) ['miR-23a', 'directly', 'bound', 'the', '3', \"'\", 'UTR', 'of', 'XIAP', ',', 'and', 'miR-23a', 'inhibition', 'led', 'to', 'an', 'increase', 'in', 'XIAP', 'mRNA', 'in', 'vitro', ',', 'demonstrating', 'that', 'XIAP', 'is', 'a']\n",
      "Counter() ['UTR', 'of', 'XIAP', ',', 'and', 'miR-23a']\n",
      "Counter({'NEG': 1, 'POS': 1}) [',', 'and', 'miR-23a', 'inhibition', 'led', 'to', 'an', 'increase', 'in', 'XIAP', 'mRNA', 'in']\n",
      "Counter({'NEG': 1, 'POS': 1}) [',', 'and', 'miR-23a', 'inhibition', 'led', 'to', 'an', 'increase', 'in', 'XIAP', 'mRNA', 'in', 'vitro', ',', 'demonstrating', 'that', 'XIAP', 'is', 'a']\n",
      "Counter() ['miR-23a', 'regulation', 'of', 'X-linked', 'inhibitor', 'of', 'apoptosis', '(', 'XIAP']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-23a regulation of X-linked inhibitor of apoptosis (XIAP) contributes to sex differences in the response to cerebral ischemia.  \n",
      "('miR-23a', 'Specific_miRNAs', (0, 7)) ('X-linked inhibitor of apoptosis', 'Genes/Proteins', (22, 53))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['miR-23a', 'regulation', 'of', 'X-linked', 'inhibitor', 'of', 'apoptosis', '(', 'XIAP', ')', 'contributes']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-23a regulation of X-linked inhibitor of apoptosis (XIAP) contributes to sex differences in the response to cerebral ischemia.  \n",
      "('miR-23a', 'Specific_miRNAs', (0, 7)) ('XIAP', 'Genes/Proteins', (55, 59))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1}) ['Transfection', 'of', 'let-7', 'miRNA', 'reduced', 'expression', 'of', 'pan-RAS', ',', 'N-RAS']\n",
      "['Transfection', 'of', 'let-7', 'miRNA', 'reduced', 'expression', 'of', 'pan-RAS', ',', 'N-RAS']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1}) ['Transfection', 'of', 'let-7', 'miRNA', 'reduced', 'expression', 'of', 'pan-RAS', ',', 'N-RAS', ',', 'and']\n",
      "['Transfection', 'of', 'let-7', 'miRNA', 'reduced', 'expression', 'of', 'pan-RAS', ',', 'N-RAS', ',', 'and']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1}) ['Transfection', 'of', 'let-7', 'miRNA', 'reduced', 'expression', 'of', 'pan-RAS', ',', 'N-RAS', ',', 'and', 'K-RAS', 'in', 'the']\n",
      "['Transfection', 'of', 'let-7', 'miRNA', 'reduced', 'expression', 'of', 'pan-RAS', ',', 'N-RAS', ',', 'and', 'K-RAS', 'in', 'the']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter() ['-actin', ',', 'SM22', ',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a']\n",
      "Counter() [',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a']\n",
      "Counter({'NEG': 1}) ['activity', 'of', 'myocardin', 'as', 'indicated', 'by', 'reduced', 'expression', 'of', 'SMC', 'contractile', 'marker', 'genes', '(', 'SM', '-actin', ',', 'SM22', ',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a']\n",
      "['activity', 'of', 'myocardin', 'as', 'indicated', 'by', 'reduced', 'expression', 'of', 'SMC', 'contractile', 'marker', 'genes', '(', 'SM', '-actin', ',', 'SM22', ',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter({'NEG': 2}) ['microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a', ')', ',', 'and', 'reduced', 'contractility', 'of', 'human', 'vascular', 'smooth', 'muscle', 'cells', '(', 'SMCs', ')', 'embedded', 'in', 'collagen', 'gel', 'lattices', ',', 'suggesting', 'that', 'myocardin', 'degradation', 'is']\n",
      "['microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a', ')', ',', 'and', 'reduced', 'contractility', 'of', 'human', 'vascular', 'smooth', 'muscle', 'cells', '(', 'SMCs', ')', 'embedded', 'in', 'collagen', 'gel', 'lattices', ',', 'suggesting', 'that', 'myocardin', 'degradation', 'is']\n",
      "Counter({'NEG': 2})\n",
      "Counter({'NEG': 3, 'POS': 1}) ['accumulation', 'of', 'myocardin', 'in', 'response', 'to', 'proteasome', 'inhibition', 'by', 'MG132', 'or', 'lactacystin', 'resulted', 'in', 'decrease', 'of', 'transcriptional', 'activity', 'of', 'myocardin', 'as', 'indicated', 'by', 'reduced', 'expression', 'of', 'SMC', 'contractile', 'marker', 'genes', '(', 'SM', '-actin', ',', 'SM22', ',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a']\n",
      "['accumulation', 'of', 'myocardin', 'in', 'response', 'to', 'proteasome', 'inhibition', 'by', 'MG132', 'or', 'lactacystin', 'resulted', 'in', 'decrease', 'of', 'transcriptional', 'activity', 'of', 'myocardin', 'as', 'indicated', 'by', 'reduced', 'expression', 'of', 'SMC', 'contractile', 'marker', 'genes', '(', 'SM', '-actin', ',', 'SM22', ',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a']\n",
      "Counter({'NEU': 5, 'NEG': 3, 'POS': 1})\n",
      "Counter() ['-actin', ',', 'SM22', ',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a']\n",
      "Counter() [',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a']\n",
      "Counter({'NEG': 1}) ['activity', 'of', 'myocardin', 'as', 'indicated', 'by', 'reduced', 'expression', 'of', 'SMC', 'contractile', 'marker', 'genes', '(', 'SM', '-actin', ',', 'SM22', ',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a', ')', ',']\n",
      "['activity', 'of', 'myocardin', 'as', 'indicated', 'by', 'reduced', 'expression', 'of', 'SMC', 'contractile', 'marker', 'genes', '(', 'SM', '-actin', ',', 'SM22', ',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a', ')', ',']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter({'NEG': 2}) ['miR-143/145', 'and', 'miR-1/133a', ')', ',', 'and', 'reduced', 'contractility', 'of', 'human', 'vascular', 'smooth', 'muscle', 'cells', '(', 'SMCs', ')', 'embedded', 'in', 'collagen', 'gel', 'lattices', ',', 'suggesting', 'that', 'myocardin', 'degradation', 'is']\n",
      "['miR-143/145', 'and', 'miR-1/133a', ')', ',', 'and', 'reduced', 'contractility', 'of', 'human', 'vascular', 'smooth', 'muscle', 'cells', '(', 'SMCs', ')', 'embedded', 'in', 'collagen', 'gel', 'lattices', ',', 'suggesting', 'that', 'myocardin', 'degradation', 'is']\n",
      "Counter({'NEG': 2})\n",
      "Counter({'NEG': 3, 'POS': 1}) ['accumulation', 'of', 'myocardin', 'in', 'response', 'to', 'proteasome', 'inhibition', 'by', 'MG132', 'or', 'lactacystin', 'resulted', 'in', 'decrease', 'of', 'transcriptional', 'activity', 'of', 'myocardin', 'as', 'indicated', 'by', 'reduced', 'expression', 'of', 'SMC', 'contractile', 'marker', 'genes', '(', 'SM', '-actin', ',', 'SM22', ',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a', ')', ',']\n",
      "['accumulation', 'of', 'myocardin', 'in', 'response', 'to', 'proteasome', 'inhibition', 'by', 'MG132', 'or', 'lactacystin', 'resulted', 'in', 'decrease', 'of', 'transcriptional', 'activity', 'of', 'myocardin', 'as', 'indicated', 'by', 'reduced', 'expression', 'of', 'SMC', 'contractile', 'marker', 'genes', '(', 'SM', '-actin', ',', 'SM22', ',', 'and', 'calponin', ')', 'and', 'muscle-enriched', 'microRNAs', '(', 'miR-143/145', 'and', 'miR-1/133a', ')', ',']\n",
      "Counter({'NEU': 5, 'NEG': 3, 'POS': 1})\n",
      "Counter({'POS': 1, 'NEG': 1}) ['Overexpression', 'of', 'mir-7', 'and', 'mir-153', 'significantly', 'reduces', 'endogenous', 'alpha-synuclein', 'levels']\n",
      "Counter({'POS': 2, 'NEG': 2}) ['Overexpression', 'of', 'mir-7', 'and', 'mir-153', 'significantly', 'reduces', 'endogenous', 'alpha-synuclein', 'levels', ',', 'whereas', 'inhibition', 'of', 'mir-7', 'and', 'mir-153', 'enhances', 'translation', 'of', 'a', 'luciferase', 'construct', 'bearing', 'the', 'alpha-synuclein', \"3'-untranslated\", 'region']\n",
      "['Overexpression', 'of', 'mir-7', 'and', 'mir-153', 'significantly', 'reduces', 'endogenous', 'alpha-synuclein', 'levels', ',', 'whereas', 'inhibition', 'of', 'mir-7', 'and', 'mir-153', 'enhances', 'translation', 'of', 'a', 'luciferase', 'construct', 'bearing', 'the', 'alpha-synuclein', \"3'-untranslated\", 'region']\n",
      "Counter({'POS': 2, 'NEG': 2, 'NEU': 2})\n",
      "Counter({'NEG': 1}) ['mir-7', 'and', 'mir-153', 'significantly', 'reduces', 'endogenous', 'alpha-synuclein', 'levels']\n",
      "['mir-7', 'and', 'mir-153', 'significantly', 'reduces', 'endogenous', 'alpha-synuclein', 'levels']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 2, 'POS': 1}) ['mir-7', 'and', 'mir-153', 'significantly', 'reduces', 'endogenous', 'alpha-synuclein', 'levels', ',', 'whereas', 'inhibition', 'of', 'mir-7', 'and', 'mir-153', 'enhances', 'translation', 'of', 'a', 'luciferase', 'construct', 'bearing', 'the', 'alpha-synuclein', \"3'-untranslated\", 'region']\n",
      "['mir-7', 'and', 'mir-153', 'significantly', 'reduces', 'endogenous', 'alpha-synuclein', 'levels', ',', 'whereas', 'inhibition', 'of', 'mir-7', 'and', 'mir-153', 'enhances', 'translation', 'of', 'a', 'luciferase', 'construct', 'bearing', 'the', 'alpha-synuclein', \"3'-untranslated\", 'region']\n",
      "Counter({'NEG': 2, 'NEU': 2, 'POS': 1})\n",
      "Counter({'NEG': 2}) ['reduces', 'endogenous', 'alpha-synuclein', 'levels', ',', 'whereas', 'inhibition', 'of', 'mir-7', 'and', 'mir-153']\n",
      "['reduces', 'endogenous', 'alpha-synuclein', 'levels', ',', 'whereas', 'inhibition', 'of', 'mir-7', 'and', 'mir-153']\n",
      "Counter({'NEG': 2, 'NEU': 1})\n",
      "Counter({'NEG': 1, 'POS': 1}) ['inhibition', 'of', 'mir-7', 'and', 'mir-153', 'enhances', 'translation', 'of', 'a', 'luciferase', 'construct', 'bearing', 'the', 'alpha-synuclein', \"3'-untranslated\", 'region']\n",
      "Counter({'NEG': 2, 'POS': 1}) ['reduces', 'endogenous', 'alpha-synuclein', 'levels', ',', 'whereas', 'inhibition', 'of', 'mir-7', 'and', 'mir-153', 'enhances', 'translation']\n",
      "Counter({'POS': 1}) ['mir-7', 'and', 'mir-153', 'enhances', 'translation', 'of', 'a', 'luciferase', 'construct', 'bearing', 'the', 'alpha-synuclein', \"3'-untranslated\", 'region']\n",
      "['mir-7', 'and', 'mir-153', 'enhances', 'translation', 'of', 'a', 'luciferase', 'construct', 'bearing', 'the', 'alpha-synuclein', \"3'-untranslated\", 'region']\n",
      "Counter({'POS': 1, 'NEU': 1})\n",
      "Counter() [',', 'namely', 'mir-7', 'and', 'mir-153', ',', 'have', 'been', 'identified', 'to', 'regulate', 'alpha-synuclein', 'levels', 'post-transcriptionally']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Here, two microRNAs, namely mir-7 and mir-153, have been identified to regulate alpha-synuclein levels post-transcriptionally. \n",
      "('mir-7', 'Specific_miRNAs', (28, 33)) ('alpha-synuclein', 'Genes/Proteins', (80, 95))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['mir-7', 'and', 'mir-153', ',', 'have', 'been', 'identified', 'to', 'regulate', 'alpha-synuclein', 'levels', 'post-transcriptionally']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Here, two microRNAs, namely mir-7 and mir-153, have been identified to regulate alpha-synuclein levels post-transcriptionally. \n",
      "('mir-153', 'Specific_miRNAs', (38, 45)) ('alpha-synuclein', 'Genes/Proteins', (80, 95))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['regulation', 'of', 'alpha-synuclein', 'expression', 'by', 'mir-7', 'and', 'mir-153']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Post-transcriptional regulation of alpha-synuclein expression by mir-7 and mir-153.  \n",
      "('mir-7', 'Specific_miRNAs', (65, 70)) ('alpha-synuclein', 'Genes/Proteins', (35, 50))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['regulation', 'of', 'alpha-synuclein', 'expression', 'by', 'mir-7', 'and', 'mir-153']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Post-transcriptional regulation of alpha-synuclein expression by mir-7 and mir-153.  \n",
      "('mir-153', 'Specific_miRNAs', (75, 82)) ('alpha-synuclein', 'Genes/Proteins', (35, 50))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1}) ['rs2507800', 'suppressed', 'angiopoietin-1', 'translation', 'by', 'facilitating', 'miR-211', 'binding', ',']\n",
      "counts between 3 Counter() NEG\n",
      "['rs2507800', 'suppressed', 'angiopoietin-1', 'translation', 'by', 'facilitating', 'miR-211', 'binding', ',']\n",
      "Counter()\n",
      "Counter() [\"3'-UTR\", 'of', 'angiopoietin-1', 'using', 'bioinformatics', ';', 'variant', 'rs2507800', 'was', 'identified', 'to', 'be', 'located', 'in', 'the', 'miR-211-binding', 'site', 'of']\n",
      "Counter({'NEG': 2}) [\"3'-UTR\", 'of', 'angiopoietin-1', 'might', 'reduce', 'the', 'risk', 'of', 'stroke', 'by', 'interfering', 'with', 'miR-211', 'binding']\n",
      "counts between 3 Counter({'NEG': 2}) NEG\n",
      "[\"3'-UTR\", 'of', 'angiopoietin-1', 'might', 'reduce', 'the', 'risk', 'of', 'stroke', 'by', 'interfering', 'with', 'miR-211', 'binding']\n",
      "Counter({'NEG': 2})\n",
      "Counter() ['miR-107', 'contributes', 'to', 'GRN', 'expression', 'regulation']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These findings indicate that miR-107 contributes to GRN expression regulation with implications for brain disorders.   \n",
      "('miR-107', 'Specific_miRNAs', (29, 36)) ('GRN', 'Genes/Proteins', (52, 55))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1, 'POS': 1}) ['decreased', 'miR-107', 'with', 'augmentation', 'of', 'neuronal', 'GRN', 'expression']\n",
      "Counter() ['miR-107', 'has', 'been', 'implicated', 'in', 'Alzheimer', \"'s\", 'disease', 'pathogenesis', ',', 'and', 'sequence', 'elements', 'in', 'the', 'open', 'reading', 'frame-rather', 'than', 'the', '3', \"'\", 'untranslated', 'region-of', 'GRN', 'mRNA', 'are']\n",
      "Counter() ['untranslated', 'region-of', 'GRN', 'mRNA', 'are', 'recognized', 'by', 'miR-107']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of GRN mRNA are recognized by miR-107 and are highly conserved among vertebrate species. \n",
      "('miR-107', 'Specific_miRNAs', (182, 189)) ('GRN', 'Genes/Proteins', (155, 158))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['miR-107', 'regulates', 'granulin/progranulin', 'with', 'implications']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-107 regulates granulin/progranulin with implications for traumatic brain injury and neurodegenerative disease.  \n",
      "('miR-107', 'Specific_miRNAs', (0, 7)) ('granulin', 'Genes/Proteins', (18, 26))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['miR-107', 'regulates', 'granulin/progranulin', 'with', 'implications']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-107 regulates granulin/progranulin with implications for traumatic brain injury and neurodegenerative disease.  \n",
      "('miR-107', 'Specific_miRNAs', (0, 7)) ('progranulin', 'Genes/Proteins', (27, 38))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['with', 'Argonaute', '2', '.', 'In', 'vitro', 'and', 'in', 'vivo', 'studies', 'indicate', 'that', 'regulation', 'of', 'GRN', 'by', 'miR-107', 'may', 'be']\n",
      "Counter() ['GRN', 'mRNA', 'interacts', 'preferentially', 'with', 'Argonaute', '2', '.', 'In', 'vitro', 'and', 'in', 'vivo', 'studies', 'indicate', 'that', 'regulation', 'of', 'GRN', 'by', 'miR-107', 'may', 'be']\n",
      "Counter() ['regulation', 'of', 'GRN', 'by', 'miR-107', 'may', 'be']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "GRN mRNA interacts preferentially with Argonaute 2. In vitro and in vivo studies indicate that regulation of GRN by miR-107 may be functionally important. \n",
      "('miR-107', 'Specific_miRNAs', (116, 123)) ('GRN', 'Genes/Proteins', (109, 112))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'POS': 1, 'NEG': 1}) ['to', 'increased', 'miR-107', 'levels', 'also', 'results', 'in', 'decreased', 'GRN', 'expression', ',']\n",
      "Counter({'POS': 2, 'NEG': 2}) ['to', 'increased', 'miR-107', 'levels', 'also', 'results', 'in', 'decreased', 'GRN', 'expression', ',', 'including', 'changes', 'in', 'cell', 'compartmentation', 'and', 'decreased', 'secretion', 'of', 'GRN', 'protein']\n",
      "['to', 'increased', 'miR-107', 'levels', 'also', 'results', 'in', 'decreased', 'GRN', 'expression', ',', 'including', 'changes', 'in', 'cell', 'compartmentation', 'and', 'decreased', 'secretion', 'of', 'GRN', 'protein']\n",
      "Counter({'NEU': 4, 'POS': 2, 'NEG': 2})\n",
      "Counter({'NEG': 1}) ['MiR-21', 'protected', 'human', 'glioblastoma', 'U87MG', 'cells', 'from', 'chemotherapeutic', 'drug', 'temozolomide', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and', 'caspase-3', 'activity']\n",
      "['MiR-21', 'protected', 'human', 'glioblastoma', 'U87MG', 'cells', 'from', 'chemotherapeutic', 'drug', 'temozolomide', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and', 'caspase-3', 'activity']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1}) ['MiR-21', 'protected', 'human', 'glioblastoma', 'U87MG', 'cells', 'from', 'chemotherapeutic', 'drug', 'temozolomide', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and']\n",
      "['MiR-21', 'protected', 'human', 'glioblastoma', 'U87MG', 'cells', 'from', 'chemotherapeutic', 'drug', 'temozolomide', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and']\n",
      "Counter({'NEG': 1})\n",
      "Counter({'NEG': 1}) ['MiR-21', 'protected', 'human', 'glioblastoma', 'U87MG', 'cells', 'from', 'chemotherapeutic', 'drug', 'temozolomide', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and']\n",
      "['MiR-21', 'protected', 'human', 'glioblastoma', 'U87MG', 'cells', 'from', 'chemotherapeutic', 'drug', 'temozolomide', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and']\n",
      "Counter({'NEG': 1})\n",
      "Counter({'POS': 1, 'NEG': 1}) ['cells', 'with', 'miR-21', 'overexpression', 'before', ',', 'which', 'appeared', 'to', 'downregulate', 'the', 'Bax', 'expression']\n",
      "Counter({'POS': 2, 'NEG': 1}) ['cells', 'with', 'miR-21', 'overexpression', 'before', ',', 'which', 'appeared', 'to', 'downregulate', 'the', 'Bax', 'expression', ',', 'upregulate', 'the', 'Bcl-2', 'expression', 'and']\n",
      "Counter({'POS': 2, 'NEG': 2}) ['cells', 'with', 'miR-21', 'overexpression', 'before', ',', 'which', 'appeared', 'to', 'downregulate', 'the', 'Bax', 'expression', ',', 'upregulate', 'the', 'Bcl-2', 'expression', 'and', 'decrease', 'caspase-3', 'activity', '.']\n",
      "Counter({'NEG': 2, 'POS': 1}) ['that', 'over-express', 'miR-21', 'could', 'inhibit', 'TMZ-induced', 'apoptosis', 'in', 'U87MG', 'cells', ',', 'at', 'least', 'in', 'part', ',', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and']\n",
      "Counter({'NEG': 2, 'POS': 1}) ['that', 'over-express', 'miR-21', 'could', 'inhibit', 'TMZ-induced', 'apoptosis', 'in', 'U87MG', 'cells', ',', 'at', 'least', 'in', 'part', ',', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and']\n",
      "Counter({'NEG': 2, 'POS': 1}) ['that', 'over-express', 'miR-21', 'could', 'inhibit', 'TMZ-induced', 'apoptosis', 'in', 'U87MG', 'cells', ',', 'at', 'least', 'in', 'part', ',', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and', 'caspase-3', 'activity']\n",
      "['that', 'over-express', 'miR-21', 'could', 'inhibit', 'TMZ-induced', 'apoptosis', 'in', 'U87MG', 'cells', ',', 'at', 'least', 'in', 'part', ',', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and', 'caspase-3', 'activity']\n",
      "Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1, 'POS': 1}) ['by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and', 'caspase-3', 'activity', ',', 'which', 'highlighted', 'the', 'possibility', 'of', 'miR-21', 'overexpression', 'in']\n",
      "Counter({'NEG': 1, 'POS': 1}) ['by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and', 'caspase-3', 'activity', ',', 'which', 'highlighted', 'the', 'possibility', 'of', 'miR-21', 'overexpression', 'in']\n",
      "Counter({'POS': 1}) ['ratio', 'and', 'caspase-3', 'activity', ',', 'which', 'highlighted', 'the', 'possibility', 'of', 'miR-21', 'overexpression', 'in']\n",
      "['ratio', 'and', 'caspase-3', 'activity', ',', 'which', 'highlighted', 'the', 'possibility', 'of', 'miR-21', 'overexpression', 'in']\n",
      "Counter({'NEU': 1, 'POS': 1})\n",
      "Counter() ['in', 'vivo', 'ephrin-A3', 'and', 'neuronal', 'pentraxin', '1', 'are', 'potentially', 'regulated', 'by', 'miR-210']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Hypoxia induces microRNA miR-210 in vitro and in vivo ephrin-A3 and neuronal pentraxin 1 are potentially regulated by miR-210.  \n",
      "('miR-210', 'Specific_miRNAs', (118, 125)) ('ephrin-A3', 'Genes/Proteins', (54, 63))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['neuronal', 'pentraxin', '1', 'are', 'potentially', 'regulated', 'by', 'miR-210']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Hypoxia induces microRNA miR-210 in vitro and in vivo ephrin-A3 and neuronal pentraxin 1 are potentially regulated by miR-210.  \n",
      "('miR-210', 'Specific_miRNAs', (118, 125)) ('neuronal pentraxin 1', 'Genes/Proteins', (68, 88))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'POS': 1}) ['induces', 'microRNA', 'miR-210', 'in', 'vitro', 'and', 'in', 'vivo', 'ephrin-A3', 'and', 'neuronal']\n",
      "['induces', 'microRNA', 'miR-210', 'in', 'vitro', 'and', 'in', 'vivo', 'ephrin-A3', 'and', 'neuronal']\n",
      "Counter({'POS': 1})\n",
      "Counter({'POS': 1}) ['induces', 'microRNA', 'miR-210', 'in', 'vitro', 'and', 'in', 'vivo', 'ephrin-A3', 'and', 'neuronal', 'pentraxin', '1', 'are', 'potentially']\n",
      "['induces', 'microRNA', 'miR-210', 'in', 'vitro', 'and', 'in', 'vivo', 'ephrin-A3', 'and', 'neuronal', 'pentraxin', '1', 'are', 'potentially']\n",
      "Counter({'POS': 1})\n",
      "Counter({'POS': 1}) ['of', 'the', 'HIF-1alpha', 'induced', 'the', 'expression', 'of', 'miR-210']\n",
      "['of', 'the', 'HIF-1alpha', 'induced', 'the', 'expression', 'of', 'miR-210']\n",
      "Counter({'POS': 1, 'NEU': 1})\n",
      "Counter() ['expression', 'of', 'miR-210', ',', 'showing', 'the', 'involvement', 'of', 'the', 'HIF-1', 'signaling', 'pathway']\n",
      "Counter({'POS': 1}) ['of', 'the', 'HIF-1alpha', 'induced', 'the', 'expression', 'of', 'miR-210', ',', 'showing', 'the', 'involvement', 'of', 'the', 'HIF-1', 'signaling', 'pathway', 'in', 'miR-210', 'gene', 'transcription']\n",
      "['of', 'the', 'HIF-1alpha', 'induced', 'the', 'expression', 'of', 'miR-210', ',', 'showing', 'the', 'involvement', 'of', 'the', 'HIF-1', 'signaling', 'pathway', 'in', 'miR-210', 'gene', 'transcription']\n",
      "Counter({'NEU': 3, 'POS': 1})\n",
      "Counter() ['of', 'the', 'HIF-1', 'signaling', 'pathway', 'in', 'miR-210', 'gene', 'transcription']\n",
      "Counter() ['Furthermore', ',', 'miR-210', 'was', 'shown', 'to', 'bind', 'to', 'the', 'predicted', 'target', 'sites', 'of', 'ephrin-A3', 'or', 'neuronal']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Furthermore, miR-210 was shown to bind to the predicted target sites of ephrin-A3 or neuronal pentraxin 1, causing repression in luciferase reporter activity. \n",
      "('miR-210', 'Specific_miRNAs', (13, 20)) ('ephrin-A3', 'Genes/Proteins', (72, 81))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['Furthermore', ',', 'miR-210', 'was', 'shown', 'to', 'bind', 'to', 'the', 'predicted', 'target', 'sites', 'of', 'ephrin-A3', 'or', 'neuronal', 'pentraxin', '1']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Furthermore, miR-210 was shown to bind to the predicted target sites of ephrin-A3 or neuronal pentraxin 1, causing repression in luciferase reporter activity. \n",
      "('miR-210', 'Specific_miRNAs', (13, 20)) ('neuronal pentraxin 1', 'Genes/Proteins', (85, 105))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1}) ['miR-124', 'directly', 'inhibited', 'the', 'transcription', 'factor', 'CCAAT/enhancer-binding', 'protein-', '(', 'C/EBP-']\n",
      "['miR-124', 'directly', 'inhibited', 'the', 'transcription', 'factor', 'CCAAT/enhancer-binding', 'protein-', '(', 'C/EBP-']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1}) ['miR-124', 'directly', 'inhibited', 'the', 'transcription', 'factor', 'CCAAT/enhancer-binding', 'protein-', '(', 'C/EBP-', ')', 'and']\n",
      "['miR-124', 'directly', 'inhibited', 'the', 'transcription', 'factor', 'CCAAT/enhancer-binding', 'protein-', '(', 'C/EBP-', ')', 'and']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter({'NEG': 1}) ['miR-124', 'directly', 'inhibited', 'the', 'transcription', 'factor', 'CCAAT/enhancer-binding', 'protein-', '(', 'C/EBP-', ')', 'and', 'its', 'downstream', 'target', 'PU.1']\n",
      "['miR-124', 'directly', 'inhibited', 'the', 'transcription', 'factor', 'CCAAT/enhancer-binding', 'protein-', '(', 'C/EBP-', ')', 'and', 'its', 'downstream', 'target', 'PU.1']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter({'NEG': 1, 'POS': 1}) ['MiR-451', 'levels', 'are', 'regulated', 'by', 'glucose', ';', 'under', 'conditions', 'of', 'abundant', 'energy', 'miR-451', 'expression', 'is', 'high', ',', 'and', 'the', 'suppression', 'of', 'AMPK', 'signaling', 'allows', 'cells', 'to', 'maintain', 'elevated', 'proliferation', 'rates', 'via', 'unrestrained', 'mTOR', 'activation', '.']\n",
      "['MiR-451', 'levels', 'are', 'regulated', 'by', 'glucose', ';', 'under', 'conditions', 'of', 'abundant', 'energy', 'miR-451', 'expression', 'is', 'high', ',', 'and', 'the', 'suppression', 'of', 'AMPK', 'signaling', 'allows', 'cells', 'to', 'maintain', 'elevated', 'proliferation', 'rates', 'via', 'unrestrained', 'mTOR', 'activation', '.']\n",
      "Counter({'NEU': 5, 'NEG': 1, 'POS': 1})\n",
      "Counter({'NEG': 1, 'POS': 1}) ['abundant', 'energy', 'miR-451', 'expression', 'is', 'high', ',', 'and', 'the', 'suppression', 'of', 'AMPK', 'signaling', 'allows', 'cells', 'to', 'maintain', 'elevated', 'proliferation', 'rates', 'via', 'unrestrained', 'mTOR', 'activation', '.']\n",
      "Counter() ['miR-206', 'mediates', 'these', 'effects', 'at', 'least', 'in', 'part', 'through', 'histone', 'deacetylase', '4', 'and', 'fibroblast']\n",
      "Counter() ['CDK6', 'is', 'regulated', 'by', 'microRNA', '124', 'in', 'medulloblastoma']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Our data strongly indicate that CDK6 is regulated by microRNA 124 in medulloblastoma and that miR 124 modulates medulloblastoma cell growth.   \n",
      "('microRNA 124', 'Specific_miRNAs', (53, 65)) ('CDK6', 'Genes/Proteins', (32, 36))\n",
      "('MIR_GENE', 'NEU') return passive? True\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'TRUE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['indicate', 'that', 'CDK6', 'is', 'regulated', 'by', 'microRNA', '124', 'in', 'medulloblastoma', 'and', 'that', 'miR-124', 'modulates', 'medulloblastoma']\n",
      "Counter() ['in', 'the', 'CDK6', 'including', 'microRNA', '124a', ',', 'a']\n",
      "Counter({'NEG': 1}) ['re-expression', 'of', 'miR-124a', 'in', 'medulloblastoma', 'cells', 'decreased', 'expression', 'of', 'CDK6', 'protein']\n",
      "['re-expression', 'of', 'miR-124a', 'in', 'medulloblastoma', 'cells', 'decreased', 'expression', 'of', 'CDK6', 'protein']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "Counter() ['association', 'between', 'miR-124a', 'and', 'CDK6', 'in', 'medulloblastoma']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays. \n",
      "('miR 124a', 'Specific_miRNAs', (31, 39)) ('CDK6', 'Genes/Proteins', (44, 48))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['dependent', 'kinase', '6', 'by', 'microRNA', '124', 'in', 'medulloblastoma']\n",
      "Counter() ['the', 'microRNA', 'miR-128a', ',', 'which', 'can', 'tune', 'paired', 'helical', 'filament', 'Tau', 'levels', 'in']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons. \n",
      "('miR-128a', 'Specific_miRNAs', (73, 81)) ('Tau', 'Genes/Proteins', (122, 125))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['BAG2', 'levels', 'in', 'cells', 'are', 'under', 'the', 'physiological', 'control', 'of', 'the', 'microRNA', 'miR-128a', ',', 'which']\n",
      "Counter({'POS': 1}) ['overexpression', 'of', 'Hsa-miR-30b', ',', 'Hsa-miR-30d', 'and', 'KHDRBS3', 'at', '8q24.22-q24.23']\n",
      "['overexpression', 'of', 'Hsa-miR-30b', ',', 'Hsa-miR-30d', 'and', 'KHDRBS3', 'at', '8q24.22-q24.23']\n",
      "Counter({'POS': 1})\n",
      "Counter() ['Hsa-miR-30b', ',', 'Hsa-miR-30d', 'and', 'KHDRBS3', 'at', '8q24.22-q24.23']\n",
      "Counter() ['expression', 'of', 'hsa-miR-30b', ',', 'hsa-miR-30d', 'and', 'KHDRBS3', 'correlated', 'with']\n",
      "Counter() ['hsa-miR-30b', ',', 'hsa-miR-30d', 'and', 'KHDRBS3', 'correlated', 'with']\n",
      "Counter() ['genes', '(', 'ZFAT1', ',', 'LOC286094', ',', 'KHDRBS3', ')', 'and', 'two', 'genes', 'encoding', 'micro-RNAs', '(', 'hsa-miR-30b', ',', 'hsa-miR-30d']\n",
      "Counter() ['LOC286094', ',', 'KHDRBS3', ')', 'and', 'two', 'genes', 'encoding', 'micro-RNAs', '(', 'hsa-miR-30b', ',', 'hsa-miR-30d']\n",
      "Counter() ['genes', '(', 'ZFAT1', ',', 'LOC286094', ',', 'KHDRBS3', ')', 'and', 'two', 'genes', 'encoding', 'micro-RNAs', '(', 'hsa-miR-30b', ',', 'hsa-miR-30d']\n",
      "Counter() ['LOC286094', ',', 'KHDRBS3', ')', 'and', 'two', 'genes', 'encoding', 'micro-RNAs', '(', 'hsa-miR-30b', ',', 'hsa-miR-30d']\n",
      "Counter() ['data', 'implicate', 'hsa-miR-30b', ',', 'hsa-miR-30d', 'and', 'KHDRBS3', 'as', 'putative']\n",
      "Counter() ['hsa-miR-30b', ',', 'hsa-miR-30d', 'and', 'KHDRBS3', 'as', 'putative']\n",
      "Counter() ['miR-29b', 'and', 'miR-125a', 'regulate', 'podoplanin']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.  \n",
      "('miR-29b', 'Specific_miRNAs', (0, 7)) ('podoplanin', 'Genes/Proteins', (30, 40))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['miR-29b', 'and', 'miR-125a', 'regulate', 'podoplanin']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.  \n",
      "('miR-125a', 'Specific_miRNAs', (12, 20)) ('podoplanin', 'Genes/Proteins', (30, 40))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter() ['focused', 'on', 'miR-29b', 'and', 'miR-125a', ',', 'which', 'were', 'predicted', 'to', 'regulate', 'PDPN']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "In this study, we focused on miR-29b and miR-125a, which were predicted to regulate PDPN, and demonstrated that these microRNAs directly target the 3' untranslated region of PDPN and inhibit invasion, apoptosis, and proliferation of glioblastomas. \n",
      "('miR-29b', 'Specific_miRNAs', (29, 36)) ('PDPN', 'Genes/Proteins', (84, 88))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1}) ['focused', 'on', 'miR-29b', 'and', 'miR-125a', ',', 'which', 'were', 'predicted', 'to', 'regulate', 'PDPN', ',', 'and', 'demonstrated', 'that', 'these', 'microRNAs', 'directly', 'target', 'the', '3', \"'\", 'untranslated', 'region', 'of', 'PDPN', 'and', 'inhibit']\n",
      "['focused', 'on', 'miR-29b', 'and', 'miR-125a', ',', 'which', 'were', 'predicted', 'to', 'regulate', 'PDPN', ',', 'and', 'demonstrated', 'that', 'these', 'microRNAs', 'directly', 'target', 'the', '3', \"'\", 'untranslated', 'region', 'of', 'PDPN', 'and', 'inhibit']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter() ['miR-29b', 'and', 'miR-125a', ',', 'which', 'were', 'predicted', 'to', 'regulate', 'PDPN']\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "In this study, we focused on miR-29b and miR-125a, which were predicted to regulate PDPN, and demonstrated that these microRNAs directly target the 3' untranslated region of PDPN and inhibit invasion, apoptosis, and proliferation of glioblastomas. \n",
      "('miR-125a', 'Specific_miRNAs', (41, 49)) ('PDPN', 'Genes/Proteins', (84, 88))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Counter({'NEG': 1}) ['miR-29b', 'and', 'miR-125a', ',', 'which', 'were', 'predicted', 'to', 'regulate', 'PDPN', ',', 'and', 'demonstrated', 'that', 'these', 'microRNAs', 'directly', 'target', 'the', '3', \"'\", 'untranslated', 'region', 'of', 'PDPN', 'and', 'inhibit']\n",
      "['miR-29b', 'and', 'miR-125a', ',', 'which', 'were', 'predicted', 'to', 'regulate', 'PDPN', ',', 'and', 'demonstrated', 'that', 'these', 'microRNAs', 'directly', 'target', 'the', '3', \"'\", 'untranslated', 'region', 'of', 'PDPN', 'and', 'inhibit']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "Counter({'POS': 1}) ['site', 'for', 'microRNA-433', ',', 'increasing', 'translation', 'of', 'FGF20', 'in', 'vitro']\n",
      "['site', 'for', 'microRNA-433', ',', 'increasing', 'translation', 'of', 'FGF20', 'in', 'vitro']\n",
      "Counter({'POS': 1, 'NEU': 1})\n",
      "Total:      232\n",
      "Correct:    218 0.9396551724137931\n",
      "Incorrect:  14 0.0603448275862069\n",
      "classes Counter({(True, False): 1116, (False, True): 850})\n",
      "Correct classified:      128 0.5517241379310345 0.927536231884058\n",
      "Incorrect classified:    10 0.04310344827586207 0.07246376811594203\n",
      "Random classified:  52 0.22413793103448276 0.37681159420289856\n",
      "Counter({'counts between': 7, 'return': 3})\n",
      "Correct interaction&classified:      121 0.521551724137931 0.8768115942028986\n",
      "Classified by:  Counter({'return': 57, 'counts between': 33, 'compartment mir gene': 12, 'counts opp': 9, 'between gene mir': 9, 'counts between equal': 8, 'between mir gene': 4, 'compartment mir gene neg corr': 4, 'counts between alternating': 1, 'compartment gene mir': 1})\n",
      "[[[132   2]\n",
      "  [  0   4]]\n",
      "\n",
      " [[137   0]\n",
      "  [  0   1]]\n",
      "\n",
      " [[ 83   3]\n",
      "  [  5  47]]\n",
      "\n",
      " [[ 55   2]\n",
      "  [  5  76]]\n",
      "\n",
      " [[135   3]\n",
      "  [  0   0]]]\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      " GENE_MIR_NEU       0.67      1.00      0.80         4\n",
      "  GENE_MIR_UP       1.00      1.00      1.00         1\n",
      "MIR_GENE_DOWN       0.94      0.90      0.92        52\n",
      " MIR_GENE_NEU       0.97      0.94      0.96        81\n",
      "  MIR_GENE_UP       0.00      0.00      0.00         0\n",
      "\n",
      "     accuracy                           0.93       138\n",
      "    macro avg       0.72      0.77      0.74       138\n",
      " weighted avg       0.95      0.93      0.94       138\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "test_results = runCheck(-1, \"TEST\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 We PRON nsubj show 3\n",
      "3 1 show VERB ROOT show 3\n",
      "8 2 that SCONJ mark regulate 31\n",
      "13 3 SRF PROPN nsubj regulate 31\n",
      "17 4 and CCONJ cc SRF 13\n",
      "21 5 myocardin PROPN conj SRF 13\n",
      "31 6 regulate VERB ccomp show 3\n",
      "40 7 a DET det cluster 83\n",
      "42 8 cardiovascular-specific ADJ amod cluster 83\n",
      "66 9 microRNA NOUN compound cluster 83\n",
      "75 10 ( PUNCT punct miRNA 76\n",
      "76 11 miRNA NOUN compound cluster 83\n",
      "81 12 ) PUNCT punct miRNA 76\n",
      "83 13 cluster NOUN dobj regulate 31\n",
      "91 14 encoding VERB acl cluster 83\n",
      "100 15 miR-143 PROPN dobj encoding 91\n",
      "108 16 and CCONJ cc miR-143 100\n",
      "112 17 miR-145 PROPN conj miR-143 100\n",
      "119 18 . PUNCT punct show 3\n",
      "SRF miR-143\n",
      "miR-143 SRF\n",
      "Conjuncts [myocardin, SRF]\n",
      "neighbour myocardin regulate\n",
      "neighbour myocardin regulate\n",
      "Conjunction before fc SRF\n",
      "Conjuncts [SRF, myocardin]\n",
      "neighbour SRF and\n",
      "neighbour SRF and\n",
      "Conjunction before fc myocardin\n",
      "Conjuncts [miR-145, miR-143]\n",
      "neighbour miR-145 .\n",
      "neighbour miR-145 None\n",
      "Conjunction before fc miR-143\n",
      "Conjuncts [miR-143, miR-145]\n",
      "neighbour miR-143 and\n",
      "neighbour miR-143 and\n",
      "Conjunction before fc miR-145\n",
      "Conjunctions\n",
      "SRF [SRF, myocardin]\n",
      "miR-143 [miR-143, miR-145]\n",
      "SDP sent: We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "SDP     : [(SRF, 'PROPN', 'nsubj', False), (regulate, 'VERB', 'ccomp', False), (cluster, 'NOUN', 'dobj', False), (encoding, 'VERB', 'acl', False), (miR-143, 'PROPN', 'dobj', False)]\n",
      "deepcheck connection rule noun found regulate SRF [that, SRF, and, myocardin, regulate, a, cardiovascular-specific, microRNA, (, miRNA, ), cluster, encoding, miR-143, and, miR-145]\n",
      "0 [We, show] 2 0 We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "1 [SRF, and, myocardin, regulate, a, cardiovascular-specific, microRNA, (, miRNA, ), cluster, encoding, miR-143, and, miR-145] 15 13 SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \n",
      "Conjuncts [myocardin, SRF]\n",
      "neighbour myocardin regulate\n",
      "neighbour myocardin regulate\n",
      "Conjunction before fc SRF\n",
      "Conjuncts [SRF, myocardin]\n",
      "neighbour SRF and\n",
      "neighbour SRF and\n",
      "Conjunction before fc myocardin\n",
      "Conjuncts [miR-145, miR-143]\n",
      "neighbour miR-145 .\n",
      "neighbour miR-145 None\n",
      "Conjunction before fc miR-143\n",
      "Conjuncts [miR-143, miR-145]\n",
      "neighbour miR-143 and\n",
      "neighbour miR-143 and\n",
      "Conjunction before fc miR-145\n",
      "(True, {'conj': True, 'sdp': True, 'compartment': True, 'context': True, 'passive': False, 'negation': False})\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"785c7135de7a4a8c857020dcef519fc3-0\" class=\"displacy\" width=\"1650\" height=\"387.0\" direction=\"ltr\" style=\"max-width: none; height: 387.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">We</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">PRON</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"150\">show</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"150\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"250\">that</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"250\">SCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"350\">SRF</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"350\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"450\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"450\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"550\">myocardin</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"550\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"650\">regulate</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"650\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"750\">a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"750\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"850\">cardiovascular-specific</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"850\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"950\">microRNA (</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"950\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1050\">miRNA)</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1050\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1150\">cluster</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1150\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1250\">encoding</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1250\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1350\">miR-143</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1350\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1450\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1450\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"297.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1550\">miR-145.</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1550\">PROPN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-0\" stroke-width=\"2px\" d=\"M62,252.0 62,235.33333333333334 138.0,235.33333333333334 138.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M62,254.0 L58,246.0 66,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-1\" stroke-width=\"2px\" d=\"M262,252.0 262,185.33333333333331 647.0,185.33333333333331 647.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">mark</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M262,254.0 L258,246.0 266,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-2\" stroke-width=\"2px\" d=\"M362,252.0 362,202.0 644.0,202.0 644.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M362,254.0 L358,246.0 366,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-3\" stroke-width=\"2px\" d=\"M362,252.0 362,235.33333333333334 438.0,235.33333333333334 438.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M438.0,254.0 L442.0,246.0 434.0,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-4\" stroke-width=\"2px\" d=\"M362,252.0 362,218.66666666666666 541.0,218.66666666666666 541.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M541.0,254.0 L545.0,246.0 537.0,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-5\" stroke-width=\"2px\" d=\"M162,252.0 162,168.66666666666669 650.0,168.66666666666669 650.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">ccomp</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M650.0,254.0 L654.0,246.0 646.0,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-6\" stroke-width=\"2px\" d=\"M762,252.0 762,185.33333333333331 1147.0,185.33333333333331 1147.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M762,254.0 L758,246.0 766,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-7\" stroke-width=\"2px\" d=\"M862,252.0 862,202.0 1144.0,202.0 1144.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M862,254.0 L858,246.0 866,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-8\" stroke-width=\"2px\" d=\"M962,252.0 962,218.66666666666666 1141.0,218.66666666666666 1141.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M962,254.0 L958,246.0 966,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-9\" stroke-width=\"2px\" d=\"M1062,252.0 1062,235.33333333333334 1138.0,235.33333333333334 1138.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1062,254.0 L1058,246.0 1066,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-10\" stroke-width=\"2px\" d=\"M662,252.0 662,168.66666666666669 1150.0,168.66666666666669 1150.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1150.0,254.0 L1154.0,246.0 1146.0,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-11\" stroke-width=\"2px\" d=\"M1162,252.0 1162,235.33333333333334 1238.0,235.33333333333334 1238.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">acl</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1238.0,254.0 L1242.0,246.0 1234.0,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-12\" stroke-width=\"2px\" d=\"M1262,252.0 1262,235.33333333333334 1338.0,235.33333333333334 1338.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1338.0,254.0 L1342.0,246.0 1334.0,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-13\" stroke-width=\"2px\" d=\"M1362,252.0 1362,235.33333333333334 1438.0,235.33333333333334 1438.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-13\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1438.0,254.0 L1442.0,246.0 1434.0,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-785c7135de7a4a8c857020dcef519fc3-0-14\" stroke-width=\"2px\" d=\"M1362,252.0 1362,218.66666666666666 1541.0,218.66666666666666 1541.0,252.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-785c7135de7a4a8c857020dcef519fc3-0-14\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1541.0,254.0 L1545.0,246.0 1537.0,246.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "analyse(\n",
    "#(\"We also found that miR-326 promoted T(H)-17 differentiation by targeting Ets-1, a negative regulator of T(H)-17 differentiation.\", 4, 10)\n",
    "    #(\"In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\", 4, 10),\n",
    "    #(\"miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 11, 18),\n",
    "    #(\"Downregulation of Pdcd4 by mir-21 facilitates glioblastoma proliferation in vivo.\", 4,2),\n",
    "    (\"We show that SRF and myocardin regulate a cardiovascular-specific microRNA (miRNA) cluster encoding miR-143 and miR-145. \", 3, 15),\n",
    "    100\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "\n",
    "with open(\"scai_test_f1.pickle\", \"wb\") as fout:\n",
    "    pickle.dump(test_results, fout)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading stems\n",
      "Loading relations scai_train_rels.tsv\n",
      "Interactions 397\n",
      "Non NA Interactions 194\n",
      "Strikingly, cells with high levels of PKM2 expressed lower levels of miR-326, suggestive of endogenous regulation of PKM2 by miR-326. \n",
      "('miR-326', 'Specific_miRNAs', (69, 76)) ('PKM2', 'Genes/Proteins', (38, 42))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "counts between 2 Counter({'POS': 2}) POS\n",
      "Antagomir-mediated prevention of mir-145 expression led to an increased protein expression of its downstream target superoxide dismutase-2 in the postischemic brain. \n",
      "('mir-145', 'Specific_miRNAs', (33, 40)) ('superoxide dismutase-2', 'Genes/Proteins', (116, 138))\n",
      "('MIR_GENE', 'DOWN') counts between equal passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Aberrant microRNA expression in the brains of neurodegenerative diseases: miR-29a decreased in Alzheimer disease brains targets neurone navigator 3.  AIMS: MicroRNAs (miRNAs) are small non-coding RNAs that regulate translational repression of target mRNAs. \n",
      "('miR-29a', 'Specific_miRNAs', (74, 81)) ('neurone navigator 3', 'Genes/Proteins', (128, 147))\n",
      "('MIR_GENE', 'DOWN') counts between passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "A strong positive correlation between miRNA-125b abundance and the glial cell markers glial fibrillary acidic protein (GFAP) and vimentin, and CDKN2A down-regulation was noted in advanced Alzheimer's disease (AD) and in Down's syndrome (DS) brain, chronic neurological disorders associated with astrogliosis. \n",
      "('miRNA-125b', 'Specific_miRNAs', (38, 48)) ('CDKN2A', 'Genes/Proteins', (143, 149))\n",
      "('MIR_GENE', 'UP') compartment mir gene pos corr passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "BACE1 mRNA levels tended to increase as miR-107 levels decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA. \n",
      "('miR-107', 'Specific_miRNAs', (40, 47)) ('BACE1', 'Genes/Proteins', (0, 5))\n",
      "('GENE_MIR', 'UP') counts between passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "The expression of microRNA miR-107 decreases early in Alzheimer's disease and may accelerate disease progression through regulation of beta-site amyloid precursor protein-cleaving enzyme 1.  \n",
      "('miR-107', 'Specific_miRNAs', (27, 34)) ('beta-site amyloid precursor protein-cleaving enzyme 1', 'Genes/Proteins', (135, 188))\n",
      "('MIR_GENE', 'DOWN') counts between passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "In human cell lines, we found that miR-205 down-regulates the expression of LRP1 by targeting sequences in the 3'UTR of LRP1 mRNA. \n",
      "('miR-205', 'Specific_miRNAs', (35, 42)) ('C1D', 'Genes/Proteins', (76, 80))\n",
      "('MIR_GENE', 'NEU') compartment mir gene passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.  \n",
      "('miR-29a/b-1', 'Specific_miRNAs', (25, 36)) ('BACE1', 'Genes/Proteins', (95, 100))\n",
      "('MIR_GENE', 'UP') counts between passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.  \n",
      "('miR-29a/b-1', 'Specific_miRNAs', (25, 36)) ('beta-secretase', 'Genes/Proteins', (101, 115))\n",
      "('MIR_GENE', 'UP') counts between passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for miR-659, is a major susceptibility factor for FTLD-U. \n",
      "('miR-659', 'Specific_miRNAs', (170, 177)) ('GRN', 'Genes/Proteins', (144, 147))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression. \n",
      "('miRNA-146a', 'Specific_miRNAs', (136, 146)) ('CFH', 'Genes/Proteins', (211, 214))\n",
      "('GENE_MIR', 'DOWN') counts opp passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP. \n",
      "('miR-630', 'Specific_miRNAs', (4, 11)) ('p27', 'Genes/Proteins', (132, 135))\n",
      "('MIR_GENE', 'UP') counts between passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP. \n",
      "('miR-630', 'Specific_miRNAs', (4, 11)) ('Kip1', 'Genes/Proteins', (136, 140))\n",
      "('MIR_GENE', 'UP') counts between passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line. \n",
      "('miR-106b', 'Specific_miRNAs', (155, 163)) ('Smad6/7', 'Genes/Proteins', (125, 132))\n",
      "('GENE_MIR', 'UP') counts between passive? False\n",
      "('MIR_GENE', 'UP', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "METHODOLOGY/PRINCIPAL FINDINGS: In a screening of MB cell lines, the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1. Down-regulation of HES1 expression by miR-199b-5p negatively regulates the proliferation rate and anchorage-independent growth of MB cells. \n",
      "('miR-199b-5p', 'Specific_miRNAs', (229, 240)) ('C21ORF33', 'Genes/Proteins', (210, 214))\n",
      "('MIR_GENE', 'NEU') compartment gene mir passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "MicroRNA-199b-5p impairs cancer stem cells through negative regulation of HES1 in medulloblastoma.  \n",
      "('MicroRNA-199b-5p', 'Specific_miRNAs', (0, 16)) ('C21ORF33', 'Genes/Proteins', (74, 78))\n",
      "('MIR_GENE', 'UP') between mir gene reg corr passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Because of its known connection to glioma biology, we focused on miR-21 among seven miRNAs influenced by IFN-beta. \n",
      "('miR-21', 'Specific_miRNAs', (65, 71)) ('IFN-beta', 'Genes/Proteins', (105, 113))\n",
      "('MIR_GENE', 'NEU') return m g by passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "We did reporter assays to elucidate the IFN-beta-mediated suppression of miR-21; the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the IFN-beta-mediated suppression of miR-21, whereas STAT3-inhibiting agents inhibited the miR-21 suppression. \n",
      "('miR-21', 'Specific_miRNAs', (222, 228)) ('IFN-beta', 'Genes/Proteins', (189, 197))\n",
      "('MIR_GENE', 'DOWN') counts between equal passive? False\n",
      "('GENE_MIR', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation. \n",
      "('miR-21', 'Specific_miRNAs', (63, 69)) ('IFN-beta', 'Genes/Proteins', (110, 118))\n",
      "('MIR_GENE', 'DOWN') counts between passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "These data suggest a significantly independent regulation of IRAK-1 and IRAK-2 in AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, miRNA-146a-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of IRAK-2 expression drives an extensively sustained inflammatory response. \n",
      "('miRNA-146a', 'Specific_miRNAs', (170, 180)) ('IRAK-2', 'Genes/Proteins', (261, 267))\n",
      "('MIR_GENE', 'DOWN') counts opp passive? False\n",
      "('MIR_GENE', 'UP', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression. \n",
      "('miRNA-146a', 'Specific_miRNAs', (39, 49)) ('IRAK1', 'Genes/Proteins', (167, 171))\n",
      "('MIR_GENE', 'UP') counts between passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "We show that the expression of miR-34a is inversely correlated with the protein level of bcl2 in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells. \n",
      "('miR-34a', 'Specific_miRNAs', (31, 38)) ('bcl2', 'Genes/Proteins', (89, 93))\n",
      "('MIR_GENE', 'DOWN') compartment mir gene neg corr passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "counts between 2 Counter({'NEG': 1}) NEG\n",
      "Total:      397\n",
      "Correct:    360 0.906801007556675\n",
      "Incorrect:  37 0.09319899244332494\n",
      "classes Counter({(True, False): 176, (False, True): 147})\n",
      "Correct classified:      172 0.4332493702770781 0.8865979381443299\n",
      "Incorrect classified:    22 0.055415617128463476 0.1134020618556701\n",
      "Random classified:  53 0.13350125944584382 0.27319587628865977\n",
      "Counter({'counts between': 10, 'return': 2, 'counts between equal': 2, 'counts opp': 2, 'compartment mir gene pos corr': 1, 'compartment mir gene': 1, 'compartment gene mir': 1, 'between mir gene reg corr': 1, 'return m g by': 1, 'compartment mir gene neg corr': 1})\n",
      "Correct interaction&classified:      161 0.40554156171284633 0.8298969072164949\n",
      "Classified by:  Counter({'return': 66, 'counts between': 41, 'counts between equal': 16, 'compartment mir gene': 14, 'compartment gene mir': 12, 'between gene mir': 12, 'counts opp': 9, 'between mir gene': 7, 'compartment mir gene pos corr': 4, 'return m g by': 3, 'compartment mir gene neg corr': 3, 'return mediated': 2, 'between gene mir sites': 2, 'between mir gene reg corr': 1, 'between mir gene nrb': 1, 'counts between alternating': 1})\n",
      "[[[186   1]\n",
      "  [  1   6]]\n",
      "\n",
      " [[191   2]\n",
      "  [  1   0]]\n",
      "\n",
      " [[188   2]\n",
      "  [  0   4]]\n",
      "\n",
      " [[134   7]\n",
      "  [  9  44]]\n",
      "\n",
      " [[ 72   3]\n",
      "  [  9 110]]\n",
      "\n",
      " [[177   7]\n",
      "  [  2   8]]]\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "GENE_MIR_DOWN       0.86      0.86      0.86         7\n",
      " GENE_MIR_NEU       0.00      0.00      0.00         1\n",
      "  GENE_MIR_UP       0.67      1.00      0.80         4\n",
      "MIR_GENE_DOWN       0.86      0.83      0.85        53\n",
      " MIR_GENE_NEU       0.97      0.92      0.95       119\n",
      "  MIR_GENE_UP       0.53      0.80      0.64        10\n",
      "\n",
      "     accuracy                           0.89       194\n",
      "    macro avg       0.65      0.74      0.68       194\n",
      " weighted avg       0.90      0.89      0.89       194\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "runCheck(-1, \"TRAIN\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def getTokenDist(tk1, tk2):\n",
    "    return abs(tk1.i-tk2.i)\n",
    "\n",
    "def countStems(tokens, geneword, mirword, keepNEU=False):\n",
    "    stemTypes2Count = Counter()\n",
    "    mirgene2stem = defaultdict(lambda: Counter())\n",
    "\n",
    "    for tidx,x in enumerate(tokens):\n",
    "        tknStem, matchStem, wasFound = getstem(x)\n",
    "\n",
    "        if wasFound and (tknStem != \"NEU\" or keepNEU):\n",
    "            \n",
    "            if tidx >= len(tokens)-2 and str(tokens[tidx-1]) == \"to\":\n",
    "                continue\n",
    "            \n",
    "            geneDist = getTokenDist(x, geneword)\n",
    "            mirDist = getTokenDist(x, mirword)\n",
    "                        \n",
    "            if geneDist < mirDist and geneDist < 3:\n",
    "                mirgene2stem[\"gene\"][tknStem] += 1\n",
    "            elif mirDist < geneDist and mirDist < 3:\n",
    "                mirgene2stem[\"mir\"][tknStem] += 1\n",
    "\n",
    "        if wasFound and (tknStem != \"NEU\" or keepNEU):\n",
    "            stemTypes2Count[tknStem] += 1\n",
    "\n",
    "    if len(stemTypes2Count) == 0:\n",
    "        if \"regulated\" in str(geneword):\n",
    "            stemTypes2Count[\"NEU\"] += 1\n",
    "\n",
    "    return stemTypes2Count,mirgene2stem\n",
    "\n",
    "def getRelsForTokens(tokens, desc,verbose):\n",
    "    rels = []\n",
    "    verbSet = set([\"VERB\"])\n",
    "    for x in tokens:\n",
    "        \n",
    "        tokenStemClass, s, _ = getstem(x)\n",
    "        rels.append((s, None, tokenStemClass, x.pos_, x))\n",
    "        if verbose:\n",
    "            print(desc, x, s, tokenStemClass, x.pos_, x)\n",
    "                    \n",
    "    hasVerb = len(list(set([x[3] for x in rels]) & verbSet)) > 0\n",
    "    \n",
    "    if hasVerb:\n",
    "        rels = [x for x in rels if x[3] in verbSet]\n",
    "    \n",
    "    return rels\n",
    "\n",
    "def evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg):\n",
    "\n",
    "    invcor = False\n",
    "    regdir = None\n",
    "\n",
    "    #if dirReg == 0:\n",
    "    if (mirPos > 0 and geneNeg > 0) or (genePos > 0 and mirNeg > 0):\n",
    "        return \"DOWN\", True\n",
    "\n",
    "    #else:\n",
    "    if ((mirPos > 0 and genePos > 0) and(mirNeg == 0 and geneNeg == 0)) or ((mirNeg > 0 and geneNeg >0) and(mirPos == 0 and genePos == 0)):\n",
    "        return \"UP\", False\n",
    "\n",
    "    return None, None\n",
    "\n",
    "def scoreWord(tkn):\n",
    "    \n",
    "    jtkstem, s, _ = getstem(tkn)\n",
    "                        \n",
    "    if jtkstem == None:\n",
    "\n",
    "        if jtk.startswith((\"loss\", \"decreas\", \"low\")):\n",
    "            return 0,1\n",
    "\n",
    "        elif jtk.startswith((\"increas\", \"high\", \"posit\")):\n",
    "            return 1,0\n",
    "\n",
    "    else:\n",
    "        if jtkstem == \"POS\":\n",
    "            return 1,0\n",
    "        elif jtkstem == \"NEG\":\n",
    "            return 0,1\n",
    "        \n",
    "    return 0,0\n",
    "\n",
    "\n",
    "def getTextBetween(tks, lword, rword):\n",
    "    tokenBetween = [x for x in tks if lword.i-3 < x.i < rword.i+3]\n",
    "    \n",
    "    lit = lword\n",
    "    while lit.pos_ in [\"nmod\", \"conj\"]:\n",
    "        if not lit in tokenBetween:\n",
    "            tokenBetween.append(lit)\n",
    "            \n",
    "        lit = lit.head\n",
    "        \n",
    "    rit = rword\n",
    "    while rit.pos_ in [\"nmod\", \"conj\"]:\n",
    "        if not rit in tokenBetween:\n",
    "            tokenBetween.append(rit)\n",
    "            \n",
    "        rit = rit.head\n",
    "\n",
    "\n",
    "    tokenBetween = list(sorted(tokenBetween, key=lambda x: x.i))\n",
    "    tokenBetweenStrict = list(sorted([x for x in tks if lword.i < x.i < rword.i], key=lambda x: x.i))\n",
    "    textBetween = ' '.join(\" \".join([str(x) for x in tokenBetween]).split())\n",
    "    textAfter = ' '.join(\" \".join([str(x) for x in [y for y in rword.doc if rword.i <= y.i < rword.i+5]]).split())\n",
    "    textBefore = ' '.join(\" \".join([str(x) for x in [y for y in rword.doc if lword.i-5 <= y.i < lword.i]]).split())\n",
    "\n",
    "    return tokenBetween, tokenBetweenStrict, textBetween, textBefore, textAfter\n",
    "\n",
    "dir2opp = {\"MIR_GENE\": \"GENE_MIR\", \"GENE_MIR\": \"MIR_GENE\"}\n",
    "idir2opp = {\"POS\": \"NEG\", \"NEG\": \"POS\"}\n",
    "rdir2opp = {\"DOWN\": \"UP\", \"UP\": \"DOWN\"}\n",
    "\n",
    "def testRelClassifier2(doc, mirword, geneword, relchecker, verbose):\n",
    "    isPassive = relchecker.checkPassive(doc, mirword, geneword, verbose=False)\n",
    "    compartment = relchecker.getCompartment(doc, mirword, geneword, verbose=False)\n",
    "    \n",
    "    if verbose:\n",
    "        print(\"passive?\", isPassive)\n",
    "        \n",
    "        \n",
    "    sdpRes = relchecker._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "        \n",
    "    stemTypes2Count = Counter()\n",
    "    \n",
    "    for x in sdpRes:\n",
    "        tknStem, matchStem, _ = getstem(x[0])\n",
    "        \n",
    "        if tknStem != \"NEU\":\n",
    "            stemTypes2Count[tknStem] += 1\n",
    "            \n",
    "    print(stemTypes2Count)\n",
    "        \n",
    "    if len(stemTypes2Count) > 0:\n",
    "        regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "        \n",
    "        if regStem == \"NEG\":\n",
    "            regStem = \"DOWN\"\n",
    "        elif regStem == \"POS\":\n",
    "            regStem = \"UP\"\n",
    "            \n",
    "        if stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"]:\n",
    "            regStem = \"DOWN\"\n",
    "            \n",
    "        idir = \"MIR_GENE\"\n",
    "        if isPassive and mirword.i < geneword.i:\n",
    "            idir = \"GENE_MIR\"\n",
    "            \n",
    "        #return {\"regulation_dir\": regStem, \"interaction_dir\": idir, \"passive\": isPassive, \"reg_detect\": \"sdp counts\"}\n",
    "        \n",
    "    targetWords = [\"targets of\", \"target of\", \"regulator of\", \"targeting\", \"direct regulator\"]\n",
    "        \n",
    "    if compartment != None:\n",
    "        tokenBetween = compartment\n",
    "        textBetween = ' '.join(\" \".join([str(x) for x in tokenBetween]).split())\n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"compartment gene mir pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"compartment gene mir neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment gene mir\"}\n",
    "            \n",
    "        elif mirword.i < geneword.i:\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment mir gene pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment mir gene neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"compartment mir gene\"}\n",
    "        \n",
    "\n",
    "\n",
    "    if True:\n",
    "        \n",
    "        tokSrc = compartment if compartment != None else doc\n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "\n",
    "            tokenBetween, tokenBetweenStrict, textBetween, textBefore, textAfter = getTextBetween(tokSrc, geneword, mirword)\n",
    "\n",
    "            print(textBetween)\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between gene mir pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between gene mir neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"negative regulation of\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between gene mir reg corr\"}\n",
    "      \n",
    "            if any([x in textBefore for x in [\"binding site\", \"binding sites\", \"binding to\", \"binding directly to\"]]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between gene mir sites\"}\n",
    "            \n",
    "            \n",
    "    \n",
    "            if any([x in textBetween for x in targetWords]) or any([x in textBetween for x in [\"target\", \"targeted by\", \"binding efficiency\", \"binding site for\", \"recognizing\"]]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between gene mir\"}\n",
    "            \n",
    "\n",
    "            #stemTypes2Count = Counter()\n",
    "            #for x in tokenBetween:\n",
    "            #    tknStem, matchStem = getstem(x)\n",
    "            #    if tknStem != \"NEU\":\n",
    "            #        stemTypes2Count[tknStem] += 1\n",
    "                    \n",
    "            if False and len(stemTypes2Count) > 0:\n",
    "                regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                if regStem == \"NEG\":\n",
    "                    regStem = \"DOWN\"\n",
    "                elif regStem == \"POS\":\n",
    "                    regStem = \"UP\"\n",
    "                    \n",
    "                if stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"]:\n",
    "                    regStem = \"DOWN\"\n",
    "\n",
    "                idir = \"MIR_GENE\"\n",
    "                if isPassive and geneword.i < mirword.i:\n",
    "                    idir = \"GENE_MIR\"\n",
    "\n",
    "                return {\"regulation_dir\": regStem, \"interaction_dir\": idir, \"passive\": isPassive, \"reg_detect\": \"counts between gene mir\"}\n",
    "            \n",
    "        else:\n",
    "            #mirword.i < geneword.i\n",
    "            tokenBetween, tokenBetweenStrict, textBetween, textBefore, textAfter = getTextBetween(tokSrc, mirword, geneword)\n",
    "                        \n",
    "            print(textBetween)\n",
    "            \n",
    "            if any([x in textBetween for x in [\"positive correlation\"]]) and stemTypes2Count.get(\"NEG\", 0) == 0:\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene pos corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"negative regulation of\"]]):\n",
    "                return {\"regulation_dir\": \"UP\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene reg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"inverse correlation\", \"inversely correlated\", \"inversely regulated\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene neg corr\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"directly inhibits\", \"directly represses\", \"translational inhibition\" ]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene\"}\n",
    "            \n",
    "            if any([x in textBetween for x in targetWords + [\"directly regulated\", \"its target gene\", \"binding site\"]]):\n",
    "                return {\"regulation_dir\": \"NEU\", \"interaction_dir\": \"MIR_GENE\", \"passive\": isPassive, \"reg_detect\": \"between mir gene\"}\n",
    "            \n",
    "            if any([x in textBetween for x in [\"negatively regulated by\", \"negative regulation of\"]]):\n",
    "                return {\"regulation_dir\": \"DOWN\", \"interaction_dir\": \"GENE_MIR\", \"passive\": isPassive, \"reg_detect\": \"between mir gene nrb\"}\n",
    "            \n",
    "            \n",
    "        if len(tokenBetween) > 0:\n",
    "                       \n",
    "            stemTypes2Count,mirgene2stem = countStems(tokenBetween, geneword, mirword)\n",
    "                    \n",
    "            if len(stemTypes2Count) > 0:\n",
    "                regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                \n",
    "                if len(stemTypes2Count) >= 2 and len(mirgene2stem[\"gene\"]) >0 and len(mirgene2stem[\"mir\"])>0 and mirgene2stem[\"gene\"].most_common(1)[0][0] == idir2opp[mirgene2stem[\"mir\"].most_common(1)[0][0]]:\n",
    "                    print(mirgene2stem)\n",
    "                    if geneword.i < mirword.i:\n",
    "                        if mirgene2stem[\"gene\"].most_common(1)[0][0] == \"POS\":\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                        else:\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                    else:\n",
    "                                                \n",
    "                        if mirgene2stem[\"mir\"].most_common(1)[0][0] == \"NEG\":\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                                                        \n",
    "                        else:\n",
    "                            regDir = \"MIR_GENE\"\n",
    "                            regStem = \"DOWN\"\n",
    "                            \n",
    "                        if \"after\" in textBetween:\n",
    "                            regDir = dir2opp[regDir]\n",
    "                            \n",
    "                    if isPassive:\n",
    "                        regDir = dir2opp[regDir]\n",
    "                        regStem = rdir2opp[regStem]\n",
    "                        \n",
    "                    return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts opp\"}\n",
    "                    \n",
    "                    \n",
    "                else:\n",
    "                    stemTypes2Count,mirgene2stem = countStems(tokenBetween, geneword, mirword, keepNEU=True)\n",
    "                    print(\"counts between 1\", stemTypes2Count)\n",
    "                    regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                    \n",
    "                    if stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"] == 1 and stemTypes2Count[\"NEU\"] <= 3:\n",
    "                        regDir = \"MIR_GENE\"\n",
    "                        regStem = \"DOWN\"\n",
    "                        \n",
    "                        if any(x in textBetween for x in [\"after\"]):\n",
    "                            regDir = dir2opp[regDir]\n",
    "                        \n",
    "                        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts between equal\"}\n",
    "                    \n",
    "                    if stemTypes2Count[\"POS\"] == 2 and stemTypes2Count[\"NEG\"] == 1 and stemTypes2Count[\"NEU\"] <= 2:\n",
    "                        regDir = \"MIR_GENE\"\n",
    "                        regStem = \"DOWN\"\n",
    "                        \n",
    "                        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts between alternating\"}\n",
    "                    \n",
    "                    if regStem == \"NEU\":\n",
    "                        if stemTypes2Count[\"POS\"] > 0 and stemTypes2Count[\"POS\"] > stemTypes2Count[\"NEG\"]:\n",
    "                            regStem = \"POS\"\n",
    "                            \n",
    "                        elif stemTypes2Count[\"NEG\"] > 0 and stemTypes2Count[\"NEG\"] > stemTypes2Count[\"POS\"]:\n",
    "                            regStem = \"NEG\"\n",
    "                        \n",
    "                    \n",
    "                    #regStem = \"NEU\"\n",
    "\n",
    "                    if mirword.i < geneword.i:\n",
    "                        regDir = \"MIR_GENE\"\n",
    "                    else:\n",
    "                        regDir = \"GENE_MIR\"\n",
    "                        \n",
    "                    if mirword.i < geneword.i and \"-stressed\" in textBetween:\n",
    "                        regDir = \"GENE_MIR\"\n",
    "                        \n",
    "                    if mirword.i < geneword.i and str(doc[geneword.i-1]) in [\"by\", \"with\"]:\n",
    "                        regDir = \"GENE_MIR\"\n",
    "\n",
    "                        stemTypes2Count,mirgene2stem = countStems(tokenBetweenStrict, geneword, mirword)\n",
    "\n",
    "                        if len(stemTypes2Count) > 0:\n",
    "                            regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                        #else:\n",
    "                        #    regStem = \"NEU\"\n",
    "                        print(\"counts between 2\", stemTypes2Count, regStem)\n",
    "                            \n",
    "                    elif geneword.i < mirword.i and any([str(doc[mirword.i-j]) in [\"by\", \"with\"] for j in [1,2]]):\n",
    "                        regDir = \"MIR_GENE\"\n",
    "\n",
    "                        stemTypes2Count,mirgene2stem = countStems(tokenBetweenStrict, geneword, mirword)\n",
    "                        if len(stemTypes2Count) > 0:\n",
    "                            regStem = stemTypes2Count.most_common(1)[0][0]\n",
    "                        #else:\n",
    "                        #    regStem = \"NEU\"\n",
    "                        print(\"counts between 2\", stemTypes2Count, regStem)\n",
    "                            \n",
    "                    elif isPassive and geneword.i < mirword.i:\n",
    "                        regDir = dir2opp[regDir]\n",
    "\n",
    "                    if stemTypes2Count[\"POS\"] > 0 and stemTypes2Count[\"POS\"] == stemTypes2Count[\"NEG\"]:\n",
    "                        regStem = \"NEU\"                       \n",
    "                        \n",
    "                    \n",
    "                        #regStem = idir2opp[regStem]\n",
    "                    \n",
    "                    if geneword.i < mirword.i:\n",
    "                        if str(geneword).endswith(\"-regulating\"):\n",
    "                            regStem = \"NEU\"\n",
    "                            \n",
    "                    if mirword.i < geneword.i and \"after the addition\" in textBetween:\n",
    "                        regStem = \"DOWN\"\n",
    "                        regDir = \"GENE_MIR\"\n",
    "                            \n",
    "                        \n",
    "                    if regStem == \"NEG\":\n",
    "                        regStem = \"DOWN\"\n",
    "                    elif regStem == \"POS\":\n",
    "                        regStem = \"UP\"\n",
    "\n",
    "                        \n",
    "                    \n",
    "                    return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"counts between\"}\n",
    "    \n",
    "        \n",
    "    regDir = \"MIR_GENE\"\n",
    "    regStem = \"NEU\"\n",
    "    \n",
    "    regeluateWords = [\"mediated\",]\n",
    "    \n",
    "    if mirword.i < geneword.i and \"mediated\" in str(mirword):\n",
    "        regStem = \"DOWN\"\n",
    "        \n",
    "    if geneword.i < mirword.i and (\"mediated\" in str(mirword) or any([str(doc[mirword.i-j]) in [\"mediated\",] for j in [1]])):\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "        \n",
    "        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"return mediated\"}\n",
    "        \n",
    "    if mirword.i<geneword.i and any([str(doc[geneword.i-j]) in [\"by\", \"with\"] for j in [1,2,3]]):\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "        \n",
    "        return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"return m g by\"}\n",
    "        \n",
    "    if geneword.i < mirword.i:\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"GENE_MIR\"\n",
    "    \n",
    "    if geneword.i < mirword.i and any([str(doc[mirword.i-j]) in [\"by\", \"with\"] for j in [1,2,3]]):\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "        \n",
    "    if mirword.i < geneword.i and str(doc[mirword.i+1]) == \"target\":\n",
    "        regStem = \"NEU\"\n",
    "        regDir = \"MIR_GENE\"\n",
    "    \n",
    "    #if mirword.i < geneword.i:\n",
    "    #    regDir = \"MIR_GENE\"\n",
    "    #else:\n",
    "    #    regDir = \"GENE_MIR\"\n",
    "    \n",
    "    #if isPassive:\n",
    "    #    if mirword.i < geneword.i:\n",
    "    #        regDir = \"GENE_MIR\"\n",
    "    #    else:\n",
    "    #        regDir = \"MIR_GENE\"\n",
    "        \n",
    "    return {\"regulation_dir\": regStem, \"interaction_dir\": regDir, \"passive\": isPassive, \"reg_detect\": \"return\"}\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getstem(tkn):\n",
    "    tknstr = str(tkn).lower()\n",
    "    for sidx, s in enumerate(stem2class):\n",
    "        if tknstr.startswith(s):\n",
    "            \n",
    "            if tknstr in [\"enhancer\",\"inhibitor\"]:\n",
    "                continue\n",
    "\n",
    "            tokenStemClass = stem2class[s]\n",
    "            \n",
    "            if tokenStemClass == \"NEU\":\n",
    "                for i in range(1,2):\n",
    "\n",
    "                    if tkn.i-i<0:\n",
    "                        continue\n",
    "\n",
    "                    #tknPre = str(tkn.doc[tkn.i-i])\n",
    "                    #if tknPre.startswith((\"high\", \"elevat\", \"increas\")):\n",
    "                    #    tokenStemClass = \"POS\"\n",
    "\n",
    "                    #if tknPre.startswith((\"low\", \"decreas\")):\n",
    "                    #    tokenStemClass = \"NEG\"\n",
    "                                                \n",
    "            return tokenStemClass, s, True\n",
    "        \n",
    "        elif tknstr.endswith((\"-stressed\")):\n",
    "            return \"POS\", s, True\n",
    "        \n",
    "    return \"NEU\", str(tkn), False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading stems\n",
      "NEG\n",
      "True\n",
      "expressed ('NEU', 'express', True)\n",
      ", ('NEU', ',', False)\n",
      "miR-34a ('NEU', 'miR-34a', False)\n",
      ", ('NEU', ',', False)\n",
      "was ('NEU', 'was', False)\n",
      "studied ('NEU', 'studied', False)\n",
      "further ('NEU', 'further', False)\n",
      "because ('NEU', 'because', False)\n",
      "sequence ('NEU', 'sequence', False)\n",
      "analysis ('NEU', 'analysis', False)\n",
      "suggested ('NEU', 'suggested', False)\n",
      "a ('NEU', 'a', False)\n",
      "likely ('NEU', 'likely', False)\n",
      "interaction ('NEU', 'interact', True)\n",
      "with ('NEU', 'with', False)\n",
      "the ('NEU', 'the', False)\n",
      "3'-untranslated ('NEU', \"3'-untranslated\", False)\n",
      "region ('NEU', 'region', False)\n",
      "of ('NEU', 'of', False)\n",
      "bcl2 ('NEU', 'bcl2', False)\n",
      "mRNA ('NEU', 'mRNA', False)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(Counter(),\n",
       " defaultdict(<function __main__.countStems.<locals>.<lambda>()>, {}))"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allStems, allRels, stem2class = loadStems()\n",
    "\n",
    "print(stem2class[\"underexpress\"])\n",
    "print(\"underexpression\".startswith(\"underexpress\"))\n",
    "\n",
    "doct = nlp(\"expressed , miR-34a , was studied further because sequence analysis suggested a likely interaction with the 3'-untranslated region of bcl2 mRNA\")\n",
    "\n",
    "toks = [x for x in doct]\n",
    "\n",
    "for tok in toks:\n",
    "    print(tok, getstem(tok))\n",
    "\n",
    "countStems(toks, toks[0], toks[-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading stems\n",
      "Loading relations scai_train_rels.tsv\n",
      "Interactions 397\n",
      "Non NA Interactions 194\n",
      "Strikingly, cells with high levels of PKM2 expressed lower levels of miR-326, suggestive of endogenous regulation of PKM2 by miR-326. \n",
      "('miR-326', 'Specific_miRNAs', (69, 76)) ('PKM2', 'Genes/Proteins', (38, 42))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "counts between 2 Counter({'POS': 2}) POS\n",
      "Antagomir-mediated prevention of mir-145 expression led to an increased protein expression of its downstream target superoxide dismutase-2 in the postischemic brain. \n",
      "('mir-145', 'Specific_miRNAs', (33, 40)) ('superoxide dismutase-2', 'Genes/Proteins', (116, 138))\n",
      "('MIR_GENE', 'DOWN') counts between equal passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Aberrant microRNA expression in the brains of neurodegenerative diseases: miR-29a decreased in Alzheimer disease brains targets neurone navigator 3.  AIMS: MicroRNAs (miRNAs) are small non-coding RNAs that regulate translational repression of target mRNAs. \n",
      "('miR-29a', 'Specific_miRNAs', (74, 81)) ('neurone navigator 3', 'Genes/Proteins', (128, 147))\n",
      "('MIR_GENE', 'DOWN') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "A strong positive correlation between miRNA-125b abundance and the glial cell markers glial fibrillary acidic protein (GFAP) and vimentin, and CDKN2A down-regulation was noted in advanced Alzheimer's disease (AD) and in Down's syndrome (DS) brain, chronic neurological disorders associated with astrogliosis. \n",
      "('miRNA-125b', 'Specific_miRNAs', (38, 48)) ('CDKN2A', 'Genes/Proteins', (143, 149))\n",
      "('MIR_GENE', 'UP') compartment mir gene pos corr passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "BACE1 mRNA levels tended to increase as miR-107 levels decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA. \n",
      "('miR-107', 'Specific_miRNAs', (40, 47)) ('BACE1', 'Genes/Proteins', (0, 5))\n",
      "('GENE_MIR', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "The expression of microRNA miR-107 decreases early in Alzheimer's disease and may accelerate disease progression through regulation of beta-site amyloid precursor protein-cleaving enzyme 1.  \n",
      "('miR-107', 'Specific_miRNAs', (27, 34)) ('beta-site amyloid precursor protein-cleaving enzyme 1', 'Genes/Proteins', (135, 188))\n",
      "('MIR_GENE', 'DOWN') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "In human cell lines, we found that miR-205 down-regulates the expression of LRP1 by targeting sequences in the 3'UTR of LRP1 mRNA. \n",
      "('miR-205', 'Specific_miRNAs', (35, 42)) ('C1D', 'Genes/Proteins', (76, 80))\n",
      "('MIR_GENE', 'NEU') compartment mir gene passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.  \n",
      "('miR-29a/b-1', 'Specific_miRNAs', (25, 36)) ('BACE1', 'Genes/Proteins', (95, 100))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.  \n",
      "('miR-29a/b-1', 'Specific_miRNAs', (25, 36)) ('beta-secretase', 'Genes/Proteins', (101, 115))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for miR-659, is a major susceptibility factor for FTLD-U. \n",
      "('miR-659', 'Specific_miRNAs', (170, 177)) ('GRN', 'Genes/Proteins', (144, 147))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression. \n",
      "('miRNA-146a', 'Specific_miRNAs', (136, 146)) ('CFH', 'Genes/Proteins', (211, 214))\n",
      "('GENE_MIR', 'DOWN') counts opp passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP. \n",
      "('miR-630', 'Specific_miRNAs', (4, 11)) ('p27', 'Genes/Proteins', (132, 135))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP. \n",
      "('miR-630', 'Specific_miRNAs', (4, 11)) ('Kip1', 'Genes/Proteins', (136, 140))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line. \n",
      "('miR-106b', 'Specific_miRNAs', (155, 163)) ('Smad6/7', 'Genes/Proteins', (125, 132))\n",
      "('GENE_MIR', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'UP', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "METHODOLOGY/PRINCIPAL FINDINGS: In a screening of MB cell lines, the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1. Down-regulation of HES1 expression by miR-199b-5p negatively regulates the proliferation rate and anchorage-independent growth of MB cells. \n",
      "('miR-199b-5p', 'Specific_miRNAs', (229, 240)) ('C21ORF33', 'Genes/Proteins', (210, 214))\n",
      "('MIR_GENE', 'NEU') compartment gene mir passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "MicroRNA-199b-5p impairs cancer stem cells through negative regulation of HES1 in medulloblastoma.  \n",
      "('MicroRNA-199b-5p', 'Specific_miRNAs', (0, 16)) ('C21ORF33', 'Genes/Proteins', (74, 78))\n",
      "('MIR_GENE', 'UP') between mir gene reg corr passive? False\n",
      "('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Because of its known connection to glioma biology, we focused on miR-21 among seven miRNAs influenced by IFN-beta. \n",
      "('miR-21', 'Specific_miRNAs', (65, 71)) ('IFN-beta', 'Genes/Proteins', (105, 113))\n",
      "('MIR_GENE', 'NEU') return m g by passive? False\n",
      "('GENE_MIR', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "We did reporter assays to elucidate the IFN-beta-mediated suppression of miR-21; the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the IFN-beta-mediated suppression of miR-21, whereas STAT3-inhibiting agents inhibited the miR-21 suppression. \n",
      "('miR-21', 'Specific_miRNAs', (222, 228)) ('IFN-beta', 'Genes/Proteins', (189, 197))\n",
      "('MIR_GENE', 'DOWN') counts between equal passive? False\n",
      "('GENE_MIR', 'DOWN', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation. \n",
      "('miR-21', 'Specific_miRNAs', (63, 69)) ('IFN-beta', 'Genes/Proteins', (110, 118))\n",
      "('MIR_GENE', 'DOWN') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "These data suggest a significantly independent regulation of IRAK-1 and IRAK-2 in AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, miRNA-146a-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of IRAK-2 expression drives an extensively sustained inflammatory response. \n",
      "('miRNA-146a', 'Specific_miRNAs', (170, 180)) ('IRAK-2', 'Genes/Proteins', (261, 267))\n",
      "('MIR_GENE', 'DOWN') counts opp passive? False\n",
      "('MIR_GENE', 'UP', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression. \n",
      "('miRNA-146a', 'Specific_miRNAs', (39, 49)) ('IRAK1', 'Genes/Proteins', (167, 171))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "We show that the expression of miR-34a is inversely correlated with the protein level of bcl2 in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells. \n",
      "('miR-34a', 'Specific_miRNAs', (31, 38)) ('bcl2', 'Genes/Proteins', (89, 93))\n",
      "('MIR_GENE', 'DOWN') compartment mir gene neg corr passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "counts between 2 Counter({'NEG': 1}) NEG\n",
      "Total:      397\n",
      "Correct:    360 0.906801007556675\n",
      "Incorrect:  37 0.09319899244332494\n",
      "classes Counter({(True, False): 28, (False, True): 25})\n",
      "Correct classified:    172 0.4332493702770781\n",
      "Incorrect classified:    22 0.055415617128463476\n",
      "Counter({'counts between ': 10, 'return': 2, 'counts between equal': 2, 'counts opp': 2, 'compartment mir gene pos corr': 1, 'compartment mir gene': 1, 'compartment gene mir': 1, 'between mir gene reg corr': 1, 'return m g by': 1, 'compartment mir gene neg corr': 1})\n"
     ]
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "runCheck(-1, \"TRAIN\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading stems\n",
      "Loading relations scai_test_rels.tsv\n",
      "Interactions 232\n",
      "Non NA Interactions 138\n",
      "Three bioinformatics-verified miRNA-128 targets, angiopoietin-related growth factor protein 5 (ARP5; ANGPTL6), a transcription suppressor that promotes stem cell renewal and inhibits the expression of known tumor suppressor genes involved in senescence and differentiation, Bmi-1, and a transcription factor critical for the control of cell-cycle progression, E2F-3a, were found to be up-regulated. \n",
      "('miRNA-128', 'Specific_miRNAs', (30, 39)) ('Bmi-1', 'Genes/Proteins', (274, 279))\n",
      "('MIR_GENE', 'DOWN') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration. \n",
      "('miR-433', 'Specific_miRNAs', (33, 40)) ('FGF20', 'Genes/Proteins', (81, 86))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration. \n",
      "('miR-433', 'Specific_miRNAs', (33, 40)) ('alpha-synuclein', 'Genes/Proteins', (118, 133))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "counts between 2 Counter() NEG\n",
      "counts between 2 Counter() NEG\n",
      "Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-synuclein 3'-untranslated region in primary neurons. \n",
      "('mir-153', 'Specific_miRNAs', (125, 132)) ('alpha-synuclein', 'Genes/Proteins', (192, 207))\n",
      "('MIR_GENE', 'UP') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "counts between 2 Counter() NEG\n",
      "We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele. \n",
      "('miR-211', 'Specific_miRNAs', (94, 101)) ('angiopoietin-1', 'Genes/Proteins', (51, 65))\n",
      "('MIR_GENE', 'DOWN') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "counts between 2 Counter({'NEG': 2}) NEG\n",
      "Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.   \n",
      "('miR-211', 'Specific_miRNAs', (146, 153)) ('angiopoietin-1', 'Genes/Proteins', (79, 93))\n",
      "('MIR_GENE', 'DOWN') counts between  passive? False\n",
      "('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein. \n",
      "('miR-107', 'Specific_miRNAs', (66, 73)) ('GRN', 'Genes/Proteins', (193, 196))\n",
      "('MIR_GENE', 'NEU') counts between  passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA. \n",
      "('microRNA 124a', 'Specific_miRNAs', (55, 68)) ('CDK6', 'Genes/Proteins', (40, 44))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.  \n",
      "('microRNA 124', 'Specific_miRNAs', (43, 55)) ('cyclin dependent kinase 6', 'Genes/Proteins', (14, 39))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons. \n",
      "('miR-128a', 'Specific_miRNAs', (73, 81)) ('BAG2', 'Genes/Proteins', (0, 4))\n",
      "('GENE_MIR', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "Total:      232\n",
      "Correct:    218 0.9396551724137931\n",
      "Incorrect:  14 0.0603448275862069\n",
      "classes Counter({(True, False): 68, (False, True): 55})\n",
      "Correct classified:      128 0.5517241379310345\n",
      "Incorrect classified:    10 0.04310344827586207\n",
      "Counter({'counts between ': 7, 'return': 3})\n",
      "[[[132.   6.]\n",
      "  [  0.   0.]]\n",
      "\n",
      " [[134.   0.]\n",
      "  [  4.   0.]]\n",
      "\n",
      " [[137.   1.]\n",
      "  [  0.   0.]]\n",
      "\n",
      " [[137.   0.]\n",
      "  [  1.   0.]]\n",
      "\n",
      " [[ 88.  50.]\n",
      "  [  0.   0.]]\n",
      "\n",
      " [[112.   0.]\n",
      "  [ 26.   0.]]\n",
      "\n",
      " [[134.   0.]\n",
      "  [  4.   0.]]\n",
      "\n",
      " [[116.   0.]\n",
      "  [ 22.   0.]]\n",
      "\n",
      " [[ 60.  78.]\n",
      "  [  0.   0.]]\n",
      "\n",
      " [[ 72.   0.]\n",
      "  [ 66.   0.]]\n",
      "\n",
      " [[137.   0.]\n",
      "  [  1.   0.]]\n",
      "\n",
      " [[126.   0.]\n",
      "  [ 12.   0.]]\n",
      "\n",
      " [[136.   0.]\n",
      "  [  2.   0.]]\n",
      "\n",
      " [[135.   3.]\n",
      "  [  0.   0.]]]\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'classification_report' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-33-39f6ebbe07cb>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mrelClassifier\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mSentenceRelationClassifier\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m \u001b[0mrunCheck\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"TEST\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-32-a48f289a64e1>\u001b[0m in \u001b[0;36mrunCheck\u001b[0;34m(numelems, base)\u001b[0m\n\u001b[1;32m    206\u001b[0m             \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclassification_report\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlClassifyTrue\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mlClassifyPred\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    207\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 208\u001b[0;31m         \u001b[0mprintStats\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstdout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    209\u001b[0m         \u001b[0;31m#printStats(sys.stderr)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    210\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-32-a48f289a64e1>\u001b[0m in \u001b[0;36mprintStats\u001b[0;34m(outfile)\u001b[0m\n\u001b[1;32m    204\u001b[0m             \u001b[0mmatrix\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmultilabel_confusion_matrix\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlClassifyTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlClassifyPred\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mallLabels\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    205\u001b[0m             \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmatrix\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 206\u001b[0;31m             \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclassification_report\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlClassifyTrue\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mlClassifyPred\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    207\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    208\u001b[0m         \u001b[0mprintStats\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstdout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'classification_report' is not defined"
     ]
    }
   ],
   "source": [
    "%aimport textmining.MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier()\n",
    "\n",
    "runCheck(-1, \"TEST\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "def testRelClassifier(doc, mirword, geneword, relchecker, verbose):\n",
    "    \n",
    "    if verbose:\n",
    "        print()\n",
    "        print()\n",
    "        print()\n",
    "        print(\"Rel Classifier\")\n",
    "    \n",
    "        print(doc.text_with_ws)\n",
    "    \n",
    "    foundStems = set()\n",
    "    sdpRes = relchecker._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "    if verbose:\n",
    "        for x in sdpRes:\n",
    "            print(x)\n",
    "        \n",
    "        \n",
    "    sdpTokens = [x[0] for x in sdpRes]\n",
    "    #print(sdpRes)\n",
    "    \n",
    "    sdpVerbRels = set()\n",
    "\n",
    "    for sidx, x in enumerate(sdpRes):\n",
    "        if x[1] == \"VERB\":\n",
    "            for s in allStems:\n",
    "                if s in str(x[0]):\n",
    "                    sdpVerbRels.add((s, sidx, stem2class[s]))\n",
    "                    if verbose:\n",
    "                        print(\"sdpVerbRels\", x[0], s, sidx, stem2class[s])\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "            \n",
    "    docRels = set()\n",
    "    for tkn in doc:\n",
    "\n",
    "        if tkn in sdpTokens:\n",
    "            continue\n",
    "\n",
    "        for s in allStems:\n",
    "            if str(tkn).lower().startswith(s):\n",
    "                docRels.add((s, sidx, stem2class[s]))\n",
    "                if verbose:    \n",
    "                    print(\"docRels\", tkn, s, sidx, stem2class[s])\n",
    "            \n",
    "            \n",
    "    mirAncests = [x for x in mirword.ancestors]\n",
    "    geneAncests = [x for x in geneword.ancestors if not x in mirAncests]\n",
    "    \n",
    "    if len(mirAncests) > 1:\n",
    "        useMirAncests = mirAncests[0:1]\n",
    "    else:\n",
    "        useMirAncests = mirAncests\n",
    "        \n",
    "    if len(geneAncests) > 1:\n",
    "        useGeneAncests = geneAncests[0:1]\n",
    "    else:\n",
    "        useGeneAncests = geneAncests\n",
    "        \n",
    "    useGeneAncests = [x for x in useGeneAncests if not x in useMirAncests]\n",
    "    \n",
    "    if verbose:\n",
    "        print(\"mirc\", mirAncests)\n",
    "        print(\"genec\", geneAncests)\n",
    "    \n",
    "    invCor = False\n",
    "    regdir = \"NEU\"\n",
    "    intdir = \"MIR_GENE\"\n",
    "    isPassive = relchecker.checkPassive(doc, mirword, geneword, verbose=False)\n",
    "    if verbose:\n",
    "        print(\"passive?\", isPassive)\n",
    "    \n",
    "    ancestHandled = False\n",
    "    #if len(mirAncests) > 0 and len(geneAncests) > 0:\n",
    "    if True:\n",
    "        mirRels = []\n",
    "        geneRels = []\n",
    "        \n",
    "        mirRels = getRelsForTokens(useMirAncests, \"ancestRels mir\", verbose)     \n",
    "        neuMirs = [x for x in mirRels if x[2] in [\"NEU\"]]\n",
    "        #if len(neuMirs) > 0 or len(mirRels) == 0:\n",
    "        #    mirRels = getRelsForTokens(mirAncests, \"ancestRels all mir\", verbose)\n",
    "            \n",
    "        geneRels = getRelsForTokens(useGeneAncests, \"ancestRels gene\", verbose)       \n",
    "        neuGenes = [x for x in geneRels if x[2] in [\"NEU\"]]\n",
    "                \n",
    "        if False and ((len(neuGenes) == len(geneRels)) or len(geneRels) == 0):\n",
    "            geneRels = getRelsForTokens([x for x in geneAncests if not x in useMirAncests], \"ancestRels all gene\", verbose)\n",
    "            neuGenes = [x for x in geneRels if x[2] in [\"NEU\"]]\n",
    "            \n",
    "            if (len(neuGenes) == len(geneRels)) or len(geneRels) == 0:\n",
    "                \n",
    "                if mirword.i < geneword.i:\n",
    "                    tks = [x for x in doc if geneword.i -8 < x.i < geneword.i if not x in useMirAncests]\n",
    "                else:\n",
    "                    tks = [x for x in doc if geneword.i < x.i < geneword.i+5 if not x in useMirAncests]\n",
    "                \n",
    "                geneRels = getRelsForTokens(tks, \"ancestRels tks gene\", verbose)\n",
    "                \n",
    "                                        \n",
    "        posRegs = [x for x in mirRels+geneRels if x[2] in [\"POS\"]]\n",
    "        negRegs = [x for x in mirRels+geneRels if x[2] in [\"NEG\"]]\n",
    "        \n",
    "        mirPos = len([x for x in mirRels if x[2] in [\"POS\"]])\n",
    "        mirNeu = [x for x in mirRels if x[2] in [\"NEU\"]]\n",
    "        mirNeg = len([x for x in mirRels if x[2] in [\"NEG\"]])\n",
    "        \n",
    "        genePos = len([x for x in geneRels if x[2] in [\"POS\"]])\n",
    "        geneNeu = [x for x in geneRels if x[2] in [\"NEU\"]]\n",
    "        geneNeg = len([x for x in geneRels if x[2] in [\"NEG\"]])\n",
    "    \n",
    "        dirReg = (mirPos - geneNeg) + (genePos-mirNeg)\n",
    "        \n",
    "        invCor = False\n",
    "        \n",
    "        if verbose:\n",
    "            print(mirword, geneword)\n",
    "            print(\"ancestRel counts\", mirPos, mirNeg, genePos, geneNeg)\n",
    "                          \n",
    "            \n",
    "        nregdir, ninvcor = evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg)\n",
    "        if verbose:\n",
    "            print(\"evalcount\")\n",
    "            print(nregdir, ninvcor)\n",
    "        \n",
    "        if nregdir != None:\n",
    "            regdir = nregdir\n",
    "            invcor = ninvcor\n",
    "            \n",
    "            ancestHandled = True\n",
    "            \n",
    "            if verbose:\n",
    "                print(\"accepted by evalcount\")\n",
    "            \n",
    "        else:\n",
    "            lookBehind = 8\n",
    "\n",
    "                                \n",
    "            if len(mirNeu) > 0 or len(geneNeu) > 0:\n",
    "                if verbose:\n",
    "                    print(\"ancestRel neu part\")\n",
    "\n",
    "                if mirword.i < geneword.i:\n",
    "                    for x in mirNeu:\n",
    "                        if x[0].startswith((\"correlat\", \"affect\")):\n",
    "\n",
    "                            for j in range(1,lookBehind):\n",
    "                                if str(doc[x[4].i-j]).startswith((\"posit\", \"high\")):\n",
    "                                    mirPos += 1\n",
    "                                    \n",
    "                                elif str(doc[x[4].i-j]).startswith((\"negat\", \"low\")):\n",
    "                                    mirNeg += 1\n",
    "                else:\n",
    "                    for x in geneNeu:\n",
    "                        if x[0].startswith((\"correlat\")):\n",
    "                            \n",
    "                            for j in range(1,lookBehind):\n",
    "                                if str(doc[x[4].i-j]).startswith((\"posit\", \"high\")):\n",
    "                                    genePos += 1\n",
    "                                elif str(doc[x[4].i-j]).startswith((\"negat\", \"low\")):\n",
    "                                    geneNeg += 1    \n",
    "            \n",
    "                                \n",
    "            if not ancestHandled:    \n",
    "                \n",
    "                                \n",
    "                    \n",
    "                if verbose:\n",
    "                    print(\"ancestRel counts stem check\", mirPos, mirNeg, genePos, geneNeg)\n",
    "                    \n",
    "                nregdir, ninvcor = evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg)\n",
    "                if nregdir != None:\n",
    "                    regdir = nregdir\n",
    "                    invcor = ninvcor\n",
    "                    ancestHandled = True\n",
    "                    \n",
    "                    if verbose:\n",
    "                        print(\"accept after before stem check\")\n",
    "                    \n",
    "                if not ancestHandled:\n",
    "                    for j in range(1,lookBehind):\n",
    "                        \n",
    "                        if mirword.i-j < 0:\n",
    "                            continue\n",
    "                        \n",
    "                        jtk = doc[mirword.i-j]\n",
    "                        if verbose:\n",
    "                            print(\"word check mir\", j, jtk)\n",
    "                        pos, neg = scoreWord(jtk)\n",
    "                        \n",
    "                        mirPos += pos\n",
    "                        mirNeg += neg\n",
    "                            \n",
    "                    for j in range(1,lookBehind):\n",
    "                        \n",
    "                        if geneword.i-j < 0:\n",
    "                            continue\n",
    "                        \n",
    "                        jtk = doc[geneword.i-j]\n",
    "                        if verbose:\n",
    "                            print(\"word check gene\", j, jtk)\n",
    "                            \n",
    "                        pos, neg = scoreWord(jtk)\n",
    "                        \n",
    "                        genePos += pos\n",
    "                        geneNeg += neg\n",
    "                            \n",
    "                            \n",
    "                if verbose:\n",
    "                    print(\"ancestRel counts word check\", mirPos, mirNeg, genePos, geneNeg)\n",
    "                \n",
    "                nregdir, ninvcor = evalCounts(dirReg, mirPos, mirNeg, genePos, geneNeg)\n",
    "                if nregdir != None:\n",
    "                    regdir = nregdir\n",
    "                    invcor = ninvcor\n",
    "                    ancestHandled = True\n",
    "                    \n",
    "                    if verbose:\n",
    "                        print(\"accept after before word check\")\n",
    "                    \n",
    "           \n",
    "\n",
    "                \n",
    "                \n",
    "        if verbose:\n",
    "            print(\"ancestRels\", dirReg, invCor)\n",
    "            \n",
    "    if ancestHandled:\n",
    "        \n",
    "        if isPassive:\n",
    "            intdir = \"GENE_MIR\"\n",
    "            \n",
    "    \n",
    "    if not ancestHandled:\n",
    "        \n",
    "        if verbose:\n",
    "            print(\"ancestHandled part\")\n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "            lword = geneword\n",
    "            rword = mirword\n",
    "            useTks = [x for x in doc if geneword.i < x.i < mirword.i]\n",
    "        else:\n",
    "            lword = mirword\n",
    "            rword = geneword\n",
    "            useTks = [x for x in doc if mirword.i < x.i < geneword.i]\n",
    "            \n",
    "        if verbose:\n",
    "            print(useTks)\n",
    "        \n",
    "        betRels = getRelsForTokens(useTks, \"ancestRels useTks\", verbose)\n",
    "        \n",
    "        if verbose:\n",
    "            print(betRels)\n",
    "        \n",
    "        betPos = len([x for x in betRels if x[2] in [\"POS\"]])\n",
    "        betNeg = len([x for x in betRels if x[2] in [\"NEG\"]])\n",
    "        \n",
    "        if betPos + betNeg == 1:\n",
    "            ancestHandled = True\n",
    "            \n",
    "            if verbose:\n",
    "                print(\"ancest bet handled\", mirword.i, geneword.i, isPassive)\n",
    "            \n",
    "            if geneword.i < mirword.i:\n",
    "                \n",
    "                if isPassive:\n",
    "                    intdir = \"MIR_GENE\"\n",
    "                    regdir = \"DOWN\" if betNeg > 0 else \"UP\"\n",
    "                    \n",
    "                else:\n",
    "                    intdir = \"GENE_MIR\"\n",
    "                    regdir = \"UP\" if betPos > 0 else \"DOWN\"\n",
    "                    \n",
    "            else:\n",
    "                \n",
    "                if isPassive:\n",
    "                    intdir = \"GENE_MIR\"\n",
    "                    regdir = \"UP\" if betPos > 0 else \"DOWN\"\n",
    "                    \n",
    "                else:\n",
    "                    intdir = \"MIR_GENE\"\n",
    "                    regdir = \"UP\" if betPos > 0 else \"DOWN\"\n",
    "                    \n",
    "        else:\n",
    "            \n",
    "            isMediated = False\n",
    "            \n",
    "            for x in [lword] + useTks[0:2]:\n",
    "                for y in [\"mediated\", \"-mediated\"]:\n",
    "                    \n",
    "                    if y in str(x):\n",
    "                        isMediated = True\n",
    "                        break\n",
    "                if isMediated:\n",
    "                    break\n",
    "                                    \n",
    "            if isMediated:\n",
    "                \n",
    "                ancestHandled = True\n",
    "                \n",
    "                if verbose:\n",
    "                    print(\"ancest mediated handled\")\n",
    "                \n",
    "                if geneword.i < mirword.i:\n",
    "                    intdir = \"GENE_MIR\"\n",
    "                else:\n",
    "                    intdir = \"MIR_GENE\"\n",
    "                    \n",
    "                if betPos > betNeg:\n",
    "                    regdir = \"UP\"\n",
    "                elif betNeg > betPos:\n",
    "                    regdir = \"DOWN\"\n",
    "                        \n",
    "            \n",
    "            \n",
    "            \n",
    "\n",
    "    #classCounter = Counter()\n",
    "    #if len(classCounter) > 0:\n",
    "    #    regdir = classCounter.most_common(1)[0][0]\n",
    "    #else:\n",
    "    #    regdir = \"NEU\"\n",
    "        \n",
    "    #if invCor:\n",
    "    #    regdir = \"DOWN\"       \n",
    "              \n",
    "    \n",
    "    if not ancestHandled:\n",
    "        \n",
    "        if verbose:\n",
    "            print(\"in final resolution\")\n",
    "        \n",
    "        \n",
    "        if geneword.i < mirword.i:\n",
    "        \n",
    "            textBetween = \" \".join([str(x) for x in doc if geneword.i < x.i < mirword.i])\n",
    "            textBetween = ' '.join(textBetween.split())\n",
    "            \n",
    "            if verbose:\n",
    "                print(textBetween)\n",
    "            \n",
    "\n",
    "            if len(textBetween) < 10 and any([x in textBetween for x in [\"by\"]]):\n",
    "                sdpRels = set()\n",
    "                for sidx,x  in enumerate(sdpRes):\n",
    "                    for s in allStems:\n",
    "                        if str(x[0]).lower().startswith(s):\n",
    "                            sdpRels.add((s, sidx, stem2class[s]))\n",
    "                            if verbose:\n",
    "                                print(\"sdpRels\", x[0], s, sidx, stem2class[s])\n",
    "\n",
    "                posRegs = [x for x in sdpRels if x[2] in [\"POS\"]]\n",
    "                negRegs = [x for x in sdpRels if x[2] in [\"NEG\"]]\n",
    "\n",
    "                dirReg = len(posRegs)-len(negRegs)\n",
    "\n",
    "                if dirReg > 0:\n",
    "                    regdir  = \"UP\"\n",
    "                elif dirReg < 0:\n",
    "                    regdir = \"DOWN\"\n",
    "                else:\n",
    "                    regdir = \"NEU\"\n",
    "                \n",
    "                intdir = \"MIR_GENE\"\n",
    "\n",
    "            elif any([x in textBetween for x in [\"target of\", \"target for\", \"targeting\"]]):\n",
    "                intdir = \"MIR_GENE\"\n",
    "                regdir = \"NEU\"\n",
    "\n",
    "            else:                \n",
    "                intdir = \"GENE_MIR\"\n",
    "    \n",
    "        elif mirword.i < geneword.i:\n",
    "            textBetween = \" \".join([str(x) for x in doc if geneword.i-5 < x.i < geneword.i])\n",
    "\n",
    "            if any([x in textBetween for x in [\"by\"]]):\n",
    "                intdir = \"GENE_MIR\"\n",
    "                regdir = \"UP\"\n",
    "        \n",
    "    \n",
    "    regstems = foundStems\n",
    "    \n",
    "    #displacy.render(doc, style=\"dep\")\n",
    "    \n",
    "    if verbose:\n",
    "        print((intdir, regdir, \"NA\", isPassive))\n",
    "        \n",
    "    return {\"regulation_dir\": regdir, \"interaction_dir\": intdir, \"regulation_keys\": regstems}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
